diff --git a/build_overrides/build.gni b/build_overrides/build.gni
index 0115814931c760e935dbef5d8432a63e68b355f2..367e05003c87b7e25d07ff78890cde7a3d32686d 100644
--- a/build_overrides/build.gni
+++ b/build_overrides/build.gni
@@ -13,6 +13,11 @@ import("//build/config/gclient_args.gni")
 # Some non-Chromium builds don't support building java targets.
 enable_java_templates = true
 
+# FlyWeb: needed by V8 12.3's gni/v8.gni (engine level 123); Chromium 35990ea269bc.
+# Variable that can be used to support multiple build scenarios, like when
+# V8 is embedded within a target.
+build_with_v8_embedder = build_with_chromium
+
 # Enables assertions on safety checks in libc++.
 #
 # This should be set to true whenever building with Chrome's custom libc++ in
