diff --git a/chrome/renderer/extensions/api/app_hooks_delegate.cc b/chrome/renderer/extensions/api/app_hooks_delegate.cc
index fe4ea590cd939ea136beaa6bfc10ea7d47a20000..a7912e9a2258fb465c80199d53f37e097a6a78e6 100644
--- a/chrome/renderer/extensions/api/app_hooks_delegate.cc
+++ b/chrome/renderer/extensions/api/app_hooks_delegate.cc
@@ -32,7 +32,7 @@ namespace extensions {
 
 // static
 void AppHooksDelegate::IsInstalledGetterCallback(
-    v8::Local<v8::String> property,
+    v8::Local<v8::Name> property,
     const v8::PropertyCallbackInfo<v8::Value>& info) {
   v8::HandleScope handle_scope(info.GetIsolate());
   v8::Local<v8::Context> context = info.Holder()->GetCreationContextChecked();
