diff --git a/chrome/renderer/extensions/api/app_hooks_delegate.h b/chrome/renderer/extensions/api/app_hooks_delegate.h
index d6a4a052fa8fdb31083ac497c965c60dc05f2178..6d58c85ed914297aec32c7282942c88056b70272 100644
--- a/chrome/renderer/extensions/api/app_hooks_delegate.h
+++ b/chrome/renderer/extensions/api/app_hooks_delegate.h
@@ -47,7 +47,7 @@ class AppHooksDelegate : public APIBindingHooksDelegate {
 
  private:
   static void IsInstalledGetterCallback(
-      v8::Local<v8::String> property,
+      v8::Local<v8::Name> property,
       const v8::PropertyCallbackInfo<v8::Value>& info);
 
   // Returns the manifest of the extension associated with the frame.
