diff --git a/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.h b/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.h
index c172f33df91978e26989a59e44965f233f696178..5181363725329796aeef5ede41a9d7730411efc5 100644
--- a/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.h
+++ b/content/app_shim_remote_cocoa/render_widget_host_view_cocoa.h
@@ -123,8 +123,9 @@ struct DidOverscrollParams;
   // Represents the input-method attributes supported by this object.
   base::scoped_nsobject<NSArray> _validAttributesForMarkedText;
 
-  // Indicates if we are currently handling a key down event.
-  BOOL _handlingKeyDown;
+  // keyCode value of an NSEvent. This field has a value while we're handling
+  // a key down event.
+  absl::optional<unsigned short> _currentKeyDownCode;
 
   // Indicates if a reconversion (which means a piece of committed text becomes
   // part of the composition again) is triggered in Japanese IME when Live
@@ -195,7 +196,7 @@ struct DidOverscrollParams;
   // The set of key codes from key down events that we haven't seen the matching
   // key up events yet.
   // Used for filtering out non-matching NSEventTypeKeyUp events.
-  std::set<unsigned short> _keyDownCodes;
+  std::set<unsigned short> _unmatchedKeyDownCodes;
 
   // The filter used to guide touch events towards a horizontal or vertical
   // orientation.
