diff --git a/content/renderer/dom_automation_controller.cc b/content/renderer/dom_automation_controller.cc
index e18f7cd524017ae93341c707576d497059fc6850..ac322ebb9376284b49a15e274aa25a3aee3a2b41 100644
--- a/content/renderer/dom_automation_controller.cc
+++ b/content/renderer/dom_automation_controller.cc
@@ -13,6 +13,7 @@
 #include "third_party/blink/public/common/associated_interfaces/associated_interface_provider.h"
 #include "third_party/blink/public/web/blink.h"
 #include "third_party/blink/public/web/web_local_frame.h"
+#include "v8/include/v8-context.h"
 
 namespace content {
 
