diff --git a/content/browser/renderer_host/navigation_controller_impl.h b/content/browser/renderer_host/navigation_controller_impl.h
index 2859e778eabe8a4527ba1b7b88d424c3ca9147cb..f9f1ddd44bbe5c7cf4d1999518db81a33649b544 100644
--- a/content/browser/renderer_host/navigation_controller_impl.h
+++ b/content/browser/renderer_host/navigation_controller_impl.h
@@ -854,7 +854,8 @@ class CONTENT_EXPORT NavigationControllerImpl : public NavigationController {
   void BroadcastHistoryOffsetAndLength();
 
   // Used by PopulateNavigationApiHistoryEntryVectors to initialize a single
-  // vector.
+  // vector. `last_index_checked` is an out parameter that indicates the last
+  // entry index walked in `direction` before stopping.
   std::vector<blink::mojom::NavigationApiHistoryEntryPtr>
   PopulateSingleNavigationApiHistoryEntryVector(
       Direction direction,
@@ -863,7 +864,8 @@ class CONTENT_EXPORT NavigationControllerImpl : public NavigationController {
       FrameTreeNode* node,
       SiteInstance* site_instance,
       int64_t pending_item_sequence_number,
-      int64_t pending_document_sequence_number);
+      int64_t pending_document_sequence_number,
+      int& last_index_checked);
   // Helper for NavigateToNavigationApiKey(). Ensures that we only navigate to
   // |target_entry| if it matches |current_entry|'s origin and site instance, as
   // well as having |navigation_api_key| as its key.
