diff --git a/gin/shell_runner.cc b/gin/shell_runner.cc
index 11d2468998a2d9a479268fe92a35d2e2104b123e..0c51bf7dc226f2b89a6bd43aa36f97ce6a320d63 100644
--- a/gin/shell_runner.cc
+++ b/gin/shell_runner.cc
@@ -66,7 +66,7 @@ v8::MaybeLocal<v8::Value> ShellRunner::Run(const std::string& source,
                                            const std::string& resource_name) {
   v8::Isolate* isolate = GetContextHolder()->isolate();
   TryCatch try_catch(isolate);
-  v8::ScriptOrigin origin(isolate, StringToV8(isolate, resource_name));
+  v8::ScriptOrigin origin(StringToV8(isolate, resource_name));
   auto maybe_script = Script::Compile(GetContextHolder()->context(),
                                       StringToV8(isolate, source), &origin);
   v8::Local<Script> script;
