diff --git a/gin/v8_foreground_task_runner_with_locker.h b/gin/v8_foreground_task_runner_with_locker.h
index d3242de2f2484452ea9097a22924b093a6280b81..288ed4f8f175b6287f141e58c360101517495a11 100644
--- a/gin/v8_foreground_task_runner_with_locker.h
+++ b/gin/v8_foreground_task_runner_with_locker.h
@@ -23,19 +23,20 @@ class V8ForegroundTaskRunnerWithLocker : public V8ForegroundTaskRunnerBase {
 
   ~V8ForegroundTaskRunnerWithLocker() override;
 
-  // v8::Platform implementation.
-  void PostTask(std::unique_ptr<v8::Task> task) override;
-
-  void PostNonNestableTask(std::unique_ptr<v8::Task> task) override;
-
-  void PostDelayedTask(std::unique_ptr<v8::Task> task,
-                       double delay_in_seconds) override;
-
-  void PostIdleTask(std::unique_ptr<v8::IdleTask> task) override;
-
   bool NonNestableTasksEnabled() const override;
 
  private:
+  // v8::Platform implementation.
+  void PostTaskImpl(std::unique_ptr<v8::Task> task,
+                    const v8::SourceLocation& location) override;
+  void PostNonNestableTaskImpl(std::unique_ptr<v8::Task> task,
+                               const v8::SourceLocation& location) override;
+  void PostDelayedTaskImpl(std::unique_ptr<v8::Task> task,
+                           double delay_in_seconds,
+                           const v8::SourceLocation& location) override;
+  void PostIdleTaskImpl(std::unique_ptr<v8::IdleTask> task,
+                        const v8::SourceLocation& location) override;
+
   raw_ptr<v8::Isolate> isolate_;
   scoped_refptr<base::SingleThreadTaskRunner> task_runner_;
 };
