diff --git a/gin/isolate_holder.cc b/gin/isolate_holder.cc
index e8608bb868c82c9af5426cd6f8512ce6b18ff39f..d530169c359e180fc94aba9258e76f7d199b3e18 100644
--- a/gin/isolate_holder.cc
+++ b/gin/isolate_holder.cc
@@ -165,7 +165,6 @@ IsolateHolder::getDefaultIsolateParams() {
   params->array_buffer_allocator = g_array_buffer_allocator;
   params->allow_atomics_wait = true;
   params->external_references = g_reference_table;
-  params->only_terminate_in_safe_scope = true;
   params->embedder_wrapper_type_index = kWrapperInfoIndex;
   params->embedder_wrapper_object_index = kEncodedValueIndex;
   params->fatal_error_callback = g_fatal_error_callback;
