diff --git a/gin/v8_platform.cc b/gin/v8_platform.cc
index a0bc00a3d1397934276517c19e0a76098b7099f5..b5c68c2a8546ff506ce69b7459082f9578c40143 100644
--- a/gin/v8_platform.cc
+++ b/gin/v8_platform.cc
@@ -25,6 +25,7 @@
 #include "base/trace_event/trace_event.h"
 #include "base/tracing_buildflags.h"
 #include "build/build_config.h"
+#include "gin/converter.h"
 #include "gin/per_isolate_data.h"
 #include "gin/thread_isolation.h"
 #include "gin/v8_platform_thread_isolated_allocator.h"
@@ -136,11 +137,6 @@ base::TaskPriority ToBaseTaskPriority(v8::TaskPriority priority) {
   }
 }
 
-base::Location ToBaseLocation(const v8::SourceLocation& location) {
-  return base::Location::Current(location.Function(), location.FileName(),
-                                 location.Line());
-}
-
 class JobDelegateImpl : public v8::JobDelegate {
  public:
   explicit JobDelegateImpl(base::JobDelegate* delegate) : delegate_(delegate) {}
@@ -381,7 +377,7 @@ void V8Platform::PostTaskOnWorkerThreadImpl(
     v8::TaskPriority priority,
     std::unique_ptr<v8::Task> task,
     const v8::SourceLocation& location) {
-  base::ThreadPool::PostTask(ToBaseLocation(location),
+  base::ThreadPool::PostTask(V8ToBaseLocation(location),
                              {ToBaseTaskPriority(priority)},
                              base::BindOnce(&v8::Task::Run, std::move(task)));
 }
@@ -392,7 +388,7 @@ void V8Platform::PostDelayedTaskOnWorkerThreadImpl(
     double delay_in_seconds,
     const v8::SourceLocation& location) {
   base::ThreadPool::PostDelayedTask(
-      ToBaseLocation(location), {ToBaseTaskPriority(priority)},
+      V8ToBaseLocation(location), {ToBaseTaskPriority(priority)},
       base::BindOnce(&v8::Task::Run, std::move(task)),
       base::Seconds(delay_in_seconds));
 }
@@ -405,7 +401,7 @@ std::unique_ptr<v8::JobHandle> V8Platform::CreateJobImpl(
   // |max_concurrency_callback| uses an unretained pointer.
   auto* job_task_ptr = job_task.get();
   auto handle =
-      base::CreateJob(ToBaseLocation(location), {ToBaseTaskPriority(priority)},
+      base::CreateJob(V8ToBaseLocation(location), {ToBaseTaskPriority(priority)},
                       base::BindRepeating(
                           [](const std::unique_ptr<v8::JobTask>& job_task,
                              base::JobDelegate* delegate) {
