diff --git a/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py b/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
index 0ad6495c4285f842c006edd3539b16b068195fc9..109bed5d59beb668c746dfc9b8b75d12b299bd29 100644
--- a/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
+++ b/third_party/blink/renderer/bindings/scripts/web_idl/__init__.py
@@ -57,6 +57,8 @@ from .extended_attribute import ExtendedAttributes
 from .function_like import FunctionLike
 from .function_like import OverloadGroup
 from .idl_type import IdlType
+from .interface import AsyncIterable
+from .interface import AsyncIterator
 from .interface import IndexedAndNamedProperties
 from .interface import Interface
 from .interface import Iterable
