diff --git a/third_party/blink/renderer/core/view_transition/view_transition_supplement.h b/third_party/blink/renderer/core/view_transition/view_transition_supplement.h
index 3bfc9174b7642e077c3db1e3499677f4d986e8a1..e03d6caa0f55df37500170b50ac758e2690e7513 100644
--- a/third_party/blink/renderer/core/view_transition/view_transition_supplement.h
+++ b/third_party/blink/renderer/core/view_transition/view_transition_supplement.h
@@ -16,6 +16,7 @@
 namespace blink {
 class ViewTransition;
 class V8ViewTransitionCallback;
+class ViewTransitionOptions;
 
 class CORE_EXPORT ViewTransitionSupplement
     : public GarbageCollected<ViewTransitionSupplement>,
@@ -34,6 +35,15 @@ class CORE_EXPORT ViewTransitionSupplement
                                              Document&,
                                              V8ViewTransitionCallback* callback,
                                              ExceptionState&);
+  // FlyWeb (Chromium 125): with options (update callback and types), and
+  // without arguments.
+  static ViewTransition* startViewTransition(ScriptState*,
+                                             Document&,
+                                             ViewTransitionOptions* options,
+                                             ExceptionState&);
+  static ViewTransition* startViewTransition(ScriptState*,
+                                             Document&,
+                                             ExceptionState&);
 
   // Creates a ViewTransition to cache the state of a Document before a
   // navigation. The cached state is provided to the caller using the
@@ -74,9 +84,16 @@ class CORE_EXPORT ViewTransitionSupplement
   void WillInsertBody();
 
  private:
+  static ViewTransition* StartViewTransitionInternal(
+      ScriptState*,
+      Document&,
+      V8ViewTransitionCallback* callback,
+      const absl::optional<Vector<String>>& types,
+      ExceptionState&);
   ViewTransition* StartTransition(ScriptState* script_state,
                                   Document& document,
                                   V8ViewTransitionCallback* callback,
+                                  const absl::optional<Vector<String>>& types,
                                   ExceptionState& exception_state);
   void StartTransition(Document& document,
                        ViewTransition::ViewTransitionStateCallback callback);
