diff --git a/third_party/blink/renderer/platform/graphics/gpu/drawing_buffer.cc b/third_party/blink/renderer/platform/graphics/gpu/drawing_buffer.cc
index a2fcea91c6047d888a6660a939640249d2abea0d..dd8fa6dcaebcf343127f0e5e0bbd619408eff449 100644
--- a/third_party/blink/renderer/platform/graphics/gpu/drawing_buffer.cc
+++ b/third_party/blink/renderer/platform/graphics/gpu/drawing_buffer.cc
@@ -169,7 +169,7 @@ scoped_refptr<DrawingBuffer> DrawingBuffer::Create(
   data_size *= size.width();
   data_size *= size.height();
   if (!data_size.IsValid() ||
-      data_size.ValueOrDie() > v8::TypedArray::kMaxLength) {
+      data_size.ValueOrDie() > v8::TypedArray::kMaxByteLength) {
     return nullptr;
   }
 
