diff --git a/third_party/blink/renderer/core/intersection_observer/intersection_observation.cc b/third_party/blink/renderer/core/intersection_observer/intersection_observation.cc
index 170301508e14973eee49c7916f3787f5a5f267a3..fdd62875eb54ad7b9605f27a6dd352d0c52d1842 100644
--- a/third_party/blink/renderer/core/intersection_observer/intersection_observation.cc
+++ b/third_party/blink/renderer/core/intersection_observer/intersection_observation.cc
@@ -49,10 +49,10 @@ int64_t IntersectionObservation::ComputeIntersection(
     absl::optional<base::TimeTicks>& monotonic_time) {
   return ComputeIntersectionInternal(
       [this, &root_geometry](unsigned geometry_flags) {
-        return IntersectionGeometry(root_geometry, *observer_->root(),
-                                    *Target(), observer_->thresholds(),
-                                    observer_->TargetMargin(), geometry_flags,
-                                    cached_rects_.get());
+        return IntersectionGeometry(
+            root_geometry, *observer_->root(), *Target(),
+            observer_->thresholds(), observer_->TargetMargin(),
+            observer_->ScrollMargin(), geometry_flags, cached_rects_.get());
       },
       compute_flags, monotonic_time);
 }
@@ -64,7 +64,8 @@ int64_t IntersectionObservation::ComputeIntersection(
       [this](unsigned geometry_flags) {
         return IntersectionGeometry(
             observer_->root(), *Target(), observer_->RootMargin(),
-            observer_->thresholds(), observer_->TargetMargin(), geometry_flags);
+            observer_->thresholds(), observer_->TargetMargin(),
+            observer_->ScrollMargin(), geometry_flags);
       },
       compute_flags, monotonic_time);
 }
@@ -143,7 +144,7 @@ bool IntersectionObservation::CanUseCachedRectsForTesting() const {
   if (cached_rects_) {
     cached_rects_copy = *cached_rects_;
   }
-  IntersectionGeometry geometry(observer_->root(), *target_, {}, {}, {}, 0,
+  IntersectionGeometry geometry(observer_->root(), *target_, {}, {}, {}, {}, 0,
                                 cached_rects_ ? &cached_rects_copy : nullptr);
   return geometry.CanUseCachedRectsForTesting();
 }
