diff --git a/third_party/blink/renderer/core/paint/svg_mask_painter.h b/third_party/blink/renderer/core/paint/svg_mask_painter.h
index 5f8981acdc51fcc647739be23cf8c0602198e5ff..ae861faf9a4984bfdab8f83ce8b4fa04d1726e54 100644
--- a/third_party/blink/renderer/core/paint/svg_mask_painter.h
+++ b/third_party/blink/renderer/core/paint/svg_mask_painter.h
@@ -5,13 +5,22 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_CORE_PAINT_SVG_MASK_PAINTER_H_
 #define THIRD_PARTY_BLINK_RENDERER_CORE_PAINT_SVG_MASK_PAINTER_H_
 
+#include "third_party/blink/renderer/platform/graphics/paint/paint_record.h"
 #include "third_party/blink/renderer/platform/wtf/allocator/allocator.h"
 
+namespace gfx {
+class RectF;
+}  // namespace gfx
+
 namespace blink {
 
 class DisplayItemClient;
 class GraphicsContext;
 class LayoutObject;
+class SVGResource;
+class SVGResourceClient;
+
+enum class EMaskType : unsigned;
 
 class SVGMaskPainter {
   STATIC_ONLY(SVGMaskPainter);
@@ -20,6 +29,16 @@ class SVGMaskPainter {
   static void Paint(GraphicsContext& context,
                     const LayoutObject& layout_object,
                     const DisplayItemClient& display_item_client);
+  static PaintRecord PaintResource(SVGResource* mask_resource,
+                                   SVGResourceClient& client,
+                                   const gfx::RectF& reference_box,
+                                   float zoom);
+  static gfx::RectF ResourceBounds(SVGResource* mask_resource,
+                                   SVGResourceClient& client,
+                                   const gfx::RectF& reference_box,
+                                   float zoom);
+  static EMaskType MaskType(SVGResource* mask_resource,
+                            SVGResourceClient& client);
 };
 
 }  // namespace blink
