diff --git a/third_party/blink/renderer/core/animation/css_overlay_interpolation_type.h b/third_party/blink/renderer/core/animation/css_overlay_interpolation_type.h
index e8c13765f9b442403cae5af0e895dca60a5b58c6..2c1debba0a08a394c661d1d94f90628141eb53d7 100644
--- a/third_party/blink/renderer/core/animation/css_overlay_interpolation_type.h
+++ b/third_party/blink/renderer/core/animation/css_overlay_interpolation_type.h
@@ -30,6 +30,7 @@ class CSSOverlayInterpolationType : public CSSInterpolationType {
   void ApplyStandardPropertyValue(const InterpolableValue&,
                                   const NonInterpolableValue*,
                                   StyleResolverState&) const final;
+  bool IsDiscrete() const override { return true; }
 
  private:
   InterpolationValue CreateOverlayValue(EOverlay) const;
