diff --git a/third_party/blink/renderer/core/css/resolver/element_style_resources.cc b/third_party/blink/renderer/core/css/resolver/element_style_resources.cc
index d018339055f2a02686043a1761beaf5212b6765a..7ff256bf3cfee359812f8ded54afc3b73f5abc3d 100644
--- a/third_party/blink/renderer/core/css/resolver/element_style_resources.cc
+++ b/third_party/blink/renderer/core/css/resolver/element_style_resources.cc
@@ -48,6 +48,7 @@
 #include "third_party/blink/renderer/core/style/style_image.h"
 #include "third_party/blink/renderer/core/style/style_image_set.h"
 #include "third_party/blink/renderer/core/style/style_pending_image.h"
+#include "third_party/blink/renderer/core/style/style_svg_mask_reference_image.h"
 #include "third_party/blink/renderer/core/svg/svg_resource.h"
 #include "third_party/blink/renderer/core/svg/svg_tree_scope_resources.h"
 #include "third_party/blink/renderer/platform/geometry/length.h"
@@ -305,6 +306,25 @@ static CSSValue* PendingCssValue(StyleImage* style_image) {
   return nullptr;
 }
 
+StyleImage* ElementStyleResources::LoadMaskSource(CSSValue& pending_value) {
+  if (!RuntimeEnabledFeatures::CSSMaskingInteropEnabled()) {
+    return nullptr;
+  }
+  auto* image_value = DynamicTo<CSSImageValue>(pending_value);
+  if (!image_value) {
+    return nullptr;
+  }
+  if (image_value->IsLocal(element_.GetDocument())) {
+    SVGTreeScopeResources& tree_scope_resources =
+        element_.OriginatingTreeScope().EnsureSVGTreeScopedResources();
+    SVGResource* resource = tree_scope_resources.ResourceForId(
+        image_value->NormalizedFragmentIdentifier());
+    return MakeGarbageCollected<StyleSVGMaskReferenceImage>(resource,
+                                                            image_value);
+  }
+  return nullptr;
+}
+
 void ElementStyleResources::LoadPendingImages(ComputedStyleBuilder& builder) {
   // We must loop over the properties and then look at the style to see if
   // a pending image exists, and only load that image. For example:
@@ -401,13 +421,21 @@ void ElementStyleResources::LoadPendingImages(ComputedStyleBuilder& builder) {
         }
         break;
       }
+      case CSSPropertyID::kMaskImage:
       case CSSPropertyID::kWebkitMaskImage: {
         for (FillLayer* mask_layer = &builder.AccessMaskLayers(); mask_layer;
              mask_layer = mask_layer->Next()) {
           if (auto* pending_value = PendingCssValue(mask_layer->GetImage())) {
-            mask_layer->SetImage(loader.Load(
-                *pending_value, FetchParameters::ImageRequestBehavior::kNone,
-                kCrossOriginAttributeAnonymous));
+            StyleImage* image = nullptr;
+            if (property == CSSPropertyID::kMaskImage) {
+              image = LoadMaskSource(*pending_value);
+            }
+            if (!image) {
+              image = loader.Load(*pending_value,
+                                  FetchParameters::ImageRequestBehavior::kNone,
+                                  kCrossOriginAttributeAnonymous);
+            }
+            mask_layer->SetImage(image);
           }
         }
         break;
