diff --git a/third_party/blink/renderer/core/animation/css_font_size_adjust_interpolation_type.cc b/third_party/blink/renderer/core/animation/css_font_size_adjust_interpolation_type.cc
index 2ceff7e2ef149fdda19cffe2c012a2f457cf0552..be9bae45644fba4aec6bd44ce27539e01aa9fa5c 100644
--- a/third_party/blink/renderer/core/animation/css_font_size_adjust_interpolation_type.cc
+++ b/third_party/blink/renderer/core/animation/css_font_size_adjust_interpolation_type.cc
@@ -103,7 +103,7 @@ InterpolationValue CSSFontSizeAdjustInterpolationType::MaybeConvertValue(
     const StyleResolverState* state,
     ConversionCheckers& conversion_checkers) const {
   return CreateFontSizeAdjustValue(
-      StyleBuilderConverterBase::ConvertFontSizeAdjust(value));
+      StyleBuilderConverterBase::ConvertFontSizeAdjust(*state, value));
 }
 
 PairwiseInterpolationValue
