diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.h b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.h
index 0152a043d3e50b9561d605b9d102fde05a3494e2..ea25cae2e8be28c78d0e80c471aa0971c823ec84 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.h
@@ -44,6 +44,10 @@ class LayoutSVGResourceContainer : public LayoutSVGHiddenContainer {
 
   virtual void RemoveAllClientsFromCache() = 0;
 
+  // Like RemoveAllClientsFromCache(), but predicated on if layout has been
+  // performed at all.
+  void InvalidateCache();
+
   // Remove any cached data for the |client|, and return true if so.
   virtual bool RemoveClientFromCache(SVGResourceClient&) {
     NOT_DESTROYED();
@@ -67,9 +71,6 @@ class LayoutSVGResourceContainer : public LayoutSVGHiddenContainer {
            resource_type == kRadialGradientResourceType;
   }
 
-  void InvalidateCacheAndMarkForLayout(LayoutInvalidationReasonForTracing);
-  void InvalidateCacheAndMarkForLayout();
-
   bool FindCycle() const;
 
   static void InvalidateDependentElements(LayoutObject&, bool needs_layout);
@@ -131,30 +132,22 @@ struct DowncastTraits<LayoutSVGResourceContainer> {
   }
 };
 
-template <typename ContainerType>
-inline bool IsResourceOfType(const LayoutSVGResourceContainer* container) {
-  return container->ResourceType() == ContainerType::kResourceType;
-}
-
 template <typename ContainerType>
 inline ContainerType* GetSVGResourceAsType(SVGResourceClient& client,
                                            const SVGResource* resource) {
-  if (!resource)
+  if (!resource) {
     return nullptr;
-  if (LayoutSVGResourceContainer* container =
-          resource->ResourceContainer(client)) {
-    if (IsResourceOfType<ContainerType>(container))
-      return static_cast<ContainerType*>(container);
   }
-  return nullptr;
+  return DynamicTo<ContainerType>(resource->ResourceContainer(client));
 }
 
 template <typename ContainerType>
 inline ContainerType* GetSVGResourceAsType(
     SVGResourceClient& client,
     const StyleSVGResource* style_resource) {
-  if (!style_resource)
+  if (!style_resource) {
     return nullptr;
+  }
   return GetSVGResourceAsType<ContainerType>(client,
                                              style_resource->Resource());
 }
