diff --git a/third_party/blink/renderer/core/html/canvas/text_metrics.idl b/third_party/blink/renderer/core/html/canvas/text_metrics.idl
index 9c58a6eef43f48ae57bf2d1319b1681722bf45fe..b65d6d67381a5f2869a4dfcfd5413d65c7de6eb1 100644
--- a/third_party/blink/renderer/core/html/canvas/text_metrics.idl
+++ b/third_party/blink/renderer/core/html/canvas/text_metrics.idl
@@ -38,7 +38,9 @@ interface TextMetrics {
     readonly attribute double fontBoundingBoxDescent;
     readonly attribute double actualBoundingBoxAscent;
     readonly attribute double actualBoundingBoxDescent;
+    [RuntimeEnabled=TextMetricsBaselines] readonly attribute double hangingBaseline;
+    [RuntimeEnabled=TextMetricsBaselines] readonly attribute double alphabeticBaseline;
+    [RuntimeEnabled=TextMetricsBaselines] readonly attribute double ideographicBaseline;
     [RuntimeEnabled=ExtendedTextMetrics] readonly attribute double emHeightAscent;
     [RuntimeEnabled=ExtendedTextMetrics] readonly attribute double emHeightDescent;
-    [RuntimeEnabled=ExtendedTextMetrics] Baselines getBaselines();
 };
