diff --git a/third_party/blink/renderer/core/svg/svg_image_element.idl b/third_party/blink/renderer/core/svg/svg_image_element.idl
index d99d7eed9f52b49da204acd558dd7c9df209e78c..cebf1c437368100d2eeb163e8ae72188d7e0fe57 100644
--- a/third_party/blink/renderer/core/svg/svg_image_element.idl
+++ b/third_party/blink/renderer/core/svg/svg_image_element.idl
@@ -35,6 +35,8 @@
     [MeasureAs=SVG1DOMImageElement] readonly attribute SVGAnimatedLength height;
     [MeasureAs=SVG1DOMImageElement] readonly attribute SVGAnimatedPreserveAspectRatio preserveAspectRatio;
     [CEReactions, Reflect, ReflectOnly=("async", "sync", "auto"), ReflectMissing="auto", ReflectInvalid="auto"] attribute DOMString decoding;
+    [CEReactions, Reflect, ReflectOnly=("anonymous","use-credentials"), ReflectEmpty="anonymous", ReflectInvalid="anonymous", RuntimeEnabled=SvgCrossOriginAttribute] attribute DOMString? crossOrigin;
+
     // Follow HTMLImageElement's spec
     [CallWith=ScriptState, RaisesException] Promise<void> decode();
 };
