diff --git a/third_party/blink/renderer/core/css/css_variable_data.cc b/third_party/blink/renderer/core/css/css_variable_data.cc
index 22d513f84be64ee39686b58c73ae406752d1001e..982be97893fbd87d5e32007789aee8efff947392 100644
--- a/third_party/blink/renderer/core/css/css_variable_data.cc
+++ b/third_party/blink/renderer/core/css/css_variable_data.cc
@@ -23,6 +23,7 @@ static bool IsFontUnitToken(CSSParserToken token) {
     case CSSPrimitiveValue::UnitType::kChs:
     case CSSPrimitiveValue::UnitType::kExs:
     case CSSPrimitiveValue::UnitType::kIcs:
+    case CSSPrimitiveValue::UnitType::kCaps:
       return true;
     default:
       return false;
@@ -39,6 +40,7 @@ static bool IsRootFontUnitToken(CSSParserToken token) {
     case CSSPrimitiveValue::UnitType::kRchs:
     case CSSPrimitiveValue::UnitType::kRics:
     case CSSPrimitiveValue::UnitType::kRlhs:
+    case CSSPrimitiveValue::UnitType::kRcaps:
       return true;
     default:
       return false;
