diff --git a/third_party/blink/renderer/platform/geometry/calculation_value.h b/third_party/blink/renderer/platform/geometry/calculation_value.h
index f1e809b6dd81a61a1c68691e8bfc6c1e8424a672..565ef459ded2f866beb427afd97708442e4a2388 100644
--- a/third_party/blink/renderer/platform/geometry/calculation_value.h
+++ b/third_party/blink/renderer/platform/geometry/calculation_value.h
@@ -92,6 +92,7 @@ class PLATFORM_EXPORT CalculationValue : public RefCounted<CalculationValue> {
                                               double progress,
                                               Length::ValueRange) const;
   scoped_refptr<const CalculationValue> SubtractFromOneHundredPercent() const;
+  scoped_refptr<const CalculationValue> Add(const CalculationValue&) const;
   scoped_refptr<const CalculationValue> Zoom(double factor) const;
 
  private:
