diff --git a/third_party/blink/renderer/core/mathml/mathml_element.h b/third_party/blink/renderer/core/mathml/mathml_element.h
index 5702618fd70df28b80fc313403dbf1951b66a10e..60e8dc2f23aad6de7bc0be0081d5401197c03b23 100644
--- a/third_party/blink/renderer/core/mathml/mathml_element.h
+++ b/third_party/blink/renderer/core/mathml/mathml_element.h
@@ -62,8 +62,6 @@ class CORE_EXPORT MathMLElement : public Element {
 
   // https://w3c.github.io/mathml-core/#dfn-boolean
   absl::optional<bool> BooleanAttribute(const QualifiedName& name) const;
-
-  LayoutObject* CreateLayoutObject(const ComputedStyle&) override;
 };
 
 template <typename T>
