diff --git a/third_party/blink/renderer/core/paint/ng/ng_text_painter.cc b/third_party/blink/renderer/core/paint/ng/ng_text_painter.cc
index 6a75e9f175de46211d600dfe0fc0f8b8f35ed439..e25a56d969716c5d7af234eaf71e61edcf4a439b 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_text_painter.cc
+++ b/third_party/blink/renderer/core/paint/ng/ng_text_painter.cc
@@ -23,6 +23,7 @@
 #include "third_party/blink/renderer/core/paint/svg_object_painter.h"
 #include "third_party/blink/renderer/core/paint/timing/paint_timing_detector.h"
 #include "third_party/blink/renderer/core/style/computed_style.h"
+#include "third_party/blink/renderer/core/style/paint_order_array.h"
 #include "third_party/blink/renderer/core/style/shadow_list.h"
 #include "third_party/blink/renderer/core/svg/svg_element.h"
 #include "third_party/blink/renderer/platform/fonts/font.h"
@@ -437,10 +438,11 @@ void NGTextPainter::PaintSvgTextFragment(
       GetSvgStyleToPaint(state, SvgPaintMode::kText, selection_style_scope,
                          has_fill, has_visible_stroke);
 
-  for (int i = 0; i < 3; i++) {
+  const PaintOrderArray paint_order(state.Style().PaintOrder());
+  for (unsigned i = 0; i < 3; i++) {
     absl::optional<LayoutSVGResourceMode> resource_mode;
 
-    switch (state.Style().PaintOrderType(i)) {
+    switch (paint_order[i]) {
       case PT_FILL:
         if (has_fill)
           resource_mode = kApplyToFillMode;
@@ -484,10 +486,11 @@ void NGTextPainter::PaintSvgDecorationsExceptLineThrough(
       GetSvgStyleToPaint(state, SvgPaintMode::kTextDecoration,
                          selection_style_scope, has_fill, has_visible_stroke);
 
-  for (int i = 0; i < 3; i++) {
+  const PaintOrderArray paint_order(state.Style().PaintOrder());
+  for (unsigned i = 0; i < 3; i++) {
     absl::optional<LayoutSVGResourceMode> resource_mode;
 
-    switch (state.Style().PaintOrderType(i)) {
+    switch (paint_order[i]) {
       case PT_FILL:
         if (has_fill)
           resource_mode = kApplyToFillMode;
@@ -529,10 +532,11 @@ void NGTextPainter::PaintSvgDecorationsOnlyLineThrough(
       GetSvgStyleToPaint(state, SvgPaintMode::kTextDecoration,
                          selection_style_scope, has_fill, has_visible_stroke);
 
-  for (int i = 0; i < 3; i++) {
+  const PaintOrderArray paint_order(state.Style().PaintOrder());
+  for (unsigned i = 0; i < 3; i++) {
     absl::optional<LayoutSVGResourceMode> resource_mode;
 
-    switch (state.Style().PaintOrderType(i)) {
+    switch (paint_order[i]) {
       case PT_FILL:
         if (has_fill)
           resource_mode = kApplyToFillMode;
