diff --git a/third_party/blink/renderer/modules/geolocation/geoposition.h b/third_party/blink/renderer/modules/geolocation/geoposition.h
index 89773cbf3ba19528e74f8bd3e5887bf47f8fca62..66a721668a325902ee7c71673999dc3aabdf280e 100644
--- a/third_party/blink/renderer/modules/geolocation/geoposition.h
+++ b/third_party/blink/renderer/modules/geolocation/geoposition.h
@@ -26,6 +26,7 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_MODULES_GEOLOCATION_GEOPOSITION_H_
 #define THIRD_PARTY_BLINK_RENDERER_MODULES_GEOLOCATION_GEOPOSITION_H_
 
+#include "third_party/blink/renderer/bindings/core/v8/script_value.h"
 #include "third_party/blink/renderer/core/timing/epoch_time_stamp.h"
 #include "third_party/blink/renderer/modules/event_modules.h"
 #include "third_party/blink/renderer/modules/geolocation/geolocation_coordinates.h"
@@ -50,6 +51,8 @@ class Geoposition final : public ScriptWrappable {
 
   EpochTimeStamp timestamp() const { return timestamp_; }
   GeolocationCoordinates* coords() const { return coordinates_; }
+  // FlyWeb (Chromium 733c562491c, Chrome 126).
+  ScriptValue toJSON(ScriptState* script_state) const;
 
  private:
   Member<GeolocationCoordinates> coordinates_;
