diff --git a/third_party/blink/renderer/core/intersection_observer/intersection_observer.idl b/third_party/blink/renderer/core/intersection_observer/intersection_observer.idl
index c263fccb5561ca09a7440c741cd8c2009c68b00e..94ec456686f5ed76c70ad0f0a3a521a577509aba 100644
--- a/third_party/blink/renderer/core/intersection_observer/intersection_observer.idl
+++ b/third_party/blink/renderer/core/intersection_observer/intersection_observer.idl
@@ -13,6 +13,8 @@ callback IntersectionObserverCallback = void (sequence<IntersectionObserverEntry
     [CallWith=ScriptState, RaisesException, MeasureAs=IntersectionObserver_Constructor] constructor(IntersectionObserverCallback callback, optional IntersectionObserverInit options = {});
     readonly attribute Node? root;
     readonly attribute DOMString rootMargin;
+    [RuntimeEnabled=IntersectionObserverScrollMargin]
+    readonly attribute DOMString scrollMargin;
     // https://github.com/WICG/IntersectionObserver/issues/114
     readonly attribute FrozenArray<double> thresholds;
     readonly attribute DOMHighResTimeStamp delay;
