diff --git a/third_party/blink/renderer/modules/clipboard/clipboard_item.h b/third_party/blink/renderer/modules/clipboard/clipboard_item.h
index 9712a2587a45cd4a259af74b9ec1961d2fd8cb91..05c15ca58a87dabfd6bfccede1b010bd7053d1f3 100644
--- a/third_party/blink/renderer/modules/clipboard/clipboard_item.h
+++ b/third_party/blink/renderer/modules/clipboard/clipboard_item.h
@@ -27,6 +27,7 @@ class ClipboardItem final : public ScriptWrappable {
   ScriptPromise getType(ScriptState* script_state,
                         const String& type,
                         ExceptionState& exception_state) const;
+  static bool supports(const String& type);
 
   const HeapVector<std::pair<String, ScriptPromise>>& GetItems() const {
     return items_;
