diff --git a/third_party/blink/renderer/core/paint/box_painter_base.cc b/third_party/blink/renderer/core/paint/box_painter_base.cc
index d7f4e22df5554470489ff80fc7221cd756ff9be1..ecdf664bf69b0f073a5002a03b3260a2ed282ba8 100644
--- a/third_party/blink/renderer/core/paint/box_painter_base.cc
+++ b/third_party/blink/renderer/core/paint/box_painter_base.cc
@@ -5,6 +5,7 @@
 #include "third_party/blink/renderer/core/paint/box_painter_base.h"
 
 #include "base/containers/adapters.h"
+#include "cc/paint/color_filter.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
 #include "third_party/blink/renderer/core/animation/element_animations.h"
 #include "third_party/blink/renderer/core/css/background_color_paint_image_generator.h"
@@ -21,17 +22,20 @@
 #include "third_party/blink/renderer/core/paint/paint_layer.h"
 #include "third_party/blink/renderer/core/paint/rounded_border_geometry.h"
 #include "third_party/blink/renderer/core/paint/rounded_inner_rect_clipper.h"
+#include "third_party/blink/renderer/core/paint/svg_mask_painter.h"
 #include "third_party/blink/renderer/core/paint/timing/image_element_timing.h"
 #include "third_party/blink/renderer/core/paint/timing/paint_timing_detector.h"
 #include "third_party/blink/renderer/core/style/border_edge.h"
 #include "third_party/blink/renderer/core/style/computed_style.h"
 #include "third_party/blink/renderer/core/style/shadow_list.h"
 #include "third_party/blink/renderer/core/style/style_fetched_image.h"
+#include "third_party/blink/renderer/core/style/style_svg_mask_reference_image.h"
 #include "third_party/blink/renderer/platform/geometry/layout_rect.h"
 #include "third_party/blink/renderer/platform/graphics/bitmap_image.h"
 #include "third_party/blink/renderer/platform/graphics/graphics_context.h"
 #include "third_party/blink/renderer/platform/graphics/graphics_context_state_saver.h"
 #include "third_party/blink/renderer/platform/graphics/paint/paint_controller.h"
+#include "third_party/blink/renderer/platform/graphics/paint_generated_image.h"
 #include "third_party/blink/renderer/platform/graphics/scoped_interpolation_quality.h"
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "ui/gfx/geometry/rect_conversions.h"
@@ -425,7 +429,9 @@ BoxPainterBase::FillLayerInfo::FillLayerInfo(
       respect_image_orientation(respect_image_orientation),
       sides_to_include(sides_to_include),
       is_bottom_layer(!layer.Next()),
-      is_border_fill(layer.Clip() == EFillBox::kBorder),
+      is_border_fill(layer.Clip() == EFillBox::kStrokeBox ||
+                     layer.Clip() == EFillBox::kViewBox ||
+                     layer.Clip() == EFillBox::kBorder),
       is_clipped_with_local_scrolling(is_scroll_container &&
                                       layer.Attachment() ==
                                           EFillAttachment::kLocal) {
@@ -461,6 +467,7 @@ BoxPainterBase::FillLayerInfo::FillLayerInfo(
   // BackgroundBleedClip{Only,Layer}.
   is_rounded_fill =
       has_rounded_border && !is_painting_background_in_contents_space &&
+      (layer.Clip() != EFillBox::kNoClip) &&
       (is_clipped_with_local_scrolling ||
        !(is_border_fill && BleedAvoidanceIsClipping(bleed_avoidance)));
 
@@ -961,7 +968,8 @@ FloatRoundedRect RoundedBorderRectForClip(
   // Clip to the padding or content boxes as necessary.
   // Use FastAndLossyFromRectF because we know it has been pixel snapped.
   PhysicalRect border_rect = PhysicalRect::FastAndLossyFromRectF(border.Rect());
-  if (bg_layer.Clip() == EFillBox::kContent) {
+  if (bg_layer.Clip() == EFillBox::kFillBox ||
+      bg_layer.Clip() == EFillBox::kContent) {
     border = RoundedBorderGeometry::PixelSnappedRoundedBorderWithOutsets(
         style, border_rect, border_padding_insets, info.sides_to_include);
     // Background of 'background-attachment: local' without visible/clip
@@ -1039,6 +1047,73 @@ bool ShouldApplyBlendOperation(const BoxPainterBase::FillLayerInfo& info,
   return !info.is_bottom_layer || layer.GetType() != EFillLayerType::kMask;
 }
 
+bool NeedsMaskLuminanceLayer(const FillLayer& layer, StyleImage* image) {
+  if (layer.GetType() != EFillLayerType::kMask) {
+    return false;
+  }
+  if (layer.MaskMode() == EFillMaskMode::kLuminance) {
+    return true;
+  }
+  if (layer.MaskMode() == EFillMaskMode::kMatchSource) {
+    if (image->IsSVGMaskReference()) {
+      const auto& svg_reference = To<StyleSVGMaskReferenceImage>(*image);
+      EMaskType type = SVGMaskPainter::MaskType(
+          svg_reference.GetSVGResource(), svg_reference.GetSVGResourceClient());
+      return type == EMaskType::kLuminance;
+    }
+  }
+  return false;
+}
+
+class ScopedMaskLuminanceLayer {
+  STACK_ALLOCATED();
+
+ public:
+  ScopedMaskLuminanceLayer(GraphicsContext& context, SkBlendMode composite_op)
+      : context_(context) {
+    context.BeginLayer(cc::ColorFilter::MakeLuma(), &composite_op);
+  }
+  ~ScopedMaskLuminanceLayer() { context_.EndLayer(); }
+
+ private:
+  GraphicsContext& context_;
+};
+
+scoped_refptr<Image> UpdateGeometryAndGetImageForLayer(
+    const PaintInfo& paint_info,
+    const ComputedStyle& style,
+    const FillLayer& layer,
+    StyleImage* image,
+    const PhysicalRect& paint_rect,
+    BackgroundImageGeometry& geometry) {
+  const ComputedStyle& image_style = geometry.ImageStyle(style);
+
+  // If the "image" referenced by the FillLayer is an SVG <mask> reference (and
+  // this is a layer for a mask), then repeat, position, clip, origin and size
+  // should have no effect.
+  if (layer.GetType() == EFillLayerType::kMask && image->IsSVGMaskReference()) {
+    const auto& svg_reference = To<StyleSVGMaskReferenceImage>(*image);
+
+    const PhysicalRect positioning_area =
+        geometry.ComputePositioningArea(paint_info, layer, paint_rect);
+    const gfx::RectF reference_box(gfx::SizeF(positioning_area.size));
+    const float zoom = image_style.EffectiveZoom();
+
+    const gfx::RectF mask_area = SVGMaskPainter::ResourceBounds(
+        svg_reference.GetSVGResource(), svg_reference.GetSVGResourceClient(),
+        reference_box, zoom);
+    geometry.SetGeometryForSVGMask(mask_area, positioning_area.offset);
+
+    PaintRecord record = SVGMaskPainter::PaintResource(
+        svg_reference.GetSVGResource(), svg_reference.GetSVGResourceClient(),
+        reference_box, zoom);
+    return PaintGeneratedImage::Create(std::move(record), mask_area.size());
+  }
+  geometry.Calculate(paint_info, layer, paint_rect);
+  return image->GetImage(geometry.ImageClient(), geometry.ImageDocument(),
+                         image_style, gfx::SizeF(geometry.TileSize()));
+}
+
 }  // anonymous namespace
 
 NGPhysicalBoxStrut BoxPainterBase::ComputeSnappedBorders() const {
@@ -1082,11 +1157,11 @@ void BoxPainterBase::PaintFillLayer(const PaintInfo& paint_info,
   scoped_refptr<Image> image;
   SkBlendMode composite_op = SkBlendMode::kSrcOver;
   absl::optional<ScopedInterpolationQuality> interpolation_quality_context;
+  absl::optional<ScopedMaskLuminanceLayer> mask_luminance_scope;
   if (fill_layer_info.should_paint_image) {
-    geometry.Calculate(paint_info, bg_layer, scrolled_paint_rect);
-    image = fill_layer_info.image->GetImage(
-        geometry.ImageClient(), geometry.ImageDocument(),
-        geometry.ImageStyle(style_), gfx::SizeF(geometry.TileSize()));
+    image = UpdateGeometryAndGetImageForLayer(paint_info, style_, bg_layer,
+                                              fill_layer_info.image,
+                                              scrolled_paint_rect, geometry);
     interpolation_quality_context.emplace(context,
                                           geometry.ImageInterpolationQuality());
 
@@ -1094,6 +1169,13 @@ void BoxPainterBase::PaintFillLayer(const PaintInfo& paint_info,
       composite_op = WebCoreCompositeToSkiaComposite(bg_layer.Composite(),
                                                      bg_layer.GetBlendMode());
     }
+
+    if (NeedsMaskLuminanceLayer(bg_layer, fill_layer_info.image)) {
+      mask_luminance_scope.emplace(context, composite_op);
+      // The mask luminance layer will apply `composite_op`, so reset it to
+      // avoid applying it twice.
+      composite_op = SkBlendMode::kSrcOver;
+    }
   }
 
   NGPhysicalBoxStrut border = ComputeSnappedBorders();
@@ -1138,6 +1220,10 @@ void BoxPainterBase::PaintFillLayer(const PaintInfo& paint_info,
   absl::optional<GraphicsContextStateSaver> background_clip_state_saver;
   if (!geometry.CanCompositeBackgroundAttachmentFixed()) {
     switch (bg_layer.Clip()) {
+      case EFillBox::kFillBox:
+      // Spec: For elements with associated CSS layout box, the used values for
+      // fill-box compute to content-box.
+      // https://drafts.fxtf.org/css-masking/#the-mask-clip
       case EFillBox::kPadding:
       case EFillBox::kContent: {
         if (fill_layer_info.is_rounded_fill) {
@@ -1148,7 +1234,8 @@ void BoxPainterBase::PaintFillLayer(const PaintInfo& paint_info,
         PhysicalRect clip_rect = scrolled_paint_rect;
         clip_rect.Contract(
             AdjustOutsetsForEdgeInclusion(border, fill_layer_info));
-        if (bg_layer.Clip() == EFillBox::kContent) {
+        if (bg_layer.Clip() == EFillBox::kFillBox ||
+            bg_layer.Clip() == EFillBox::kContent) {
           clip_rect.Contract(
               AdjustOutsetsForEdgeInclusion(padding, fill_layer_info));
         }
@@ -1156,6 +1243,12 @@ void BoxPainterBase::PaintFillLayer(const PaintInfo& paint_info,
         context.Clip(ToPixelSnappedRect(clip_rect));
         break;
       }
+      case EFillBox::kStrokeBox:
+      case EFillBox::kViewBox:
+      // Spec: For elements with associated CSS layout box, ... stroke-box and
+      // view-box compute to border-box.
+      // https://drafts.fxtf.org/css-masking/#the-mask-clip
+      case EFillBox::kNoClip:
       case EFillBox::kBorder:
         break;
       case EFillBox::kText:  // fall through
