diff --git a/third_party/blink/renderer/core/css/rule_feature_set.cc b/third_party/blink/renderer/core/css/rule_feature_set.cc
index c320f4a9da5ff5958f0f0a90d6e038d2862f33dc..3caf4028f7883439042c514ab184daca86d5a16c 100644
--- a/third_party/blink/renderer/core/css/rule_feature_set.cc
+++ b/third_party/blink/renderer/core/css/rule_feature_set.cc
@@ -89,6 +89,7 @@ bool SupportsInvalidation(CSSSelector::PseudoType type) {
     case CSSSelector::kPseudoNthLastOfType:
     case CSSSelector::kPseudoPart:
     case CSSSelector::kPseudoState:
+    case CSSSelector::kPseudoStateDeprecatedSyntax:
     case CSSSelector::kPseudoLink:
     case CSSSelector::kPseudoVisited:
     case CSSSelector::kPseudoAny:
@@ -114,6 +115,8 @@ bool SupportsInvalidation(CSSSelector::PseudoType type) {
     case CSSSelector::kPseudoRequired:
     case CSSSelector::kPseudoReadOnly:
     case CSSSelector::kPseudoReadWrite:
+    case CSSSelector::kPseudoUserInvalid:
+    case CSSSelector::kPseudoUserValid:
     case CSSSelector::kPseudoValid:
     case CSSSelector::kPseudoInvalid:
     case CSSSelector::kPseudoIndeterminate:
@@ -196,6 +199,8 @@ bool SupportsInvalidation(CSSSelector::PseudoType type) {
     case CSSSelector::kPseudoViewTransitionNew:
     case CSSSelector::kPseudoViewTransitionOld:
     case CSSSelector::kPseudoToggle:
+    case CSSSelector::kPseudoActiveViewTransition:
+    case CSSSelector::kPseudoActiveViewTransitionType:
       return true;
     case CSSSelector::kPseudoUnknown:
     case CSSSelector::kPseudoLeftPage:
@@ -714,6 +719,9 @@ InvalidationSet* RuleFeatureSet::InvalidationSetForSimpleSelector(
       case CSSSelector::kPseudoReadOnly:
       case CSSSelector::kPseudoReadWrite:
       case CSSSelector::kPseudoState:
+      case CSSSelector::kPseudoStateDeprecatedSyntax:
+      case CSSSelector::kPseudoUserInvalid:
+      case CSSSelector::kPseudoUserValid:
       case CSSSelector::kPseudoValid:
       case CSSSelector::kPseudoInvalid:
       case CSSSelector::kPseudoIndeterminate:
@@ -741,6 +749,8 @@ InvalidationSet* RuleFeatureSet::InvalidationSetForSimpleSelector(
       case CSSSelector::kPseudoModal:
       case CSSSelector::kPseudoSelectorFragmentAnchor:
       case CSSSelector::kPseudoToggle:
+      case CSSSelector::kPseudoActiveViewTransition:
+      case CSSSelector::kPseudoActiveViewTransitionType:
         return &EnsurePseudoInvalidationSet(selector.GetPseudoType(), type,
                                             position);
       case CSSSelector::kPseudoFirstOfType:
