diff --git a/third_party/blink/renderer/core/css/style_property_serializer.h b/third_party/blink/renderer/core/css/style_property_serializer.h
index 2f47dec4ae8e92ac0ce64213dff9629b65e57c23..d349627bccac7ed704bce10299c34e5753116b51 100644
--- a/third_party/blink/renderer/core/css/style_property_serializer.h
+++ b/third_party/blink/renderer/core/css/style_property_serializer.h
@@ -81,7 +81,6 @@ class CORE_EXPORT StylePropertySerializer {
                                           StringBuilder& result) const;
   String OffsetValue() const;
   String TextDecorationValue() const;
-  String BackgroundRepeatPropertyValue() const;
   String ContainIntrinsicSizeValue() const;
   String WhiteSpaceValue() const;
   String ScrollStartValue() const;
