diff --git a/third_party/blink/renderer/core/html/parser/html_construction_site.cc b/third_party/blink/renderer/core/html/parser/html_construction_site.cc
index fe92f7fc26dfab633625b8403103c855f35877af..185012b06f027813c1b426c4dc1e4af0eb7a052b 100644
--- a/third_party/blink/renderer/core/html/parser/html_construction_site.cc
+++ b/third_party/blink/renderer/core/html/parser/html_construction_site.cc
@@ -822,10 +822,11 @@ void HTMLConstructionSite::InsertHTMLTemplateElement(
   HTMLStackItem* template_stack_item =
       HTMLStackItem::Create(template_element, token);
   bool should_attach_template = true;
-  if (declarative_shadow_root_type ==
-          DeclarativeShadowRootType::kStreamingOpen ||
-      declarative_shadow_root_type ==
-          DeclarativeShadowRootType::kStreamingClosed) {
+  if ((declarative_shadow_root_type ==
+           DeclarativeShadowRootType::kStreamingOpen ||
+       declarative_shadow_root_type ==
+           DeclarativeShadowRootType::kStreamingClosed) &&
+      IsA<Element>(open_elements_.TopStackItem()->GetNode())) {
     DCHECK(RuntimeEnabledFeatures::StreamingDeclarativeShadowDOMEnabled());
     // Attach the shadow root now
     auto focus_delegation = template_stack_item->GetAttributeItem(
