diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_block.h b/third_party/blink/renderer/core/layout/svg/layout_svg_block.h
index 381d11769f0181ce6a0c30dbf9f58386581bcf90..4326eb47718e8fcefa0121ef7d82a4b5c662a731 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_block.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_block.h
@@ -82,6 +82,7 @@ class LayoutSVGBlock : public LayoutBlockFlow {
   }
 
   bool CheckForImplicitTransformChange(bool bbox_changed) const;
+  void UpdateTransformBeforeLayout();
   bool UpdateTransformAfterLayout(bool bounds_changed);
   void StyleDidChange(StyleDifference, const ComputedStyle* old_style) override;
   void UpdateFromStyle() override;
