diff --git a/third_party/blink/renderer/bindings/scripts/generate_bindings.py b/third_party/blink/renderer/bindings/scripts/generate_bindings.py
index 18b2b21c56e8f3f70ef5308024c0f4a41404417c..1f8552c712fe51f8ad3fae75f686408cc82a5359 100644
--- a/third_party/blink/renderer/bindings/scripts/generate_bindings.py
+++ b/third_party/blink/renderer/bindings/scripts/generate_bindings.py
@@ -75,6 +75,7 @@ def parse_options(valid_tasks):
 
 def main():
     dispatch_table = {
+        'async_iterator': bind_gen.generate_async_iterators,
         'callback_function': bind_gen.generate_callback_functions,
         'callback_interface': bind_gen.generate_callback_interfaces,
         'dictionary': bind_gen.generate_dictionaries,
