diff --git a/third_party/blink/renderer/core/paint/box_model_object_painter.h b/third_party/blink/renderer/core/paint/box_model_object_painter.h
index b4c6d35002399c3525a1d2fff884a42f1e98b483..3d2945e55babc6e4678426b23733c91b2b51bcf0 100644
--- a/third_party/blink/renderer/core/paint/box_model_object_painter.h
+++ b/third_party/blink/renderer/core/paint/box_model_object_painter.h
@@ -28,6 +28,7 @@ class BoxModelObjectPainter : public BoxPainterBase {
  protected:
   NGPhysicalBoxStrut ComputeBorders() const override;
   NGPhysicalBoxStrut ComputePadding() const override;
+  NGPhysicalBoxStrut ComputeMargins() const override;
   BoxPainterBase::FillLayerInfo GetFillLayerInfo(
       const Color&,
       const FillLayer&,
