diff --git a/third_party/blink/renderer/core/html/forms/date_time_edit_element.cc b/third_party/blink/renderer/core/html/forms/date_time_edit_element.cc
index 15442d8e1e975d4249d055201a0df86fe1c87b15..4e301924989085711099fd10ca2de0719d636302 100644
--- a/third_party/blink/renderer/core/html/forms/date_time_edit_element.cc
+++ b/third_party/blink/renderer/core/html/forms/date_time_edit_element.cc
@@ -604,7 +604,11 @@ DateTimeEditElement::CustomStyleForLayoutObject(
     }
   }
   ComputedStyleBuilder builder(*original_style);
-  builder.SetWidth(Length::Fixed(ceilf(width)));
+  if (original_style->IsHorizontalWritingMode()) {
+    builder.SetWidth(Length::Fixed(ceilf(width)));
+  } else {
+    builder.SetHeight(Length::Fixed(ceilf(width)));
+  }
   builder.SetCustomStyleCallbackDependsOnFont();
   return builder.TakeStyle();
 }
