diff --git a/third_party/blink/renderer/core/script/modulator_impl_base.h b/third_party/blink/renderer/core/script/modulator_impl_base.h
index fafd98fdd279c8e0a2d8ce60b807aa3bb0536fc3..4534ff070d692f90554df0ea535152b59c71d22f 100644
--- a/third_party/blink/renderer/core/script/modulator_impl_base.h
+++ b/third_party/blink/renderer/core/script/modulator_impl_base.h
@@ -84,6 +84,9 @@ class ModulatorImplBase : public Modulator {
 
   ModuleImportMeta HostGetImportMetaProperties(
       v8::Local<v8::Module>) const override;
+
+  String GetIntegrityMetadataString(const KURL&) const override;
+  IntegrityMetadataSet GetIntegrityMetadata(const KURL&) const override;
   ModuleType ModuleTypeFromRequest(
       const ModuleRequest& module_request) const override;
 
