diff --git a/third_party/blink/renderer/core/dom/shadow_root_init.idl b/third_party/blink/renderer/core/dom/shadow_root_init.idl
index b48d0de6ca1f5f747e993466d20a5a75be49c2f6..9bf37f48a040e460445fed7de263721d09dc6d7b 100644
--- a/third_party/blink/renderer/core/dom/shadow_root_init.idl
+++ b/third_party/blink/renderer/core/dom/shadow_root_init.idl
@@ -12,4 +12,6 @@ dictionary ShadowRootInit {
     boolean delegatesFocus;
     SlotAssignmentMode slotAssignment;
     [RuntimeEnabled=ScopedCustomElementRegistry] CustomElementRegistry registry;
+    [RuntimeEnabled=ShadowRootClonable] boolean clonable;
+    [RuntimeEnabled=ElementGetHTML] boolean serializable;
 };
