diff --git a/third_party/blink/renderer/bindings/scripts/web_idl/user_defined_type.py b/third_party/blink/renderer/bindings/scripts/web_idl/user_defined_type.py
index a49d818d7eaa8ef05daf9150b4a05ad64075dae0..27a9832591564cb7cd3d98858eec7502f613ca28 100644
--- a/third_party/blink/renderer/bindings/scripts/web_idl/user_defined_type.py
+++ b/third_party/blink/renderer/bindings/scripts/web_idl/user_defined_type.py
@@ -51,6 +51,11 @@ class UserDefinedType(WithIdentifier):
         """Returns True if this is an IDL namespace."""
         return False
 
+    @property
+    def is_async_iterator(self):
+        """Returns True if this is an async iterator."""
+        return False
+
     @property
     def is_sync_iterator(self):
         """Returns True if this is a sync iterator."""
