diff --git a/third_party/blink/renderer/core/css/rule_feature_set.cc b/third_party/blink/renderer/core/css/rule_feature_set.cc
index c320f4a9da5ff5958f0f0a90d6e038d2862f33dc..53dd0bc088a0182921a2728784eee917cdfd7096 100644
--- a/third_party/blink/renderer/core/css/rule_feature_set.cc
+++ b/third_party/blink/renderer/core/css/rule_feature_set.cc
@@ -114,6 +114,8 @@ bool SupportsInvalidation(CSSSelector::PseudoType type) {
     case CSSSelector::kPseudoRequired:
     case CSSSelector::kPseudoReadOnly:
     case CSSSelector::kPseudoReadWrite:
+    case CSSSelector::kPseudoUserInvalid:
+    case CSSSelector::kPseudoUserValid:
     case CSSSelector::kPseudoValid:
     case CSSSelector::kPseudoInvalid:
     case CSSSelector::kPseudoIndeterminate:
@@ -714,6 +716,8 @@ InvalidationSet* RuleFeatureSet::InvalidationSetForSimpleSelector(
       case CSSSelector::kPseudoReadOnly:
       case CSSSelector::kPseudoReadWrite:
       case CSSSelector::kPseudoState:
+      case CSSSelector::kPseudoUserInvalid:
+      case CSSSelector::kPseudoUserValid:
       case CSSSelector::kPseudoValid:
       case CSSSelector::kPseudoInvalid:
       case CSSSelector::kPseudoIndeterminate:
