diff --git a/third_party/blink/renderer/core/paint/box_model_object_painter.cc b/third_party/blink/renderer/core/paint/box_model_object_painter.cc
index 1c49cb782046e2d367800eb2439841032edd6424..e078cc56630ae30d77455c1d69ed3be356a27fd7 100644
--- a/third_party/blink/renderer/core/paint/box_model_object_painter.cc
+++ b/third_party/blink/renderer/core/paint/box_model_object_painter.cc
@@ -68,6 +68,10 @@ NGPhysicalBoxStrut BoxModelObjectPainter::ComputePadding() const {
   return box_model_.PaddingOutsets();
 }
 
+NGPhysicalBoxStrut BoxModelObjectPainter::ComputeMargins() const {
+  return box_model_.MarginOutsets();
+}
+
 BoxPainterBase::FillLayerInfo BoxModelObjectPainter::GetFillLayerInfo(
     const Color& color,
     const FillLayer& bg_layer,
