diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_shape.h b/third_party/blink/renderer/core/layout/svg/layout_svg_shape.h
index b2b70cde801f5497fff49004e69511cc971c51b9..5871c752a50e5f63df7786ebecdc8a53cb02333b 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_shape.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_shape.h
@@ -171,7 +171,7 @@ class LayoutSVGShape : public LayoutSVGModelObject {
                                           const WindRule) const;
 
   gfx::RectF fill_bounding_box_;
-  gfx::RectF stroke_bounding_box_;
+  gfx::RectF decorated_bounding_box_;
 
   LayoutSVGShapeRareData& EnsureRareData() const;
 
@@ -198,9 +198,13 @@ class LayoutSVGShape : public LayoutSVGModelObject {
                     const HitTestLocation&,
                     PointerEventsHitRules);
 
-  gfx::RectF StrokeBoundingBox() const final {
+  // Calculates the tight stroke bounding box of the shape excluding any dash
+  // pattern.
+  gfx::RectF StrokeBoundingBox() const final;
+
+  gfx::RectF DecoratedBoundingBox() const final {
     NOT_DESTROYED();
-    return stroke_bounding_box_;
+    return decorated_bounding_box_;
   }
 
   // Calculates an inclusive bounding box of this shape as if this shape has a
