diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index e77f3afb2cc9e94b35d7456daa7f7e7554d56bd8..6f53ed1f429730856f6949dd6b299609104fc55a 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -815,7 +815,8 @@
     },
     {
       name: "CSSContainIntrinsicSizeAutoNone",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     // Ablation study to try to measure very roughly memory savings from not saving
     // pretokenized strings in CSS custom properties.
@@ -909,7 +910,8 @@
     },
     {
       name: "CSSInitialPseudo",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -1152,7 +1154,8 @@
     // `text-wrap: pretty`. crbug.com/1432798
     {
       name: "CSSTextWrapPretty",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       depends_on: ["CSSTextWrap", "CSSWhiteSpaceShorthand"],
     },
     // Support for CSS Toggles, https://tabatkins.github.io/css-toggle/
@@ -1170,12 +1173,14 @@
       //
       // See: https://github.com/w3c/csswg-drafts/issues/8189
       name: "CSSTopLayerForTransitions",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
       name: "CSSTransitionDiscrete",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     {
       // A kill switch for the fix for part of https://crbug.com/1180387 .
@@ -2137,7 +2142,8 @@
     },
     {
       name: "LayoutNGSubgrid",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
