diff --git a/third_party/blink/renderer/core/css/resolver/font_builder.cc b/third_party/blink/renderer/core/css/resolver/font_builder.cc
index bddbb984182f09ebe0d841748064ff68154a2775..feb75e52edd4075ad09487d35b3bf9e1c2c383f3 100644
--- a/third_party/blink/renderer/core/css/resolver/font_builder.cc
+++ b/third_party/blink/renderer/core/css/resolver/font_builder.cc
@@ -375,6 +375,17 @@ void FontBuilder::UpdateAdjustedSize(FontDescription& font_description,
     return;
   }
 
+  FontSizeAdjust size_adjust = font_description.SizeAdjust();
+  if (size_adjust.IsFromFont() &&
+      size_adjust.Value() == FontSizeAdjust::kFontSizeAdjustNone) {
+    absl::optional<float> aspect_value = FontSizeFunctions::FontAspectValue(
+        font_data, size_adjust.GetMetric(), font_description.ComputedSize());
+    font_description.SetSizeAdjust(FontSizeAdjust(
+        aspect_value.has_value() ? aspect_value.value()
+                                 : FontSizeAdjust::kFontSizeAdjustNone,
+        size_adjust.GetMetric(), FontSizeAdjust::ValueType::kFromFont));
+  }
+
   if (auto adjusted_size = FontSizeFunctions::MetricsMultiplierAdjustedFontSize(
           font_data, font_description)) {
     font_description.SetAdjustedSize(adjusted_size.value());
