diff --git a/third_party/blink/renderer/core/style/build.gni b/third_party/blink/renderer/core/style/build.gni
index 31b5deae1d34f57132dcd66ec098d1b145ec6246..bcd1f4bf1e5e892299a001658ed628e73cbcc122 100644
--- a/third_party/blink/renderer/core/style/build.gni
+++ b/third_party/blink/renderer/core/style/build.gni
@@ -51,6 +51,7 @@ blink_core_sources_style = [
   "nine_piece_image.h",
   "ordered_named_grid_lines.h",
   "paint_images.h",
+  "paint_order_array.h",
   "quad_length_value.h",
   "reference_clip_path_operation.cc",
   "reference_clip_path_operation.h",
