diff --git a/third_party/blink/renderer/core/css/style_property_serializer.cc b/third_party/blink/renderer/core/css/style_property_serializer.cc
index f32faf3a9bbfdeda831980efd36919c571227e0f..3df98577319a781e5274ce901b484fae49d43fc3 100644
--- a/third_party/blink/renderer/core/css/style_property_serializer.cc
+++ b/third_party/blink/renderer/core/css/style_property_serializer.cc
@@ -399,6 +399,7 @@ static bool AllowInitialInShorthand(CSSPropertyID property_id) {
     case CSSPropertyID::kTextDecoration:
     case CSSPropertyID::kTextEmphasis:
     case CSSPropertyID::kWebkitMask:
+    case CSSPropertyID::kAlternativeMask:
     case CSSPropertyID::kWebkitTextStroke:
     case CSSPropertyID::kAlternativeWhiteSpace:
       return true;
@@ -504,8 +505,6 @@ String StylePropertySerializer::SerializeShorthand(
       return Get2Values(borderSpacingShorthand());
     case CSSPropertyID::kBackgroundPosition:
       return GetLayeredShorthandValue(backgroundPositionShorthand());
-    case CSSPropertyID::kBackgroundRepeat:
-      return BackgroundRepeatPropertyValue();
     case CSSPropertyID::kBackground:
       return GetLayeredShorthandValue(backgroundShorthand());
     case CSSPropertyID::kBorder:
@@ -623,12 +622,14 @@ String StylePropertySerializer::SerializeShorthand(
       return GetLayeredShorthandValue(transitionShorthand());
     case CSSPropertyID::kListStyle:
       return GetShorthandValue(listStyleShorthand());
+    case CSSPropertyID::kMaskPosition:
+      return GetLayeredShorthandValue(maskPositionShorthand());
     case CSSPropertyID::kWebkitMaskPosition:
       return GetLayeredShorthandValue(webkitMaskPositionShorthand());
-    case CSSPropertyID::kWebkitMaskRepeat:
-      return GetLayeredShorthandValue(webkitMaskRepeatShorthand());
     case CSSPropertyID::kWebkitMask:
       return GetLayeredShorthandValue(webkitMaskShorthand());
+    case CSSPropertyID::kAlternativeMask:
+      return GetLayeredShorthandValue(alternativeMaskShorthand());
     case CSSPropertyID::kTextEmphasis:
       return GetShorthandValue(textEmphasisShorthand());
     case CSSPropertyID::kWebkitTextStroke:
@@ -1506,6 +1507,48 @@ String StylePropertySerializer::Get4Values(
   return result.ReleaseString();
 }
 
+namespace {
+
+// Serialize clip and origin (https://drafts.fxtf.org/css-masking/#the-mask):
+// * If one <geometry-box> value and the no-clip keyword are present then
+//   <geometry-box> sets mask-origin and no-clip sets mask-clip to that value.
+// * If one <geometry-box> value and no no-clip keyword are present then
+//   <geometry-box> sets both mask-origin and mask-clip to that value.
+// * If two <geometry-box> values are present, then the first sets mask-origin
+//   and the second mask-clip.
+// Additionally, omits components when possible (see:
+// https://drafts.csswg.org/cssom/#serialize-a-css-value).
+void SerializeMaskOriginAndClip(StringBuilder& result,
+                                const CSSValueID& origin_id,
+                                const CSSValueID& clip_id) {
+  // If both values are border-box, omit everything as it is the default.
+  if (origin_id == CSSValueID::kBorderBox &&
+      clip_id == CSSValueID::kBorderBox) {
+    return;
+  }
+
+  if (!result.empty()) {
+    result.Append(' ');
+  }
+  if (origin_id == clip_id) {
+    // If the values are the same, only emit one value. Note that mask-origin
+    // does not support no-clip, so there is no need to consider no-clip
+    // special cases.
+    result.Append(getValueName(origin_id));
+  } else if (origin_id == CSSValueID::kBorderBox &&
+             clip_id == CSSValueID::kNoClip) {
+    // Mask-origin does not support no-clip, so mask-origin can be omitted if it
+    // is the default.
+    result.Append(getValueName(clip_id));
+  } else {
+    result.Append(getValueName(origin_id));
+    result.Append(' ');
+    result.Append(getValueName(clip_id));
+  }
+}
+
+}  // namespace
+
 String StylePropertySerializer::GetLayeredShorthandValue(
     const StylePropertyShorthand& shorthand) const {
   const unsigned size = shorthand.length();
@@ -1530,11 +1573,9 @@ String StylePropertySerializer::GetLayeredShorthandValue(
   // Now stitch the properties together.
   for (wtf_size_t layer = 0; layer < num_layers; layer++) {
     StringBuilder layer_result;
-    bool use_repeat_x_shorthand = false;
-    bool use_repeat_y_shorthand = false;
-    bool use_single_word_shorthand = false;
-    bool found_position_xcss_property = false;
-    bool found_position_ycss_property = false;
+    bool found_position_x_css_property = false;
+    bool found_position_y_css_property = false;
+    CSSValueID mask_origin_value = CSSValueID::kBorderBox;
 
     for (unsigned property_index = 0; property_index < size; property_index++) {
       const CSSValue* value = nullptr;
@@ -1561,45 +1602,7 @@ String StylePropertySerializer::GetLayeredShorthandValue(
         continue;
       }
 
-      // Special case for background-repeat.
-      if (property->IDEquals(CSSPropertyID::kBackgroundRepeatX) ||
-          property->IDEquals(CSSPropertyID::kWebkitMaskRepeatX)) {
-        DCHECK(shorthand.properties()[property_index + 1]->IDEquals(
-                   CSSPropertyID::kBackgroundRepeatY) ||
-               shorthand.properties()[property_index + 1]->IDEquals(
-                   CSSPropertyID::kWebkitMaskRepeatY));
-        auto* value_list =
-            DynamicTo<CSSValueList>(values[property_index + 1].Get());
-        const CSSValue& y_value =
-            value_list ? value_list->Item(layer) : *values[property_index + 1];
-
-        // FIXME: At some point we need to fix this code to avoid returning an
-        // invalid shorthand, since some longhand combinations are not
-        // serializable into a single shorthand.
-        if (!IsA<CSSIdentifierValue>(value) ||
-            !IsA<CSSIdentifierValue>(y_value)) {
-          continue;
-        }
-
-        CSSValueID x_id = To<CSSIdentifierValue>(value)->GetValueID();
-        CSSValueID y_id = To<CSSIdentifierValue>(y_value).GetValueID();
-        // Maybe advance propertyIndex to look at the next CSSValue in the list
-        // for the checks below.
-        if (x_id == y_id) {
-          use_single_word_shorthand = true;
-          property = shorthand.properties()[++property_index];
-        } else if (x_id == CSSValueID::kRepeat &&
-                   y_id == CSSValueID::kNoRepeat) {
-          use_repeat_x_shorthand = true;
-          property = shorthand.properties()[++property_index];
-        } else if (x_id == CSSValueID::kNoRepeat &&
-                   y_id == CSSValueID::kRepeat) {
-          use_repeat_y_shorthand = true;
-          property = shorthand.properties()[++property_index];
-        }
-      }
-
-      bool is_initial_value = value->IsInitialValue();
+      bool omit_value = value->IsInitialValue();
 
       // The shorthand can not represent the following properties if they have
       // non-initial values. This is because they are always reset to their
@@ -1616,7 +1619,7 @@ String StylePropertySerializer::GetLayeredShorthandValue(
           DCHECK(RuntimeEnabledFeatures::ScrollTimelineEnabled());
           return g_empty_string;
         }
-        is_initial_value = true;
+        omit_value = true;
       }
       if (property->IDEquals(CSSPropertyID::kAnimationDelayEnd)) {
         if (CSSToStyleMap::MapAnimationDelayEnd(*value) !=
@@ -1624,7 +1627,7 @@ String StylePropertySerializer::GetLayeredShorthandValue(
             layer > 0) {
           return g_empty_string;
         }
-        is_initial_value = true;
+        omit_value = true;
       }
       if (property->IDEquals(CSSPropertyID::kAnimationRangeStart)) {
         auto* ident = DynamicTo<CSSIdentifierValue>(value);
@@ -1633,7 +1636,7 @@ String StylePropertySerializer::GetLayeredShorthandValue(
           DCHECK(RuntimeEnabledFeatures::ScrollTimelineEnabled());
           return g_empty_string;
         }
-        is_initial_value = true;
+        omit_value = true;
       }
       if (property->IDEquals(CSSPropertyID::kAnimationRangeEnd)) {
         auto* ident = DynamicTo<CSSIdentifierValue>(value);
@@ -1642,13 +1645,60 @@ String StylePropertySerializer::GetLayeredShorthandValue(
           DCHECK(RuntimeEnabledFeatures::ScrollTimelineEnabled());
           return g_empty_string;
         }
-        is_initial_value = true;
+        omit_value = true;
+      }
+      if (property->IDEquals(CSSPropertyID::kTransitionBehavior)) {
+        auto* ident = DynamicTo<CSSIdentifierValue>(value);
+        CHECK(ident) << " transition-behavior should only have a "
+                        "CSSIdentifierValue for a value. CssText: "
+                     << value->CssText();
+        if (ident->GetValueID() == CSSValueID::kNormal) {
+          // transition-behavior overrides InitialValue to return "normal"
+          // instead of "initial", but we don't want to include "normal" in the
+          // shorthand serialization, so this special case is needed.
+          // TODO(http://crbug.com/501673): We should have a better solution
+          // before fixing all CSS properties to fix the above bug.
+          omit_value = true;
+        }
+      }
+      if (shorthand.id() == CSSPropertyID::kAlternativeMask) {
+        if (property->IDEquals(CSSPropertyID::kMaskOrigin)) {
+          if (auto* ident = DynamicTo<CSSIdentifierValue>(value)) {
+            mask_origin_value = ident->GetValueID();
+          }
+          // Omit this value as it is serialized alongside mask-clip.
+          omit_value = true;
+        } else if (property->IDEquals(CSSPropertyID::kMaskClip)) {
+          CSSValueID mask_clip_id = CSSValueID::kBorderBox;
+          if (auto* ident = DynamicTo<CSSIdentifierValue>(value)) {
+            mask_clip_id = To<CSSIdentifierValue>(ident)->GetValueID();
+          }
+          SerializeMaskOriginAndClip(layer_result, mask_origin_value,
+                                     mask_clip_id);
+          omit_value = true;
+        } else if (property->IDEquals(CSSPropertyID::kMaskComposite)) {
+          if (auto* ident = DynamicTo<CSSIdentifierValue>(value)) {
+            if (To<CSSIdentifierValue>(ident)->GetValueID() ==
+                CSSValueID::kAdd) {
+              omit_value = true;
+            }
+          }
+        } else if (property->IDEquals(CSSPropertyID::kMaskMode)) {
+          if (auto* ident = DynamicTo<CSSIdentifierValue>(value)) {
+            if (To<CSSIdentifierValue>(ident)->GetValueID() ==
+                CSSValueID::kMatchSource) {
+              omit_value = true;
+            }
+          }
+        }
+        // TODO(pdr): Omit default values for mask properties.
       }
 
-      if (!is_initial_value) {
+      if (!omit_value) {
         if (property->IDEquals(CSSPropertyID::kBackgroundSize) ||
-            property->IDEquals(CSSPropertyID::kWebkitMaskSize)) {
-          if (found_position_ycss_property || found_position_xcss_property) {
+            property->IDEquals(CSSPropertyID::kWebkitMaskSize) ||
+            property->IDEquals(CSSPropertyID::kMaskSize)) {
+          if (found_position_y_css_property || found_position_x_css_property) {
             layer_result.Append(" / ");
           } else {
             layer_result.Append(" 0% 0% / ");
@@ -1659,30 +1709,24 @@ String StylePropertySerializer::GetLayeredShorthandValue(
           layer_result.Append(' ');
         }
 
-        if (use_repeat_x_shorthand) {
-          use_repeat_x_shorthand = false;
-          layer_result.Append(getValueName(CSSValueID::kRepeatX));
-        } else if (use_repeat_y_shorthand) {
-          use_repeat_y_shorthand = false;
-          layer_result.Append(getValueName(CSSValueID::kRepeatY));
-        } else {
-          if (use_single_word_shorthand) {
-            use_single_word_shorthand = false;
-          }
-          layer_result.Append(value->CssText());
-        }
+        layer_result.Append(value->CssText());
+
         if (property->IDEquals(CSSPropertyID::kBackgroundPositionX) ||
             property->IDEquals(CSSPropertyID::kWebkitMaskPositionX)) {
-          found_position_xcss_property = true;
+          found_position_x_css_property = true;
         }
         if (property->IDEquals(CSSPropertyID::kBackgroundPositionY) ||
             property->IDEquals(CSSPropertyID::kWebkitMaskPositionY)) {
-          found_position_ycss_property = true;
+          found_position_y_css_property = true;
           // background-position is a special case. If only the first offset is
           // specified, the second one defaults to "center", not the same value.
         }
       }
     }
+    if (shorthand.id() == CSSPropertyID::kAlternativeMask &&
+        layer_result.empty()) {
+      result.Append(getValueName(CSSValueID::kNone));
+    }
     if (!layer_result.empty()) {
       if (!result.empty()) {
         result.Append(", ");
@@ -2132,79 +2176,6 @@ String StylePropertySerializer::BorderRadiusValue() const {
   return builder.ReleaseString();
 }
 
-static void AppendBackgroundRepeatValue(StringBuilder& builder,
-                                        const CSSValue& repeat_xcss_value,
-                                        const CSSValue& repeat_ycss_value) {
-  // FIXME: Ensure initial values do not appear in CSS_VALUE_LISTS.
-  DEFINE_STATIC_LOCAL(const Persistent<CSSIdentifierValue>,
-                      initial_repeat_value,
-                      (CSSIdentifierValue::Create(CSSValueID::kRepeat)));
-  const CSSIdentifierValue& repeat_x =
-      repeat_xcss_value.IsInitialValue()
-          ? *initial_repeat_value
-          : To<CSSIdentifierValue>(repeat_xcss_value);
-  const CSSIdentifierValue& repeat_y =
-      repeat_ycss_value.IsInitialValue()
-          ? *initial_repeat_value
-          : To<CSSIdentifierValue>(repeat_ycss_value);
-  CSSValueID repeat_x_value_id = repeat_x.GetValueID();
-  CSSValueID repeat_y_value_id = repeat_y.GetValueID();
-  if (repeat_x_value_id == repeat_y_value_id) {
-    builder.Append(repeat_x.CssText());
-  } else if (repeat_x_value_id == CSSValueID::kNoRepeat &&
-             repeat_y_value_id == CSSValueID::kRepeat) {
-    builder.Append("repeat-y");
-  } else if (repeat_x_value_id == CSSValueID::kRepeat &&
-             repeat_y_value_id == CSSValueID::kNoRepeat) {
-    builder.Append("repeat-x");
-  } else {
-    builder.Append(repeat_x.CssText());
-    builder.Append(' ');
-    builder.Append(repeat_y.CssText());
-  }
-}
-
-String StylePropertySerializer::BackgroundRepeatPropertyValue() const {
-  const CSSValue& repeat_x =
-      *property_set_.GetPropertyCSSValue(GetCSSPropertyBackgroundRepeatX());
-  const CSSValue& repeat_y =
-      *property_set_.GetPropertyCSSValue(GetCSSPropertyBackgroundRepeatY());
-
-  const auto* repeat_x_list = DynamicTo<CSSValueList>(repeat_x);
-  int repeat_x_length = 1;
-  if (repeat_x_list) {
-    repeat_x_length = repeat_x_list->length();
-  } else if (!repeat_x.IsIdentifierValue()) {
-    return String();
-  }
-
-  const auto* repeat_y_list = DynamicTo<CSSValueList>(repeat_y);
-  int repeat_y_length = 1;
-  if (repeat_y_list) {
-    repeat_y_length = repeat_y_list->length();
-  } else if (!repeat_y.IsIdentifierValue()) {
-    return String();
-  }
-
-  size_t shorthand_length =
-      LowestCommonMultiple(repeat_x_length, repeat_y_length);
-  StringBuilder builder;
-  for (size_t i = 0; i < shorthand_length; ++i) {
-    if (i) {
-      builder.Append(", ");
-    }
-
-    const CSSValue& x_value =
-        repeat_x_list ? repeat_x_list->Item(i % repeat_x_list->length())
-                      : repeat_x;
-    const CSSValue& y_value =
-        repeat_y_list ? repeat_y_list->Item(i % repeat_y_list->length())
-                      : repeat_y;
-    AppendBackgroundRepeatValue(builder, x_value, y_value);
-  }
-  return builder.ReleaseString();
-}
-
 String StylePropertySerializer::PageBreakPropertyValue(
     const StylePropertyShorthand& shorthand) const {
   const CSSValue* value =
