diff --git a/third_party/blink/renderer/core/dom/document.idl b/third_party/blink/renderer/core/dom/document.idl
index a2913255fd80766992f0de50a503507e985d5042..1f684bcf02b5093909b3d1250f581508ca9a2ea7 100644
--- a/third_party/blink/renderer/core/dom/document.idl
+++ b/third_party/blink/renderer/core/dom/document.idl
@@ -80,6 +80,9 @@ typedef (HTMLScriptElement or SVGScriptElement) HTMLOrSVGScriptElement;
     // HTML
     // https://html.spec.whatwg.org/C/#the-document-object
 
+    // https://github.com/whatwg/html/pull/9538
+    [RuntimeEnabled=HTMLUnsafeMethods,CallWith=ExecutionContext,MeasureAs=ParseHTMLUnsafe] static Document parseHTMLUnsafe(HTMLString html);
+
     // resource metadata management
 
     [PutForwards=href, LegacyUnforgeable] readonly attribute Location? location;
