diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_image.h b/third_party/blink/renderer/core/layout/svg/layout_svg_image.h
index ea6294800e7dd0763f59b9b690bc3de598e1cf6a..43f9232e5625de2a3cf6fc4dad15ee8021d91f5e 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_image.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_image.h
@@ -88,6 +88,11 @@ class LayoutSVGImage final : public LayoutSVGModelObject {
     return object_bounding_box_;
   }
 
+  gfx::RectF DecoratedBoundingBox() const override {
+    NOT_DESTROYED();
+    return object_bounding_box_;
+  }
+
   void ImageChanged(WrappedImagePtr, CanDeferInvalidation) override;
 
   void UpdateLayout() override;
