diff --git a/third_party/blink/renderer/core/css/css_selector.h b/third_party/blink/renderer/core/css/css_selector.h
index d4dfa0e334d7c82ccf0f51f4dd26160268ae6edf..8903f4b4f2254b8a5ca05b91b3572230da20a6f5 100644
--- a/third_party/blink/renderer/core/css/css_selector.h
+++ b/third_party/blink/renderer/core/css/css_selector.h
@@ -215,6 +215,9 @@ class CORE_EXPORT CSSSelector {
 
   enum PseudoType {
     kPseudoActive,
+    // FlyWeb (Chromium 125).
+    kPseudoActiveViewTransition,
+    kPseudoActiveViewTransitionType,
     kPseudoAfter,
     kPseudoAny,
     kPseudoAnyLink,
@@ -286,13 +289,20 @@ class CORE_EXPORT CSSSelector {
     kPseudoSelectorFragmentAnchor,
     kPseudoSingleButton,
     kPseudoStart,
+    // kPseudoState is for :state(foo). kPseudoStateDeprecated is for :--foo.
+    // :--foo is deprecated and is replacing :state(foo).
+    // TODO(crbug.com/1514397): Remove kPseudoStateDeprecatedSyntax after the
+    // deprecation is done.
     kPseudoState,
+    kPseudoStateDeprecatedSyntax,
     kPseudoTarget,
     kPseudoUnknown,
     // Something that was unparsable, but contained either a nesting
     // selector (&), or a :scope pseudo-class, and must therefore be kept
     // for serialization purposes.
     kPseudoUnparsed,
+    kPseudoUserInvalid,
+    kPseudoUserValid,
     kPseudoValid,
     kPseudoVertical,
     kPseudoVisited,
