diff --git a/third_party/blink/renderer/core/layout/shapes/shape.cc b/third_party/blink/renderer/core/layout/shapes/shape.cc
index c8862c43b0e537128bb42cf56098266f46afa3b7..c1a019ec769b25ea26eebba2eefda27e0f0dd077 100644
--- a/third_party/blink/renderer/core/layout/shapes/shape.cc
+++ b/third_party/blink/renderer/core/layout/shapes/shape.cc
@@ -224,7 +224,7 @@ static bool ExtractImageData(Image* image,
   size_t dst_size_bytes = dst_info.computeMinByteSize();
   {
     if (SkImageInfo::ByteSizeOverflowed(dst_size_bytes) ||
-        dst_size_bytes > v8::TypedArray::kMaxLength) {
+        dst_size_bytes > v8::TypedArray::kMaxByteLength) {
       return false;
     }
     ArrayBufferContents result(dst_size_bytes, 1,
