diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.cc
index fdd9767b7e7c738c882772676ace18033a9931c7..179fa00fe2b4923b37e1f1e6dc1dad86ad9dbd54 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.cc
@@ -31,9 +31,7 @@
 namespace blink {
 
 LayoutSVGResourceMarker::LayoutSVGResourceMarker(SVGMarkerElement* node)
-    : LayoutSVGResourceContainer(node),
-      needs_transform_update_(true),
-      is_in_layout_(false) {}
+    : LayoutSVGResourceContainer(node), is_in_layout_(false) {}
 
 LayoutSVGResourceMarker::~LayoutSVGResourceMarker() = default;
 
@@ -145,18 +143,12 @@ bool LayoutSVGResourceMarker::ShouldPaint() const {
 
 void LayoutSVGResourceMarker::SetNeedsTransformUpdate() {
   NOT_DESTROYED();
-  // The transform paint property relies on the SVG transform being up-to-date
-  // (see: PaintPropertyTreeBuilder::updateTransformForNonRootSVG).
-  SetNeedsPaintPropertyUpdate();
-  needs_transform_update_ = true;
+  LayoutSVGContainer::SetNeedsTransformUpdate();
 }
 
-SVGTransformChange LayoutSVGResourceMarker::CalculateLocalTransform(
-    bool bounds_changed) {
+SVGTransformChange LayoutSVGResourceMarker::UpdateLocalTransform(
+    const gfx::RectF& reference_box) {
   NOT_DESTROYED();
-  if (!needs_transform_update_)
-    return SVGTransformChange::kNone;
-
   auto* marker = To<SVGMarkerElement>(GetElement());
   DCHECK(marker);
 
@@ -167,8 +159,6 @@ SVGTransformChange LayoutSVGResourceMarker::CalculateLocalTransform(
 
   SVGTransformChangeDetector change_detector(local_to_parent_transform_);
   local_to_parent_transform_ = marker->ViewBoxToViewTransform(viewport_size_);
-
-  needs_transform_update_ = false;
   return change_detector.ComputeChange(local_to_parent_transform_);
 }
 
