diff --git a/third_party/blink/renderer/core/css/resolver/css_to_style_map.h b/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
index 14e8b1191afc6832552efa768d4a539f00ecc6f0..9cf6f2f69263b0896a720a23f1831e5762f52034 100644
--- a/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
+++ b/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
@@ -80,6 +80,9 @@ class CSSToStyleMap {
   static double MapAnimationIterationCount(StyleResolverState&,
                                            const CSSValue&);
   static AtomicString MapAnimationName(StyleResolverState&, const CSSValue&);
+  static CSSTransitionData::TransitionBehavior MapAnimationBehavior(
+      StyleResolverState&,
+      const CSSValue&);
   static StyleTimeline MapAnimationTimeline(StyleResolverState&,
                                             const CSSValue&);
   static EAnimPlayState MapAnimationPlayState(StyleResolverState&,
