diff --git a/third_party/blink/renderer/core/css/parser/css_parser_token_stream.cc b/third_party/blink/renderer/core/css/parser/css_parser_token_stream.cc
index 2f2d5cbaaaa7a016a034a2deaaf406008d869b5d..47555c00e393b246ec4aef3dcaaa11d79cedce5f 100644
--- a/third_party/blink/renderer/core/css/parser/css_parser_token_stream.cc
+++ b/third_party/blink/renderer/core/css/parser/css_parser_token_stream.cc
@@ -59,6 +59,28 @@ void CSSParserTokenStream::UncheckedConsumeComponentValue() {
   } while (!PeekInternal().IsEOF() && nesting_level);
 }
 
+CSSParserTokenRange CSSParserTokenStream::ConsumeComponentValue() {
+  EnsureLookAhead();
+
+  buffer_.Shrink(0);
+
+  if (AtEnd()) {
+    return CSSParserTokenRange(base::span<CSSParserToken>{});
+  }
+
+  unsigned nesting_level = 0;
+  do {
+    buffer_.push_back(UncheckedConsumeInternal());
+    if (buffer_.back().GetBlockType() == CSSParserToken::kBlockStart) {
+      nesting_level++;
+    } else if (buffer_.back().GetBlockType() == CSSParserToken::kBlockEnd) {
+      nesting_level--;
+    }
+  } while (!PeekInternal().IsEOF() && nesting_level);
+
+  return CSSParserTokenRange(buffer_);
+}
+
 void CSSParserTokenStream::UncheckedSkipToEndOfBlock() {
   DCHECK(HasLookAhead());
 
