diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_root.h b/third_party/blink/renderer/core/layout/svg/layout_svg_root.h
index a5afe32364cfda593e432ff814a49de2a6cf0225..c91e7c3764530f1480446e0f6d1ba94ab5c535fe 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_root.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_root.h
@@ -158,13 +158,13 @@ class CORE_EXPORT LayoutSVGRoot final : public LayoutReplaced {
     NOT_DESTROYED();
     return content_.ObjectBoundingBox();
   }
-  gfx::RectF StrokeBoundingBox() const override {
+  gfx::RectF DecoratedBoundingBox() const override {
     NOT_DESTROYED();
-    return content_.StrokeBoundingBox();
+    return content_.DecoratedBoundingBox();
   }
   gfx::RectF VisualRectInLocalSVGCoordinates() const override {
     NOT_DESTROYED();
-    return content_.StrokeBoundingBox();
+    return content_.DecoratedBoundingBox();
   }
 
   bool NodeAtPoint(HitTestResult&,
