diff --git a/third_party/blink/renderer/core/html/forms/password_input_type.cc b/third_party/blink/renderer/core/html/forms/password_input_type.cc
index 21cee6a7bf410af0b2f302cd26d6d6832dad421f..80d325fe4b92468d28851b2d682fac4e88a895a2 100644
--- a/third_party/blink/renderer/core/html/forms/password_input_type.cc
+++ b/third_party/blink/renderer/core/html/forms/password_input_type.cc
@@ -239,4 +239,8 @@ bool PasswordInputType::SupportsInputModeAttribute() const {
   return true;
 }
 
+bool PasswordInputType::ShouldAutoDirUseValue() const {
+  return false;
+}
+
 }  // namespace blink
