diff --git a/third_party/blink/renderer/core/layout/ng/inline/ng_inline_layout_algorithm.cc b/third_party/blink/renderer/core/layout/ng/inline/ng_inline_layout_algorithm.cc
index 77ea1583e3b6d1a7d3775556204ba5e317d4d9eb..33074613195bf486ccdc63910469699e7f774d9f 100644
--- a/third_party/blink/renderer/core/layout/ng/inline/ng_inline_layout_algorithm.cc
+++ b/third_party/blink/renderer/core/layout/ng/inline/ng_inline_layout_algorithm.cc
@@ -146,7 +146,7 @@ class NGLineBreakStrategy {
     DCHECK(!opportunities.back().HasShapeExclusions());
 
     // Try `NGScoreLineBreaker` first if it's applicable.
-    if (use_score_line_break_) {
+    if (use_score_line_break_ && score_line_break_context_->IsActive()) {
       DCHECK(RuntimeEnabledFeatures::CSSTextWrapBalanceByScoreEnabled());
       DCHECK(score_line_break_context_->LineBreakPoints().empty());
       DCHECK_EQ(score_line_break_context_->LineBreakPointsIndex(), 0u);
