diff --git a/third_party/blink/renderer/core/url/url_search_params.h b/third_party/blink/renderer/core/url/url_search_params.h
index df7430372e20f5896af5d6b9afc3db69a288ff2a..60abea350cd4f040c6fb1a47dd41bd28c390c361 100644
--- a/third_party/blink/renderer/core/url/url_search_params.h
+++ b/third_party/blink/renderer/core/url/url_search_params.h
@@ -53,14 +53,17 @@ class CORE_EXPORT URLSearchParams final
   String toString() const;
   uint32_t size() const;
   void append(const String& name, const String& value);
-  void deleteAllWithName(ScriptState*, const String&);
-  void deleteAllWithName(ScriptState*,
-                         const String&,
-                         const ScriptValue& ignored);
+  void deleteAllWithNameOrTuple(ExecutionContext* execution_context,
+                                const String& name);
+  void deleteAllWithNameOrTuple(ExecutionContext* execution_context,
+                                const String& name,
+                                const String& val);
   String get(const String&) const;
   Vector<String> getAll(const String&) const;
-  bool has(ScriptState*, const String&) const;
-  bool has(ScriptState*, const String&, const ScriptValue& ignored) const;
+  bool has(ExecutionContext* execution_context, const String& name) const;
+  bool has(ExecutionContext* execution_context,
+           const String& name,
+           const String& val) const;
   void set(const String& name, const String& value);
   void sort();
   void SetInputWithoutUpdate(const String&);
