diff --git a/third_party/blink/renderer/core/html/forms/input_type.cc b/third_party/blink/renderer/core/html/forms/input_type.cc
index 3c8125ea9f7ea1816e37f75b10bb152dd7cc6722..d7fe53812aaf5d0313812df6dfbfa0f054989fb3 100644
--- a/third_party/blink/renderer/core/html/forms/input_type.cc
+++ b/third_party/blink/renderer/core/html/forms/input_type.cc
@@ -145,6 +145,10 @@ bool InputType::IsTextField() const {
   return false;
 }
 
+bool InputType::ShouldAutoDirUseValue() const {
+  return false;
+}
+
 template <typename T>
 bool ValidateInputType(const T& input_type, const String& value) {
   if (!input_type.CanSetStringValue()) {
