diff --git a/third_party/blink/renderer/core/mathml/mathml_table_cell_element.h b/third_party/blink/renderer/core/mathml/mathml_table_cell_element.h
new file mode 100644
index 0000000000000000000000000000000000000000..3de938d0448999f9215e1d788778350e0dda21d9
--- /dev/null
+++ b/third_party/blink/renderer/core/mathml/mathml_table_cell_element.h
@@ -0,0 +1,26 @@
+// Copyright 2023 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef THIRD_PARTY_BLINK_RENDERER_CORE_MATHML_MATHML_TABLE_CELL_ELEMENT_H_
+#define THIRD_PARTY_BLINK_RENDERER_CORE_MATHML_MATHML_TABLE_CELL_ELEMENT_H_
+
+#include "third_party/blink/renderer/core/mathml/mathml_element.h"
+
+namespace blink {
+
+class CORE_EXPORT MathMLTableCellElement final : public MathMLElement {
+ public:
+  explicit MathMLTableCellElement(Document&);
+
+  unsigned colSpan() const;
+  unsigned rowSpan() const;
+
+ private:
+  void ParseAttribute(const AttributeModificationParams&) final;
+  LayoutObject* CreateLayoutObject(const ComputedStyle&) final;
+};
+
+}  // namespace blink
+
+#endif  // THIRD_PARTY_BLINK_RENDERER_CORE_MATHML_MATHML_TABLE_CELL_ELEMENT_H_
