diff --git a/third_party/blink/renderer/modules/accessibility/ax_node_object.cc b/third_party/blink/renderer/modules/accessibility/ax_node_object.cc
index bab762a111725a735b0ab8cfd9d10eca10506235..f6f244137f62c12a17fc087a13de7a5cbc052938 100644
--- a/third_party/blink/renderer/modules/accessibility/ax_node_object.cc
+++ b/third_party/blink/renderer/modules/accessibility/ax_node_object.cc
@@ -795,8 +795,9 @@ absl::optional<String> AXNodeObject::GetCSSAltText(const Node* node) {
   if (node->IsPseudoElement()) {
     for (const ContentData* content_data = style->GetContentData();
          content_data; content_data = content_data->Next()) {
-      if (content_data->IsAltText())
-        return To<AltTextContentData>(content_data)->GetText();
+      if (auto* css_alt = DynamicTo<AltTextContentData>(content_data)) {
+        return css_alt->ConcatenateAltText();
+      }
     }
     return absl::nullopt;
   }
@@ -807,7 +808,7 @@ absl::optional<String> AXNodeObject::GetCSSAltText(const Node* node) {
   const ContentData* content_data = style->GetContentData();
   if (content_data && content_data->IsImage() && content_data->Next() &&
       content_data->Next()->IsAltText()) {
-    return To<AltTextContentData>(content_data->Next())->GetText();
+    return To<AltTextContentData>(content_data->Next())->ConcatenateAltText();
   }
 
   return absl::nullopt;
