diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
index a8737eac306351e6cbfbe1cda00390497d9c5bcc..933e93711c14c670bebfcda81810ef014505528a 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_container.cc
@@ -197,19 +197,11 @@ void LayoutSVGResourceContainer::MarkAllClientsForInvalidation(
   is_invalidating_ = false;
 }
 
-void LayoutSVGResourceContainer::InvalidateCacheAndMarkForLayout(
-    LayoutInvalidationReasonForTracing reason) {
+void LayoutSVGResourceContainer::InvalidateCache() {
   NOT_DESTROYED();
-  SetNeedsLayoutAndFullPaintInvalidation(reason, kMarkContainerChain);
-
-  if (EverHadLayout())
+  if (EverHadLayout()) {
     RemoveAllClientsFromCache();
-}
-
-void LayoutSVGResourceContainer::InvalidateCacheAndMarkForLayout() {
-  NOT_DESTROYED();
-  InvalidateCacheAndMarkForLayout(
-      layout_invalidation_reason::kSvgResourceInvalidated);
+  }
 }
 
 static inline void RemoveFromCacheAndInvalidateDependencies(
