diff --git a/third_party/blink/renderer/core/paint/document_marker_painter.cc b/third_party/blink/renderer/core/paint/document_marker_painter.cc
index 2e1f7e2ca687586abcb8a597ded48df290d3bc87..a1322a6e1b6c2c8690305b3dcfa028d519fc85dc 100644
--- a/third_party/blink/renderer/core/paint/document_marker_painter.cc
+++ b/third_party/blink/renderer/core/paint/document_marker_painter.cc
@@ -296,6 +296,7 @@ TextPaintStyle DocumentMarkerPainter::ComputeTextPaintStyleFrom(
   text_style.stroke_width = style.TextStrokeWidth();
   text_style.color_scheme = style.UsedColorScheme();
   text_style.shadow = nullptr;
+  text_style.paint_order = style.PaintOrder();
   if (marker.GetType() == DocumentMarker::kTextMatch)
     return text_style;
   return HighlightStyleUtils::HighlightPaintingStyle(
