diff --git a/third_party/blink/renderer/bindings/scripts/bind_gen/codegen_context.py b/third_party/blink/renderer/bindings/scripts/bind_gen/codegen_context.py
index c3fd4d1b09c7cd48c1dab3d1043ccc00f77ad406..16caaf2bf51b080bc0edf02ce2186542d4b1d743 100644
--- a/third_party/blink/renderer/bindings/scripts/bind_gen/codegen_context.py
+++ b/third_party/blink/renderer/bindings/scripts/bind_gen/codegen_context.py
@@ -78,6 +78,7 @@ class CodeGenContext(object):
         #   attribute name: default value
         cls._context_attrs = {
             # Top-level definition
+            "async_iterator": None,
             "callback_function": None,
             "callback_interface": None,
             "dictionary": None,
@@ -230,8 +231,9 @@ class CodeGenContext(object):
 
     @property
     def class_like(self):
-        return (self.callback_interface or self.dictionary or self.interface
-                or self.namespace or self.sync_iterator)
+        return (self.async_iterator or self.callback_interface
+                or self.dictionary or self.interface or self.namespace
+                or self.sync_iterator)
 
     @property
     def does_override_idl_return_type(self):
