diff --git a/third_party/blink/renderer/core/css/font_size_functions.cc b/third_party/blink/renderer/core/css/font_size_functions.cc
index 5629009dfa1c775c329b0e59a6304b63562e7f14..cf626db14febe8dc96535bb8997c273318216047 100644
--- a/third_party/blink/renderer/core/css/font_size_functions.cc
+++ b/third_party/blink/renderer/core/css/font_size_functions.cc
@@ -199,22 +199,15 @@ int FontSizeFunctions::LegacyFontSize(const Document* document,
                                           medium_size);
 }
 
-absl::optional<float> FontSizeFunctions::MetricsMultiplierAdjustedFontSize(
-    const SimpleFontData* font_data,
-    const FontDescription& font_description) {
-  DCHECK(font_data);
-  const float computed_size = font_description.ComputedSize();
-  if (!computed_size) {
-    return absl::nullopt;
-  }
-
-  const FontSizeAdjust size_adjust = font_description.SizeAdjust();
-  const FontMetrics& font_metrics = font_data->GetFontMetrics();
-
+static absl::optional<float> AspectValue(const SimpleFontData& font_data,
+                                        FontSizeAdjust::Metric metric,
+                                        float computed_size) {
+  DCHECK(computed_size);
+  const FontMetrics& font_metrics = font_data.GetFontMetrics();
   // FIXME: The behavior for missing metrics has yet to be defined.
   // https://github.com/w3c/csswg-drafts/issues/6384
-  float aspect_value = 1.0;
-  switch (size_adjust.GetMetric()) {
+  absl::optional<float> aspect_value = 1.0;
+  switch (metric) {
     case FontSizeAdjust::Metric::kCapHeight:
       if (font_metrics.CapHeight() > 0) {
         aspect_value = font_metrics.CapHeight() / computed_size;
@@ -226,22 +219,60 @@ absl::optional<float> FontSizeFunctions::MetricsMultiplierAdjustedFontSize(
       }
       break;
     case FontSizeAdjust::Metric::kIcWidth:
-      if (font_metrics.IdeographicFullWidth().has_value()) {
-        aspect_value =
-            font_metrics.IdeographicFullWidth().value() / computed_size;
+      if (const absl::optional<float>& size =
+              font_data.IdeographicAdvanceWidth()) {
+        aspect_value = *size / computed_size;
       }
       break;
+    case FontSizeAdjust::Metric::kIcHeight: {
+      const absl::optional<float>& size = font_data.IdeographicAdvanceHeight();
+      aspect_value = font_data.HasVerticalMetrics() && size
+                         ? absl::make_optional<float>(*size / computed_size)
+                         : absl::nullopt;
+      break;
+    }
     case FontSizeAdjust::Metric::kExHeight:
     default:
       if (font_metrics.HasXHeight()) {
         aspect_value = font_metrics.XHeight() / computed_size;
       }
   }
+  return aspect_value;
+}
+
+absl::optional<float> FontSizeFunctions::FontAspectValue(
+    const SimpleFontData* font_data,
+    FontSizeAdjust::Metric metric,
+    float computed_size) {
+  if (!font_data || !computed_size) {
+    return absl::nullopt;
+  }
+
+  absl::optional<float> aspect_value =
+      AspectValue(*font_data, metric, computed_size);
+  if (!aspect_value || !aspect_value.value()) {
+    return absl::nullopt;
+  }
+  return aspect_value;
+}
+
+absl::optional<float> FontSizeFunctions::MetricsMultiplierAdjustedFontSize(
+    const SimpleFontData* font_data,
+    const FontDescription& font_description) {
+  DCHECK(font_data);
+  const float computed_size = font_description.ComputedSize();
+  const FontSizeAdjust size_adjust = font_description.SizeAdjust();
+  if (!computed_size ||
+      size_adjust.Value() == FontSizeAdjust::kFontSizeAdjustNone) {
+    return absl::nullopt;
+  }
 
-  if (!aspect_value) {
+  absl::optional<float> aspect_value =
+      AspectValue(*font_data, size_adjust.GetMetric(), computed_size);
+  if (!aspect_value || !aspect_value.value()) {
     return absl::nullopt;
   }
-  return (size_adjust.Value() / aspect_value) * computed_size;
+  return (size_adjust.Value() / aspect_value.value()) * computed_size;
 }
 
 }  // namespace blink
