diff --git a/third_party/blink/renderer/core/css/css_numeric_literal_value.cc b/third_party/blink/renderer/core/css/css_numeric_literal_value.cc
index 81e212b7c68089947087a30c67514df513515272..97da6a41b1d1e2edb0c93b297f1ced726da74cc4 100644
--- a/third_party/blink/renderer/core/css/css_numeric_literal_value.cc
+++ b/third_party/blink/renderer/core/css/css_numeric_literal_value.cc
@@ -212,6 +212,8 @@ String CSSNumericLiteralValue::CustomCSSText() const {
     case UnitType::kRics:
     case UnitType::kChs:
     case UnitType::kIcs:
+    case UnitType::kCaps:
+    case UnitType::kRcaps:
     case UnitType::kLhs:
     case UnitType::kRlhs:
     case UnitType::kPixels:
