diff --git a/third_party/blink/renderer/bindings/generated_in_modules.gni b/third_party/blink/renderer/bindings/generated_in_modules.gni
index d479a0547c8e36bcc8500581b00cdf390b958972..1da204c362a764c33b11f67edc7fe7fa88d265b7 100644
--- a/third_party/blink/renderer/bindings/generated_in_modules.gni
+++ b/third_party/blink/renderer/bindings/generated_in_modules.gni
@@ -2796,6 +2796,7 @@ generated_interface_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_webgl_sub_image.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_xr_webgl_sub_image.h",
 ]
+generated_interface_sources_in_modules += brave_generated_interface_sources_in_modules
 
 generated_namespace_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_css.cc",
@@ -2814,6 +2815,8 @@ generated_namespace_sources_in_modules = [
 
 generated_observable_array_sources_in_modules = []
 
+generated_async_iterator_sources_in_modules = []
+
 generated_sync_iterator_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sync_iterator_audio_param_map.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_sync_iterator_audio_param_map.h",
