diff --git a/third_party/blink/renderer/core/animation/css/css_animations.h b/third_party/blink/renderer/core/animation/css/css_animations.h
index f1e39b0b535f591ea822bd38bd7da1e2f45fec1d..ee318f36f7afd30dc4021cd114b3b008fdffcf63 100644
--- a/third_party/blink/renderer/core/animation/css/css_animations.h
+++ b/third_party/blink/renderer/core/animation/css/css_animations.h
@@ -274,18 +274,18 @@ class CORE_EXPORT CSSAnimations final {
   static void CalculateTransitionUpdateForProperty(
       TransitionUpdateState&,
       const CSSTransitionData::TransitionProperty&,
-      size_t transition_index,
+      wtf_size_t transition_index,
       WritingDirectionMode);
 
   static void CalculateTransitionUpdateForCustomProperty(
       TransitionUpdateState&,
       const CSSTransitionData::TransitionProperty&,
-      size_t transition_index);
+      wtf_size_t transition_index);
 
   static void CalculateTransitionUpdateForStandardProperty(
       TransitionUpdateState&,
       const CSSTransitionData::TransitionProperty&,
-      size_t transition_index,
+      wtf_size_t transition_index,
       WritingDirectionMode);
 
   static bool CanCalculateTransitionUpdateForProperty(
@@ -295,7 +295,7 @@ class CORE_EXPORT CSSAnimations final {
   static void CalculateTransitionUpdateForPropertyHandle(
       TransitionUpdateState&,
       const PropertyHandle&,
-      size_t transition_index,
+      wtf_size_t transition_index,
       bool animate_all);
 
   static void CalculateAnimationActiveInterpolations(
