diff --git a/third_party/blink/renderer/core/css/css_numeric_literal_value.h b/third_party/blink/renderer/core/css/css_numeric_literal_value.h
index 7feba2a5b4be2a02fbda937eba5d9d16c05bd1a9..84188243dec0a93c38294c2a6365c6ad94e0feef 100644
--- a/third_party/blink/renderer/core/css/css_numeric_literal_value.h
+++ b/third_party/blink/renderer/core/css/css_numeric_literal_value.h
@@ -35,6 +35,8 @@ class CORE_EXPORT CSSNumericLiteralValue : public CSSPrimitiveValue {
       case UnitType::kChs:
       case UnitType::kIcs:
       case UnitType::kLhs:
+      case UnitType::kCaps:
+      case UnitType::kRcaps:
       case UnitType::kRexs:
       case UnitType::kRchs:
       case UnitType::kRics:
