diff --git a/third_party/blink/renderer/core/css/resolver/element_style_resources.h b/third_party/blink/renderer/core/css/resolver/element_style_resources.h
index 04c534d8c45e9eca83fd8d4511c8c437092801a6..4220ffff80480fe0ffd22178d7846c15e2f860c0 100644
--- a/third_party/blink/renderer/core/css/resolver/element_style_resources.h
+++ b/third_party/blink/renderer/core/css/resolver/element_style_resources.h
@@ -82,6 +82,8 @@ class ElementStyleResources {
   bool IsPending(const CSSValue&) const;
   StyleImage* CachedStyleImage(const CSSValue&) const;
 
+  StyleImage* LoadMaskSource(CSSValue&);
+
   void LoadPendingSVGResources(ComputedStyleBuilder&);
   void LoadPendingImages(ComputedStyleBuilder&);
 
