diff --git a/third_party/blink/renderer/core/css/parser/css_supports_parser.cc b/third_party/blink/renderer/core/css/parser/css_supports_parser.cc
index 7c799f0c0bf5b4bf4bfc748b2a9f163d32f55ffd..c69f6ea1dfcadbe6e400dd42f581b4466a30e34d 100644
--- a/third_party/blink/renderer/core/css/parser/css_supports_parser.cc
+++ b/third_party/blink/renderer/core/css/parser/css_supports_parser.cc
@@ -259,6 +259,25 @@ CSSSupportsParser::Result CSSSupportsParser::ConsumeSupportsDecl(
   if (parser_.ConsumeSupportsDeclaration(stream)) {
     return Result::kSupported;
   }
+
+  // ConsumeSupportsDeclaration can leave the stream in various states,
+  // see documentation near CSSParserImpl::ConsumeDeclaration.
+  if (!stream.AtEnd()) {
+    // If there are remaining tokens, then ConsumeSupportsDeclaration backed
+    // out early due to a missing colon or invalid property.
+    // This normally means it's either an invalid property (kUnsupported),
+    // or some other unknown construct (also kUnsupported). However, the unknown
+    // construct must not violate the rules of <general-enclosed>. If that
+    // happens, it is instead a kParseFailure.
+    CSSParserTokenRange remaining = stream.ConsumeUntilPeekedTypeIs<>();
+    // TODO(crbug.com/1361240): This is the same check as
+    // ConsumeGeneralEnclosed. It would be cleaner to just restart and actually
+    // call that function.
+    if (!ConsumeAnyValue(remaining) || !remaining.AtEnd()) {
+      return Result::kParseFailure;
+    }
+  }
+
   return Result::kUnsupported;
 }
 
