diff --git a/third_party/blink/renderer/core/scroll/scrollbar.h b/third_party/blink/renderer/core/scroll/scrollbar.h
index eeae88d612520d466330c30acb2fa179e368a27b..22320da6c78724fd582afd5913a47974d5fe0139 100644
--- a/third_party/blink/renderer/core/scroll/scrollbar.h
+++ b/third_party/blink/renderer/core/scroll/scrollbar.h
@@ -31,6 +31,7 @@
 #include "third_party/blink/renderer/core/core_export.h"
 #include "third_party/blink/renderer/core/scroll/scroll_types.h"
 #include "third_party/blink/renderer/core/style/computed_style_constants.h"
+#include "third_party/blink/renderer/core/style/style_scrollbar_color.h"
 #include "third_party/blink/renderer/platform/graphics/compositor_element_id.h"
 #include "third_party/blink/renderer/platform/graphics/paint/display_item_client.h"
 #include "third_party/blink/renderer/platform/heap/garbage_collected.h"
@@ -44,8 +45,8 @@ class Rect;
 
 namespace blink {
 
-class Element;
 class GraphicsContext;
+class LayoutObject;
 class ScrollableArea;
 class ScrollbarTheme;
 class WebGestureEvent;
@@ -59,14 +60,11 @@ class CORE_EXPORT Scrollbar : public GarbageCollected<Scrollbar>,
   // scrollbar.
   static Scrollbar* CreateForTesting(ScrollableArea* scrollable_area,
                                      ScrollbarOrientation orientation,
-                                     ScrollbarTheme* theme) {
-    return MakeGarbageCollected<Scrollbar>(scrollable_area, orientation,
-                                           nullptr, theme);
-  }
+                                     ScrollbarTheme* theme);
 
   Scrollbar(ScrollableArea*,
             ScrollbarOrientation,
-            Element* style_source,
+            const LayoutObject* style_source,
             ScrollbarTheme* = nullptr);
   ~Scrollbar() override;
 
@@ -211,11 +209,18 @@ class CORE_EXPORT Scrollbar : public GarbageCollected<Scrollbar>,
 
   // scrollbar-width CSS property
   EScrollbarWidth CSSScrollbarWidth() const;
-
-  // The Element that supplies our style information. If the scrollbar is
-  // for a document, this is either the <body> or <html> element. Otherwise, it
-  // is the element that owns our PaintLayerScrollableArea.
-  Element* StyleSource() const { return style_source_.Get(); }
+  // scrollbar-color CSS property
+  absl::optional<blink::Color> ScrollbarThumbColor() const;
+  absl::optional<blink::Color> ScrollbarTrackColor() const;
+
+  // The LayoutObject that supplies our style information. If the scrollbar is
+  // for a document, this is:
+  // 1. the LayoutView (with some scrollbar related styles propagated from the
+  //    document element and/or the <body>), or
+  // 2. the <body> or document element's layout object if it has webkit custom
+  //    scrollbar styles.
+  // Otherwise, it is the LayoutBox that owns our PaintLayerScrollableArea.
+  const LayoutObject* StyleSource() const { return style_source_.Get(); }
 
   mojom::blink::ColorScheme UsedColorScheme() const;
 
@@ -278,7 +283,7 @@ class CORE_EXPORT Scrollbar : public GarbageCollected<Scrollbar>,
   bool scrollbar_manipulation_in_progress_on_cc_thread_;
 
   gfx::Rect frame_rect_;
-  Member<Element> style_source_;
+  WeakMember<const LayoutObject> style_source_;
 
   // Tracks scroll delta that has been injected into the compositor thread as a
   // GestureScrollUpdate but hasn't yet updated the scroll position on main.
