diff --git a/third_party/blink/renderer/core/layout/layout_custom_scrollbar_part.cc b/third_party/blink/renderer/core/layout/layout_custom_scrollbar_part.cc
index 87d369ebb589f559d828b5322e56f85a661e8171..17e65218e5a9bbe2d27c3fe6e1298f3e1af90019 100644
--- a/third_party/blink/renderer/core/layout/layout_custom_scrollbar_part.cc
+++ b/third_party/blink/renderer/core/layout/layout_custom_scrollbar_part.cc
@@ -192,29 +192,33 @@ static LayoutUnit ComputeMargin(const Length& style_margin) {
 
 LayoutUnit LayoutCustomScrollbarPart::MarginTop() const {
   NOT_DESTROYED();
-  if (scrollbar_->Orientation() == kHorizontalScrollbar)
+  if (scrollbar_ && scrollbar_->Orientation() == kHorizontalScrollbar) {
     return LayoutUnit();
+  }
   return ComputeMargin(StyleRef().MarginTop());
 }
 
 LayoutUnit LayoutCustomScrollbarPart::MarginBottom() const {
   NOT_DESTROYED();
-  if (scrollbar_->Orientation() == kHorizontalScrollbar)
+  if (scrollbar_ && scrollbar_->Orientation() == kHorizontalScrollbar) {
     return LayoutUnit();
+  }
   return ComputeMargin(StyleRef().MarginBottom());
 }
 
 LayoutUnit LayoutCustomScrollbarPart::MarginLeft() const {
   NOT_DESTROYED();
-  if (scrollbar_->Orientation() == kVerticalScrollbar)
+  if (scrollbar_ && scrollbar_->Orientation() == kVerticalScrollbar) {
     return LayoutUnit();
+  }
   return ComputeMargin(StyleRef().MarginLeft());
 }
 
 LayoutUnit LayoutCustomScrollbarPart::MarginRight() const {
   NOT_DESTROYED();
-  if (scrollbar_->Orientation() == kVerticalScrollbar)
+  if (scrollbar_ && scrollbar_->Orientation() == kVerticalScrollbar) {
     return LayoutUnit();
+  }
   return ComputeMargin(StyleRef().MarginRight());
 }
 
