diff --git a/third_party/blink/renderer/core/intersection_observer/intersection_observer.cc b/third_party/blink/renderer/core/intersection_observer/intersection_observer.cc
index a9b32c80294e4acef5ceae501ea5d8aaa3054b07..ec408a38af3fbd9d6eebb04820eed32d90065bb4 100644
--- a/third_party/blink/renderer/core/intersection_observer/intersection_observer.cc
+++ b/third_party/blink/renderer/core/intersection_observer/intersection_observer.cc
@@ -92,9 +92,10 @@ class IntersectionObserverDelegateImpl final
   bool needs_initial_observation_with_detached_target_;
 };
 
-void ParseMargin(String margin_parameter,
+void ParseMargin(const String& margin_parameter,
                  Vector<Length>& margin,
-                 ExceptionState& exception_state) {
+                 ExceptionState& exception_state,
+                 const String& marginName) {
   // TODO(szager): Make sure this exact syntax and behavior is spec-ed
   // somewhere.
 
@@ -104,6 +105,7 @@ void ParseMargin(String margin_parameter,
   // "1px 2px" = top/bottom left/right
   // "1px 2px 3px" = top left/right bottom
   // "1px 2px 3px 4px" = top left right bottom
+
   CSSTokenizer tokenizer(margin_parameter);
   const auto tokens = tokenizer.TokenizeToEOF();
   CSSParserTokenRange token_range(tokens);
@@ -113,7 +115,7 @@ void ParseMargin(String margin_parameter,
     if (margin.size() == 4) {
       exception_state.ThrowDOMException(
           DOMExceptionCode::kSyntaxError,
-          "Extra text found at the end of rootMargin.");
+          "Extra text found at the end of " + marginName + "Margin.");
       break;
     }
     const CSSParserToken& token = token_range.ConsumeIncludingWhitespace();
@@ -133,13 +135,13 @@ void ParseMargin(String margin_parameter,
           default:
             exception_state.ThrowDOMException(
                 DOMExceptionCode::kSyntaxError,
-                "rootMargin must be specified in pixels or percent.");
+                marginName + "Margin must be specified in pixels or percent.");
         }
         break;
       default:
         exception_state.ThrowDOMException(
             DOMExceptionCode::kSyntaxError,
-            "rootMargin must be specified in pixels or percent.");
+            marginName + "Margin must be specified in pixels or percent.");
     }
   }
 }
@@ -174,6 +176,69 @@ void ParseThresholds(
   std::sort(thresholds.begin(), thresholds.end());
 }
 
+// Returns a Vector of 4 margins (top, right, bottom, left) following
+// https://drafts.csswg.org/css-box-4/#margin-shorthand
+Vector<Length> NormalizeMargins(const Vector<Length>& margins) {
+  Vector<Length> normalized_margins(4, Length::Fixed(0));
+
+  switch (margins.size()) {
+    case 0:
+      break;
+    case 1:
+      normalized_margins[0] = normalized_margins[1] = normalized_margins[2] =
+          normalized_margins[3] = margins[0];
+      break;
+    case 2:
+      normalized_margins[0] = normalized_margins[2] = margins[0];
+      normalized_margins[1] = normalized_margins[3] = margins[1];
+      break;
+    case 3:
+      normalized_margins[0] = margins[0];
+      normalized_margins[1] = normalized_margins[3] = margins[1];
+      normalized_margins[2] = margins[2];
+      break;
+    case 4:
+      normalized_margins[0] = margins[0];
+      normalized_margins[1] = margins[1];
+      normalized_margins[2] = margins[2];
+      normalized_margins[3] = margins[3];
+      break;
+    default:
+      NOTREACHED();
+      break;
+  }
+
+  return normalized_margins;
+}
+
+String StringifyMargin(const Vector<Length>& margin) {
+  StringBuilder string_builder;
+
+  const auto append_length = [&](const Length& length) {
+    string_builder.AppendNumber(length.IntValue());
+    if (length.IsPercent()) {
+      string_builder.Append('%');
+    } else {
+      string_builder.Append("px", 2);
+    }
+  };
+
+  if (margin.empty()) {
+    string_builder.Append("0px 0px 0px 0px");
+  } else {
+    DCHECK_EQ(margin.size(), 4u);
+    append_length(margin[0]);
+    string_builder.Append(' ');
+    append_length(margin[1]);
+    string_builder.Append(' ');
+    append_length(margin[2]);
+    string_builder.Append(' ');
+    append_length(margin[3]);
+  }
+
+  return string_builder.ToString();
+}
+
 }  // anonymous namespace
 
 static bool throttle_delay_enabled = true;
@@ -215,18 +280,29 @@ IntersectionObserver* IntersectionObserver::Create(
   }
 
   Vector<Length> margin;
-  ParseMargin(observer_init->rootMargin(), margin, exception_state);
-  if (exception_state.HadException())
+  ParseMargin(observer_init->rootMargin(), margin, exception_state, "root");
+  if (exception_state.HadException()) {
     return nullptr;
+  }
+
+  Vector<Length> scroll_margin;
+  if (RuntimeEnabledFeatures::IntersectionObserverScrollMarginEnabled()) {
+    ParseMargin(observer_init->scrollMargin(), scroll_margin, exception_state,
+                "scroll");
+    if (exception_state.HadException()) {
+      return nullptr;
+    }
+  }
 
   Vector<float> thresholds;
   ParseThresholds(observer_init->threshold(), thresholds, exception_state);
-  if (exception_state.HadException())
+  if (exception_state.HadException()) {
     return nullptr;
+  }
 
   return MakeGarbageCollected<IntersectionObserver>(
-      delegate, root, margin, thresholds, kFractionOfTarget, delay,
-      track_visibility, false, kApplyMarginToRoot, false);
+      delegate, root, margin, scroll_margin, thresholds, kFractionOfTarget,
+      delay, track_visibility, false, kApplyMarginToRoot, false);
 }
 
 IntersectionObserver* IntersectionObserver::Create(
@@ -264,8 +340,9 @@ IntersectionObserver* IntersectionObserver::Create(
           document->GetExecutionContext(), std::move(callback), ukm_metric_id,
           behavior, needs_initial_observation_with_detached_target);
   return MakeGarbageCollected<IntersectionObserver>(
-      *intersection_observer_delegate, nullptr, margin, thresholds, semantics,
-      delay, track_visibility, always_report_root_bounds, margin_target,
+      *intersection_observer_delegate, nullptr, margin,
+      /* scroll_margin */ Vector<Length>(), thresholds, semantics, delay,
+      track_visibility, always_report_root_bounds, margin_target,
       use_overflow_clip_edge);
 }
 
@@ -273,6 +350,7 @@ IntersectionObserver::IntersectionObserver(
     IntersectionObserverDelegate& delegate,
     Node* root,
     const Vector<Length>& margin,
+    const Vector<Length>& scroll_margin,
     const Vector<float>& thresholds,
     ThresholdInterpretation semantics,
     DOMHighResTimeStamp delay,
@@ -286,38 +364,20 @@ IntersectionObserver::IntersectionObserver(
       root_(root),
       thresholds_(thresholds),
       delay_(delay),
-      margin_(4, Length::Fixed(0)),
       margin_target_(margin_target),
       root_is_implicit_(root ? 0 : 1),
       track_visibility_(track_visibility),
       track_fraction_of_root_(semantics == kFractionOfRoot),
       always_report_root_bounds_(always_report_root_bounds),
       use_overflow_clip_edge_(use_overflow_clip_edge) {
-  switch (margin.size()) {
-    case 0:
-      break;
-    case 1:
-      margin_[0] = margin_[1] = margin_[2] = margin_[3] = margin[0];
-      break;
-    case 2:
-      margin_[0] = margin_[2] = margin[0];
-      margin_[1] = margin_[3] = margin[1];
-      break;
-    case 3:
-      margin_[0] = margin[0];
-      margin_[1] = margin_[3] = margin[1];
-      margin_[2] = margin[2];
-      break;
-    case 4:
-      margin_[0] = margin[0];
-      margin_[1] = margin[1];
-      margin_[2] = margin[2];
-      margin_[3] = margin[3];
-      break;
-    default:
-      NOTREACHED();
-      break;
+  margin_ = NormalizeMargins(margin);
+
+  scroll_margin_ = NormalizeMargins(scroll_margin);
+  if (std::all_of(scroll_margin_.begin(), scroll_margin_.end(),
+                  [](const auto& m) { return m.IsZero(); })) {
+    scroll_margin_.clear();
   }
+
   if (root) {
     if (root->IsDocumentNode()) {
       To<Document>(root)
@@ -433,30 +493,12 @@ HeapVector<Member<IntersectionObserverEntry>> IntersectionObserver::takeRecords(
   return entries;
 }
 
-static void AppendLength(StringBuilder& string_builder, const Length& length) {
-  string_builder.AppendNumber(length.IntValue());
-  if (length.IsPercent())
-    string_builder.Append('%');
-  else
-    string_builder.Append("px", 2);
+String IntersectionObserver::rootMargin() const {
+  return StringifyMargin(RootMargin());
 }
 
-String IntersectionObserver::rootMargin() const {
-  StringBuilder string_builder;
-  const auto& margin = RootMargin();
-  if (margin.empty()) {
-    string_builder.Append("0px 0px 0px 0px");
-  } else {
-    DCHECK_EQ(margin.size(), 4u);
-    AppendLength(string_builder, margin[0]);
-    string_builder.Append(' ');
-    AppendLength(string_builder, margin[1]);
-    string_builder.Append(' ');
-    AppendLength(string_builder, margin[2]);
-    string_builder.Append(' ');
-    AppendLength(string_builder, margin[3]);
-  }
-  return string_builder.ToString();
+String IntersectionObserver::scrollMargin() const {
+  return StringifyMargin(ScrollMargin());
 }
 
 DOMHighResTimeStamp IntersectionObserver::GetEffectiveDelay() const {
