diff --git a/third_party/blink/renderer/core/streams/readable_stream.idl b/third_party/blink/renderer/core/streams/readable_stream.idl
index d18fa3eb5ec5bf7b79c21adab4beaa379657c12c..43d6ac5422d3ff8259c8512297e1e1c448f8fe0b 100644
--- a/third_party/blink/renderer/core/streams/readable_stream.idl
+++ b/third_party/blink/renderer/core/streams/readable_stream.idl
@@ -12,6 +12,8 @@ enum ReadableStreamType { "bytes" };
 [
     Exposed=(Window,Worker,Worklet)
 ] interface ReadableStream {
+    [RuntimeEnabled=ReadableStreamAsyncIterable, HasAsyncIteratorReturnAlgorithm] async iterable<any>(optional ReadableStreamIteratorOptions options = {});
+
     [CallWith=ScriptState, RaisesException] constructor(optional any underlyingSource, optional any strategy);
     readonly attribute boolean locked;
 
