diff --git a/third_party/blink/renderer/core/css/selector_checker.h b/third_party/blink/renderer/core/css/selector_checker.h
index ba4ebf8f0b5f59d41cd502278eab60f5bd111eb8..fa1359a98a10a92812d5ee57baf53efa011db51a 100644
--- a/third_party/blink/renderer/core/css/selector_checker.h
+++ b/third_party/blink/renderer/core/css/selector_checker.h
@@ -98,6 +98,7 @@ class CORE_EXPORT SelectorChecker {
       : scrollbar_(style_request.scrollbar),
         part_names_(part_names),
         pseudo_argument_(style_request.pseudo_argument),
+        pseudo_ident_list_(style_request.pseudo_ident_list),
         scrollbar_part_(style_request.scrollbar_part),
         mode_(mode),
         is_ua_rule_(is_ua_rule) {}
@@ -331,6 +332,7 @@ class CORE_EXPORT SelectorChecker {
   CustomScrollbar* scrollbar_;
   PartNames* part_names_;
   const String pseudo_argument_;
+  const Vector<AtomicString> pseudo_ident_list_;
   ScrollbarPart scrollbar_part_;
   Mode mode_;
   bool is_ua_rule_;
