diff --git a/third_party/blink/renderer/core/url_pattern/url_pattern.idl b/third_party/blink/renderer/core/url_pattern/url_pattern.idl
index 84b25dda37e84db2b9d9a5914d28b136c096a937..7e055263ab363df420b106176f6a869d4e0b5515 100644
--- a/third_party/blink/renderer/core/url_pattern/url_pattern.idl
+++ b/third_party/blink/renderer/core/url_pattern/url_pattern.idl
@@ -32,6 +32,8 @@ enum URLPatternComponent { "protocol", "username", "password", "hostname",
   readonly attribute USVString search;
   readonly attribute USVString hash;
 
+  [RuntimeEnabled=URLPatternHasRegExpGroups] readonly attribute boolean hasRegExpGroups;
+
   [RuntimeEnabled=URLPatternCompareComponent, Measure]
   static short compareComponent(URLPatternComponent component,
                                 URLPattern left, URLPattern right);
