diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_mac.cc b/third_party/blink/renderer/platform/theme/web_theme_engine_mac.cc
index b91d226cc158fb9afb5a3d348873ad580d924b40..d55b78ca0f7d36298bbd75b48f2ad758e3b8f8e8 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_mac.cc
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_mac.cc
@@ -54,6 +54,10 @@ void WebThemeEngineMac::PaintMacScrollBarParts(
       extra_params->scrollbar_extra.is_overlay;
   native_theme_extra_params.scrollbar_extra.scale_from_dip =
       extra_params->scrollbar_extra.scale_from_dip;
+  native_theme_extra_params.scrollbar_extra.track_color =
+      extra_params->scrollbar_extra.track_color;
+  native_theme_extra_params.scrollbar_extra.thumb_color =
+      extra_params->scrollbar_extra.thumb_color;
   switch (extra_params->scrollbar_extra.orientation) {
     case WebThemeEngine::kVerticalOnRight:
       native_theme_extra_params.scrollbar_extra.orientation =
