diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index e77f3afb2cc9e94b35d7456daa7f7e7554d56bd8..8a5e54022c54af61a567aaec05f0122103d476f0 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -621,6 +621,16 @@
       status: "experimental",
       base_feature: "none",
     },
+    {
+      // https://drafts.fxtf.org/css-masking/#typedef-geometry-box
+      name: "ClipPathGeometryBox",
+      status: "stable",
+    },
+    {
+      // https://www.w3.org/TR/css-shapes/#supported-basic-shapes
+      name: "ClipPathXYWHAndRect",
+      status: "stable",
+    },
     {
       name: "CloseWatcher",
     },
@@ -796,6 +806,10 @@
       name: "CSSCalcSimplificationAndSerialization",
       base_feature: "none",
     },
+    {
+      name: "CSSCapFontUnits",
+      status: "stable",
+    },
     {
       // Support case-sensitive attribute selector modifier
       // https://drafts.csswg.org/selectors-4/#attribute-case
@@ -815,7 +829,8 @@
     },
     {
       name: "CSSContainIntrinsicSizeAutoNone",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     // Ablation study to try to measure very roughly memory savings from not saving
     // pretokenized strings in CSS custom properties.
@@ -909,7 +924,8 @@
     },
     {
       name: "CSSInitialPseudo",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -923,7 +939,7 @@
     },
     {
       name: "CSSLogical",
-      status: "experimental",
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -976,6 +992,13 @@
       // Support nested selectors that start with idents.
       name: "CSSNestingIdent",
     },
+    {
+      // Support for typed om numeric factory methods for various units that
+      // were otherwise shipped: rex, rch, ic, ric, lh, rlh, x (cap and rcap
+      // covered by separate runtime flag).
+      name: "CSSNumericFactoryCompleteness",
+      status: "stable",
+    },
     {
       name: "CSSObjectViewBox",
       status: "stable",
@@ -1152,7 +1175,8 @@
     // `text-wrap: pretty`. crbug.com/1432798
     {
       name: "CSSTextWrapPretty",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       depends_on: ["CSSTextWrap", "CSSWhiteSpaceShorthand"],
     },
     // Support for CSS Toggles, https://tabatkins.github.io/css-toggle/
@@ -1170,12 +1194,21 @@
       //
       // See: https://github.com/w3c/csswg-drafts/issues/8189
       name: "CSSTopLayerForTransitions",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
+    {
+      // Support for the 'border-box', 'content-box' and 'stroke-box' keywords
+      // for the 'transform-box' property.
+      // FlyWeb: as shipped in Chromium 118 (engine level 118).
+      name: "CSSTransformBoxAdditionalKeywords",
+      status: "stable",
+    },
     {
       name: "CSSTransitionDiscrete",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     {
       // A kill switch for the fix for part of https://crbug.com/1180387 .
@@ -1811,8 +1844,7 @@
     },
     {
       name: "FontVariantPosition",
-      status: "experimental",
-      base_feature: "none",
+      status: "stable",
     },
     {
       // TODO(crbug.com/1231644): This flag is being kept (even though the
@@ -1942,6 +1974,11 @@
       },
       base_feature: "none",
     },
+    {
+      name: "HasUAVisualTransition",
+      status: "stable",
+      base_feature: "none",
+    },
     // HighlightAPI's custom highlights use HighlightInheritance's behavior even
     // if the HighlightInheritance feature is not enabled.
     {
@@ -1996,6 +2033,11 @@
       name: "HTMLPopoverHint",
       status: "experimental",
     },
+    {
+      // FlyWeb: <search>, Chrome 118 (engine level 118).
+      name: "HTMLSearchElement",
+      status: "stable",
+    },
     {
       name: "HTMLSelectMenuElement",
       status: "experimental",
@@ -2137,7 +2179,8 @@
     },
     {
       name: "LayoutNGSubgrid",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -3449,10 +3492,19 @@
       name: "StylusHandwriting",
       base_feature: "none",
     },
+    {
+      // FlyWeb: stable, as in Chromium 118 (engine level 118).
+      name: "SvgCrossOriginAttribute",
+      status: "stable",
+    },
     {
       name: "SvgMapLocalToAncestorFix",
       status: "stable",
     },
+    {
+      name: "SvgNoPixelSnappingScaleAdjustment",
+      status: "stable",
+    },
     {
       name: "SvgRasterOptimizations",
       status: "experimental",
@@ -3513,6 +3565,10 @@
       status: {"Android": "stable"},
       base_feature: "none",
     },
+    {
+      name: "TextMetricsBaselines",
+      status: "stable",
+    },
     {
       name: "ThirdPartyCookiesUserBypass",
       status: "experimental",
@@ -3643,6 +3699,10 @@
       status: "experimental",
       base_feature: "none",
     },
+    {
+      name: "URLSearchParamsHasAndDeleteMultipleArgs",
+      status: "stable",
+    },
     {
       // Makes the global root scrollbars follow the preferred color scheme
       // by default. Please see more details: crbug.com/1423087.
@@ -3657,6 +3717,11 @@
       name: "UserAgentClientHint",
       status: "stable",
     },
+    {
+      // https://chromestatus.com/feature/5132477781245952
+      name: "UserValidUserInvalid",
+      status: "stable",
+    },
     {
       name: "V8IdleTasks",
       base_feature: "none",
@@ -4167,6 +4232,12 @@
       name: "WindowPlacementFullscreenOnScreensChange",
       status: "experimental",
     },
+    {
+      // Update xywh()/rect() computed value to use inset() per the spec:
+      // https://drafts.csswg.org/css-shapes-1/#basic-shape-computed-values
+      name: "XYWHAndRectComputedValue",
+      status: "stable",
+    },
     {
       // If enabled, the `getDisplayMedia()` family of APIs will ask for NV12
       // frames, which should trigger a zero-copy path in the tab capture code.
