diff --git a/third_party/blink/renderer/core/style/content_data.h b/third_party/blink/renderer/core/style/content_data.h
index 4619d46d9110bf51cbc0af69b342267f1c9c1c3b..ff9ad9c5b32fcb7401bd8bd1b00e256fad4068ad 100644
--- a/third_party/blink/renderer/core/style/content_data.h
+++ b/third_party/blink/renderer/core/style/content_data.h
@@ -32,6 +32,7 @@
 #include "third_party/blink/renderer/platform/heap/member.h"
 #include "third_party/blink/renderer/platform/wtf/casting.h"
 #include "third_party/blink/renderer/platform/wtf/text/atomic_string.h"
+#include "third_party/blink/renderer/platform/wtf/text/string_builder.h"
 
 namespace blink {
 
@@ -158,6 +159,14 @@ class AltTextContentData final : public ContentData {
   explicit AltTextContentData(const String& text) : text_(text) {}
 
   String GetText() const { return text_; }
+  String ConcatenateAltText() const {
+    StringBuilder alt_text;
+    for (const ContentData* content_data = this; content_data;
+         content_data = content_data->Next()) {
+      alt_text.Append(To<AltTextContentData>(content_data)->GetText());
+    }
+    return alt_text.ToString();
+  }
   void SetText(const String& text) { text_ = text; }
 
   bool IsAltText() const override { return true; }
