diff --git a/third_party/blink/public/platform/web_theme_engine.h b/third_party/blink/public/platform/web_theme_engine.h
index 97f1dbf351c6251907db86f69b062721db92d94c..07252442d0d8fcffbf1823e78a0328eb5f88c1ac 100644
--- a/third_party/blink/public/platform/web_theme_engine.h
+++ b/third_party/blink/public/platform/web_theme_engine.h
@@ -205,6 +205,11 @@ class WebThemeEngine {
     mojom::ColorScheme scrollbar_theme;
     ScrollbarOrientation orientation;
     float scale_from_dip;
+    absl::optional<SkColor> thumb_color;
+    absl::optional<SkColor> track_color;
+
+    ScrollbarExtraParams()
+        : thumb_color(absl::nullopt), track_color(absl::nullopt) {}
   };
 #endif
 
@@ -220,6 +225,8 @@ class WebThemeEngine {
     ScrollbarButtonExtraParams scrollbar_button;
 #if BUILDFLAG(IS_MAC)
     ScrollbarExtraParams scrollbar_extra;
+
+    ExtraParams() : scrollbar_extra() {}
 #endif
   };
 
