diff --git a/third_party/blink/renderer/platform/geometry/calculation_value.cc b/third_party/blink/renderer/platform/geometry/calculation_value.cc
index 6b84f1567510b45ef55e54193784bf67b1923520..1dd1bd63dc06614a71594c05f73a5efc406a60e4 100644
--- a/third_party/blink/renderer/platform/geometry/calculation_value.cc
+++ b/third_party/blink/renderer/platform/geometry/calculation_value.cc
@@ -116,6 +116,15 @@ CalculationValue::SubtractFromOneHundredPercent() const {
                           Length::ValueRange::kAll);
 }
 
+scoped_refptr<const CalculationValue> CalculationValue::Add(
+    const CalculationValue& other) const {
+  CHECK_EQ(GetValueRange(), other.GetValueRange());
+  auto result_expression = CalculationExpressionOperationNode::CreateSimplified(
+      {GetOrCreateExpression(), other.GetOrCreateExpression()},
+      CalculationOperator::kAdd);
+  return CreateSimplified(result_expression, GetValueRange());
+}
+
 scoped_refptr<const CalculationValue> CalculationValue::Zoom(
     double factor) const {
   if (!IsExpression()) {
