diff --git a/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.h b/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.h
index 579fcd6d45a3188525fa6b35313e76a29924121a..0ad43c74375114824a4811a171b043a3c5ea914e 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.h
+++ b/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.h
@@ -78,6 +78,7 @@ class CORE_EXPORT NGBoxFragmentPainter : public BoxPainterBase {
  protected:
   NGPhysicalBoxStrut ComputeBorders() const override;
   NGPhysicalBoxStrut ComputePadding() const override;
+  NGPhysicalBoxStrut ComputeMargins() const override;
   BoxPainterBase::FillLayerInfo GetFillLayerInfo(
       const Color&,
       const FillLayer&,
