diff --git a/third_party/blink/renderer/core/css/rule_set.cc b/third_party/blink/renderer/core/css/rule_set.cc
index 0f10a1f3aa6471fcae9e14cc5b59121cc3477a33..d277dee9d47301e9c39520299aaf444114c41236 100644
--- a/third_party/blink/renderer/core/css/rule_set.cc
+++ b/third_party/blink/renderer/core/css/rule_set.cc
@@ -736,6 +736,9 @@ void RuleSet::AddRulesFromSheet(StyleSheetContents* sheet,
       sheet->ImportRules();
   for (unsigned i = 0; i < import_rules.size(); ++i) {
     StyleRuleImport* import_rule = import_rules[i].Get();
+    if (!import_rule->IsSupported()) {
+      continue;
+    }
     if (!MatchMediaForAddRules(medium, import_rule->MediaQueries())) {
       continue;
     }
