diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_clipper.h b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_clipper.h
index 1d807678b32fc3bba4b8503827005c93d57d6e23..29ce89e7b2c8eb74c0ee3d6c09b668e96914837c 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_clipper.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_clipper.h
@@ -49,7 +49,9 @@ class LayoutSVGResourceClipper final : public LayoutSVGResourceContainer {
     return kResourceType;
   }
 
-  bool HitTestClipContent(const gfx::RectF&, const HitTestLocation&) const;
+  bool HitTestClipContent(const gfx::RectF& reference_box,
+                          const LayoutObject& reference_box_object,
+                          const HitTestLocation&) const;
 
   SVGUnitTypes::SVGUnitType ClipPathUnits() const;
   AffineTransform CalculateClipTransform(const gfx::RectF& reference_box) const;
