diff --git a/third_party/blink/renderer/core/navigation_api/navigate_event.cc b/third_party/blink/renderer/core/navigation_api/navigate_event.cc
index 6f5c3f42e7463da8670cfb2a693772ca7d076f4f..ab2ab5c4886998dd4249e55dc4a6d095b4a38813 100644
--- a/third_party/blink/renderer/core/navigation_api/navigate_event.cc
+++ b/third_party/blink/renderer/core/navigation_api/navigate_event.cc
@@ -65,7 +65,8 @@ NavigateEvent::NavigateEvent(ExecutionContext* context,
       info_(init->hasInfo()
                 ? init->info()
                 : ScriptValue(context->GetIsolate(),
-                              v8::Undefined(context->GetIsolate()))) {
+                              v8::Undefined(context->GetIsolate()))),
+      has_ua_visual_transition_(init->hasUAVisualTransition()) {
   CHECK(IsA<LocalDOMWindow>(context));
 }
 
