diff --git a/third_party/blink/renderer/core/css/style_change_reason.h b/third_party/blink/renderer/core/css/style_change_reason.h
index 82461b49db74050528ebf261b584a4abc695ea71..7e7735b38d16fd2f58550424913545cef0d1f5fb 100644
--- a/third_party/blink/renderer/core/css/style_change_reason.h
+++ b/third_party/blink/renderer/core/css/style_change_reason.h
@@ -58,6 +58,8 @@ typedef const char StyleChangeReasonString[];
 
 namespace style_change_extra_data {
 extern const AtomicString& g_active;
+extern const AtomicString& g_active_view_transition;
+extern const AtomicString& g_active_view_transition_type;
 extern const AtomicString& g_disabled;
 extern const AtomicString& g_drag;
 extern const AtomicString& g_focus;
