diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_inline.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_inline.cc
index f4be270aafd00ff0c1da6dca3936bde8640e5ec5..1024acc1036891a34c63e9d31d87d1539347ff2c 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_inline.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_inline.cc
@@ -99,7 +99,7 @@ gfx::RectF LayoutSVGInline::ObjectBoundingBox() const {
   return bounds;
 }
 
-gfx::RectF LayoutSVGInline::StrokeBoundingBox() const {
+gfx::RectF LayoutSVGInline::DecoratedBoundingBox() const {
   NOT_DESTROYED();
   if (!IsObjectBoundingBoxValid())
     return gfx::RectF();
