diff --git a/third_party/blink/renderer/core/style/build.gni b/third_party/blink/renderer/core/style/build.gni
index 31b5deae1d34f57132dcd66ec098d1b145ec6246..c7776f8ff9b29e469808750edf7148cb1be65e13 100644
--- a/third_party/blink/renderer/core/style/build.gni
+++ b/third_party/blink/renderer/core/style/build.gni
@@ -36,6 +36,7 @@ blink_core_sources_style = [
   "filter_operations.cc",
   "filter_operations.h",
   "font_size_style.h",
+  "geometry_box_clip_path_operation.h",
   "grid_area.h",
   "grid_enums.h",
   "grid_length.h",
@@ -51,6 +52,7 @@ blink_core_sources_style = [
   "nine_piece_image.h",
   "ordered_named_grid_lines.h",
   "paint_images.h",
+  "paint_order_array.h",
   "quad_length_value.h",
   "reference_clip_path_operation.cc",
   "reference_clip_path_operation.h",
@@ -111,6 +113,8 @@ blink_core_sources_style = [
   "style_scrollbar_color.cc",
   "style_scrollbar_color.h",
   "style_self_alignment_data.h",
+  "style_svg_mask_reference_image.cc",
+  "style_svg_mask_reference_image.h",
   "style_svg_resource.cc",
   "style_svg_resource.h",
   "style_timeline.h",
