diff --git a/third_party/blink/renderer/core/navigation_api/navigation_activation.idl b/third_party/blink/renderer/core/navigation_api/navigation_activation.idl
new file mode 100644
index 0000000000000000000000000000000000000000..73230142f7b8a395d533b855a690729e4d4e0837
--- /dev/null
+++ b/third_party/blink/renderer/core/navigation_api/navigation_activation.idl
@@ -0,0 +1,13 @@
+// Copyright 2023 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// https://wicg.github.io/navigation-api/
+[
+  Exposed=Window,
+  RuntimeEnabled=NavigationActivation
+] interface NavigationActivation {
+  readonly attribute NavigationHistoryEntry? entry;
+  readonly attribute NavigationHistoryEntry? from;
+  readonly attribute NavigationType navigationType;
+};
