diff --git a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_track_collection.h b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_track_collection.h
index 17405e90a3bddc06c6a939fb9175dc2c65d5fba1..802799a7f62dd6af0a94c12f0b7b5b358e08f524 100644
--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_track_collection.h
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_track_collection.h
@@ -423,9 +423,8 @@ class CORE_EXPORT NGGridSizingTrackCollection final
   LayoutUnit TotalTrackSize() const;
 
   void BuildSets(const ComputedStyle& grid_style,
-                 LayoutUnit grid_available_size);
-  void InitializeSets(LayoutUnit grid_available_size = kIndefiniteSize,
-                      LayoutUnit gutter_size = LayoutUnit());
+                 LayoutUnit grid_available_size,
+                 LayoutUnit gutter_size);
   void SetIndefiniteGrowthLimitsToBaseSize();
 
   // Caches the geometry of definite sets; this is useful when building the sets
@@ -447,10 +446,11 @@ class CORE_EXPORT NGGridSizingTrackCollection final
   friend class NGGridLayoutAlgorithmTest;
   friend class NGGridTrackCollectionTest;
 
-  // This private version of |BuildSets| is directly used in testing.
+  // These methods are internal implementations also used in testing.
   void BuildSets(const NGGridTrackList& explicit_track_list,
                  const NGGridTrackList& implicit_track_list,
                  bool is_available_size_indefinite = true);
+  void InitializeSets(LayoutUnit grid_available_size = kIndefiniteSize);
 
   wtf_size_t non_collapsed_track_count_{0};
 
