diff --git a/third_party/blink/renderer/core/layout/ng/ng_block_node.cc b/third_party/blink/renderer/core/layout/ng/ng_block_node.cc
index 368a745eda9491456cdb14281e1d2eddf15f942d..015da3958005092707cab1ba31b06e39a8012c89 100644
--- a/third_party/blink/renderer/core/layout/ng/ng_block_node.cc
+++ b/third_party/blink/renderer/core/layout/ng/ng_block_node.cc
@@ -79,6 +79,7 @@
 #include "third_party/blink/renderer/core/mathml/mathml_under_over_element.h"
 #include "third_party/blink/renderer/core/paint/paint_layer.h"
 #include "third_party/blink/renderer/core/paint/paint_layer_scrollable_area.h"
+#include "third_party/blink/renderer/core/paint/transform_utils.h"
 #include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 #include "third_party/blink/renderer/platform/text/writing_mode.h"
 #include "third_party/blink/renderer/platform/wtf/allocator/allocator.h"
@@ -316,14 +317,18 @@ absl::optional<LayoutUnit> ContentMinimumInlineSize(
   else
     inline_size += border_padding.InlineSum();
 
-  if (block_node.IsTextControl())
+  const bool apply_form_sizing = style.ApplyControlFixedSize(node);
+  if (block_node.IsTextControl() && apply_form_sizing) {
     return inline_size;
-  if (IsA<HTMLSelectElement>(node))
+  }
+  if (IsA<HTMLSelectElement>(node) && apply_form_sizing) {
     return inline_size;
+  }
   if (const auto* input_element = DynamicTo<HTMLInputElement>(node)) {
     const AtomicString& type = input_element->type();
-    if (type == input_type_names::kFile)
+    if (type == input_type_names::kFile && apply_form_sizing) {
       return inline_size;
+    }
     if (type == input_type_names::kRange)
       return inline_size;
   }
@@ -1719,8 +1724,9 @@ absl::optional<gfx::Transform> NGBlockNode::GetTransformForChildFragment(
     // us. Calculate it now.
     fragment_transform.emplace();
     fragment_transform->MakeIdentity();
+    const PhysicalRect reference_box = ComputeReferenceBox(child_fragment);
     child_fragment.Style().ApplyTransform(
-        *fragment_transform, box_, child_fragment.Size(),
+        *fragment_transform, box_, reference_box,
         ComputedStyle::kIncludeTransformOperations,
         ComputedStyle::kIncludeTransformOrigin,
         ComputedStyle::kIncludeMotionPath,
