diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_shape.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_shape.cc
index 006c06a3a69974b24c48081490a86c052b992a76..ff708ce0b17cb4e3950dedc4828fcc747fd34f13 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_shape.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_shape.cc
@@ -35,6 +35,7 @@
 #include "third_party/blink/renderer/core/layout/svg/svg_resources.h"
 #include "third_party/blink/renderer/core/layout/svg/transform_helper.h"
 #include "third_party/blink/renderer/core/layout/svg/transformed_hit_test_location.h"
+#include "third_party/blink/renderer/core/paint/clip_path_clipper.h"
 #include "third_party/blink/renderer/core/paint/svg_shape_painter.h"
 #include "third_party/blink/renderer/core/svg/svg_geometry_element.h"
 #include "third_party/blink/renderer/core/svg/svg_length_context.h"
@@ -79,18 +80,27 @@ void LayoutSVGShape::StyleDidChange(StyleDifference diff,
 
   TransformHelper::UpdateOffsetPath(*GetElement(), old_style);
   transform_uses_reference_box_ =
-      TransformHelper::DependsOnReferenceBox(StyleRef());
+      TransformHelper::UpdateReferenceBoxDependency(*this);
   SVGResources::UpdatePaints(*this, old_style, StyleRef());
 
-  // Most of the stroke attributes (caps, joins, miters, width, etc.) will cause
-  // a re-layout which will clear the stroke-path cache; however, there are a
-  // couple of additional properties that *won't* cause a layout, but are
-  // significant enough to require invalidating the cache.
-  if (!diff.NeedsFullLayout() && old_style && stroke_path_cache_) {
+  if (old_style) {
     const ComputedStyle& style = StyleRef();
-    if (old_style->StrokeDashOffset() != style.StrokeDashOffset() ||
-        *old_style->StrokeDashArray() != *style.StrokeDashArray()) {
-      stroke_path_cache_.reset();
+    // Most of the stroke attributes (caps, joins, miters, width, etc.) will
+    // cause a re-layout which will clear the stroke-path cache; however, there
+    // are a couple of additional properties that *won't* cause a layout, but
+    // are significant enough to require invalidating the cache.
+    if (!diff.NeedsFullLayout() && stroke_path_cache_) {
+      if (old_style->StrokeDashOffset() != style.StrokeDashOffset() ||
+          *old_style->StrokeDashArray() != *style.StrokeDashArray()) {
+        stroke_path_cache_.reset();
+      }
+    }
+
+    if (transform_uses_reference_box_ && !needs_transform_update_) {
+      if (TransformHelper::CheckReferenceBoxDependencies(*old_style, style)) {
+        SetNeedsTransformUpdate();
+        SetNeedsPaintPropertyUpdate();
+      }
     }
   }
 
@@ -136,12 +146,13 @@ void LayoutSVGShape::UpdateShapeFromElement() {
 
   if (HasNonScalingStroke()) {
     // NonScalingStrokeTransform may depend on LocalTransform which in turn may
-    // depend on ObjectBoundingBox, thus we need to call them in this order.
-    local_transform_ = CalculateLocalTransform();
+    // depend on the reference box, thus we need to call them in this order.
+    local_transform_ =
+        TransformHelper::ComputeTransformIncludingMotion(*GetElement());
     UpdateNonScalingStrokeData();
   }
 
-  stroke_bounding_box_ = CalculateStrokeBoundingBox();
+  decorated_bounding_box_ = CalculateStrokeBoundingBox();
 }
 
 namespace {
@@ -188,10 +199,37 @@ gfx::RectF LayoutSVGShape::ApproximateStrokeBoundingBox(
 gfx::RectF LayoutSVGShape::HitTestStrokeBoundingBox() const {
   NOT_DESTROYED();
   if (StyleRef().HasStroke())
-    return stroke_bounding_box_;
+    return decorated_bounding_box_;
   return ApproximateStrokeBoundingBox(fill_bounding_box_);
 }
 
+gfx::RectF LayoutSVGShape::StrokeBoundingBox() const {
+  NOT_DESTROYED();
+  if (!StyleRef().HasStroke()) {
+    return fill_bounding_box_;
+  }
+  // If no Path object has been created for the shape, assume that it is
+  // 'simple' and thus the approximation is accurate.
+  if (!HasPath()) {
+    DCHECK_EQ(geometry_class_, kSimple);
+    return decorated_bounding_box_;
+  }
+  StrokeData stroke_data;
+  SVGLayoutSupport::ApplyStrokeStyleToStrokeData(stroke_data, StyleRef(), *this,
+                                                 DashScaleFactor());
+  // Reset the dash pattern.
+  //
+  // "...set box to be the union of box and the tightest rectangle in
+  // coordinate system space that contains the stroke shape of the element,
+  // with the assumption that the element has no dash pattern."
+  //
+  // (https://www.w3.org/TR/SVG2/coords.html#TermStrokeBoundingBox)
+  DashArray dashes;
+  stroke_data.SetLineDash(dashes, 0);
+  const gfx::RectF stroke_bounds = GetPath().StrokeBoundingRect(stroke_data);
+  return gfx::UnionRects(fill_bounding_box_, stroke_bounds);
+}
+
 bool LayoutSVGShape::ShapeDependentStrokeContains(
     const HitTestLocation& location) {
   NOT_DESTROYED();
@@ -274,8 +312,10 @@ bool LayoutSVGShape::StrokeContains(const HitTestLocation& location,
     return false;
 
   if (requires_stroke) {
-    if (!StrokeBoundingBox().InclusiveContains(location.TransformedPoint()))
+    if (!DecoratedBoundingBox().InclusiveContains(
+            location.TransformedPoint())) {
       return false;
+    }
 
     if (!HasPaintServer(*this, StyleRef().StrokePaint()))
       return false;
@@ -328,7 +368,8 @@ void LayoutSVGShape::UpdateLayout() {
   }
 
   if (needs_transform_update_) {
-    local_transform_ = CalculateLocalTransform();
+    local_transform_ =
+        TransformHelper::ComputeTransformIncludingMotion(*GetElement());
     needs_transform_update_ = false;
     update_parent_boundaries = true;
   }
@@ -412,9 +453,9 @@ bool LayoutSVGShape::NodeAtPoint(HitTestResult& result,
                                             LocalToSVGParentTransform());
   if (!local_location)
     return false;
-  if (!SVGLayoutSupport::IntersectsClipPath(*this, fill_bounding_box_,
-                                            *local_location))
+  if (HasClipPath() && !ClipPathClipper::HitTest(*this, *local_location)) {
     return false;
+  }
 
   if (HitTestShape(result.GetHitTestRequest(), *local_location, hit_rules)) {
     UpdateHitTestResult(result, PhysicalOffset::FromPointFRound(
