diff --git a/third_party/blink/renderer/bindings/idl_in_core.gni b/third_party/blink/renderer/bindings/idl_in_core.gni
index 08410ec268174507329fe03ccc3c151a03886bd2..55b54b886b963efe5efec08870ed5f2c67522a3c 100644
--- a/third_party/blink/renderer/bindings/idl_in_core.gni
+++ b/third_party/blink/renderer/bindings/idl_in_core.gni
@@ -520,6 +520,7 @@ static_idl_files_in_core = get_path_info(
           "//third_party/blink/renderer/core/navigation_api/navigate_event.idl",
           "//third_party/blink/renderer/core/navigation_api/navigate_event_init.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation.idl",
+  "//third_party/blink/renderer/core/navigation_api/navigation_activation.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_current_entry_change_event.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_current_entry_change_event_init.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_destination.idl",
