diff --git a/third_party/blink/renderer/core/css/css_properties.json5 b/third_party/blink/renderer/core/css/css_properties.json5
index 871edc0924403a04bf16f4d5602f421d3882473c..c62afe37fd750bacc13eb43ee4aa3320c08d7a0d 100644
--- a/third_party/blink/renderer/core/css/css_properties.json5
+++ b/third_party/blink/renderer/core/css/css_properties.json5
@@ -991,6 +991,19 @@
       valid_for_marker: true,
       is_animation_property: true,
     },
+    {
+      name: "transition-behavior",
+      property_methods: ["ParseSingleValue", "CSSValueFromComputedStyleInternal", "InitialValue"],
+      typedom_types: ["Keyword"],
+      separator: ",",
+      style_builder_template: "transition",
+      style_builder_template_args: {
+        attribute: "Behavior",
+      },
+      valid_for_marker: true,
+      is_animation_property: true,
+      runtime_flag: "CSSTransitionDiscrete",
+    },
     {
       name: "transition-timing-function",
       property_methods: ["ParseSingleValue", "CSSValueFromComputedStyleInternal", "InitialValue"],
@@ -7387,7 +7400,8 @@
       name: "transition",
       longhands: [
         "transition-property", "transition-duration",
-        "transition-timing-function", "transition-delay"
+        "transition-timing-function", "transition-delay",
+        "transition-behavior",
       ],
       property_methods: ["ParseShorthand", "CSSValueFromComputedStyleInternal"],
     },
