diff --git a/third_party/blink/renderer/core/frame/local_frame_mojo_handler.h b/third_party/blink/renderer/core/frame/local_frame_mojo_handler.h
index 348b5a8e47d6f286713d2021f218781c3eb64394..1d7bd131c1802551bec504ee97b7df691b44573d 100644
--- a/third_party/blink/renderer/core/frame/local_frame_mojo_handler.h
+++ b/third_party/blink/renderer/core/frame/local_frame_mojo_handler.h
@@ -194,7 +194,8 @@ class LocalFrameMojoHandler
   void GetOpenGraphMetadata(GetOpenGraphMetadataCallback callback) final;
 
   void SetNavigationApiHistoryEntriesForRestore(
-      mojom::blink::NavigationApiHistoryEntryArraysPtr) final;
+      mojom::blink::NavigationApiHistoryEntryArraysPtr,
+      mojom::blink::NavigationApiEntryRestoreReason) final;
   void NotifyNavigationApiOfDisposedEntries(
       const WTF::Vector<WTF::String>&) final;
   void TraverseCancelled(const String& navigation_api_key,
