diff --git a/third_party/blink/renderer/core/css/properties/shorthands/shorthands_custom.cc b/third_party/blink/renderer/core/css/properties/shorthands/shorthands_custom.cc
index 420cb4301e98f574101ceb2750a4f60d09619430..c9e7f94f11649e0fc44c31fae2c3e2bed7a247ab 100644
--- a/third_party/blink/renderer/core/css/properties/shorthands/shorthands_custom.cc
+++ b/third_party/blink/renderer/core/css/properties/shorthands/shorthands_custom.cc
@@ -3442,6 +3442,12 @@ CSSValue* ConsumeTransitionValue(CSSPropertyID property,
       return css_parsing_utils::ConsumeTransitionProperty(range, context);
     case CSSPropertyID::kTransitionTimingFunction:
       return css_parsing_utils::ConsumeAnimationTimingFunction(range, context);
+    case CSSPropertyID::kTransitionBehavior:
+      if (css_parsing_utils::IsValidTransitionBehavior(range.Peek().Id())) {
+        return CSSIdentifierValue::Create(
+            range.ConsumeIncludingWhitespace().Id());
+      }
+      return nullptr;
     default:
       NOTREACHED();
       return nullptr;
@@ -3508,6 +3514,11 @@ const CSSValue* Transition::CSSValueFromComputedStyleInternal(
                                      i)));
       list->Append(*ComputedStyleUtils::ValueForAnimationDelayStart(
           CSSTimingData::GetRepeated(transition_data->DelayStartList(), i)));
+      if (CSSTimingData::GetRepeated(transition_data->BehaviorList(), i) !=
+          CSSTransitionData::InitialBehavior()) {
+        list->Append(*ComputedStyleUtils::CreateTransitionBehaviorValue(
+            transition_data->BehaviorList()[i]));
+      }
       transitions_list->Append(*list);
     }
     return transitions_list;
