diff --git a/third_party/blink/renderer/core/css/parser/css_proto_converter.cc b/third_party/blink/renderer/core/css/parser/css_proto_converter.cc
index 06e68bf48bd34c4d767d1bd65cdd071e9293fb6a..aab7caf1644809b7b039a292d4f5cd06830c727a 100644
--- a/third_party/blink/renderer/core/css/parser/css_proto_converter.cc
+++ b/third_party/blink/renderer/core/css/parser/css_proto_converter.cc
@@ -118,6 +118,8 @@ const std::string Converter::kPseudoLookupTable[] = {
     "single-button",
     "start",
     "target",
+    "user-invalid",
+    "user-valid",
     "valid",
     "vertical",
     "visited",
