diff --git a/third_party/blink/renderer/core/intersection_observer/intersection_observer_init.idl b/third_party/blink/renderer/core/intersection_observer/intersection_observer_init.idl
index 9321162ef8df0d3152bbfcfff5453b78ee99eeba..6b25f20263c9af35bf1b7f500bbfb7dc68264eb2 100644
--- a/third_party/blink/renderer/core/intersection_observer/intersection_observer_init.idl
+++ b/third_party/blink/renderer/core/intersection_observer/intersection_observer_init.idl
@@ -7,6 +7,7 @@
 dictionary IntersectionObserverInit {
     (Element or Document)? root = null;
     DOMString rootMargin = "0px";
+    DOMString scrollMargin = "0px";
     (double or sequence<double>) threshold = 0;
     DOMHighResTimeStamp delay = 0;
     boolean trackVisibility = false;
