diff --git a/third_party/blink/renderer/modules/geolocation/geolocation_position.idl b/third_party/blink/renderer/modules/geolocation/geolocation_position.idl
index e6daf9b980483ef1c0f3767a527c2f0ab5670fd9..9a07d7125b76d67a1f82083cdd79a993b470f478 100644
--- a/third_party/blink/renderer/modules/geolocation/geolocation_position.idl
+++ b/third_party/blink/renderer/modules/geolocation/geolocation_position.idl
@@ -31,4 +31,5 @@
 ] interface GeolocationPosition {
     readonly attribute GeolocationCoordinates coords;
     readonly attribute EpochTimeStamp timestamp;
+    [CallWith=ScriptState] object toJSON();
 };
