diff --git a/third_party/blink/renderer/core/html/html_template_element.idl b/third_party/blink/renderer/core/html/html_template_element.idl
index 8f5ca1474c7182d1c80bca161d907aa00b26bbd5..04ecdef415304877813e226b68e71cdd1254e340 100644
--- a/third_party/blink/renderer/core/html/html_template_element.idl
+++ b/third_party/blink/renderer/core/html/html_template_element.idl
@@ -53,4 +53,6 @@
     // This is the "new" declarative Shadow DOM IDL attribute, which reflects
     // the 'shadowrootmode' content attribute.
     [RuntimeEnabled=StreamingDeclarativeShadowDOM,Reflect,ReflectOnly=("open","closed")] attribute DOMString? shadowRootMode;
+    [CEReactions,Reflect,RuntimeEnabled=TemplateDelegatesFocusReflection] attribute boolean shadowRootDelegatesFocus;
+    [CEReactions,Reflect,RuntimeEnabled=ShadowRootClonable] attribute boolean shadowRootClonable;
 };
