diff --git a/third_party/blink/renderer/core/css/css_math_expression_node.cc b/third_party/blink/renderer/core/css/css_math_expression_node.cc
index 00af81fe62eb0d2471a491d4821388fc3f38b92a..7c4d9a1708488b4c28594ee511255ac2d810babd 100644
--- a/third_party/blink/renderer/core/css/css_math_expression_node.cc
+++ b/third_party/blink/renderer/core/css/css_math_expression_node.cc
@@ -85,6 +85,10 @@ static CalculationResultCategory UnitCategory(
     case CSSPrimitiveValue::UnitType::kIcs:
     case CSSPrimitiveValue::UnitType::kLhs:
       return kCalcLength;
+    case CSSPrimitiveValue::UnitType::kCaps:
+    case CSSPrimitiveValue::UnitType::kRcaps:
+      return RuntimeEnabledFeatures::CSSCapFontUnitsEnabled() ? kCalcLength
+                                                              : kCalcOther;
     case CSSPrimitiveValue::UnitType::kViewportInlineSize:
     case CSSPrimitiveValue::UnitType::kViewportBlockSize:
     case CSSPrimitiveValue::UnitType::kSmallViewportWidth:
@@ -152,6 +156,8 @@ static bool HasDoubleValue(CSSPrimitiveValue::UnitType type) {
     case CSSPrimitiveValue::UnitType::kChs:
     case CSSPrimitiveValue::UnitType::kIcs:
     case CSSPrimitiveValue::UnitType::kLhs:
+    case CSSPrimitiveValue::UnitType::kCaps:
+    case CSSPrimitiveValue::UnitType::kRcaps:
     case CSSPrimitiveValue::UnitType::kRlhs:
     case CSSPrimitiveValue::UnitType::kRems:
     case CSSPrimitiveValue::UnitType::kRexs:
