diff --git a/third_party/blink/renderer/core/script/classic_script.cc b/third_party/blink/renderer/core/script/classic_script.cc
index f06b9fbe5792cd67c369712b22acafc58833a538..27f06d933f862366f0e7393c7f560d196a12faf1 100644
--- a/third_party/blink/renderer/core/script/classic_script.cc
+++ b/third_party/blink/renderer/core/script/classic_script.cc
@@ -201,7 +201,7 @@ v8::ScriptOrigin ClassicScript::CreateScriptOrigin(v8::Isolate* isolate) const {
   // start at 1 only for historic reasons now. I guess we could change it, but
   // there's not much benefit doing so.
   return v8::ScriptOrigin(
-      isolate, V8String(isolate, SourceUrl()),
+      V8String(isolate, SourceUrl()),
       StartPosition().line_.ZeroBasedInt(),
       StartPosition().column_.ZeroBasedInt(),
       GetSanitizeScriptErrors() == SanitizeScriptErrors::kDoNotSanitize, -1,
