diff --git a/third_party/blink/renderer/modules/websockets/websocket_common.cc b/third_party/blink/renderer/modules/websockets/websocket_common.cc
index 9e543e41518ff50cb137078d177eafa27fdfc043..c5e22d9014f854b376aeb08aa5c49e233e9e1d49 100644
--- a/third_party/blink/renderer/modules/websockets/websocket_common.cc
+++ b/third_party/blink/renderer/modules/websockets/websocket_common.cc
@@ -38,7 +38,18 @@ WebSocketCommon::ConnectResult WebSocketCommon::Connect(
     const Vector<String>& protocols,
     WebSocketChannel* channel,
     ExceptionState& exception_state) {
-  url_ = KURL(NullURL(), url);
+  // CompleteURL is not used here because this is expected to always be UTF-8,
+  // and not match document encoding.
+  url_ = KURL(execution_context->BaseURL(), url);
+
+  if (RuntimeEnabledFeatures::WebSocketHTTPURLEnabled(execution_context) &&
+      url_.IsValid()) {
+    if (url_.ProtocolIs("http")) {
+      url_.SetProtocol("ws");
+    } else if (url_.ProtocolIs("https")) {
+      url_.SetProtocol("wss");
+    }
+  }
 
   bool upgrade_insecure_requests_set =
       (execution_context->GetSecurityContext().GetInsecureRequestPolicy() &
@@ -65,8 +76,8 @@ WebSocketCommon::ConnectResult WebSocketCommon::Connect(
     state_ = kClosed;
     exception_state.ThrowDOMException(
         DOMExceptionCode::kSyntaxError,
-        "The URL's scheme must be either 'ws' or 'wss'. '" + url_.Protocol() +
-            "' is not allowed.");
+        "The URL's scheme must be either 'http', 'https', 'ws', or 'wss'. '" +
+            url_.Protocol() + "' is not allowed.");
     return ConnectResult::kException;
   }
 
