diff --git a/third_party/blink/renderer/core/css/resolver/css_to_style_map.h b/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
index 14e8b1191afc6832552efa768d4a539f00ecc6f0..a9d8477954c23577cad4e19c28f60a8297d67cd9 100644
--- a/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
+++ b/third_party/blink/renderer/core/css/resolver/css_to_style_map.h
@@ -48,16 +48,16 @@ class CSSToStyleMap {
                                 FillLayer*,
                                 const CSSValue&);
   static void MapFillClip(StyleResolverState&, FillLayer*, const CSSValue&);
-  static void MapFillComposite(StyleResolverState&,
-                               FillLayer*,
-                               const CSSValue&);
+  static void MapFillCompositingOperator(StyleResolverState&,
+                                         FillLayer*,
+                                         const CSSValue&);
   static void MapFillBlendMode(StyleResolverState&,
                                FillLayer*,
                                const CSSValue&);
   static void MapFillOrigin(StyleResolverState&, FillLayer*, const CSSValue&);
   static void MapFillImage(StyleResolverState&, FillLayer*, const CSSValue&);
-  static void MapFillRepeatX(StyleResolverState&, FillLayer*, const CSSValue&);
-  static void MapFillRepeatY(StyleResolverState&, FillLayer*, const CSSValue&);
+  static void MapFillRepeat(StyleResolverState&, FillLayer*, const CSSValue&);
+  static void MapFillMaskMode(StyleResolverState&, FillLayer*, const CSSValue&);
   static void MapFillSize(StyleResolverState&, FillLayer*, const CSSValue&);
   static void MapFillPositionX(StyleResolverState&,
                                FillLayer*,
@@ -80,6 +80,9 @@ class CSSToStyleMap {
   static double MapAnimationIterationCount(StyleResolverState&,
                                            const CSSValue&);
   static AtomicString MapAnimationName(StyleResolverState&, const CSSValue&);
+  static CSSTransitionData::TransitionBehavior MapAnimationBehavior(
+      StyleResolverState&,
+      const CSSValue&);
   static StyleTimeline MapAnimationTimeline(StyleResolverState&,
                                             const CSSValue&);
   static EAnimPlayState MapAnimationPlayState(StyleResolverState&,
