diff --git a/third_party/blink/renderer/core/css/cssom/css_numeric_value_type.cc b/third_party/blink/renderer/core/css/cssom/css_numeric_value_type.cc
index 63a74431374221f445d7d454e69ce5ab9b32bb1f..658595fdc01d9353bd47f96a22422b3bc5757ed1 100644
--- a/third_party/blink/renderer/core/css/cssom/css_numeric_value_type.cc
+++ b/third_party/blink/renderer/core/css/cssom/css_numeric_value_type.cc
@@ -67,6 +67,8 @@ CSSNumericValueType::BaseType UnitTypeToBaseType(
     case UnitType::kIcs:
     case UnitType::kLhs:
     case UnitType::kRlhs:
+    case UnitType::kCaps:
+    case UnitType::kRcaps:
       return BaseType::kLength;
     case UnitType::kMilliseconds:
     case UnitType::kSeconds:
