diff --git a/third_party/blink/renderer/core/navigation_api/navigation_api.cc b/third_party/blink/renderer/core/navigation_api/navigation_api.cc
index ba58528718d975b4e6ef5bcdf6fd968f4522dbc7..f4d0a542298cfaab5958b4ab8b56f5a66908fc96 100644
--- a/third_party/blink/renderer/core/navigation_api/navigation_api.cc
+++ b/third_party/blink/renderer/core/navigation_api/navigation_api.cc
@@ -32,11 +32,13 @@
 #include "third_party/blink/renderer/core/loader/document_loader.h"
 #include "third_party/blink/renderer/core/loader/frame_load_request.h"
 #include "third_party/blink/renderer/core/navigation_api/navigate_event.h"
+#include "third_party/blink/renderer/core/navigation_api/navigation_activation.h"
 #include "third_party/blink/renderer/core/navigation_api/navigation_api_navigation.h"
 #include "third_party/blink/renderer/core/navigation_api/navigation_current_entry_change_event.h"
 #include "third_party/blink/renderer/core/navigation_api/navigation_destination.h"
 #include "third_party/blink/renderer/core/navigation_api/navigation_history_entry.h"
 #include "third_party/blink/renderer/core/navigation_api/navigation_transition.h"
+#include "third_party/blink/renderer/core/page/page.h"
 #include "third_party/blink/renderer/core/timing/soft_navigation_heuristics.h"
 #include "third_party/blink/renderer/platform/bindings/exception_context.h"
 #include "third_party/blink/renderer/platform/bindings/exception_state.h"
@@ -95,7 +97,13 @@ String DetermineNavigationType(WebFrameLoadType type) {
   NOTREACHED_NORETURN();
 }
 
-NavigationApi::NavigationApi(LocalDOMWindow* window) : window_(window) {}
+NavigationApi::NavigationApi(LocalDOMWindow* window)
+    : window_(window),
+      activation_(MakeGarbageCollected<NavigationActivation>()) {}
+
+NavigationActivation* NavigationApi::activation() const {
+  return HasEntriesAndEventsDisabled() ? nullptr : activation_;
+}
 
 void NavigationApi::setOnnavigate(EventListener* listener) {
   UseCounter::Count(window_, WebFeature::kAppHistory);
@@ -108,13 +116,42 @@ void NavigationApi::PopulateKeySet() {
     keys_to_indices_.insert(entries_[i]->key(), i);
 }
 
+void NavigationApi::UpdateActivation(HistoryItem* previous_item,
+                                     WebFrameLoadType load_type) {
+  NavigationHistoryEntry* previous_history_entry = nullptr;
+  if (previous_item) {
+    if (auto* entry =
+            GetExistingEntryFor(previous_item->GetNavigationApiKey(),
+                                previous_item->GetNavigationApiId())) {
+      previous_history_entry = entry;
+    } else {
+      previous_history_entry = MakeEntryFromItem(*previous_item);
+    }
+  }
+  String navigation_type = window_->GetFrame()->GetPage()->IsPrerendering()
+                               ? "push"
+                               : DetermineNavigationType(load_type);
+  activation_->Update(currentEntry(), previous_history_entry, navigation_type);
+}
+
+NavigationHistoryEntry* NavigationApi::GetExistingEntryFor(const String& key,
+                                                           const String& id) {
+  const auto& it = keys_to_indices_.find(key);
+  if (it == keys_to_indices_.end()) {
+    return nullptr;
+  }
+  NavigationHistoryEntry* existing_entry = entries_[it->value];
+  return existing_entry->id() == id ? existing_entry : nullptr;
+}
+
 void NavigationApi::InitializeForNewWindow(
     HistoryItem& current,
     WebFrameLoadType load_type,
     CommitReason commit_reason,
     NavigationApi* previous,
     const WebVector<WebHistoryItem>& back_entries,
-    const WebVector<WebHistoryItem>& forward_entries) {
+    const WebVector<WebHistoryItem>& forward_entries,
+    HistoryItem* previous_entry) {
   CHECK(entries_.empty());
 
   // This can happen even when commit_reason is not kInitialization, e.g. when
@@ -160,6 +197,7 @@ void NavigationApi::InitializeForNewWindow(
   for (const auto& entry : forward_entries)
     entries_.emplace_back(MakeEntryFromItem(*entry));
   PopulateKeySet();
+  UpdateActivation(previous_entry, load_type);
 }
 
 void NavigationApi::UpdateForNavigation(HistoryItem& item,
@@ -248,11 +286,11 @@ void NavigationApi::UpdateForNavigation(HistoryItem& item,
 
 NavigationHistoryEntry* NavigationApi::GetEntryForRestore(
     const mojom::blink::NavigationApiHistoryEntryPtr& entry) {
-  const auto& it = keys_to_indices_.find(entry->key);
-  if (it != keys_to_indices_.end()) {
-    NavigationHistoryEntry* existing_entry = entries_[it->value];
-    if (existing_entry->id() == entry->id)
-      return existing_entry;
+  if (!entry) {
+    return nullptr;
+  }
+  if (auto* existing_entry = GetExistingEntryFor(entry->key, entry->id)) {
+    return existing_entry;
   }
   return MakeGarbageCollected<NavigationHistoryEntry>(
       window_, entry->key, entry->id, KURL(entry->url),
@@ -269,7 +307,8 @@ void FireDisposeEventsAsync(
 }
 
 void NavigationApi::SetEntriesForRestore(
-    const mojom::blink::NavigationApiHistoryEntryArraysPtr& entry_arrays) {
+    const mojom::blink::NavigationApiHistoryEntryArraysPtr& entry_arrays,
+    mojom::blink::NavigationApiEntryRestoreReason restore_reason) {
   // If this window HasEntriesAndEventsDisabled(), we shouldn't attempt to
   // restore anything.
   if (HasEntriesAndEventsDisabled())
@@ -291,6 +330,27 @@ void NavigationApi::SetEntriesForRestore(
   keys_to_indices_.clear();
   PopulateKeySet();
 
+  String navigation_type;
+  switch (restore_reason) {
+    case mojom::blink::NavigationApiEntryRestoreReason::kBFCache:
+      navigation_type = "traverse";
+      break;
+    case mojom::blink::NavigationApiEntryRestoreReason::
+        kPrerenderActivationPush:
+      navigation_type = "push";
+      break;
+    case mojom::blink::NavigationApiEntryRestoreReason::
+        kPrerenderActivationReplace:
+      navigation_type = "replace";
+      break;
+    default:
+      NOTREACHED();
+      break;
+  }
+  activation_->Update(currentEntry(),
+                      GetEntryForRestore(entry_arrays->previous_entry),
+                      navigation_type);
+
   // |new_entries| now contains the previous entries_. Find the ones that are no
   // longer in entries_ so they can be disposed.
   HeapVector<Member<NavigationHistoryEntry>>* disposed_entries =
@@ -959,6 +1019,7 @@ void NavigationApi::Trace(Visitor* visitor) const {
   visitor->Trace(window_);
   visitor->Trace(entries_);
   visitor->Trace(transition_);
+  visitor->Trace(activation_);
   visitor->Trace(ongoing_navigation_);
   visitor->Trace(upcoming_traversals_);
   visitor->Trace(upcoming_non_traversal_navigation_);
