diff --git a/third_party/blink/renderer/core/url/url_search_params.idl b/third_party/blink/renderer/core/url/url_search_params.idl
index c87ece2d0c63a1fa0af1fac9a49f476fef0f3cc7..2a55e582b4995784d5bc1b4bfafdb1b3d8530677 100644
--- a/third_party/blink/renderer/core/url/url_search_params.idl
+++ b/third_party/blink/renderer/core/url/url_search_params.idl
@@ -10,10 +10,10 @@
     [RaisesException] constructor(optional (sequence<sequence<USVString>> or record<USVString, USVString> or USVString) init = "");
     readonly attribute unsigned long size;
     void append(USVString name, USVString value);
-    [ImplementedAs=deleteAllWithName, CallWith=ScriptState] void delete(USVString name, optional any ignored);
+    [ImplementedAs=deleteAllWithNameOrTuple, CallWith=ExecutionContext] void delete(USVString name, optional USVString value);
     USVString? get(USVString name);
     sequence<USVString> getAll(USVString name);
-    [CallWith=ScriptState] boolean has(USVString name, optional any ignored);
+    [CallWith=ExecutionContext] boolean has(USVString name, optional USVString value);
     void set(USVString name, USVString value);
 
     void sort();
