diff --git a/third_party/blink/renderer/core/html/html_template_element.idl b/third_party/blink/renderer/core/html/html_template_element.idl
index 8f5ca1474c7182d1c80bca161d907aa00b26bbd5..10d2b6350a0cbaa3ee17bd88de4630255a9625a8 100644
--- a/third_party/blink/renderer/core/html/html_template_element.idl
+++ b/third_party/blink/renderer/core/html/html_template_element.idl
@@ -53,4 +53,5 @@
     // This is the "new" declarative Shadow DOM IDL attribute, which reflects
     // the 'shadowrootmode' content attribute.
     [RuntimeEnabled=StreamingDeclarativeShadowDOM,Reflect,ReflectOnly=("open","closed")] attribute DOMString? shadowRootMode;
+    [CEReactions,Reflect,RuntimeEnabled=TemplateDelegatesFocusReflection] attribute boolean shadowRootDelegatesFocus;
 };
