diff --git a/third_party/blink/renderer/core/view_transition/view_transition_options.idl b/third_party/blink/renderer/core/view_transition/view_transition_options.idl
new file mode 100644
index 0000000000000000000000000000000000000000..2e0bc6b0175d3dac284b10dc89e2b5048cac682c
--- /dev/null
+++ b/third_party/blink/renderer/core/view_transition/view_transition_options.idl
@@ -0,0 +1,10 @@
+// Copyright 2023 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// https://drafts.csswg.org/css-view-transitions-2/#dictdef-startviewtransitionoptions
+// FlyWeb: Chromium 125.
+dictionary ViewTransitionOptions {
+  ViewTransitionCallback? update = null;
+  sequence<DOMString>? types = null;
+};
