diff --git a/third_party/blink/renderer/bindings/generated_in_core.gni b/third_party/blink/renderer/bindings/generated_in_core.gni
index 0b7cbc66244f34e7ea738c31bed3d52507f6759b..b7a46d06c1f37815bec9e9a3a5916e63e68faded 100644
--- a/third_party/blink/renderer/bindings/generated_in_core.gni
+++ b/third_party/blink/renderer/bindings/generated_in_core.gni
@@ -1157,6 +1157,8 @@ generated_interface_sources_in_core = [
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigate_event.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation.h",
+  "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation_activation.cc",
+  "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation_activation.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation_current_entry_change_event.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation_current_entry_change_event.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/core/v8/v8_navigation_destination.cc",
