diff --git a/third_party/blink/renderer/core/css/css_primitive_value_units.json5 b/third_party/blink/renderer/core/css/css_primitive_value_units.json5
index c28be859c04916c7db541c1d4ed047c3f83463c4..edb2100c92ca0fdb2bed21ed9b0446074f8b669b 100644
--- a/third_party/blink/renderer/core/css/css_primitive_value_units.json5
+++ b/third_party/blink/renderer/core/css/css_primitive_value_units.json5
@@ -246,6 +246,14 @@
       name: "rlh",
       unit_type: "kRlhs",
     },
+    {
+      name: "cap",
+      unit_type: "kCaps",
+    },
+    {
+      name: "rcap",
+      unit_type: "kRcaps",
+    },
     {
       name: "__qem",
       unit_type: "kQuirkyEms",
