diff --git a/third_party/blink/renderer/platform/theme/web_theme_engine_android.cc b/third_party/blink/renderer/platform/theme/web_theme_engine_android.cc
index 9e996d26704698b3712777a4726384fcfc2a8bfe..fb498a997427035cea8abf827e2c84799436da66 100644
--- a/third_party/blink/renderer/platform/theme/web_theme_engine_android.cc
+++ b/third_party/blink/renderer/platform/theme/web_theme_engine_android.cc
@@ -89,6 +89,9 @@ static void GetNativeThemeExtraParams(
           extra_params->inner_spin.spin_up;
       native_theme_extra_params->inner_spin.read_only =
           extra_params->inner_spin.read_only;
+      native_theme_extra_params->inner_spin.spin_arrows_direction =
+          static_cast<ui::NativeTheme::SpinArrowsDirection>(
+              extra_params->inner_spin.spin_arrows_direction);
       break;
     case WebThemeEngine::kPartProgressBar:
       native_theme_extra_params->progress_bar.determinate =
