diff --git a/third_party/blink/renderer/core/css/css_container_values.cc b/third_party/blink/renderer/core/css/css_container_values.cc
index f995ccc576bd2e2a32278c3395e4806685ea5d44..9197c26d4a5ee784d3044dec1acbac78d820ba8a 100644
--- a/third_party/blink/renderer/core/css/css_container_values.cc
+++ b/third_party/blink/renderer/core/css/css_container_values.cc
@@ -77,6 +77,14 @@ float CSSContainerValues::RootLineHeight(float zoom) const {
   return line_height_size_.Rlh(zoom);
 }
 
+float CSSContainerValues::CapFontSize(float zoom) const {
+  return font_sizes_.Cap(zoom);
+}
+
+float CSSContainerValues::RcapFontSize(float zoom) const {
+  return font_sizes_.Rcap(zoom);
+}
+
 double CSSContainerValues::ContainerWidth() const {
   return container_sizes_.Width().value_or(SmallViewportWidth());
 }
