diff --git a/third_party/blink/public/mojom/navigation/navigation_api_history_entry_arrays.mojom b/third_party/blink/public/mojom/navigation/navigation_api_history_entry_arrays.mojom
index 7fda00c4d10232476e95c44ed1b2c25a49fcbb3a..8b3c13be286d6e7af7629f5c969fb92626f78b3a 100644
--- a/third_party/blink/public/mojom/navigation/navigation_api_history_entry_arrays.mojom
+++ b/third_party/blink/public/mojom/navigation/navigation_api_history_entry_arrays.mojom
@@ -26,4 +26,5 @@ struct NavigationApiHistoryEntry {
 struct NavigationApiHistoryEntryArrays {
   array<NavigationApiHistoryEntry> back_entries;
   array<NavigationApiHistoryEntry> forward_entries;
+  NavigationApiHistoryEntry? previous_entry;
 };
