diff --git a/third_party/blink/renderer/core/style/clip_path_operation.h b/third_party/blink/renderer/core/style/clip_path_operation.h
index f984c9ab377bc083925f541db06a25bb5b4cc01e..2e69b8f4dd85517e59b3a17d784d8fbc4e5ebdaa 100644
--- a/third_party/blink/renderer/core/style/clip_path_operation.h
+++ b/third_party/blink/renderer/core/style/clip_path_operation.h
@@ -40,7 +40,7 @@ class ClipPathOperation : public RefCounted<ClipPathOperation> {
   USING_FAST_MALLOC(ClipPathOperation);
 
  public:
-  enum OperationType { kReference, kShape };
+  enum OperationType { kReference, kShape, kGeometryBox };
 
   virtual ~ClipPathOperation() = default;
 
