diff --git a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_item.h b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_item.h
index fc8eb0184764bc9a1d427f69e0a95d0dd6b17a1c..c64e0a4f97578b3e40bd8e41d1312c829c440844 100644
--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_item.h
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_item.h
@@ -31,6 +31,8 @@ struct CORE_EXPORT GridItemData {
 
  public:
   GridItemData() = delete;
+  GridItemData(const GridItemData&) = default;
+  GridItemData& operator=(const GridItemData&) = default;
 
   GridItemData(NGBlockNode node,
                const ComputedStyle& root_grid_style,
@@ -166,6 +168,13 @@ struct CORE_EXPORT GridItemData {
                : is_opposite_direction_in_root_grid_rows;
   }
 
+  bool MustCachePlacementIndices(
+      GridTrackSizingDirection track_direction) const {
+    return !is_subgridded_to_parent_grid ||
+           IsConsideredForSizing(track_direction) ||
+           MustConsiderGridItemsForSizing(track_direction);
+  }
+
   bool MustConsiderGridItemsForSizing(
       GridTrackSizingDirection track_direction) const {
     return (track_direction == kForColumns)
@@ -217,7 +226,7 @@ struct CORE_EXPORT GridItemData {
 
   void Trace(Visitor* visitor) const { visitor->Trace(node); }
 
-  const NGBlockNode node;
+  NGBlockNode node;
   GridArea resolved_position;
 
   bool has_subgridded_columns : 1;
@@ -326,6 +335,16 @@ class CORE_EXPORT GridItems {
   typedef IteratorBase<false> Iterator;
   typedef IteratorBase<true> ConstIterator;
 
+  GridItems() = default;
+  GridItems(GridItems&&) = default;
+  GridItems& operator=(GridItems&&) = default;
+
+  GridItems(const GridItems& other);
+
+  GridItems& operator=(const GridItems& other) {
+    return *this = GridItems(other);
+  }
+
   Iterator begin() { return {&item_data_, 0}; }
   Iterator end() { return {&item_data_, item_data_.size()}; }
 
