diff --git a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.h b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.h
index 4b1531c594a33d1b4d744b6521ec9d133927b61e..bf8c17e67de19f0994c0862b4510290f4a8354ca 100644
--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.h
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_layout_algorithm.h
@@ -246,8 +246,7 @@ class CORE_EXPORT NGGridLayoutAlgorithm
   // This is used for fragmentation which requires us to know the final offset
   // of each item before fragmentation occurs.
   void PlaceGridItems(
-      const GridItems& grid_items,
-      const NGGridLayoutSubtree& layout_subtree,
+      const NGGridSizingTree& sizing_tree,
       Vector<EBreakBetween>* out_row_break_between,
       Vector<GridItemPlacementData>* out_grid_items_placement_data = nullptr);
 
@@ -257,9 +256,8 @@ class CORE_EXPORT NGGridLayoutAlgorithm
   // This will go through all the grid_items and place fragments which belong
   // within this fragmentainer.
   void PlaceGridItemsForFragmentation(
-      const GridItems& grid_items,
+      const NGGridSizingTree& sizing_tree,
       const Vector<EBreakBetween>& row_break_between,
-      NGGridLayoutData* layout_data,
       Vector<GridItemPlacementData>* grid_item_placement_data,
       Vector<LayoutUnit>* row_offset_adjustments,
       LayoutUnit* intrinsic_block_size,
