diff --git a/third_party/blink/renderer/core/layout/ng/inline/ng_line_info.h b/third_party/blink/renderer/core/layout/ng/inline/ng_line_info.h
index 6b241a91e27ab7c4b3045b8d7f7e6d4a3ed85f18..1809b549c9a4f2c4f43cd541acf86e4b38fec870 100644
--- a/third_party/blink/renderer/core/layout/ng/inline/ng_line_info.h
+++ b/third_party/blink/renderer/core/layout/ng/inline/ng_line_info.h
@@ -150,6 +150,9 @@ class CORE_EXPORT NGLineInfo {
   bool HasOverflow() const { return has_overflow_; }
   void SetHasOverflow(bool value = true) { has_overflow_ = value; }
 
+  // True if this line is hyphenated.
+  bool IsHyphenated() const;
+
   void SetBfcOffset(const NGBfcOffset& bfc_offset) { bfc_offset_ = bfc_offset; }
   void SetWidth(LayoutUnit available_width, LayoutUnit width) {
     available_width_ = available_width;
