diff --git a/third_party/blink/renderer/core/paint/build.gni b/third_party/blink/renderer/core/paint/build.gni
index fd75feaf315977a60b80e66b4a8ecd09d8edb7e4..c97c0a386ea68230722109167d48ed119281de92 100644
--- a/third_party/blink/renderer/core/paint/build.gni
+++ b/third_party/blink/renderer/core/paint/build.gni
@@ -210,6 +210,8 @@ blink_core_sources_paint = [
   "timing/text_element_timing.h",
   "timing/text_paint_timing_detector.cc",
   "timing/text_paint_timing_detector.h",
+  "transform_utils.cc",
+  "transform_utils.h",
   "url_metadata_utils.cc",
   "url_metadata_utils.h",
   "video_painter.cc",
