diff --git a/third_party/blink/renderer/modules/clipboard/clipboard_item.idl b/third_party/blink/renderer/modules/clipboard/clipboard_item.idl
index c67e2b7c871576fce7a14260346b743a31e7105e..f1190a74a76788d8345f8309a5d4a5713cacc7e4 100644
--- a/third_party/blink/renderer/modules/clipboard/clipboard_item.idl
+++ b/third_party/blink/renderer/modules/clipboard/clipboard_item.idl
@@ -15,4 +15,9 @@
     CallWith=ScriptState,
     RaisesException
   ] Promise<Blob> getType(DOMString type);
+
+  [
+    Measure,
+    RuntimeEnabled=ClipboardSupportedTypes
+  ] static boolean supports(DOMString type);
 };
