diff --git a/third_party/blink/renderer/bindings/scripts/validator/rules/supported_extended_attributes.py b/third_party/blink/renderer/bindings/scripts/validator/rules/supported_extended_attributes.py
index 851a730737d5245b15a9a96940ade3e95cb5102c..dc37f1c326d1999eb9152439b76139cff96bfd68 100644
--- a/third_party/blink/renderer/bindings/scripts/validator/rules/supported_extended_attributes.py
+++ b/third_party/blink/renderer/bindings/scripts/validator/rules/supported_extended_attributes.py
@@ -96,6 +96,7 @@ def _build_supported_extended_attributes():
           values=V_CALL_WITH),
         E("Global", applicable_to=[T.INTERFACE], forms=[F.IDENT,
                                                         F.IDENT_LIST]),
+        E("HasAsyncIteratorReturnAlgorithm", applicable_to=[T.OPERATION]),
         E("HighEntropy",
           applicable_to=[T.ATTRIBUTE, T.CONSTRUCTOR, T.OPERATION],
           forms=[F.NO_ARGS, F.IDENT],
