diff --git a/third_party/blink/renderer/core/css/css_default_style_sheets.h b/third_party/blink/renderer/core/css/css_default_style_sheets.h
index 6f40e7b156187f3d1046f982c246e95e85f457d0..bcd5b5925810f8efaefd8cdc0d9001486527daa5 100644
--- a/third_party/blink/renderer/core/css/css_default_style_sheets.h
+++ b/third_party/blink/renderer/core/css/css_default_style_sheets.h
@@ -98,6 +98,9 @@ class CSSDefaultStyleSheets final
   StyleSheetContents* FormControlsNotVerticalSheet() {
     return form_controls_not_vertical_style_sheet_.Get();
   }
+  StyleSheetContents* FormControlsNotVerticalTextSheet() {
+    return form_controls_not_vertical_style_text_sheet_.Get();
+  }
 
   CORE_EXPORT void PrepareForLeakDetection();
 
@@ -162,6 +165,7 @@ class CSSDefaultStyleSheets final
   Member<StyleSheetContents> marker_style_sheet_;
   Member<StyleSheetContents> forced_colors_style_sheet_;
   Member<StyleSheetContents> form_controls_not_vertical_style_sheet_;
+  Member<StyleSheetContents> form_controls_not_vertical_style_text_sheet_;
 
   std::unique_ptr<UAStyleSheetLoader> media_controls_style_sheet_loader_;
 };
