diff --git a/third_party/blink/renderer/core/css/css_value_keywords.json5 b/third_party/blink/renderer/core/css/css_value_keywords.json5
index c6d27b511248e693bb0c36eaabe0da4a92f1f4ad..e7eb11ae1c36b6a3ce7b50a239b3979241fd1196 100644
--- a/third_party/blink/renderer/core/css/css_value_keywords.json5
+++ b/third_party/blink/renderer/core/css/css_value_keywords.json5
@@ -1344,6 +1344,8 @@
     "oklch",
     // "color",
     "-internal-light-dark",
+    // FlyWeb: public light-dark() for colors (Chromium 123).
+    "light-dark",
 
     // color spaces
     // "srgb", included in color-interpolation
@@ -1681,5 +1683,13 @@
     // scroll()
     "root",
     "nearest",
+
+    // transition-behavior
+    "allow-discrete",
+
+    // (scripting:) media feature
+    "enabled",
+    "initial-only",
+    // none
   ],
 }
