diff --git a/third_party/blink/renderer/bindings/core/v8/BUILD.gn b/third_party/blink/renderer/bindings/core/v8/BUILD.gn
index 8ea09184610ea5cdc9cd27a543cd1fb2277c98c6..aa87887c53c86ccf18660ea6339147bfd36cb87e 100644
--- a/third_party/blink/renderer/bindings/core/v8/BUILD.gn
+++ b/third_party/blink/renderer/bindings/core/v8/BUILD.gn
@@ -20,6 +20,7 @@ blink_core_sources("v8") {
             generated_namespace_sources_in_core +
             generated_observable_array_sources_in_core +
             generated_sync_iterator_sources_in_core +
+            generated_async_iterator_sources_in_core +
             generated_typedef_sources_in_core + generated_union_sources_in_core
 
   deps = [
