diff --git a/third_party/blink/renderer/core/view_transition/build.gni b/third_party/blink/renderer/core/view_transition/build.gni
index eb9fa3664d684445c8cd9f65f0a8f1c68a2c5d1d..0b2e0207febbbdda3743fa55d32a70a1a5aa7ee2 100644
--- a/third_party/blink/renderer/core/view_transition/build.gni
+++ b/third_party/blink/renderer/core/view_transition/build.gni
@@ -17,6 +17,8 @@ blink_core_sources_view_transition = [
   "view_transition_style_tracker.h",
   "view_transition_supplement.cc",
   "view_transition_supplement.h",
+  "view_transition_type_set.cc",
+  "view_transition_type_set.h",
   "view_transition_utils.cc",
   "view_transition_utils.h",
 ]
