diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_container.h b/third_party/blink/renderer/core/layout/svg/layout_svg_container.h
index 7fafad54150525cdbed18ba7f34c97b933822602..c5b74005d448551f7139907a5ef4909d708a5259 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_container.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_container.h
@@ -60,6 +60,7 @@ class LayoutSVGContainer : public LayoutSVGModelObject {
     NOT_DESTROYED();
     needs_boundaries_update_ = true;
   }
+  void SetNeedsTransformUpdate() override;
   bool DidScreenScaleFactorChange() const {
     NOT_DESTROYED();
     return did_screen_scale_factor_change_;
@@ -102,13 +103,23 @@ class LayoutSVGContainer : public LayoutSVGModelObject {
   }
   void UpdateLayout() override;
 
+  void SetTransformUsesReferenceBox(bool transform_uses_reference_box) {
+    NOT_DESTROYED();
+    transform_uses_reference_box_ = transform_uses_reference_box;
+  }
+
   void AddChild(LayoutObject* child,
                 LayoutObject* before_child = nullptr) final;
   void RemoveChild(LayoutObject*) final;
 
   gfx::RectF StrokeBoundingBox() const final {
     NOT_DESTROYED();
-    return content_.StrokeBoundingBox();
+    return content_.ComputeStrokeBoundingBox();
+  }
+
+  gfx::RectF DecoratedBoundingBox() const final {
+    NOT_DESTROYED();
+    return content_.DecoratedBoundingBox();
   }
 
   bool NodeAtPoint(HitTestResult&,
@@ -117,7 +128,8 @@ class LayoutSVGContainer : public LayoutSVGModelObject {
                    HitTestPhase) override;
 
   // Called during layout to update the local transform.
-  virtual SVGTransformChange CalculateLocalTransform(bool bounds_changed);
+  virtual SVGTransformChange UpdateLocalTransform(
+      const gfx::RectF& reference_box);
 
   bool UpdateCachedBoundaries();
 
@@ -127,7 +139,9 @@ class LayoutSVGContainer : public LayoutSVGModelObject {
   SVGContentContainer content_;
   bool object_bounding_box_valid_;
   bool needs_boundaries_update_ : 1;
+  bool needs_transform_update_ : 1;
   bool did_screen_scale_factor_change_ : 1;
+  bool transform_uses_reference_box_ : 1;
   mutable bool has_non_isolated_blending_descendants_ : 1;
   mutable bool has_non_isolated_blending_descendants_dirty_ : 1;
 };
