diff --git a/third_party/blink/renderer/core/css/properties/computed_style_utils.h b/third_party/blink/renderer/core/css/properties/computed_style_utils.h
index 8e2c3fa4ff6b5d6475eecab4364d6e0de3350305..0d8e1566321224e77c464541753c7c90662bfb1b 100644
--- a/third_party/blink/renderer/core/css/properties/computed_style_utils.h
+++ b/third_party/blink/renderer/core/css/properties/computed_style_utils.h
@@ -233,7 +233,10 @@ class CORE_EXPORT ComputedStyleUtils {
   static CSSValue* ResolvedTransform(const LayoutObject*, const ComputedStyle&);
   static CSSValue* CreateTransitionPropertyValue(
       const CSSTransitionData::TransitionProperty&);
+  static CSSValue* CreateTransitionBehaviorValue(
+      const CSSTransitionData::TransitionBehavior&);
   static CSSValue* ValueForTransitionProperty(const CSSTransitionData*);
+  static CSSValue* ValueForTransitionBehavior(const CSSTransitionData*);
   static CSSValue* ValueForContentData(const ComputedStyle&,
                                        bool allow_visited_style);
 
