diff --git a/third_party/blink/renderer/core/mathml/mathml_attribute_names.json5 b/third_party/blink/renderer/core/mathml/mathml_attribute_names.json5
index 41a349a88ca8b6c1a3f000db8eab199faf6b4e9a..7c29b3bb600f60cbd68be5f7e456f358c671f824 100644
--- a/third_party/blink/renderer/core/mathml/mathml_attribute_names.json5
+++ b/third_party/blink/renderer/core/mathml/mathml_attribute_names.json5
@@ -9,6 +9,7 @@
   data: [
     "accent",
     "accentunder",
+    "columnspan",
     "definitionURL",
     "depth",
     "display",
@@ -26,6 +27,7 @@
     "maxsize",
     "minsize",
     "movablelimits",
+    "rowspan",
     "rspace",
     "scriptlevel",
     "stretchy",
