diff --git a/third_party/blink/renderer/core/css/font_size_functions.h b/third_party/blink/renderer/core/css/font_size_functions.h
index 6b317a8b2024ef52236df285d4da8621d0d35bc9..d4a9caaaf75314db1e8056dce9aa4c35706ea515 100644
--- a/third_party/blink/renderer/core/css/font_size_functions.h
+++ b/third_party/blink/renderer/core/css/font_size_functions.h
@@ -26,6 +26,7 @@
 #include "third_party/abseil-cpp/absl/types/optional.h"
 #include "third_party/blink/renderer/core/core_export.h"
 #include "third_party/blink/renderer/core/css_value_keywords.h"
+#include "third_party/blink/renderer/platform/fonts/font_size_adjust.h"
 #include "third_party/blink/renderer/platform/wtf/allocator/allocator.h"
 
 namespace blink {
@@ -84,6 +85,17 @@ class CORE_EXPORT FontSizeFunctions {
                             int pixel_font_size,
                             bool is_monospace);
 
+  // Given font data, this function returns a normalized aspect value for the
+  // specified font metric, which is calculated using the size of the font
+  // metric divided by the font size.
+  // https://www.w3.org/TR/css-fonts-5/#font-size-adjust-aspect-value
+  static absl::optional<float> FontAspectValue(const SimpleFontData*,
+                                               FontSizeAdjust::Metric,
+                                               float computed_size);
+
+  // Given font data, this function returns a font size adjusted by
+  // font-size-adjust, scaling the font size to achieve the desired aspect
+  // value of font size to metric.
   static absl::optional<float> MetricsMultiplierAdjustedFontSize(
       const SimpleFontData*,
       const FontDescription&);
