diff --git a/third_party/blink/renderer/core/streams/readable_stream.h b/third_party/blink/renderer/core/streams/readable_stream.h
index 37a3611b12444f9d6af030ef8b21310fa27f8ef3..28fbcc2b96fc831c8648952c0678bcc444df7fac 100644
--- a/third_party/blink/renderer/core/streams/readable_stream.h
+++ b/third_party/blink/renderer/core/streams/readable_stream.h
@@ -8,7 +8,10 @@
 #include <stdint.h>
 #include <memory>
 
+#include "third_party/blink/renderer/bindings/core/v8/async_iterable.h"
 #include "third_party/blink/renderer/bindings/core/v8/script_value.h"
+#include "third_party/blink/renderer/bindings/core/v8/v8_async_iterator_readable_stream.h"
+#include "third_party/blink/renderer/bindings/core/v8/v8_readable_stream_iterator_options.h"
 #include "third_party/blink/renderer/bindings/core/v8/v8_typedefs.h"
 #include "third_party/blink/renderer/core/core_export.h"
 #include "third_party/blink/renderer/core/streams/readable_stream_default_reader.h"
@@ -45,7 +48,10 @@ class WritableStream;
 
 // C++ implementation of ReadableStream.
 // See https://streams.spec.whatwg.org/#rs-model for background.
-class CORE_EXPORT ReadableStream : public ScriptWrappable {
+class CORE_EXPORT ReadableStream
+    : public ScriptWrappable,
+      public ValueAsyncIterable<ReadableStream,
+                                ReadableStreamIteratorOptions*> {
   DEFINE_WRAPPERTYPEINFO();
 
  public:
@@ -282,6 +288,8 @@ class CORE_EXPORT ReadableStream : public ScriptWrappable {
   class PullAlgorithm;
   class CancelAlgorithm;
   class ReadHandleImpl;
+  class IterationSource;
+  class IterationReadRequest;
 
   // https://streams.spec.whatwg.org/#rs-constructor
   void InitInternal(ScriptState*,
@@ -352,6 +360,17 @@ class CORE_EXPORT ReadableStream : public ScriptWrappable {
   Member<ReadableStreamGenericReader> reader_;
   TraceWrapperV8Reference<v8::Value> stored_error_;
   std::unique_ptr<ReadableStreamTransferringOptimizer> transferring_optimizer_;
+
+  // ValueAsyncIterable<ReadableStream> overrides:
+  // FlyWeb: backported from Chromium 124 (async iteration of ReadableStream).
+  using IterationSourceBase =
+      ValueAsyncIterable<ReadableStream,
+                         ReadableStreamIteratorOptions*>::IterationSource;
+  IterationSourceBase* CreateIterationSource(
+      ScriptState* script_state,
+      IterationSourceBase::Kind kind,
+      ReadableStreamIteratorOptions* options,
+      ExceptionState& exception_state) override;
 };
 
 }  // namespace blink
