diff --git a/third_party/blink/renderer/core/paint/css_mask_painter.cc b/third_party/blink/renderer/core/paint/css_mask_painter.cc
index 2dbe8cb7d3c18b05a68afab5dc12f2c7f94ee65f..c6129a1797a17dbc18008ac7aa50eb348c2e8b25 100644
--- a/third_party/blink/renderer/core/paint/css_mask_painter.cc
+++ b/third_party/blink/renderer/core/paint/css_mask_painter.cc
@@ -8,6 +8,7 @@
 #include "third_party/blink/renderer/core/layout/layout_inline.h"
 #include "third_party/blink/renderer/core/layout/svg/layout_svg_resource_masker.h"
 #include "third_party/blink/renderer/core/layout/svg/svg_resources.h"
+#include "third_party/blink/renderer/core/paint/paint_layer.h"
 
 namespace blink {
 
@@ -17,6 +18,8 @@ absl::optional<gfx::RectF> CSSMaskPainter::MaskBoundingBox(
   if (!object.IsBoxModelObject() && !object.IsSVGChild())
     return absl::nullopt;
 
+  // TODO(pdr): The SVG bounding box does not account for mask-clip values like
+  // `no-clip`.
   const ComputedStyle& style = object.StyleRef();
   if (object.IsSVG()) {
     if (SVGResourceClient* client = SVGResources::GetClient(object)) {
@@ -40,17 +43,30 @@ absl::optional<gfx::RectF> CSSMaskPainter::MaskBoundingBox(
     return absl::nullopt;
 
   PhysicalRect maximum_mask_region;
-  // For HTML/CSS objects, the extent of the mask is known as "mask
-  // painting area", which is determined by CSS mask-clip property.
-  // We don't implement mask-clip:margin-box or no-clip currently,
-  // so the maximum we can get is border-box.
+  EFillBox maximum_mask_clip = style.MaskLayers().LayersClipMax();
   if (object.IsBox()) {
-    maximum_mask_region = To<LayoutBox>(object).PhysicalBorderBoxRect();
+    if (maximum_mask_clip == EFillBox::kNoClip) {
+      maximum_mask_region =
+          To<LayoutBox>(object)
+              .Layer()
+              ->LocalBoundingBoxIncludingSelfPaintingDescendants();
+    } else {
+      // We could use a tighter rect for padding-box/content-box.
+      maximum_mask_region = To<LayoutBox>(object).PhysicalBorderBoxRect();
+    }
   } else {
     // For inline elements, depends on the value of box-decoration-break
     // there could be one box in multiple fragments or multiple boxes.
     // Either way here we are only interested in the bounding box of them.
-    maximum_mask_region = To<LayoutInline>(object).PhysicalLinesBoundingBox();
+    if (maximum_mask_clip == EFillBox::kNoClip) {
+      maximum_mask_region =
+          To<LayoutInline>(object)
+              .Layer()
+              ->LocalBoundingBoxIncludingSelfPaintingDescendants();
+    } else {
+      // We could use a tighter rect for padding-box/content-box.
+      maximum_mask_region = To<LayoutInline>(object).PhysicalLinesBoundingBox();
+    }
   }
   if (style.HasMaskBoxImageOutsets())
     maximum_mask_region.Expand(style.MaskBoxImageOutsets());
