diff --git a/third_party/blink/renderer/core/events/mutation_event.h b/third_party/blink/renderer/core/events/mutation_event.h
index b5bdd3351f2e6754adce7b745a59b65817dbb372..37db7252b10b74ce5f8813f9eb3903ee912821ae 100644
--- a/third_party/blink/renderer/core/events/mutation_event.h
+++ b/third_party/blink/renderer/core/events/mutation_event.h
@@ -29,6 +29,8 @@
 
 namespace blink {
 
+class EventDispatcher;
+
 class MutationEvent final : public Event {
   DEFINE_WRAPPERTYPEINFO();
 
@@ -80,6 +82,8 @@ class MutationEvent final : public Event {
 
   const AtomicString& InterfaceName() const override;
 
+  DispatchEventResult DispatchEvent(EventDispatcher&) override;
+
   void Trace(Visitor*) const override;
 
  private:
