diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.h b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.h
index 872bcb212b3de76a418c866181064ed56fc71e76..58ce1b45790c291cc3c51aa367c2ad5c6dba748b 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_resource_marker.h
@@ -75,12 +75,12 @@ class LayoutSVGResourceMarker final : public LayoutSVGResourceContainer {
 
  private:
   void UpdateLayout() override;
-  SVGTransformChange CalculateLocalTransform(bool bounds_changed) final;
+  SVGTransformChange UpdateLocalTransform(
+      const gfx::RectF& reference_box) final;
   bool FindCycleFromSelf() const override;
 
   AffineTransform local_to_parent_transform_;
   gfx::SizeF viewport_size_;
-  bool needs_transform_update_;
   bool is_in_layout_;
 };
 
