diff --git a/third_party/blink/renderer/core/animation/BUILD.gn b/third_party/blink/renderer/core/animation/BUILD.gn
index 89e988ab3efdc690c3800f4c257fc6493ebfddc0..6eb8aec91ca9edbc0e87bf1c06710a771a453e9c 100644
--- a/third_party/blink/renderer/core/animation/BUILD.gn
+++ b/third_party/blink/renderer/core/animation/BUILD.gn
@@ -149,6 +149,8 @@ blink_core_sources("animation") {
     "css_rotate_interpolation_type.h",
     "css_scale_interpolation_type.cc",
     "css_scale_interpolation_type.h",
+    "css_scrollbar_color_interpolation_type.cc",
+    "css_scrollbar_color_interpolation_type.h",
     "css_shadow_list_interpolation_type.cc",
     "css_shadow_list_interpolation_type.h",
     "css_size_list_interpolation_type.cc",
@@ -203,6 +205,8 @@ blink_core_sources("animation") {
     "interpolable_grid_track_size.h",
     "interpolable_length.cc",
     "interpolable_length.h",
+    "interpolable_scrollbar_color.cc",
+    "interpolable_scrollbar_color.h",
     "interpolable_shadow.cc",
     "interpolable_shadow.h",
     "interpolable_transform_list.cc",
