diff --git a/third_party/blink/renderer/core/url/url.idl b/third_party/blink/renderer/core/url/url.idl
index c914c23f2f678eb8ee33ea2483228b186dbae7bb..b90a0b68e397b03917930020f3e68e59919e31ff 100644
--- a/third_party/blink/renderer/core/url/url.idl
+++ b/third_party/blink/renderer/core/url/url.idl
@@ -33,6 +33,8 @@
 ] interface URL {
     [RaisesException] constructor(USVString url, optional USVString base);
 
+    [RuntimeEnabled=URLCanParse] static boolean canParse(USVString url, optional USVString base);
+
     [RaisesException=Setter] stringifier attribute USVString href;
     readonly attribute USVString origin;
     attribute USVString protocol;
