diff --git a/third_party/blink/renderer/bindings/modules/v8/BUILD.gn b/third_party/blink/renderer/bindings/modules/v8/BUILD.gn
index 5f1951cde876d174ab6196e6cb8bd6c0abdb1c4c..6a33cb975877d99e619361d4221337f3b8f3e69b 100644
--- a/third_party/blink/renderer/bindings/modules/v8/BUILD.gn
+++ b/third_party/blink/renderer/bindings/modules/v8/BUILD.gn
@@ -20,6 +20,7 @@ blink_modules_sources("v8") {
       generated_namespace_sources_in_modules +
       generated_observable_array_sources_in_modules +
       generated_sync_iterator_sources_in_modules +
+      generated_async_iterator_sources_in_modules +
       generated_typedef_sources_in_modules + generated_union_sources_in_modules
 
   deps = [
