diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_rect.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_rect.cc
index 9f4a3c7c64f931d5708780e5c4ed711fab586acb..02a4ea70ab0bd52a564866a6fc0f87a11720a542 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_rect.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_rect.cc
@@ -41,7 +41,7 @@ LayoutSVGRect::~LayoutSVGRect() = default;
 void LayoutSVGRect::UpdateShapeFromElement() {
   NOT_DESTROYED();
 
-  stroke_bounding_box_ = gfx::RectF();
+  decorated_bounding_box_ = gfx::RectF();
   use_path_fallback_ = false;
 
   SVGLengthContext length_context(GetElement());
@@ -77,7 +77,7 @@ void LayoutSVGRect::UpdateShapeFromElement() {
   if (!use_path_fallback_)
     ClearPath();
 
-  stroke_bounding_box_ = CalculateStrokeBoundingBox();
+  decorated_bounding_box_ = CalculateStrokeBoundingBox();
 }
 
 bool LayoutSVGRect::ShapeDependentStrokeContains(
