diff --git a/third_party/blink/renderer/core/css/css_segmented_font_face.h b/third_party/blink/renderer/core/css/css_segmented_font_face.h
index ed5f7691f7604cf503365d6716f9e7e07a3dbc3c..bab6ab0ba3e7ce57335a1240ed27627daba11786 100644
--- a/third_party/blink/renderer/core/css/css_segmented_font_face.h
+++ b/third_party/blink/renderer/core/css/css_segmented_font_face.h
@@ -108,7 +108,7 @@ class CSSSegmentedFontFace final
 
   scoped_refptr<FontData> GetFontData(const FontDescription&);
 
-  bool CheckFont(const String&) const;
+  bool CheckFont(UChar32) const;
   void Match(const String&, HeapVector<Member<FontFace>>*) const;
   void WillUseFontData(const FontDescription&, const String& text);
   void WillUseRange(const FontDescription&, const blink::FontDataForRangeSet&);
