diff --git a/third_party/blink/renderer/core/layout/layout_theme_default.cc b/third_party/blink/renderer/core/layout/layout_theme_default.cc
index ae79d13b3df78a1f02f19b406ba3f1afc8bf939d..8e3f9cd4273835828501bd5f70f5b74d0e9b43f6 100644
--- a/third_party/blink/renderer/core/layout/layout_theme_default.cc
+++ b/third_party/blink/renderer/core/layout/layout_theme_default.cc
@@ -192,8 +192,13 @@ void LayoutThemeDefault::AdjustInnerSpinButtonStyle(
       WebThemeEngine::kPartInnerSpinButton);
 
   float zoom_level = style.EffectiveZoom();
-  style.SetWidth(Length::Fixed(size.width() * zoom_level));
-  style.SetMinWidth(Length::Fixed(size.width() * zoom_level));
+  if (IsHorizontalWritingMode(style.GetWritingMode())) {
+    style.SetWidth(Length::Fixed(size.width() * zoom_level));
+    style.SetMinWidth(Length::Fixed(size.width() * zoom_level));
+  } else {
+    style.SetHeight(Length::Fixed(size.width() * zoom_level));
+    style.SetMinHeight(Length::Fixed(size.width() * zoom_level));
+  }
 }
 
 Color LayoutThemeDefault::PlatformFocusRingColor() const {
