diff --git a/third_party/blink/renderer/core/css/css_keyframes_rule.idl b/third_party/blink/renderer/core/css/css_keyframes_rule.idl
index 0372406bd40831f9b69c90fb7af9e3d5f0119956..0f1ef7ecc0276720276891a5178572b302be2236 100644
--- a/third_party/blink/renderer/core/css/css_keyframes_rule.idl
+++ b/third_party/blink/renderer/core/css/css_keyframes_rule.idl
@@ -33,11 +33,10 @@
 ] interface CSSKeyframesRule : CSSRule {
     attribute DOMString name;
     readonly attribute CSSRuleList cssRules;
+    [RuntimeEnabled=CSSKeyframesRuleLength] readonly attribute unsigned long length;
 
+    getter CSSKeyframeRule (unsigned long index);
     [CallWith=ExecutionContext] void appendRule(DOMString rule);
     [CallWith=ExecutionContext] void deleteRule(DOMString select);
     [CallWith=ExecutionContext] CSSKeyframeRule? findRule(DOMString select);
-
-    // Non-standard APIs
-    [NotEnumerable] getter CSSKeyframeRule (unsigned long index);
 };
