diff --git a/third_party/blink/renderer/core/css/css_default_style_sheets.cc b/third_party/blink/renderer/core/css/css_default_style_sheets.cc
index 4e2b25680e65b29dd08887f04ca8132d413e7e92..e6cab870145574def0e82f34d15f7dd0d88e662a 100644
--- a/third_party/blink/renderer/core/css/css_default_style_sheets.cc
+++ b/third_party/blink/renderer/core/css/css_default_style_sheets.cc
@@ -38,6 +38,8 @@
 #include "third_party/blink/renderer/core/execution_context/security_context.h"
 #include "third_party/blink/renderer/core/frame/settings.h"
 #include "third_party/blink/renderer/core/html/forms/html_button_element.h"
+#include "third_party/blink/renderer/core/html/forms/html_input_element.h"
+#include "third_party/blink/renderer/core/html/forms/html_select_element.h"
 #include "third_party/blink/renderer/core/html/forms/html_text_area_element.h"
 #include "third_party/blink/renderer/core/html/html_anchor_element.h"
 #include "third_party/blink/renderer/core/html/html_html_element.h"
@@ -149,6 +151,7 @@ void CSSDefaultStyleSheets::PrepareForLeakDetection() {
   selectmenu_style_sheet_.Clear();
   marker_style_sheet_.Clear();
   form_controls_not_vertical_style_sheet_.Clear();
+  form_controls_not_vertical_style_text_sheet_.Clear();
   // Recreate the default style sheet to clean up possible SVG resources.
   String default_rules = UncompressResourceAsASCIIString(IDR_UASTYLE_HTML_CSS) +
                          OverflowForReplacedElementRules() +
@@ -356,15 +359,17 @@ bool CSSDefaultStyleSheets::EnsureDefaultStyleSheetsForElement(
     changed_default_style = true;
   }
 
-  // TODO(crbug.com/681917): The FormControlsVerticalWritingModeSupport
-  // flag enables vertical writing mode to be used on form controls. When
-  // it is *disabled*, we need to force horizontal writing mode.
+  // TODO(crbug.com/681917, crbug.com/484651): We enable vertical writing mode
+  // on form controls using features FormControlsVerticalWritingModeSupport
+  // and FormControlsVerticalWritingModeTextSupport. When it is *disabled*,
+  // we need to force horizontal writing mode.
+  const auto* input = DynamicTo<HTMLInputElement>(element);
   if (!RuntimeEnabledFeatures::
           FormControlsVerticalWritingModeSupportEnabled() &&
       !form_controls_not_vertical_style_sheet_ &&
       (IsA<HTMLProgressElement>(element) || IsA<HTMLMeterElement>(element) ||
-       IsA<HTMLInputElement>(element) || IsA<HTMLTextAreaElement>(element) ||
-       IsA<HTMLButtonElement>(element))) {
+       IsA<HTMLButtonElement>(element) || IsA<HTMLSelectElement>(element) ||
+       (input && !input->IsTextField()))) {
     form_controls_not_vertical_style_sheet_ =
         ParseUASheet(UncompressResourceAsASCIIString(
             IDR_UASTYLE_FORM_CONTROLS_NOT_VERTICAL_CSS));
@@ -372,6 +377,17 @@ bool CSSDefaultStyleSheets::EnsureDefaultStyleSheetsForElement(
                                  NamespaceType::kHTML);
     changed_default_style = true;
   }
+  if (!RuntimeEnabledFeatures::
+          FormControlsVerticalWritingModeTextSupportEnabled() &&
+      !form_controls_not_vertical_style_text_sheet_ &&
+      (IsA<HTMLTextAreaElement>(element) || (input && input->IsTextField()))) {
+    form_controls_not_vertical_style_text_sheet_ =
+        ParseUASheet(UncompressResourceAsASCIIString(
+            IDR_UASTYLE_FORM_CONTROLS_NOT_VERTICAL_CSS_TEXT));
+    AddRulesToDefaultStyleSheets(form_controls_not_vertical_style_text_sheet_,
+                                 NamespaceType::kHTML);
+    changed_default_style = true;
+  }
 
   DCHECK(!default_html_style_->Features().HasIdsInSelectors());
   return changed_default_style;
@@ -488,6 +504,7 @@ void CSSDefaultStyleSheets::Trace(Visitor* visitor) const {
   visitor->Trace(selectmenu_style_sheet_);
   visitor->Trace(marker_style_sheet_);
   visitor->Trace(form_controls_not_vertical_style_sheet_);
+  visitor->Trace(form_controls_not_vertical_style_text_sheet_);
 }
 
 }  // namespace blink
