diff --git a/third_party/blink/renderer/core/navigation_api/navigate_event_init.idl b/third_party/blink/renderer/core/navigation_api/navigate_event_init.idl
index 814c14005ab57d3bb19e7e1f72316078679d176a..620d52a244ba89c9dfcb3cdd3424a6105a12cab6 100644
--- a/third_party/blink/renderer/core/navigation_api/navigate_event_init.idl
+++ b/third_party/blink/renderer/core/navigation_api/navigate_event_init.idl
@@ -21,4 +21,5 @@ dictionary NavigateEventInit : EventInit {
   FormData? formData = null;
   DOMString? downloadRequest = null;
   any info;
+  boolean hasUAVisualTransition = false;
 };
