diff --git a/third_party/blink/renderer/core/navigation_api/navigate_event.h b/third_party/blink/renderer/core/navigation_api/navigate_event.h
index 12d6778befc88703acaa37981ca6c4f2550ec709..e482abd2dd7af5b3aae46e9dae00969fc1ab6188 100644
--- a/third_party/blink/renderer/core/navigation_api/navigate_event.h
+++ b/third_party/blink/renderer/core/navigation_api/navigate_event.h
@@ -63,6 +63,7 @@ class NavigateEvent final : public Event,
   FormData* formData() const { return form_data_; }
   String downloadRequest() const { return download_request_; }
   ScriptValue info() const { return info_; }
+  bool hasUAVisualTransition() const { return has_ua_visual_transition_; }
   void intercept(NavigationInterceptOptions*, ExceptionState&);
   void commit(ExceptionState&);
 
@@ -111,6 +112,7 @@ class NavigateEvent final : public Event,
   Member<FormData> form_data_;
   String download_request_;
   ScriptValue info_;
+  bool has_ua_visual_transition_ = false;
   absl::optional<V8NavigationFocusReset> focus_reset_behavior_ = absl::nullopt;
   absl::optional<V8NavigationScrollBehavior> scroll_behavior_ = absl::nullopt;
   absl::optional<V8NavigationCommitBehavior> commit_behavior_ = absl::nullopt;
