diff --git a/third_party/blink/renderer/bindings/BUILD.gn b/third_party/blink/renderer/bindings/BUILD.gn
index 01d5bdc58ddc5e2b4e13c86bf16c8ca8e577dc72..8104cee831a019a75f02d3457c008e751a1554c0 100644
--- a/third_party/blink/renderer/bindings/BUILD.gn
+++ b/third_party/blink/renderer/bindings/BUILD.gn
@@ -174,6 +174,7 @@ template("generate_bindings") {
     args += invoker.targets
 
     deps = [ ":web_idl_database" ]
+    import("//brave/third_party/blink/renderer/bindings/sources.gni") deps += brave_third_party_blink_renderer_bindings_generate_bindings_deps
   }
 }
 
@@ -182,6 +183,9 @@ generate_bindings("generate_bindings_all") {
 
   targets = []
   outputs = []
+  targets += [ "async_iterator" ]
+  outputs += generated_async_iterator_sources_in_core +
+             generated_async_iterator_sources_in_modules
   targets += [ "callback_function" ]
   outputs += generated_callback_function_sources_in_core +
              generated_callback_function_sources_in_modules +
@@ -249,6 +253,12 @@ action_with_pydeps("check_generated_file_list") {
 
   file_list = []
   file_list += [ "--for_prod" ]
+  file_list += [
+    "--kind",
+    "async_iterator",
+  ]
+  file_list += generated_async_iterator_sources_in_core +
+               generated_async_iterator_sources_in_modules
   file_list += [
     "--kind",
     "callback_function",
