diff --git a/third_party/blink/renderer/core/html/html_details_element.h b/third_party/blink/renderer/core/html/html_details_element.h
index 8b434960ff46144f2aee1e3d53951add78f478d5..8ccba766544348923b4a33fc1ffbabf54a39f888 100644
--- a/third_party/blink/renderer/core/html/html_details_element.h
+++ b/third_party/blink/renderer/core/html/html_details_element.h
@@ -21,6 +21,7 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_CORE_HTML_HTML_DETAILS_ELEMENT_H_
 #define THIRD_PARTY_BLINK_RENDERER_CORE_HTML_HTML_DETAILS_ELEMENT_H_
 
+#include "third_party/blink/renderer/core/events/toggle_event.h"
 #include "third_party/blink/renderer/core/html/html_element.h"
 #include "third_party/blink/renderer/platform/scheduler/public/post_cancellable_task.h"
 
@@ -58,14 +59,18 @@ class HTMLDetailsElement final : public HTMLElement {
   // attribute, excluding |this|, in tree order.  If there is no such group
   // (e.g., because there is no name attribute), returns an empty list.
   HeapVector<Member<HTMLDetailsElement>> OtherElementsInNameGroup();
+  void MaybeCloseForExclusivity();
 
   LayoutObject* CreateLayoutObject(const ComputedStyle&) override;
   void ParseAttribute(const AttributeModificationParams&) override;
+  void AttributeChanged(const AttributeModificationParams&) override;
+  InsertionNotificationRequest InsertedInto(ContainerNode&) override;
   void DidAddUserAgentShadowRoot(ShadowRoot&) override;
   bool IsInteractiveContent() const override;
 
   bool is_open_ = false;
-  TaskHandle pending_event_;
+  TaskHandle pending_event_task_;
+  Member<ToggleEvent> pending_toggle_event_;
   Member<HTMLSlotElement> summary_slot_;
   Member<HTMLSlotElement> content_slot_;
 };
