diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_model_object.h b/third_party/blink/renderer/core/layout/svg/layout_svg_model_object.h
index 1052d75e15d3092f573fa69e9539bc9bb322f2ae..2456a6ecd69d9068bd98fbebcd697599b54c0980 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_model_object.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_model_object.h
@@ -52,7 +52,7 @@ class LayoutSVGModelObject : public LayoutObject {
 
   gfx::RectF VisualRectInLocalSVGCoordinates() const override {
     NOT_DESTROYED();
-    return StrokeBoundingBox();
+    return DecoratedBoundingBox();
   }
 
   void AbsoluteQuads(Vector<gfx::QuadF>&,
@@ -83,7 +83,6 @@ class LayoutSVGModelObject : public LayoutObject {
   void InsertedIntoTree() override;
   void WillBeRemovedFromTree() override;
 
-  AffineTransform CalculateLocalTransform() const;
   bool CheckForImplicitTransformChange(bool bbox_changed) const;
 
  private:
