diff --git a/third_party/blink/renderer/core/css/build.gni b/third_party/blink/renderer/core/css/build.gni
index 9ef46279d933030422c8733a790cba4bddc063b0..230bda8934191cdfc064116ec64a0ed5c5f78af5 100644
--- a/third_party/blink/renderer/core/css/build.gni
+++ b/third_party/blink/renderer/core/css/build.gni
@@ -243,6 +243,8 @@ blink_core_sources_css = [
   "css_reflect_value.cc",
   "css_reflect_value.h",
   "css_reflection_direction.h",
+  "css_repeat_style_value.cc",
+  "css_repeat_style_value.h",
   "css_resolution_units.h",
   "css_resource_fetch_restriction.h",
   "css_revert_layer_value.cc",
