diff --git a/third_party/blink/renderer/core/css/properties/computed_style_utils.cc b/third_party/blink/renderer/core/css/properties/computed_style_utils.cc
index 71ed755e000c1c86b71b0fab44228c56f5e49c2b..b5330b60c543c017aee522445d37fb9b51b4b11d 100644
--- a/third_party/blink/renderer/core/css/properties/computed_style_utils.cc
+++ b/third_party/blink/renderer/core/css/properties/computed_style_utils.cc
@@ -1984,16 +1984,27 @@ CSSValue* ComputedStyleUtils::ValueForGridTrackList(
     is_track_list_empty = positions.size() == 1;
   }
 
-  const bool is_subgrid = computed_grid_track_list.IsSubgriddedAxis();
-
-  // Even if the track list is empty or it's not actually a grid/subgrid in
-  // layout, if the author specified `subgrid`, the computed value should always
-  // begin with `subgrid` and cannot be `none`.
+  const bool is_subgrid_specified = computed_grid_track_list.IsSubgriddedAxis();
+  const bool is_subgrid_valid =
+      grid ? grid->CachedPlacementData().line_resolver.SubgridSpanSize(
+                 direction) != kNotFound
+           : false;
+  const bool is_subgrid = is_subgrid_specified && is_subgrid_valid;
+
+  // Standalone grids with empty track lists should compute to `none`, but
+  // this is not the case for subgrids. Subgrids need to account for the
+  // following: "If there is no parent grid, this value is equivalent to the
+  // initial value, `none`, and the grid container is not a subgrid."
+  // https://www.w3.org/TR/css-grid-2/#subgrid-listing
+  //
+  // Interestingly, specifying `subgrid` on a non-grid *will* compute to
+  // `subgrid` syntax.
   CSSValueList* list = CSSValueList::CreateSpaceSeparated();
-  if (is_subgrid) {
+  if (is_subgrid || (is_subgrid_specified && !grid)) {
     list->Append(
         *MakeGarbageCollected<CSSIdentifierValue>(CSSValueID::kSubgrid));
-  } else if (is_track_list_empty) {
+  } else if ((!is_subgrid_specified && is_track_list_empty) ||
+             (grid && is_subgrid_specified && !is_subgrid_valid)) {
     return CSSIdentifierValue::Create(CSSValueID::kNone);
   }
 
@@ -2046,8 +2057,8 @@ CSSValue* ComputedStyleUtils::ValueForGridTrackList(
   // Otherwise, the resolved value is the computed value, preserving repeat().
   OrderedNamedLinesCollector collector(
       computed_grid_track_list.ordered_named_grid_lines,
-      computed_grid_track_list.auto_repeat_ordered_named_grid_lines, is_subgrid,
-      !!grid);
+      computed_grid_track_list.auto_repeat_ordered_named_grid_lines,
+      is_subgrid_specified, !!grid);
   PopulateGridTrackListForNonGrid(list, collector, ng_track_list, style);
   return list;
 }
@@ -2917,6 +2928,18 @@ CSSValue* ComputedStyleUtils::CreateTransitionPropertyValue(
           .GetPropertyNameAtomicString());
 }
 
+CSSValue* ComputedStyleUtils::CreateTransitionBehaviorValue(
+    const CSSTransitionData::TransitionBehavior& type) {
+  switch (type) {
+    case CSSTransitionData::TransitionBehavior::kNormal:
+      return CSSIdentifierValue::Create(CSSValueID::kNormal);
+    case CSSTransitionData::TransitionBehavior::kAllowDiscrete:
+      return CSSIdentifierValue::Create(CSSValueID::kAllowDiscrete);
+  }
+  NOTREACHED() << " Unrecognized type: " << static_cast<unsigned>(type);
+  return nullptr;
+}
+
 CSSValue* ComputedStyleUtils::ValueForTransitionProperty(
     const CSSTransitionData* transition_data) {
   CSSValueList* list = CSSValueList::CreateCommaSeparated();
@@ -2931,6 +2954,19 @@ CSSValue* ComputedStyleUtils::ValueForTransitionProperty(
   return list;
 }
 
+CSSValue* ComputedStyleUtils::ValueForTransitionBehavior(
+    const CSSTransitionData* transition_data) {
+  CSSValueList* list = CSSValueList::CreateCommaSeparated();
+  if (transition_data) {
+    for (const auto& mode : transition_data->BehaviorList()) {
+      list->Append(*CreateTransitionBehaviorValue(mode));
+    }
+  } else {
+    list->Append(*CSSIdentifierValue::Create(CSSValueID::kNormal));
+  }
+  return list;
+}
+
 CSSValueID ValueForQuoteType(const QuoteType quote_type) {
   switch (quote_type) {
     case QuoteType::kNoOpen:
