diff --git a/third_party/blink/renderer/core/testing/dummy_modulator.cc b/third_party/blink/renderer/core/testing/dummy_modulator.cc
index c7d1531123895cb788baf7979b76797dba615ab5..1a8e400a94fb6a89c2899e0ecbfc809d65bbdd2b 100644
--- a/third_party/blink/renderer/core/testing/dummy_modulator.cc
+++ b/third_party/blink/renderer/core/testing/dummy_modulator.cc
@@ -159,4 +159,12 @@ ModuleScriptFetcher* DummyModulator::CreateModuleScriptFetcher(
 
 void DummyModulator::ProduceCacheModuleTreeTopLevel(ModuleScript*) {}
 
+String DummyModulator::GetIntegrityMetadataString(const KURL&) const {
+  return String();
+}
+
+IntegrityMetadataSet DummyModulator::GetIntegrityMetadata(const KURL&) const {
+  return IntegrityMetadataSet();
+}
+
 }  // namespace blink
