diff --git a/third_party/blink/renderer/core/css/selector_checker.cc b/third_party/blink/renderer/core/css/selector_checker.cc
index 0a5aae9deb41a6b65270600424a857e4ba8d6980..c7513b89c4e74317eae481fc37397255acc9fa67 100644
--- a/third_party/blink/renderer/core/css/selector_checker.cc
+++ b/third_party/blink/renderer/core/css/selector_checker.cc
@@ -47,6 +47,7 @@
 #include "third_party/blink/renderer/core/dom/nth_index_cache.h"
 #include "third_party/blink/renderer/core/dom/popover_data.h"
 #include "third_party/blink/renderer/core/dom/shadow_root.h"
+#include "third_party/blink/renderer/core/dom/slot_assignment_engine.h"
 #include "third_party/blink/renderer/core/dom/text.h"
 #include "third_party/blink/renderer/core/editing/frame_selection.h"
 #include "third_party/blink/renderer/core/frame/local_frame.h"
@@ -1639,6 +1640,18 @@ bool SelectorChecker::CheckPseudoClass(const SelectorCheckingContext& context,
       return element.IsOptionalFormControl();
     case CSSSelector::kPseudoRequired:
       return element.IsRequiredFormControl();
+    case CSSSelector::kPseudoUserInvalid:
+      CHECK(RuntimeEnabledFeatures::UserValidUserInvalidEnabled());
+      if (auto* form_control = DynamicTo<HTMLFormControlElement>(element)) {
+        return form_control->MatchesUserInvalidPseudo();
+      }
+      return false;
+    case CSSSelector::kPseudoUserValid:
+      CHECK(RuntimeEnabledFeatures::UserValidUserInvalidEnabled());
+      if (auto* form_control = DynamicTo<HTMLFormControlElement>(element)) {
+        return form_control->MatchesUserValidPseudo();
+      }
+      return false;
     case CSSSelector::kPseudoValid:
       return element.MatchesValidityPseudoClasses() && element.IsValidElement();
     case CSSSelector::kPseudoInvalid:
@@ -1696,10 +1709,16 @@ bool SelectorChecker::CheckPseudoClass(const SelectorCheckingContext& context,
         break;
       }
 
-      if (auto* html_element = DynamicTo<HTMLElement>(element)) {
-        return html_element->CachedDirectionality() == direction;
+      // Recomputing the slot assignment can update cached directionality.  In
+      // most cases it's OK for this code to be run when slot assignments are
+      // dirty; however for API calls like Element.matches() we should recalc
+      // them now.
+      Document& document = element.GetDocument();
+      if (mode_ == kQueryingRules && document.IsSlotAssignmentDirty()) {
+        document.GetSlotAssignmentEngine().RecalcSlotAssignments();
       }
-      break;
+
+      return element.CachedDirectionality() == direction;
     }
     case CSSSelector::kPseudoPopoverInTopLayer:
       if (auto* html_element = DynamicTo<HTMLElement>(element);
@@ -1827,10 +1846,16 @@ bool SelectorChecker::CheckPseudoClass(const SelectorCheckingContext& context,
         return false;
       }
       return !element.GetDocument().GetPage()->GetFocusController().IsActive();
-    case CSSSelector::kPseudoState: {
+    case CSSSelector::kPseudoStateDeprecatedSyntax: {
+      CHECK(RuntimeEnabledFeatures::CSSCustomStateDeprecatedSyntaxEnabled());
       return element.DidAttachInternals() &&
              element.EnsureElementInternals().HasState(selector.Value());
     }
+    case CSSSelector::kPseudoState: {
+      CHECK(RuntimeEnabledFeatures::CSSCustomStateNewSyntaxEnabled());
+      return element.DidAttachInternals() &&
+             element.EnsureElementInternals().HasState(selector.Argument());
+    }
     case CSSSelector::kPseudoHorizontal:
     case CSSSelector::kPseudoVertical:
     case CSSSelector::kPseudoDecrement:
