diff --git a/third_party/blink/renderer/core/url/dom_url.h b/third_party/blink/renderer/core/url/dom_url.h
index 663939a9f387b4f3b3740b2e50a1e77bfbcdcf50..6c8c712ca89f92edd548fae46784804c11f3ae36 100644
--- a/third_party/blink/renderer/core/url/dom_url.h
+++ b/third_party/blink/renderer/core/url/dom_url.h
@@ -58,6 +58,9 @@ class CORE_EXPORT DOMURL final : public ScriptWrappable, public DOMURLUtils {
   DOMURL(PassKey, const String& url, const KURL& base, ExceptionState&);
   ~DOMURL() override;
 
+  static bool canParse(const String& url);
+  static bool canParse(const String& url, const String& base);
+
   static String CreatePublicURL(ExecutionContext*, URLRegistrable*);
 
   KURL Url() const override { return url_; }
