diff --git a/third_party/blink/renderer/core/html/canvas/text_metrics.h b/third_party/blink/renderer/core/html/canvas/text_metrics.h
index 16ab5a7d14c7482a240cc0c4915918557e9a4db3..744c5dff4c78f80c7aaae43bb6040d727ba099c0 100644
--- a/third_party/blink/renderer/core/html/canvas/text_metrics.h
+++ b/third_party/blink/renderer/core/html/canvas/text_metrics.h
@@ -53,12 +53,14 @@ class CORE_EXPORT TextMetrics final : public ScriptWrappable {
   double fontBoundingBoxAscent() const { return font_bounding_box_ascent_; }
   double fontBoundingBoxDescent() const { return font_bounding_box_descent_; }
   double actualBoundingBoxAscent() const { return actual_bounding_box_ascent_; }
+  double alphabeticBaseline() const { return baselines_->alphabetic(); }
+  double hangingBaseline() const { return baselines_->hanging(); }
+  double ideographicBaseline() const { return baselines_->ideographic(); }
   double actualBoundingBoxDescent() const {
     return actual_bounding_box_descent_;
   }
   double emHeightAscent() const { return em_height_ascent_; }
   double emHeightDescent() const { return em_height_descent_; }
-  Baselines* getBaselines() const { return baselines_; }
 
   static float GetFontBaseline(const TextBaseline&, const SimpleFontData&);
 
