diff --git a/third_party/blink/renderer/core/editing/serializers/serialization.h b/third_party/blink/renderer/core/editing/serializers/serialization.h
index 4b5c1b9f7d430191e68d6b2a96f22c1127e78e3e..a50cbb5fd65c259623b12486446ef87902764389 100644
--- a/third_party/blink/renderer/core/editing/serializers/serialization.h
+++ b/third_party/blink/renderer/core/editing/serializers/serialization.h
@@ -65,7 +65,6 @@ DocumentFragment* CreateFragmentFromMarkupWithContext(Document&,
 DocumentFragment* CreateFragmentForInnerOuterHTML(const String&,
                                                   Element*,
                                                   ParserContentPolicy,
-                                                  const char* method,
                                                   bool include_shadow_roots,
                                                   ExceptionState&);
 DocumentFragment* CreateFragmentForTransformToFragment(
