diff --git a/third_party/blink/renderer/core/css/cssom/css_unit_values.idl b/third_party/blink/renderer/core/css/cssom/css_unit_values.idl
index 4696edd4fe7a48cd600c563c1ee619d026253785..c2b505e07feb5d9c267f5e4fb862ffcc79f40b55 100644
--- a/third_party/blink/renderer/core/css/cssom/css_unit_values.idl
+++ b/third_party/blink/renderer/core/css/cssom/css_unit_values.idl
@@ -12,9 +12,17 @@
 
   // <length>
   [NewObject] CSSUnitValue em(double value);
+  [NewObject] CSSUnitValue rem(double value);
   [NewObject] CSSUnitValue ex(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue rex(double value);
   [NewObject] CSSUnitValue ch(double value);
-  [NewObject] CSSUnitValue rem(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue rch(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue ic(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue ric(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue lh(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue rlh(double value);
+  [NewObject, RuntimeEnabled=CSSCapFontUnits] CSSUnitValue cap(double value);
+  [NewObject, RuntimeEnabled=CSSCapFontUnits] CSSUnitValue rcap(double value);
   [NewObject] CSSUnitValue vw(double value);
   [NewObject] CSSUnitValue vh(double value);
   [NewObject, RuntimeEnabled=CSSViewportUnits4] CSSUnitValue vi(double value);
@@ -57,10 +65,6 @@
   [NewObject] CSSUnitValue pc(double value);
   [NewObject] CSSUnitValue px(double value);
   [NewObject] CSSUnitValue Q(double value);
-  // TODO: Currently unsupported length units that are specified
-  // [NewObject] CSSUnitValue ic(double value);
-  // [NewObject] CSSUnitValue lh(double value);
-  // [NewObject] CSSUnitValue rlh(double value);
 
   // <angle>
   [NewObject] CSSUnitValue deg(double value);
@@ -80,6 +84,7 @@
   [NewObject] CSSUnitValue dpi(double value);
   [NewObject] CSSUnitValue dpcm(double value);
   [NewObject] CSSUnitValue dppx(double value);
+  [NewObject, RuntimeEnabled=CSSNumericFactoryCompleteness] CSSUnitValue x(double value);
 
   // <flex>
   [NewObject] CSSUnitValue fr(double value);
