diff --git a/third_party/blink/renderer/core/svg/svg_pattern_element.h b/third_party/blink/renderer/core/svg/svg_pattern_element.h
index ad6f8e67dbbd4984121626df692155a063249228..d27ccfe219873e0bc720a9a67f25bbec6597dda6 100644
--- a/third_party/blink/renderer/core/svg/svg_pattern_element.h
+++ b/third_party/blink/renderer/core/svg/svg_pattern_element.h
@@ -21,7 +21,6 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_CORE_SVG_SVG_PATTERN_ELEMENT_H_
 #define THIRD_PARTY_BLINK_RENDERER_CORE_SVG_SVG_PATTERN_ELEMENT_H_
 
-#include "third_party/blink/renderer/core/inspector/inspector_trace_events.h"
 #include "third_party/blink/renderer/core/svg/svg_animated_enumeration.h"
 #include "third_party/blink/renderer/core/svg/svg_element.h"
 #include "third_party/blink/renderer/core/svg/svg_fit_to_view_box.h"
@@ -71,7 +70,7 @@ class SVGPatternElement final : public SVGElement,
       const {
     return pattern_content_units_.Get();
   }
-  void InvalidatePattern(LayoutInvalidationReasonForTracing);
+  void InvalidatePattern();
   void InvalidateDependentPatterns();
 
   PatternAttributes CollectPatternAttributes() const;
