diff --git a/third_party/blink/renderer/core/streams/readable_stream_iterator_options.idl b/third_party/blink/renderer/core/streams/readable_stream_iterator_options.idl
new file mode 100644
index 0000000000000000000000000000000000000000..f9aac8693e039b74ebe4a273ed5479edba88140e
--- /dev/null
+++ b/third_party/blink/renderer/core/streams/readable_stream_iterator_options.idl
@@ -0,0 +1,9 @@
+// Copyright 2024 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// https://streams.spec.whatwg.org/#dictdef-readablestreamiteratoroptions
+
+dictionary ReadableStreamIteratorOptions {
+  boolean preventCancel = false;
+};
