diff --git a/third_party/blink/renderer/bindings/core/v8/use_counter_callback.cc b/third_party/blink/renderer/bindings/core/v8/use_counter_callback.cc
index 189d745f86bbbdc69825e84e760f0584acdd0f57..3e7add5a34c561689c6e9ea3bd30f7054205dc37 100644
--- a/third_party/blink/renderer/bindings/core/v8/use_counter_callback.cc
+++ b/third_party/blink/renderer/bindings/core/v8/use_counter_callback.cc
@@ -32,18 +32,12 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kBreakIterator:
       blink_feature = WebFeature::kBreakIterator;
       break;
-    case v8::Isolate::kLegacyConst:
-      blink_feature = WebFeature::kLegacyConst;
-      break;
     case v8::Isolate::kSloppyMode:
       blink_feature = WebFeature::kV8SloppyMode;
       break;
     case v8::Isolate::kStrictMode:
       blink_feature = WebFeature::kV8StrictMode;
       break;
-    case v8::Isolate::kStrongMode:
-      blink_feature = WebFeature::kV8StrongMode;
-      break;
     case v8::Isolate::kRegExpPrototypeStickyGetter:
       blink_feature = WebFeature::kV8RegExpPrototypeStickyGetter;
       break;
@@ -53,24 +47,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kRegExpPrototypeUnicodeGetter:
       blink_feature = WebFeature::kV8RegExpPrototypeUnicodeGetter;
       break;
-    case v8::Isolate::kIntlV8Parse:
-      blink_feature = WebFeature::kV8IntlV8Parse;
-      break;
-    case v8::Isolate::kIntlPattern:
-      blink_feature = WebFeature::kV8IntlPattern;
-      break;
-    case v8::Isolate::kIntlResolved:
-      blink_feature = WebFeature::kV8IntlResolved;
-      break;
-    case v8::Isolate::kPromiseChain:
-      blink_feature = WebFeature::kV8PromiseChain;
-      break;
-    case v8::Isolate::kPromiseAccept:
-      blink_feature = WebFeature::kV8PromiseAccept;
-      break;
-    case v8::Isolate::kPromiseDefer:
-      blink_feature = WebFeature::kV8PromiseDefer;
-      break;
     case v8::Isolate::kHtmlCommentInExternalScript:
       blink_feature = WebFeature::kV8HTMLCommentInExternalScript;
       break;
@@ -83,30 +59,15 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kForInInitializer:
       blink_feature = WebFeature::kV8ForInInitializer;
       break;
-    case v8::Isolate::kArrayProtectorDirtied:
-      blink_feature = WebFeature::kV8ArrayProtectorDirtied;
-      break;
     case v8::Isolate::kArraySpeciesModified:
       blink_feature = WebFeature::kV8ArraySpeciesModified;
       break;
     case v8::Isolate::kArrayPrototypeConstructorModified:
       blink_feature = WebFeature::kV8ArrayPrototypeConstructorModified;
       break;
-    case v8::Isolate::kArrayInstanceProtoModified:
-      blink_feature = WebFeature::kV8ArrayInstanceProtoModified;
-      break;
     case v8::Isolate::kArrayInstanceConstructorModified:
       blink_feature = WebFeature::kV8ArrayInstanceConstructorModified;
       break;
-    case v8::Isolate::kLegacyFunctionDeclaration:
-      blink_feature = WebFeature::kV8LegacyFunctionDeclaration;
-      break;
-    case v8::Isolate::kRegExpPrototypeSourceGetter:
-      blink_feature = WebFeature::kV8RegExpPrototypeSourceGetter;
-      break;
-    case v8::Isolate::kRegExpPrototypeOldFlagGetter:
-      blink_feature = WebFeature::kV8RegExpPrototypeOldFlagGetter;
-      break;
     case v8::Isolate::kDecimalWithLeadingZeroInStrictMode:
       blink_feature = WebFeature::kV8DecimalWithLeadingZeroInStrictMode;
       break;
@@ -128,12 +89,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kPromiseConstructorReturnedUndefined:
       blink_feature = WebFeature::kV8PromiseConstructorReturnedUndefined;
       break;
-    case v8::Isolate::kConstructorNonUndefinedPrimitiveReturn:
-      blink_feature = WebFeature::kV8ConstructorNonUndefinedPrimitiveReturn;
-      break;
-    case v8::Isolate::kLabeledExpressionStatement:
-      blink_feature = WebFeature::kV8LabeledExpressionStatement;
-      break;
     case v8::Isolate::kErrorCaptureStackTrace:
       blink_feature = WebFeature::kV8ErrorCaptureStackTrace;
       break;
@@ -149,9 +104,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kDeoptimizerDisableSpeculation:
       blink_feature = WebFeature::kV8DeoptimizerDisableSpeculation;
       break;
-    case v8::Isolate::kArrayPrototypeSortJSArrayModifiedPrototype:
-      blink_feature = WebFeature::kV8ArrayPrototypeSortJSArrayModifiedPrototype;
-      break;
     case v8::Isolate::kFunctionTokenOffsetTooLongForToString:
       blink_feature = WebFeature::kV8FunctionTokenOffsetTooLongForToString;
       break;
@@ -191,9 +143,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kStringLocaleCompare:
       blink_feature = WebFeature::kStringLocaleCompare;
       break;
-    case v8::Isolate::kStringToLocaleUpperCase:
-      blink_feature = WebFeature::kStringToLocaleUpperCase;
-      break;
     case v8::Isolate::kStringToLocaleLowerCase:
       blink_feature = WebFeature::kStringToLocaleLowerCase;
       break;
@@ -215,9 +164,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kAttemptOverrideReadOnlyOnPrototypeStrict:
       blink_feature = WebFeature::kV8AttemptOverrideReadOnlyOnPrototypeStrict;
       break;
-    case v8::Isolate::kOptimizedFunctionWithOneShotBytecode:
-      blink_feature = WebFeature::kV8OptimizedFunctionWithOneShotBytecode;
-      break;
     case v8::Isolate::kRegExpMatchIsTrueishOnNonJSRegExp:
       blink_feature = WebFeature::kV8RegExpMatchIsTrueishOnNonJSRegExp;
       break;
@@ -372,12 +318,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kWasmRefTypes:
       blink_feature = WebFeature::kV8WasmRefTypes;
       break;
-    case v8::Isolate::kWasmBulkMemory:
-      blink_feature = WebFeature::kV8WasmBulkMemory;
-      break;
-    case v8::Isolate::kWasmMultiValue:
-      blink_feature = WebFeature::kV8WasmMultiValue;
-      break;
     case v8::Isolate::kWasmExceptionHandling:
       blink_feature = WebFeature::kV8WasmExceptionHandling;
       break;
@@ -393,10 +333,6 @@ void UseCounterCallback(v8::Isolate* isolate,
     case v8::Isolate::kAsyncStackTaggingCreateTaskCall:
       blink_feature = WebFeature::kV8AsyncStackTaggingCreateTaskCall;
       break;
-    case v8::Isolate::kRegExpUnicodeSetIncompatibilitiesWithUnicodeMode:
-      blink_feature =
-          WebFeature::kV8RegExpUnicodeSetIncompatibilitiesWithUnicodeMode;
-      break;
     case v8::Isolate::kImportAssertionDeprecatedSyntax:
       blink_feature = WebFeature::kV8ImportAssertionDeprecatedSyntax;
       break;
