diff --git a/third_party/blink/renderer/core/timing/measure_memory/local_web_memory_measurer.h b/third_party/blink/renderer/core/timing/measure_memory/local_web_memory_measurer.h
index 7c6b92c9eeb456ac0c2ef3e7e0252673c2a13a98..423d79a967843a6d3a954b901bea1dae97e4f9fa 100644
--- a/third_party/blink/renderer/core/timing/measure_memory/local_web_memory_measurer.h
+++ b/third_party/blink/renderer/core/timing/measure_memory/local_web_memory_measurer.h
@@ -44,10 +44,7 @@ class LocalWebMemoryMeasurer : public v8::MeasureMemoryDelegate {
 
   // v8::MeasureMemoryDelegate overrides.
   bool ShouldMeasure(v8::Local<v8::Context> context) override;
-  void MeasurementComplete(
-      const std::vector<std::pair<v8::Local<v8::Context>, size_t>>&
-          context_sizes,
-      size_t unattributed_size) override;
+  void MeasurementComplete(v8::MeasureMemoryDelegate::Result result) override;
 
  private:
   LocalWebMemoryMeasurer(MeasureMemoryController*,
