diff --git a/third_party/blink/renderer/core/layout/svg/svg_layout_support.h b/third_party/blink/renderer/core/layout/svg/svg_layout_support.h
index 5f7853b5479d979b948fe5c3a901d88148ea5e74..c02bd249481a2c820946a9a4801fd50985d2a940 100644
--- a/third_party/blink/renderer/core/layout/svg/svg_layout_support.h
+++ b/third_party/blink/renderer/core/layout/svg/svg_layout_support.h
@@ -66,14 +66,6 @@ class CORE_EXPORT SVGLayoutSupport {
   static gfx::RectF ComputeVisualRectForText(const LayoutObject&,
                                              const gfx::RectF& text_bounds);
 
-  // Determine whether the passed location intersects a clip path referenced by
-  // the passed LayoutObject.
-  // |reference_box| is used to resolve 'objectBoundingBox' units/percentages,
-  // and can differ from the reference box of the passed LayoutObject.
-  static bool IntersectsClipPath(const LayoutObject&,
-                                 const gfx::RectF& reference_box,
-                                 const HitTestLocation&);
-
   // Important functions used by nearly all SVG layoutObjects centralizing
   // coordinate transformations / visual rect calculations
   static gfx::RectF LocalVisualRect(const LayoutObject&);
