diff --git a/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.h b/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.h
index d02e5ae9d63ad8fa44c1f2e46c56b3d481958191..7d34c39a48a8f0bfdd2c3b1beac1ebce43b87482 100644
--- a/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.h
+++ b/third_party/blink/renderer/platform/fonts/shaping/harfbuzz_face.h
@@ -44,6 +44,7 @@
 namespace blink {
 
 class FontPlatformData;
+class OpenTypeVerticalData;
 struct HarfBuzzFontData;
 
 // |HarfBuzzFace| is a thread specific data associated to |FontPlatformData|,
@@ -84,6 +85,8 @@ class HarfBuzzFace final : public RefCounted<HarfBuzzFace> {
 
   bool ShouldSubpixelPosition();
 
+  const OpenTypeVerticalData& VerticalData() const;
+
   static void Init();
 
  private:
