diff --git a/third_party/blink/renderer/core/layout/svg/transform_helper.cc b/third_party/blink/renderer/core/layout/svg/transform_helper.cc
index 46b5a7c0fbab3524f62c154a6278fe794082d25a..dbf39633f995f542a5f38e6b20afebe01eda8e6a 100644
--- a/third_party/blink/renderer/core/layout/svg/transform_helper.cc
+++ b/third_party/blink/renderer/core/layout/svg/transform_helper.cc
@@ -18,6 +18,18 @@
 
 namespace blink {
 
+namespace {
+
+bool StrokeBoundingBoxMayHaveChanged(const ComputedStyle& old_style,
+                                     const ComputedStyle& style) {
+  return old_style.StrokeWidth() != style.StrokeWidth() ||
+         old_style.CapStyle() != style.CapStyle() ||
+         old_style.StrokeMiterLimit() != style.StrokeMiterLimit() ||
+         old_style.JoinStyle() != style.JoinStyle();
+}
+
+}  // namespace
+
 static inline bool TransformOriginIsFixed(const ComputedStyle& style) {
   // If the transform box is view-box and the transform origin is absolute,
   // then is does not depend on the reference box. For fill-box, the origin
@@ -66,17 +78,57 @@ bool TransformHelper::DependsOnReferenceBox(const ComputedStyle& style) {
   return false;
 }
 
+bool TransformHelper::UpdateReferenceBoxDependency(
+    LayoutObject& layout_object) {
+  const bool transform_uses_reference_box =
+      DependsOnReferenceBox(layout_object.StyleRef());
+  UpdateReferenceBoxDependency(layout_object, transform_uses_reference_box);
+  return transform_uses_reference_box;
+}
+
+void TransformHelper::UpdateReferenceBoxDependency(
+    LayoutObject& layout_object,
+    bool transform_uses_reference_box) {
+  if (transform_uses_reference_box &&
+      layout_object.StyleRef().TransformBox() == ETransformBox::kViewBox) {
+    layout_object.SetSVGSelfOrDescendantHasViewportDependency();
+  } else {
+    layout_object.ClearSVGSelfOrDescendantHasViewportDependency();
+  }
+}
+
+bool TransformHelper::CheckReferenceBoxDependencies(
+    const ComputedStyle& old_style,
+    const ComputedStyle& style) {
+  ETransformBox transform_box = style.TransformBox();
+  // Changes to fill-box and view-box are handled by the
+  // `CheckForImplicitTransformChange()` implementations.
+  if (transform_box != ETransformBox::kStrokeBox &&
+      transform_box != ETransformBox::kBorderBox) {
+    return false;
+  }
+  return StrokeBoundingBoxMayHaveChanged(old_style, style);
+}
+
 gfx::RectF TransformHelper::ComputeReferenceBox(
     const LayoutObject& layout_object) {
   const ComputedStyle& style = layout_object.StyleRef();
   gfx::RectF reference_box;
-  if (style.TransformBox() == ETransformBox::kFillBox) {
-    reference_box = layout_object.ObjectBoundingBox();
-  } else {
-    DCHECK_EQ(style.TransformBox(), ETransformBox::kViewBox);
-    SVGLengthContext length_context(
-        DynamicTo<SVGElement>(layout_object.GetNode()));
-    reference_box.set_size(length_context.ResolveViewport());
+  switch (style.TransformBox()) {
+    case ETransformBox::kFillBox:
+    case ETransformBox::kContentBox:
+      reference_box = layout_object.ObjectBoundingBox();
+      break;
+    case ETransformBox::kStrokeBox:
+    case ETransformBox::kBorderBox:
+      reference_box = layout_object.StrokeBoundingBox();
+      break;
+    case ETransformBox::kViewBox: {
+      SVGLengthContext length_context(
+          DynamicTo<SVGElement>(layout_object.GetNode()));
+      reference_box.set_size(length_context.ResolveViewport());
+      break;
+    }
   }
   const float zoom = style.EffectiveZoom();
   if (zoom != 1)
@@ -85,12 +137,12 @@ gfx::RectF TransformHelper::ComputeReferenceBox(
 }
 
 AffineTransform TransformHelper::ComputeTransform(
-    const LayoutObject& layout_object,
+    UseCounter& use_counter,
+    const ComputedStyle& style,
+    const gfx::RectF& reference_box,
     ComputedStyle::ApplyTransformOrigin apply_transform_origin) {
-  const ComputedStyle& style = layout_object.StyleRef();
   if (DependsOnReferenceBox(style)) {
-    UseCounter::Count(layout_object.GetDocument(),
-                      WebFeature::kTransformUsesBoxSizeOnSVG);
+    UseCounter::Count(use_counter, WebFeature::kTransformUsesBoxSizeOnSVG);
   }
 
   // CSS transforms operate with pre-scaled lengths. To make this work with SVG
@@ -105,7 +157,6 @@ AffineTransform TransformHelper::ComputeTransform(
   // clipPath. See
   // https://svgwg.org/svg2-draft/coords.html#ObjectBoundingBoxUnits
   gfx::Transform transform;
-  gfx::RectF reference_box = ComputeReferenceBox(layout_object);
   style.ApplyTransform(transform, nullptr, reference_box,
                        ComputedStyle::kIncludeTransformOperations,
                        apply_transform_origin,
@@ -118,10 +169,30 @@ AffineTransform TransformHelper::ComputeTransform(
   return AffineTransform::FromTransform(transform);
 }
 
+AffineTransform TransformHelper::ComputeTransformIncludingMotion(
+    const SVGElement& element,
+    const gfx::RectF& reference_box) {
+  const LayoutObject& layout_object = *element.GetLayoutObject();
+  if (layout_object.HasTransform() || element.HasMotionTransform()) {
+    AffineTransform matrix =
+        ComputeTransform(element.GetDocument(), layout_object.StyleRef(),
+                         reference_box, ComputedStyle::kIncludeTransformOrigin);
+    element.ApplyMotionTransform(matrix);
+    return matrix;
+  }
+  return AffineTransform();
+}
+
+AffineTransform TransformHelper::ComputeTransformIncludingMotion(
+    const SVGElement& element) {
+  const LayoutObject& layout_object = *element.GetLayoutObject();
+  const gfx::RectF reference_box = ComputeReferenceBox(layout_object);
+  return ComputeTransformIncludingMotion(element, reference_box);
+}
+
 gfx::PointF TransformHelper::ComputeTransformOrigin(
-    const LayoutObject& layout_object) {
-  const auto& style = layout_object.StyleRef();
-  gfx::RectF reference_box = ComputeReferenceBox(layout_object);
+    const ComputedStyle& style,
+    const gfx::RectF& reference_box) {
   gfx::PointF origin(FloatValueForLength(style.GetTransformOrigin().X(),
                                          reference_box.width()) +
                          reference_box.x(),
