diff --git a/third_party/blink/renderer/core/view_transition/view_transition.idl b/third_party/blink/renderer/core/view_transition/view_transition.idl
index 33cc34e6d7b6d21fbb6abf1113df75e30af76814..7ba16fa93fec4960c20dead4a9bb2e1efbd88aa5 100644
--- a/third_party/blink/renderer/core/view_transition/view_transition.idl
+++ b/third_party/blink/renderer/core/view_transition/view_transition.idl
@@ -27,4 +27,8 @@
   // has successfully run. Note that if the callback promise rejects, then this
   // promise is rejected as well.
   readonly attribute Promise<void> updateCallbackDone;
+
+  // FlyWeb (Chromium 125): the active list of types for this transition,
+  // selectable with the :active-view-transition-type() pseudo-class.
+  [RuntimeEnabled=ViewTransitionTypes] readonly attribute ViewTransitionTypeSet types;
 };
