diff --git a/third_party/blink/renderer/core/testing/dummy_modulator.h b/third_party/blink/renderer/core/testing/dummy_modulator.h
index 5b1576b635d9195ff6f4ef6d420db8b793ec0508..2cdf769da4643ec861a3d9972596d483f15dda2b 100644
--- a/third_party/blink/renderer/core/testing/dummy_modulator.h
+++ b/third_party/blink/renderer/core/testing/dummy_modulator.h
@@ -71,6 +71,8 @@ class DummyModulator : public Modulator {
       ModuleScriptCustomFetchType,
       base::PassKey<ModuleScriptLoader>) override;
   void ProduceCacheModuleTreeTopLevel(ModuleScript*) override;
+  String GetIntegrityMetadataString(const KURL&) const override;
+  IntegrityMetadataSet GetIntegrityMetadata(const KURL&) const override;
 
   Member<ModuleRecordResolver> resolver_;
 };
