diff --git a/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d_state.cc b/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d_state.cc
index 0cd4c8c677df581f0019b1e4d9c468e2ce5ad17b..344952452957f3cb38f131d7f2b149797dcaf1ff 100644
--- a/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d_state.cc
+++ b/third_party/blink/renderer/modules/canvas/canvas2d/canvas_rendering_context_2d_state.cc
@@ -397,8 +397,9 @@ sk_sp<PaintFilter> CanvasRenderingContext2DState::GetFilterForOffscreenCanvas(
   cc::PaintFlags stroke_flags_for_filter;
   stroke_style_.ApplyToFlags(stroke_flags_for_filter, 1.0f);
 
+  const gfx::SizeF canvas_viewport(canvas_size);
   FilterEffectBuilder filter_effect_builder(
-      gfx::RectF(gfx::SizeF(canvas_size)),
+      gfx::RectF(canvas_viewport), canvas_viewport,
       1.0f,  // Deliberately ignore zoom on the canvas element.
       &fill_flags_for_filter, &stroke_flags_for_filter);
 
@@ -467,8 +468,9 @@ sk_sp<PaintFilter> CanvasRenderingContext2DState::GetFilter(
   cc::PaintFlags stroke_flags_for_filter;
   stroke_style_.ApplyToFlags(stroke_flags_for_filter, 1.0f);
 
+  const gfx::SizeF canvas_viewport(canvas_size);
   FilterEffectBuilder filter_effect_builder(
-      gfx::RectF(gfx::SizeF(canvas_size)),
+      gfx::RectF(canvas_viewport), canvas_viewport,
       1.0f,  // Deliberately ignore zoom on the canvas element.
       &fill_flags_for_filter, &stroke_flags_for_filter);
 
