diff --git a/third_party/blink/renderer/core/layout/layout_inline.cc b/third_party/blink/renderer/core/layout/layout_inline.cc
index ce56ead24e7995399272e44117bb38c29b4b0ec9..6529deafeb564e951dc3c305a8428950d6ec732b 100644
--- a/third_party/blink/renderer/core/layout/layout_inline.cc
+++ b/third_party/blink/renderer/core/layout/layout_inline.cc
@@ -767,22 +767,6 @@ PhysicalRect LayoutInline::PhysicalVisualOverflowRect() const {
   return overflow_rect;
 }
 
-PhysicalRect LayoutInline::ReferenceBoxForClipPath() const {
-  NOT_DESTROYED();
-  // The spec just says to use the border box as clip-path reference box. It
-  // doesn't say what to do if there are multiple lines. Gecko uses the first
-  // fragment in that case. We'll do the same here (but correctly with respect
-  // to writing-mode - Gecko has some issues there).
-  // See crbug.com/641907
-  if (IsInLayoutNGInlineFormattingContext()) {
-    NGInlineCursor cursor;
-    cursor.MoveTo(*this);
-    if (cursor)
-      return cursor.Current().RectInContainerFragment();
-  }
-  return PhysicalRect();
-}
-
 bool LayoutInline::MapToVisualRectInAncestorSpaceInternal(
     const LayoutBoxModelObject* ancestor,
     TransformState& transform_state,
