diff --git a/third_party/blink/renderer/bindings/scripts/check_generated_file_list.py b/third_party/blink/renderer/bindings/scripts/check_generated_file_list.py
index 41b1fdeb100e103a8ae64dbe6fb05a9be49ac85a..4f1f16bcaca61ddab3395d53edc70ea35bb454aa 100644
--- a/third_party/blink/renderer/bindings/scripts/check_generated_file_list.py
+++ b/third_party/blink/renderer/bindings/scripts/check_generated_file_list.py
@@ -86,6 +86,11 @@ def main():
     web_idl_database = bind_gen.package_initializer.package_initializer(
     ).web_idl_database()
     idl_definitions = {
+        "async_iterator": [
+            interface.async_iterator
+            for interface in web_idl_database.interfaces
+            if interface.async_iterator
+        ],
         "callback_function":
         web_idl_database.callback_functions,
         "callback_interface":
