diff --git a/third_party/blink/renderer/core/html/html_attribute_names.json5 b/third_party/blink/renderer/core/html/html_attribute_names.json5
index ca28e7b550cb0f3fc2b7011cd34e8372e10138a9..3083cc88082692aa1396a33ef697b0978b54c9dc 100644
--- a/third_party/blink/renderer/core/html/html_attribute_names.json5
+++ b/third_party/blink/renderer/core/html/html_attribute_names.json5
@@ -310,7 +310,9 @@
     "select",
     "selected",
     "shadowroot",
+    "shadowrootclonable",
     "shadowrootmode",
+    "shadowrootserializable",
     "shadowrootdelegatesfocus",
     "shape",
     "size",
