diff --git a/third_party/blink/renderer/core/animation/css_content_visibility_interpolation_type.h b/third_party/blink/renderer/core/animation/css_content_visibility_interpolation_type.h
index 3953f503796b451c79d09dd1495d66caf9168f59..e1a03e173b0f2e3110afe283eeb764ed4177ce1f 100644
--- a/third_party/blink/renderer/core/animation/css_content_visibility_interpolation_type.h
+++ b/third_party/blink/renderer/core/animation/css_content_visibility_interpolation_type.h
@@ -30,6 +30,7 @@ class CSSContentVisibilityInterpolationType : public CSSInterpolationType {
   void ApplyStandardPropertyValue(const InterpolableValue&,
                                   const NonInterpolableValue*,
                                   StyleResolverState&) const final;
+  bool IsDiscrete() const final { return true; }
 
  private:
   InterpolationValue CreateContentVisibilityValue(EContentVisibility) const;
