diff --git a/third_party/blink/renderer/modules/geolocation/BUILD.gn b/third_party/blink/renderer/modules/geolocation/BUILD.gn
index 2c152d285f34dedaa5074d76f51e64755029736f..853b62312304543a0ff3b85f59d4214e0c7f6bf7 100644
--- a/third_party/blink/renderer/modules/geolocation/BUILD.gn
+++ b/third_party/blink/renderer/modules/geolocation/BUILD.gn
@@ -16,6 +16,7 @@ blink_modules_sources("geolocation") {
     "geolocation_position_error.h",
     "geolocation_watchers.cc",
     "geolocation_watchers.h",
+    "geoposition.cc",
     "geoposition.h",
   ]
 }
