diff --git a/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py b/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
index 0ad1a5724f74ba343acdb69941bbc7f2188834e5..5b91787e546bd63cee1e2409c9eaaa9cfebd4333 100644
--- a/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
+++ b/third_party/blink/renderer/bindings/scripts/bind_gen/__init__.py
@@ -33,6 +33,7 @@ def _setup_sys_path():
 
 _setup_sys_path()
 
+from .async_iterator import generate_async_iterators
 from .callback_function import generate_callback_functions
 from .callback_interface import generate_callback_interfaces
 from .dictionary import generate_dictionaries
