diff --git a/third_party/blink/renderer/core/css/css_properties.json5 b/third_party/blink/renderer/core/css/css_properties.json5
index 871edc0924403a04bf16f4d5602f421d3882473c..221a470797bb9c14a92db3b4a6ade61605df0ce1 100644
--- a/third_party/blink/renderer/core/css/css_properties.json5
+++ b/third_party/blink/renderer/core/css/css_properties.json5
@@ -991,6 +991,19 @@
       valid_for_marker: true,
       is_animation_property: true,
     },
+    {
+      name: "transition-behavior",
+      property_methods: ["ParseSingleValue", "CSSValueFromComputedStyleInternal", "InitialValue"],
+      typedom_types: ["Keyword"],
+      separator: ",",
+      style_builder_template: "transition",
+      style_builder_template_args: {
+        attribute: "Behavior",
+      },
+      valid_for_marker: true,
+      is_animation_property: true,
+      runtime_flag: "CSSTransitionDiscrete",
+    },
     {
       name: "transition-timing-function",
       property_methods: ["ParseSingleValue", "CSSValueFromComputedStyleInternal", "InitialValue"],
@@ -2446,7 +2459,7 @@
       type_name: "ClipPathOperation",
       computed_style_custom_functions: ["getter", "setter"],
       converter: "ConvertClipPath",
-      keywords: ["none"],
+      keywords: ["border-box", "padding-box", "content-box", "margin-box", "fill-box", "stroke-box", "view-box", "none"],
       typedom_types: ["Keyword"],
     },
     {
@@ -5228,7 +5241,7 @@
       name: "transform-box",
       property_methods: ["CSSValueFromComputedStyleInternal"],
       field_template: "keyword",
-      keywords: ["fill-box", "view-box"],
+      keywords: ["content-box", "border-box", "fill-box", "stroke-box", "view-box"],
       default_value: "view-box",
       typedom_types: ["Keyword"],
       computable: false,
@@ -7387,7 +7400,8 @@
       name: "transition",
       longhands: [
         "transition-property", "transition-duration",
-        "transition-timing-function", "transition-delay"
+        "transition-timing-function", "transition-delay",
+        "transition-behavior",
       ],
       property_methods: ["ParseShorthand", "CSSValueFromComputedStyleInternal"],
     },
