diff --git a/third_party/blink/renderer/core/css/media_feature_names.json5 b/third_party/blink/renderer/core/css/media_feature_names.json5
index 54f04c6bbfea795e0dc0c74409663daad11dbf0f..ddd523f79d60096bb75cbd180ddc446a0d338c96 100644
--- a/third_party/blink/renderer/core/css/media_feature_names.json5
+++ b/third_party/blink/renderer/core/css/media_feature_names.json5
@@ -71,5 +71,6 @@
     "horizontal-viewport-segments",
     "vertical-viewport-segments",
     "stuck",
+    "scripting",
   ],
 }
