diff --git a/third_party/blink/renderer/core/html/custom/custom_state_set.cc b/third_party/blink/renderer/core/html/custom/custom_state_set.cc
index 56f299e94db96a2c846d448c8672c824c3b19695..a2d6391cdfa0bd8d7f5e2f473759bbaf9f17baeb 100644
--- a/third_party/blink/renderer/core/html/custom/custom_state_set.cc
+++ b/third_party/blink/renderer/core/html/custom/custom_state_set.cc
@@ -7,6 +7,7 @@
 #include "third_party/blink/renderer/core/css/css_selector.h"
 #include "third_party/blink/renderer/core/css/parser/css_parser_idioms.h"
 #include "third_party/blink/renderer/core/dom/element.h"
+#include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 
 namespace blink {
 
@@ -50,6 +51,14 @@ void CustomStateSet::Trace(Visitor* visitor) const {
 }
 
 void CustomStateSet::add(const String& value, ExceptionState& exception_state) {
+  if (RuntimeEnabledFeatures::CSSCustomStateNewSyntaxEnabled()) {
+    if (!list_.Contains(value)) {
+      list_.push_back(value);
+    }
+    InvalidateStyle();
+    return;
+  }
+
   // https://wicg.github.io/custom-state-pseudo-class/#dom-customstateset-add
 
   // 1. If value does not match to <dashed-ident>, then throw a "SyntaxError"
@@ -127,6 +136,7 @@ void CustomStateSet::InvalidateStyle() const {
   // performance in documents with various custom state pseudo classes by
   // having blink::InvalidationSet for each of states.
   element_->PseudoStateChanged(CSSSelector::kPseudoState);
+  element_->PseudoStateChanged(CSSSelector::kPseudoStateDeprecatedSyntax);
 }
 
 }  // namespace blink
