diff --git a/third_party/blink/renderer/core/mathml/mathml_tag_names.json5 b/third_party/blink/renderer/core/mathml/mathml_tag_names.json5
index 1274e5df09dbedca334a98ed753f69b355ed5b52..fe44406a1996259e69fbf8967ae118cbbcb202c9 100644
--- a/third_party/blink/renderer/core/mathml/mathml_tag_names.json5
+++ b/third_party/blink/renderer/core/mathml/mathml_tag_names.json5
@@ -109,7 +109,7 @@
     },
     {
       name: "mtd",
-      interfaceName: "MathMLElement",
+      interfaceName: "MathMLTableCellElement",
     },
     {
       name: "mtr",
