diff --git a/third_party/blink/renderer/modules/clipboard/clipboard_item.cc b/third_party/blink/renderer/modules/clipboard/clipboard_item.cc
index 723ad56e3e307de90209dedef6642ca9ca11a1ea..9cd3c8f3d7dae19d096e88d78e50a333d5201400 100644
--- a/third_party/blink/renderer/modules/clipboard/clipboard_item.cc
+++ b/third_party/blink/renderer/modules/clipboard/clipboard_item.cc
@@ -7,6 +7,7 @@
 #include "net/base/mime_util.h"
 #include "third_party/blink/public/common/features.h"
 #include "third_party/blink/renderer/bindings/core/v8/script_promise_resolver.h"
+#include "third_party/blink/renderer/core/clipboard/clipboard_mime_types.h"
 #include "third_party/blink/renderer/core/dom/dom_exception.h"
 #include "third_party/blink/renderer/modules/clipboard/clipboard.h"
 #include "third_party/blink/renderer/platform/bindings/script_state.h"
@@ -84,6 +85,15 @@ ScriptPromise ClipboardItem::getType(ScriptState* script_state,
   return ScriptPromise();
 }
 
+// static
+bool ClipboardItem::supports(const String& type) {
+  if (type == kMimeTypeImagePng || type == kMimeTypeTextPlain ||
+      type == kMimeTypeTextHTML) {
+    return true;
+  }
+  return !Clipboard::ParseWebCustomFormat(type).empty();
+}
+
 void ClipboardItem::Trace(Visitor* visitor) const {
   visitor->Trace(items_);
   ScriptWrappable::Trace(visitor);
