diff --git a/third_party/blink/renderer/core/css/css_import_rule.cc b/third_party/blink/renderer/core/css/css_import_rule.cc
index 955e0a4eba8adc7d4138541523a0108c696057f9..5b626cbdd0138ff79d1c902c7262d94645caab82 100644
--- a/third_party/blink/renderer/core/css/css_import_rule.cc
+++ b/third_party/blink/renderer/core/css/css_import_rule.cc
@@ -64,6 +64,13 @@ String CSSImportRule::cssText() const {
     }
   }
 
+  if (String supports = import_rule_->GetSupportsString();
+      supports != g_null_atom) {
+    result.Append(" supports(");
+    result.Append(supports);
+    result.Append(")");
+  }
+
   if (import_rule_->MediaQueries()) {
     String media_text = import_rule_->MediaQueries()->MediaText();
     if (!media_text.empty()) {
@@ -97,6 +104,10 @@ String CSSImportRule::layerName() const {
   return import_rule_->GetLayerNameAsString();
 }
 
+String CSSImportRule::supportsText() const {
+  return import_rule_->GetSupportsString();
+}
+
 void CSSImportRule::Reattach(StyleRuleBase*) {
   // FIXME: Implement when enabling caching for stylesheets with import rules.
   NOTREACHED();
