diff --git a/third_party/blink/renderer/core/svg/svg_attribute_names.json5 b/third_party/blink/renderer/core/svg/svg_attribute_names.json5
index 9bb14a4557f120ae00a1edaa7a00d757ede0cca6..8efa4107ae22cee6f11f1a103bf4ba71f4d9caf4 100644
--- a/third_party/blink/renderer/core/svg/svg_attribute_names.json5
+++ b/third_party/blink/renderer/core/svg/svg_attribute_names.json5
@@ -32,6 +32,7 @@
     "color-interpolation",
     "color-interpolation-filters",
     "color-rendering",
+    "crossorigin",
     "cursor",
     "cx",
     "cy",
