diff --git a/third_party/blink/renderer/core/editing/serializers/serialization.cc b/third_party/blink/renderer/core/editing/serializers/serialization.cc
index 6588f95b0eef16f51ac92445d426a8840cbf01f8..ccdd95f35d0f6293a433b7c1707ec0d8cc7834a3 100644
--- a/third_party/blink/renderer/core/editing/serializers/serialization.cc
+++ b/third_party/blink/renderer/core/editing/serializers/serialization.cc
@@ -660,7 +660,6 @@ DocumentFragment* CreateFragmentForInnerOuterHTML(
     const String& markup,
     Element* context_element,
     ParserContentPolicy parser_content_policy,
-    const char* method,
     bool include_shadow_roots,
     ExceptionState& exception_state) {
   DCHECK(context_element);
@@ -785,7 +784,7 @@ DocumentFragment* CreateContextualFragment(
   DCHECK(element);
 
   DocumentFragment* fragment = CreateFragmentForInnerOuterHTML(
-      markup, element, parser_content_policy, "createContextualFragment",
+      markup, element, parser_content_policy,
       /*include_shadow_roots=*/false, exception_state);
   if (!fragment)
     return nullptr;
