diff --git a/third_party/blink/renderer/core/dom/shadow_root.h b/third_party/blink/renderer/core/dom/shadow_root.h
index 26796e955550ddeaae310705541d84ec74a5ede3..fbb7a10a8a247af650f2e84697248f545371573f 100644
--- a/third_party/blink/renderer/core/dom/shadow_root.h
+++ b/third_party/blink/renderer/core/dom/shadow_root.h
@@ -128,6 +128,7 @@ class CORE_EXPORT ShadowRoot final : public DocumentFragment,
   String innerHTML() const;
   String getInnerHTML(const GetInnerHTMLOptions* options) const;
   void setInnerHTML(const String&, ExceptionState& = ASSERT_NO_EXCEPTION);
+  void setHTMLUnsafe(const String& html, ExceptionState&);
 
   Node* Clone(Document&, CloneChildrenFlag) const override;
 
