diff --git a/third_party/blink/renderer/core/style/content_data.h b/third_party/blink/renderer/core/style/content_data.h
index 4619d46d9110bf51cbc0af69b342267f1c9c1c3b..0c6690105070a40b0b70da9702b733c3b2872f19 100644
--- a/third_party/blink/renderer/core/style/content_data.h
+++ b/third_party/blink/renderer/core/style/content_data.h
@@ -158,6 +158,14 @@ class AltTextContentData final : public ContentData {
   explicit AltTextContentData(const String& text) : text_(text) {}
 
   String GetText() const { return text_; }
+  String ConcatenateAltText() const {
+    StringBuilder alt_text;
+    for (const ContentData* content_data = this; content_data;
+         content_data = content_data->Next()) {
+      alt_text.Append(To<AltTextContentData>(content_data)->GetText());
+    }
+    return alt_text.ToString();
+  }
   void SetText(const String& text) { text_ = text; }
 
   bool IsAltText() const override { return true; }
