diff --git a/third_party/blink/renderer/core/svg/proxy_svg_resource_client.h b/third_party/blink/renderer/core/svg/proxy_svg_resource_client.h
new file mode 100644
index 0000000000000000000000000000000000000000..9e594722ee0055db93b217bba1f72e0e12fa5622
--- /dev/null
+++ b/third_party/blink/renderer/core/svg/proxy_svg_resource_client.h
@@ -0,0 +1,54 @@
+// Copyright 2023 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef THIRD_PARTY_BLINK_RENDERER_CORE_SVG_PROXY_SVG_RESOURCE_CLIENT_H_
+#define THIRD_PARTY_BLINK_RENDERER_CORE_SVG_PROXY_SVG_RESOURCE_CLIENT_H_
+
+#include "third_party/blink/renderer/core/css/css_value.h"
+#include "third_party/blink/renderer/core/loader/resource/image_resource_observer.h"
+#include "third_party/blink/renderer/core/svg/svg_resource_client.h"
+#include "third_party/blink/renderer/platform/heap/collection_support/heap_hash_counted_set.h"
+
+namespace blink {
+
+// A SVGResourceClients that proxies notifications to ImageResourceObservers.
+class ProxySVGResourceClient final
+    : public GarbageCollected<ProxySVGResourceClient>,
+      public SVGResourceClient {
+ public:
+  explicit ProxySVGResourceClient(const CSSValue& value) : css_value_(value) {}
+
+  bool AddClient(ImageResourceObserver* observer) {
+    const bool was_empty = clients_.empty();
+    clients_.insert(observer);
+    return was_empty;
+  }
+  bool RemoveClient(ImageResourceObserver* observer) {
+    clients_.erase(observer);
+    return clients_.empty();
+  }
+
+  void Trace(Visitor* visitor) const override {
+    visitor->Trace(clients_);
+    visitor->Trace(css_value_);
+    SVGResourceClient::Trace(visitor);
+  }
+
+ private:
+  void ResourceContentChanged(SVGResource*) override {
+    for (auto& entry : clients_) {
+      // FlyWeb: WrappedImagePtr is not const in 116.
+      entry.key->ImageChanged(
+          static_cast<WrappedImagePtr>(const_cast<CSSValue*>(css_value_.Get())),
+          ImageResourceObserver::CanDeferInvalidation::kNo);
+    }
+  }
+
+  HeapHashCountedSet<Member<ImageResourceObserver>> clients_;
+  Member<const CSSValue> css_value_;
+};
+
+}  // namespace blink
+
+#endif  // THIRD_PARTY_BLINK_RENDERER_CORE_SVG_PROXY_SVG_RESOURCE_CLIENT_H_
