diff --git a/third_party/blink/renderer/core/url/url.idl b/third_party/blink/renderer/core/url/url.idl
index c914c23f2f678eb8ee33ea2483228b186dbae7bb..1892a0f68cf8bcc76a3a9deb6028df4b757a9293 100644
--- a/third_party/blink/renderer/core/url/url.idl
+++ b/third_party/blink/renderer/core/url/url.idl
@@ -33,6 +33,9 @@
 ] interface URL {
     [RaisesException] constructor(USVString url, optional USVString base);
 
+    [RuntimeEnabled=URLParse] static URL? parse(USVString url, optional USVString base);
+    [RuntimeEnabled=URLCanParse] static boolean canParse(USVString url, optional USVString base);
+
     [RaisesException=Setter] stringifier attribute USVString href;
     readonly attribute USVString origin;
     attribute USVString protocol;
