diff --git a/third_party/blink/renderer/platform/graphics/graphics_types.h b/third_party/blink/renderer/platform/graphics/graphics_types.h
index 814eb20deacb9f8e4a10d3f3bc529c60bb20e802..3f5620948761e0e715e3c888d74961a7658bf3ce 100644
--- a/third_party/blink/renderer/platform/graphics/graphics_types.h
+++ b/third_party/blink/renderer/platform/graphics/graphics_types.h
@@ -194,6 +194,8 @@ enum TextAlign {
   kRightTextAlign
 };
 
+enum TextPaintOrder { kFillStroke, kStrokeFill };
+
 enum TextDrawingMode {
   kTextModeFill = 1 << 0,
   kTextModeStroke = 1 << 1,
