diff --git a/third_party/blink/renderer/core/html/html_bdi_element.cc b/third_party/blink/renderer/core/html/html_bdi_element.cc
index 25f3ab2ec35542b5edb5d9ac9bbb292c3e4ddad9..c256b936ffe4dc2f3d6a3c05ba9d51d733513108 100644
--- a/third_party/blink/renderer/core/html/html_bdi_element.cc
+++ b/third_party/blink/renderer/core/html/html_bdi_element.cc
@@ -11,7 +11,7 @@ HTMLBDIElement::HTMLBDIElement(Document& document)
   // <bdi> defaults to dir="auto"
   // https://html.spec.whatwg.org/C/#the-bdi-element
   SetSelfOrAncestorHasDirAutoAttribute();
-  GetDocument().SetDirAttributeDirty();
+  GetDocument().SetHasDirAttribute();
 }
 
 }  // namespace blink
