diff --git a/third_party/blink/renderer/core/css/css_font_selector.cc b/third_party/blink/renderer/core/css/css_font_selector.cc
index 35c6638d9b6da4eeeb7ef761e1952b9f69e2e765..47c606cd183de5f7176ac7dac14d1528113315ab 100644
--- a/third_party/blink/renderer/core/css/css_font_selector.cc
+++ b/third_party/blink/renderer/core/css/css_font_selector.cc
@@ -31,6 +31,7 @@
 #include "third_party/blink/renderer/core/css/css_segmented_font_face.h"
 #include "third_party/blink/renderer/core/css/css_value_list.h"
 #include "third_party/blink/renderer/core/css/font_face_set_document.h"
+#include "third_party/blink/renderer/core/css/font_size_functions.h"
 #include "third_party/blink/renderer/core/css/resolver/scoped_style_resolver.h"
 #include "third_party/blink/renderer/core/css/resolver/style_resolver.h"
 #include "third_party/blink/renderer/core/dom/document.h"
@@ -38,6 +39,7 @@
 #include "third_party/blink/renderer/core/frame/settings.h"
 #include "third_party/blink/renderer/platform/fonts/font_cache.h"
 #include "third_party/blink/renderer/platform/fonts/font_selector_client.h"
+#include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 
 namespace blink {
 
@@ -265,6 +267,17 @@ scoped_refptr<FontData> CSSFontSelector::GetFontData(
 
   scoped_refptr<SimpleFontData> font_data =
       FontCache::Get().GetFontData(request_description, settings_family_name);
+  if (font_data && request_description.HasSizeAdjust()) {
+    DCHECK(RuntimeEnabledFeatures::CSSFontSizeAdjustEnabled());
+    if (auto adjusted_size =
+            FontSizeFunctions::MetricsMultiplierAdjustedFontSize(
+                font_data.get(), request_description)) {
+      FontDescription size_adjusted_description(request_description);
+      size_adjusted_description.SetAdjustedSize(adjusted_size.value());
+      font_data = FontCache::Get().GetFontData(size_adjusted_description,
+                                               settings_family_name);
+    }
+  }
 
   ReportFontLookupByUniqueOrFamilyName(settings_family_name,
                                        request_description, font_data);
