diff --git a/third_party/blink/renderer/bindings/idl_in_core.gni b/third_party/blink/renderer/bindings/idl_in_core.gni
index 08410ec268174507329fe03ccc3c151a03886bd2..5ac45d287e95e6af344163da6f58476dc5b4c268 100644
--- a/third_party/blink/renderer/bindings/idl_in_core.gni
+++ b/third_party/blink/renderer/bindings/idl_in_core.gni
@@ -520,6 +520,7 @@ static_idl_files_in_core = get_path_info(
           "//third_party/blink/renderer/core/navigation_api/navigate_event.idl",
           "//third_party/blink/renderer/core/navigation_api/navigate_event_init.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation.idl",
+  "//third_party/blink/renderer/core/navigation_api/navigation_activation.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_current_entry_change_event.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_current_entry_change_event_init.idl",
           "//third_party/blink/renderer/core/navigation_api/navigation_destination.idl",
@@ -554,6 +555,7 @@ static_idl_files_in_core = get_path_info(
           "//third_party/blink/renderer/core/streams/readable_stream_default_reader.idl",
           "//third_party/blink/renderer/core/streams/readable_stream_generic_reader.idl",
           "//third_party/blink/renderer/core/streams/readable_stream_get_reader_options.idl",
+          "//third_party/blink/renderer/core/streams/readable_stream_iterator_options.idl",
           "//third_party/blink/renderer/core/streams/readable_stream_read_result.idl",
           "//third_party/blink/renderer/core/streams/readable_writable_pair.idl",
           "//third_party/blink/renderer/core/streams/stream_pipe_options.idl",
