diff --git a/third_party/blink/renderer/core/css/css_length_resolver.h b/third_party/blink/renderer/core/css/css_length_resolver.h
index 58388e2762b46c36bea699883e06fe91058bf350..597662e4239c16e1457d18631ac9591ebaeeeaab 100644
--- a/third_party/blink/renderer/core/css/css_length_resolver.h
+++ b/third_party/blink/renderer/core/css/css_length_resolver.h
@@ -32,6 +32,8 @@ class CORE_EXPORT CSSLengthResolver {
   virtual float RicFontSize(float zoom) const = 0;
   virtual float LineHeight(float zoom) const = 0;
   virtual float RootLineHeight(float zoom) const = 0;
+  virtual float CapFontSize(float zoom) const = 0;
+  virtual float RcapFontSize(float zoom) const = 0;
 
   // Other sizes are not pre-zoomed.
   virtual double ViewportWidth() const = 0;
