diff --git a/third_party/blink/renderer/modules/peerconnection/rtc_rtp_receiver.cc b/third_party/blink/renderer/modules/peerconnection/rtc_rtp_receiver.cc
index 3a45927ca7f29520d7af50b8c1ef9f764b67c5f7..a5f275e25c1ecac03d90587df284d3d9a0f92e95 100644
--- a/third_party/blink/renderer/modules/peerconnection/rtc_rtp_receiver.cc
+++ b/third_party/blink/renderer/modules/peerconnection/rtc_rtp_receiver.cc
@@ -107,6 +107,29 @@ void RTCRtpReceiver::setPlayoutDelayHint(absl::optional<double> hint,
   receiver_->SetJitterBufferMinimumDelay(playout_delay_hint_);
 }
 
+absl::optional<double> RTCRtpReceiver::jitterBufferTarget() const {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  return jitter_buffer_target_;
+}
+
+void RTCRtpReceiver::setJitterBufferTarget(absl::optional<double> target,
+                                           ExceptionState& exception_state) {
+  DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
+  if (target.has_value() && (target.value() < 0.0 || target.value() > 4000.0)) {
+    exception_state.ThrowRangeError(
+        "jitterBufferTarget is out of expected range 0 to 4000 ms");
+    return;
+  }
+
+  jitter_buffer_target_ = target;
+  if (jitter_buffer_target_.has_value()) {
+    receiver_->SetJitterBufferMinimumDelay(jitter_buffer_target_.value() /
+                                           1000.0);
+  } else {
+    receiver_->SetJitterBufferMinimumDelay(absl::nullopt);
+  }
+}
+
 HeapVector<Member<RTCRtpSynchronizationSource>>
 RTCRtpReceiver::getSynchronizationSources(ScriptState* script_state,
                                           ExceptionState& exception_state) {
