diff --git a/third_party/blink/renderer/core/css/cssom/css_unit_values.h b/third_party/blink/renderer/core/css/cssom/css_unit_values.h
index c6ed633393f10585448b0592df7ac2578192af82..d8a58ee1fcf60a49928c868ae25326f26f768dd1 100644
--- a/third_party/blink/renderer/core/css/cssom/css_unit_values.h
+++ b/third_party/blink/renderer/core/css/cssom/css_unit_values.h
@@ -51,6 +51,14 @@ class CSSUnitValues {
     return CSSUnitValue::Create(value, CSSPrimitiveValue::UnitType::kRems);
   }
 
+  static CSSUnitValue* cap(double value) {
+    return CSSUnitValue::Create(value, CSSPrimitiveValue::UnitType::kCaps);
+  }
+
+  static CSSUnitValue* rcap(double value) {
+    return CSSUnitValue::Create(value, CSSPrimitiveValue::UnitType::kRcaps);
+  }
+
   static CSSUnitValue* rex(double value) {
     return CSSUnitValue::Create(value, CSSPrimitiveValue::UnitType::kRexs);
   }
