diff --git a/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom b/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
index a9d8b17d029ff11fcd5f0bd1f3c25bdaafafe63b..f00129d734c5fe686f2312728ee4f76ea86dbf29 100644
--- a/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
+++ b/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
@@ -81,8 +81,8 @@ enum CSSSampleId {
     kBackgroundPositionX = 28,
     kBackgroundPositionY = 29,
     kBackgroundRepeat = 30,
-    kBackgroundRepeatX = 31,
-    kBackgroundRepeatY = 32,
+    // kBackgroundRepeatX = 31,
+    // kBackgroundRepeatY = 32,
     kBackgroundSize = 33,
     kBorder = 34,
     kBorderBottom = 35,
@@ -345,8 +345,8 @@ enum CSSSampleId {
     kWebkitMaskPositionX = 292,
     kWebkitMaskPositionY = 293,
     kWebkitMaskRepeat = 294,
-    kWebkitMaskRepeatX = 295,
-    kWebkitMaskRepeatY = 296,
+    // kWebkitMaskRepeatX = 295,
+    // kWebkitMaskRepeatY = 296,
     kWebkitMaskSize = 297,
     kAliasWebkitMaxLogicalWidth = 298,
     kAliasWebkitMaxLogicalHeight = 299,
@@ -816,6 +816,16 @@ enum CSSSampleId {
     kTimelineScope = 762,
     kScrollbarColor = 763,
     kWordBoundaryDetection = 764,
+    // FlyWeb: same number as Chrome 117 (765 is not in the 116).
+    kTransitionBehavior = 766,
+    kMaskImage = 772,
+    kMaskClip = 773,
+    kMaskSize = 774,
+    kMaskOrigin = 775,
+    kMaskRepeat = 777,
+    kMaskComposite = 778,
+    kMaskPosition = 779,
+    kMaskMode = 780,
     // 1. Add new features above this line (don't change the assigned numbers of
     //    the existing items).
     // 2. Run the src/tools/metrics/histograms/update_use_counter_css.py script
