diff --git a/third_party/blink/renderer/core/animation/css/css_transition_data.cc b/third_party/blink/renderer/core/animation/css/css_transition_data.cc
index 34383a4fdaf1743d1db542c7c4b2850034a2dc96..d281a5f99e5040c7873b0a38911fa5543aeb1365 100644
--- a/third_party/blink/renderer/core/animation/css/css_transition_data.cc
+++ b/third_party/blink/renderer/core/animation/css/css_transition_data.cc
@@ -10,6 +10,7 @@ namespace blink {
 
 CSSTransitionData::CSSTransitionData() : CSSTimingData(InitialDuration()) {
   property_list_.push_back(InitialProperty());
+  behavior_list_.push_back(InitialBehavior());
 }
 
 CSSTransitionData::CSSTransitionData(const CSSTransitionData& other) = default;
