diff --git a/third_party/blink/renderer/core/css/parser/css_variable_parser.h b/third_party/blink/renderer/core/css/parser/css_variable_parser.h
index eb59a29e085d7806d1b497528a0f8ecfb8c0cd91..e5b85286a80811e389ce81c9585513be5c68c760 100644
--- a/third_party/blink/renderer/core/css/parser/css_variable_parser.h
+++ b/third_party/blink/renderer/core/css/parser/css_variable_parser.h
@@ -28,7 +28,11 @@ class CORE_EXPORT CSSVariableParser {
       const CSSTokenizedValue&,
       bool is_animation_tainted,
       const CSSParserContext&);
-  static CSSVariableReferenceValue* ParseVariableReferenceValue(
+  // Custom properties registered with universal syntax [1] are parsed with
+  // this function.
+  //
+  // https://drafts.css-houdini.org/css-properties-values-api-1/#universal-syntax-definition
+  static CSSVariableReferenceValue* ParseUniversalSyntaxValue(
       CSSTokenizedValue,
       const CSSParserContext&,
       bool is_animation_tainted);
