diff --git a/third_party/blink/renderer/core/navigation_api/navigation.idl b/third_party/blink/renderer/core/navigation_api/navigation.idl
index 2dc44f9195b53ef6e1f9e143b4606572d9a46b27..259fb1192a7a3f4f1bb248a50cd1ee201c107531 100644
--- a/third_party/blink/renderer/core/navigation_api/navigation.idl
+++ b/third_party/blink/renderer/core/navigation_api/navigation.idl
@@ -11,6 +11,7 @@
   sequence<NavigationHistoryEntry> entries();
   [RaisesException, MeasureAs=AppHistory] void updateCurrentEntry(NavigationUpdateCurrentEntryOptions options);
   readonly attribute NavigationTransition? transition;
+  [RuntimeEnabled=NavigationActivation] readonly attribute NavigationActivation? activation;
 
   readonly attribute boolean canGoBack;
   readonly attribute boolean canGoForward;
