diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_path.cc b/third_party/blink/renderer/core/layout/svg/layout_svg_path.cc
index 797b86d550a402f4009bbeb4f43641a1b1ac47f5..173f4980d71e34e8f1674865184e3be0c35e7ef0 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_path.cc
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_path.cc
@@ -116,7 +116,7 @@ void LayoutSVGPath::UpdateMarkers() {
     }
   }
 
-  stroke_bounding_box_.Union(boundaries);
+  decorated_bounding_box_.Union(boundaries);
 }
 
 }  // namespace blink
