diff --git a/third_party/blink/renderer/core/css/css_import_rule.h b/third_party/blink/renderer/core/css/css_import_rule.h
index 9618130991a3341212040a372d1e4ce3c3272a52..94caac233b75a2f0a3426326d71c344f4b1525ec 100644
--- a/third_party/blink/renderer/core/css/css_import_rule.h
+++ b/third_party/blink/renderer/core/css/css_import_rule.h
@@ -48,6 +48,8 @@ class CSSImportRule final : public CSSRule, public MediaQuerySetOwner {
 
   String layerName() const;
 
+  String supportsText() const;
+
   void Trace(Visitor*) const override;
 
  private:
