diff --git a/third_party/blink/renderer/core/svg/build.gni b/third_party/blink/renderer/core/svg/build.gni
index 69559a062e4c7fc221a97dbb5c12812dbd6f8b6e..185b0fab012ac79eb45aefac2d9f77df866722c5 100644
--- a/third_party/blink/renderer/core/svg/build.gni
+++ b/third_party/blink/renderer/core/svg/build.gni
@@ -44,6 +44,7 @@ blink_core_sources_svg = [
   "properties/svg_property_info.h",
   "properties/svg_property_tear_off.cc",
   "properties/svg_property_tear_off.h",
+  "proxy_svg_resource_client.h",
   "radial_gradient_attributes.h",
   "svg_a_element.cc",
   "svg_a_element.h",
