diff --git a/third_party/blink/renderer/core/css/css_primitive_value_mappings.h b/third_party/blink/renderer/core/css/css_primitive_value_mappings.h
index d8614b4af5e314e224d5e76e3841851a8f1a17c9..6b73ebd539e46f57c71aaf86e682941d70f1e6a0 100644
--- a/third_party/blink/renderer/core/css/css_primitive_value_mappings.h
+++ b/third_party/blink/renderer/core/css/css_primitive_value_mappings.h
@@ -48,7 +48,6 @@
 #include "third_party/blink/renderer/platform/fonts/font_smoothing_mode.h"
 #include "third_party/blink/renderer/platform/fonts/text_rendering_mode.h"
 #include "third_party/blink/renderer/platform/geometry/length.h"
-#include "third_party/blink/renderer/platform/graphics/graphics_types.h"
 #include "third_party/blink/renderer/platform/graphics/touch_action.h"
 #include "third_party/blink/renderer/platform/text/text_run.h"
 #include "third_party/blink/renderer/platform/text/writing_mode.h"
@@ -144,43 +143,56 @@ inline OutlineIsAuto CSSIdentifierValue::ConvertTo() const {
 }
 
 template <>
-inline CSSIdentifierValue::CSSIdentifierValue(CompositeOperator e)
+inline CSSIdentifierValue::CSSIdentifierValue(
+    CompositingOperator compositing_operator)
     : CSSValue(kIdentifierClass) {
-  switch (e) {
-    case kCompositeClear:
+  switch (compositing_operator) {
+    case CompositingOperator::kAdd:
+      value_id_ = CSSValueID::kAdd;
+      break;
+    case CompositingOperator::kSubtract:
+      value_id_ = CSSValueID::kSubtract;
+      break;
+    case CompositingOperator::kIntersect:
+      value_id_ = CSSValueID::kIntersect;
+      break;
+    case CompositingOperator::kExclude:
+      value_id_ = CSSValueID::kExclude;
+      break;
+    case CompositingOperator::kClear:
       value_id_ = CSSValueID::kClear;
       break;
-    case kCompositeCopy:
+    case CompositingOperator::kCopy:
       value_id_ = CSSValueID::kCopy;
       break;
-    case kCompositeSourceOver:
+    case CompositingOperator::kSourceOver:
       value_id_ = CSSValueID::kSourceOver;
       break;
-    case kCompositeSourceIn:
+    case CompositingOperator::kSourceIn:
       value_id_ = CSSValueID::kSourceIn;
       break;
-    case kCompositeSourceOut:
+    case CompositingOperator::kSourceOut:
       value_id_ = CSSValueID::kSourceOut;
       break;
-    case kCompositeSourceAtop:
+    case CompositingOperator::kSourceAtop:
       value_id_ = CSSValueID::kSourceAtop;
       break;
-    case kCompositeDestinationOver:
+    case CompositingOperator::kDestinationOver:
       value_id_ = CSSValueID::kDestinationOver;
       break;
-    case kCompositeDestinationIn:
+    case CompositingOperator::kDestinationIn:
       value_id_ = CSSValueID::kDestinationIn;
       break;
-    case kCompositeDestinationOut:
+    case CompositingOperator::kDestinationOut:
       value_id_ = CSSValueID::kDestinationOut;
       break;
-    case kCompositeDestinationAtop:
+    case CompositingOperator::kDestinationAtop:
       value_id_ = CSSValueID::kDestinationAtop;
       break;
-    case kCompositeXOR:
+    case CompositingOperator::kXOR:
       value_id_ = CSSValueID::kXor;
       break;
-    case kCompositePlusLighter:
+    case CompositingOperator::kPlusLighter:
       value_id_ = CSSValueID::kPlusLighter;
       break;
     default:
@@ -190,38 +202,46 @@ inline CSSIdentifierValue::CSSIdentifierValue(CompositeOperator e)
 }
 
 template <>
-inline CompositeOperator CSSIdentifierValue::ConvertTo() const {
+inline CompositingOperator CSSIdentifierValue::ConvertTo() const {
   switch (value_id_) {
+    case CSSValueID::kAdd:
+      return CompositingOperator::kAdd;
+    case CSSValueID::kSubtract:
+      return CompositingOperator::kSubtract;
+    case CSSValueID::kIntersect:
+      return CompositingOperator::kIntersect;
+    case CSSValueID::kExclude:
+      return CompositingOperator::kExclude;
     case CSSValueID::kClear:
-      return kCompositeClear;
+      return CompositingOperator::kClear;
     case CSSValueID::kCopy:
-      return kCompositeCopy;
+      return CompositingOperator::kCopy;
     case CSSValueID::kSourceOver:
-      return kCompositeSourceOver;
+      return CompositingOperator::kSourceOver;
     case CSSValueID::kSourceIn:
-      return kCompositeSourceIn;
+      return CompositingOperator::kSourceIn;
     case CSSValueID::kSourceOut:
-      return kCompositeSourceOut;
+      return CompositingOperator::kSourceOut;
     case CSSValueID::kSourceAtop:
-      return kCompositeSourceAtop;
+      return CompositingOperator::kSourceAtop;
     case CSSValueID::kDestinationOver:
-      return kCompositeDestinationOver;
+      return CompositingOperator::kDestinationOver;
     case CSSValueID::kDestinationIn:
-      return kCompositeDestinationIn;
+      return CompositingOperator::kDestinationIn;
     case CSSValueID::kDestinationOut:
-      return kCompositeDestinationOut;
+      return CompositingOperator::kDestinationOut;
     case CSSValueID::kDestinationAtop:
-      return kCompositeDestinationAtop;
+      return CompositingOperator::kDestinationAtop;
     case CSSValueID::kXor:
-      return kCompositeXOR;
+      return CompositingOperator::kXOR;
     case CSSValueID::kPlusLighter:
-      return kCompositePlusLighter;
+      return CompositingOperator::kPlusLighter;
     default:
       break;
   }
 
   NOTREACHED();
-  return kCompositeClear;
+  return CompositingOperator::kAdd;
 }
 
 template <>
@@ -370,6 +390,18 @@ inline CSSIdentifierValue::CSSIdentifierValue(EFillBox e)
     case EFillBox::kText:
       value_id_ = CSSValueID::kText;
       break;
+    case EFillBox::kFillBox:
+      value_id_ = CSSValueID::kFillBox;
+      break;
+    case EFillBox::kStrokeBox:
+      value_id_ = CSSValueID::kStrokeBox;
+      break;
+    case EFillBox::kViewBox:
+      value_id_ = CSSValueID::kViewBox;
+      break;
+    case EFillBox::kNoClip:
+      value_id_ = CSSValueID::kNoClip;
+      break;
   }
 }
 
@@ -387,6 +419,14 @@ inline EFillBox CSSIdentifierValue::ConvertTo() const {
       return EFillBox::kContent;
     case CSSValueID::kText:
       return EFillBox::kText;
+    case CSSValueID::kFillBox:
+      return EFillBox::kFillBox;
+    case CSSValueID::kStrokeBox:
+      return EFillBox::kStrokeBox;
+    case CSSValueID::kViewBox:
+      return EFillBox::kViewBox;
+    case CSSValueID::kNoClip:
+      return EFillBox::kNoClip;
     default:
       break;
   }
@@ -428,11 +468,44 @@ inline EFillRepeat CSSIdentifierValue::ConvertTo() const {
     default:
       break;
   }
-
   NOTREACHED();
   return EFillRepeat::kRepeatFill;
 }
 
+template <>
+inline CSSIdentifierValue::CSSIdentifierValue(EFillMaskMode e)
+    : CSSValue(kIdentifierClass) {
+  switch (e) {
+    case EFillMaskMode::kAlpha:
+      value_id_ = CSSValueID::kAlpha;
+      break;
+    case EFillMaskMode::kLuminance:
+      value_id_ = CSSValueID::kLuminance;
+      break;
+    case EFillMaskMode::kMatchSource:
+      value_id_ = CSSValueID::kMatchSource;
+      break;
+    default:
+      NOTREACHED();
+      value_id_ = CSSValueID::kMatchSource;
+  }
+}
+
+template <>
+inline EFillMaskMode CSSIdentifierValue::ConvertTo() const {
+  switch (value_id_) {
+    case CSSValueID::kAlpha:
+      return EFillMaskMode::kAlpha;
+    case CSSValueID::kLuminance:
+      return EFillMaskMode::kLuminance;
+    case CSSValueID::kMatchSource:
+      return EFillMaskMode::kMatchSource;
+    default:
+      NOTREACHED();
+      return EFillMaskMode::kMatchSource;
+  }
+}
+
 template <>
 inline CSSIdentifierValue::CSSIdentifierValue(BackgroundEdgeOrigin e)
     : CSSValue(kIdentifierClass) {
@@ -1710,6 +1783,58 @@ inline CSSIdentifierValue::CSSIdentifierValue(CoordBox coord_box)
   }
 }
 
+template <>
+inline GeometryBox CSSIdentifierValue::ConvertTo() const {
+  switch (GetValueID()) {
+    case CSSValueID::kBorderBox:
+      return GeometryBox::kBorderBox;
+    case CSSValueID::kPaddingBox:
+      return GeometryBox::kPaddingBox;
+    case CSSValueID::kContentBox:
+      return GeometryBox::kContentBox;
+    case CSSValueID::kMarginBox:
+      return GeometryBox::kMarginBox;
+    case CSSValueID::kFillBox:
+      return GeometryBox::kFillBox;
+    case CSSValueID::kStrokeBox:
+      return GeometryBox::kStrokeBox;
+    case CSSValueID::kViewBox:
+      return GeometryBox::kViewBox;
+    default:
+      break;
+  }
+  NOTREACHED();
+  return GeometryBox::kBorderBox;
+}
+
+template <>
+inline CSSIdentifierValue::CSSIdentifierValue(GeometryBox geometry_box)
+    : CSSValue(kIdentifierClass) {
+  switch (geometry_box) {
+    case GeometryBox::kBorderBox:
+      value_id_ = CSSValueID::kBorderBox;
+      break;
+    case GeometryBox::kPaddingBox:
+      value_id_ = CSSValueID::kPaddingBox;
+      break;
+    case GeometryBox::kContentBox:
+      value_id_ = CSSValueID::kContentBox;
+      break;
+    case GeometryBox::kMarginBox:
+      value_id_ = CSSValueID::kMarginBox;
+      break;
+    case GeometryBox::kFillBox:
+      value_id_ = CSSValueID::kFillBox;
+      break;
+    case GeometryBox::kStrokeBox:
+      value_id_ = CSSValueID::kStrokeBox;
+      break;
+    case GeometryBox::kViewBox:
+      value_id_ = CSSValueID::kViewBox;
+      break;
+  }
+}
+
 template <>
 inline TextUnderlinePosition CSSIdentifierValue::ConvertTo() const {
   switch (GetValueID()) {
