diff --git a/third_party/blink/renderer/core/svg/svg_gradient_element.h b/third_party/blink/renderer/core/svg/svg_gradient_element.h
index e0eea98c026717c3e7c10e903690d1a002d68e3b..dae11399b276c73932b2e38b6bf8bc273682524f 100644
--- a/third_party/blink/renderer/core/svg/svg_gradient_element.h
+++ b/third_party/blink/renderer/core/svg/svg_gradient_element.h
@@ -21,7 +21,6 @@
 #ifndef THIRD_PARTY_BLINK_RENDERER_CORE_SVG_SVG_GRADIENT_ELEMENT_H_
 #define THIRD_PARTY_BLINK_RENDERER_CORE_SVG_SVG_GRADIENT_ELEMENT_H_
 
-#include "third_party/blink/renderer/core/inspector/inspector_trace_events.h"
 #include "third_party/blink/renderer/core/svg/svg_animated_enumeration.h"
 #include "third_party/blink/renderer/core/svg/svg_element.h"
 #include "third_party/blink/renderer/core/svg/svg_unit_types.h"
@@ -59,7 +58,7 @@ class SVGGradientElement : public SVGElement, public SVGURIReference {
     return gradient_units_.Get();
   }
 
-  void InvalidateGradient(LayoutInvalidationReasonForTracing);
+  void InvalidateGradient();
   void InvalidateDependentGradients();
 
   const SVGGradientElement* ReferencedElement() const;
