diff --git a/third_party/blink/renderer/platform/fonts/web_font_decoder.cc b/third_party/blink/renderer/platform/fonts/web_font_decoder.cc
index 7a9b585583a6afaaf609978e128e91dbc11c2bce..0953dc528ddaa49ecd4a5f49a0f79972b692a93d 100644
--- a/third_party/blink/renderer/platform/fonts/web_font_decoder.cc
+++ b/third_party/blink/renderer/platform/fonts/web_font_decoder.cc
@@ -105,6 +105,7 @@ ots::TableAction BlinkOTSContext::GetTableAction(uint32_t tag) {
   const uint32_t kCff2Tag = OTS_TAG('C', 'F', 'F', '2');
   const uint32_t kSbixTag = OTS_TAG('s', 'b', 'i', 'x');
 #if HB_VERSION_ATLEAST(1, 0, 0)
+  const uint32_t kBaseTag = OTS_TAG('B', 'A', 'S', 'E');
   const uint32_t kGdefTag = OTS_TAG('G', 'D', 'E', 'F');
   const uint32_t kGposTag = OTS_TAG('G', 'P', 'O', 'S');
   const uint32_t kGsubTag = OTS_TAG('G', 'S', 'U', 'B');
@@ -133,6 +134,7 @@ ots::TableAction BlinkOTSContext::GetTableAction(uint32_t tag) {
 #if HB_VERSION_ATLEAST(1, 0, 0)
     // Let HarfBuzz handle how to deal with broken tables.
     case kAvarTag:
+    case kBaseTag:
     case kCvarTag:
     case kFvarTag:
     case kGvarTag:
