diff --git a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_sizing_tree.cc b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_sizing_tree.cc
index 4f434d3a7f39d372188b1ceca37d59aa45233651..63a43babc9d1152ae7789926bb008b0c052a604e 100644
--- a/third_party/blink/renderer/core/layout/ng/grid/ng_grid_sizing_tree.cc
+++ b/third_party/blink/renderer/core/layout/ng/grid/ng_grid_sizing_tree.cc
@@ -6,8 +6,21 @@
 
 namespace blink {
 
+NGGridSizingTree NGGridSizingTree::CopyForFragmentation() const {
+  NGGridSizingTree tree_copy;
+  tree_copy.tree_data_.ReserveInitialCapacity(tree_data_.size());
+
+  for (const auto& sizing_data : tree_data_) {
+    DCHECK(sizing_data);
+    auto& sizing_data_copy = tree_copy.CreateSizingData();
+    sizing_data_copy = *sizing_data;
+  }
+  return tree_copy;
+}
+
 scoped_refptr<const NGGridLayoutTree> NGGridSizingTree::FinalizeTree() const {
   auto layout_tree = base::MakeRefCounted<NGGridLayoutTree>(tree_data_.size());
+
   for (const auto& grid_tree_node : tree_data_) {
     layout_tree->Append(grid_tree_node->layout_data,
                         grid_tree_node->subtree_size);
@@ -20,12 +33,8 @@ NGGridSizingTree::GridTreeNode& NGGridSizingTree::CreateSizingData(
   if (subgrid_data) {
     const auto* subgrid_layout_box = subgrid_data->node.GetLayoutBox();
 
-    if (!subgrid_index_lookup_map_) {
-      subgrid_index_lookup_map_ = MakeGarbageCollected<SubgridIndexLookupMap>();
-    }
-
-    DCHECK(!subgrid_index_lookup_map_->Contains(subgrid_layout_box));
-    subgrid_index_lookup_map_->insert(subgrid_layout_box, tree_data_.size());
+    DCHECK(!subgrid_index_lookup_map_.Contains(subgrid_layout_box));
+    subgrid_index_lookup_map_.insert(subgrid_layout_box, tree_data_.size());
   }
   return *tree_data_.emplace_back(std::make_unique<GridTreeNode>());
 }
@@ -34,32 +43,25 @@ void NGGridSizingTree::AddSubgriddedItemLookupData(
     NGSubgriddedItemData&& subgridded_item_data) {
   const auto* item_layout_box = subgridded_item_data->node.GetLayoutBox();
 
-  if (!subgridded_item_data_lookup_map_) {
-    subgridded_item_data_lookup_map_ =
-        MakeGarbageCollected<SubgriddedItemDataLookupMap>();
-  }
-
-  DCHECK(!subgridded_item_data_lookup_map_->Contains(item_layout_box));
-  subgridded_item_data_lookup_map_->insert(item_layout_box,
-                                           std::move(subgridded_item_data));
+  DCHECK(!subgridded_item_data_lookup_map_.Contains(item_layout_box));
+  subgridded_item_data_lookup_map_.insert(item_layout_box,
+                                          std::move(subgridded_item_data));
 }
 
 NGSubgriddedItemData NGGridSizingTree::LookupSubgriddedItemData(
     const GridItemData& grid_item) const {
   const auto* item_layout_box = grid_item.node.GetLayoutBox();
 
-  DCHECK(subgridded_item_data_lookup_map_ &&
-         subgridded_item_data_lookup_map_->Contains(item_layout_box));
-  return subgridded_item_data_lookup_map_->at(item_layout_box);
+  DCHECK(subgridded_item_data_lookup_map_.Contains(item_layout_box));
+  return subgridded_item_data_lookup_map_.at(item_layout_box);
 }
 
 wtf_size_t NGGridSizingTree::LookupSubgridIndex(
     const GridItemData& subgrid_data) const {
   const auto* subgrid_layout_box = subgrid_data.node.GetLayoutBox();
 
-  DCHECK(subgrid_index_lookup_map_ &&
-         subgrid_index_lookup_map_->Contains(subgrid_layout_box));
-  return subgrid_index_lookup_map_->at(subgrid_layout_box);
+  DCHECK(subgrid_index_lookup_map_.Contains(subgrid_layout_box));
+  return subgrid_index_lookup_map_.at(subgrid_layout_box);
 }
 
 }  // namespace blink
