diff --git a/third_party/blink/renderer/core/html/forms/input_type.h b/third_party/blink/renderer/core/html/forms/input_type.h
index 53d5c900ce6d4bfd14e713d6e80584b10e584f32..57203ebec6b3e796b6421d55c04f423e3252b8d3 100644
--- a/third_party/blink/renderer/core/html/forms/input_type.h
+++ b/third_party/blink/renderer/core/html/forms/input_type.h
@@ -111,6 +111,7 @@ class CORE_EXPORT InputType : public GarbageCollected<InputType> {
   virtual bool IsInteractiveContent() const;
   virtual bool IsTextButton() const;
   virtual bool IsTextField() const;
+  virtual bool ShouldAutoDirUseValue() const;
 
   bool IsButtonInputType() const { return type_ == Type::kButton; }
   bool IsColorInputType() const { return type_ == Type::kColor; }
