diff --git a/third_party/blink/renderer/core/events/pop_state_event_init.idl b/third_party/blink/renderer/core/events/pop_state_event_init.idl
index 8ca38ca9f2e6b0c6fe6a1e5bf53da506b9f929e0..5f34e1c08c96e4e64cc8917be6a2d520a3b58097 100644
--- a/third_party/blink/renderer/core/events/pop_state_event_init.idl
+++ b/third_party/blink/renderer/core/events/pop_state_event_init.idl
@@ -6,4 +6,5 @@
 
 dictionary PopStateEventInit : EventInit {
     any state = null;
+    boolean hasUAVisualTransition = false;
 };
