diff --git a/third_party/blink/renderer/core/css/css_import_rule.idl b/third_party/blink/renderer/core/css/css_import_rule.idl
index d096603a71096cb2c9595b49cb3e5bc38222f863..6f4d85fbd17948677533f6505e70f98b830eef2e 100644
--- a/third_party/blink/renderer/core/css/css_import_rule.idl
+++ b/third_party/blink/renderer/core/css/css_import_rule.idl
@@ -29,4 +29,5 @@
     [SameObject] readonly attribute CSSStyleSheet styleSheet;
 
     readonly attribute CSSOMString? layerName;
+    readonly attribute CSSOMString? supportsText;
 };
