diff --git a/third_party/blink/renderer/core/frame/frame_view.cc b/third_party/blink/renderer/core/frame/frame_view.cc
index 9d6a766499b211515d34b04963e52dbb7742d335..6f8b74591362614950250610f5a6d98b42e4830d 100644
--- a/third_party/blink/renderer/core/frame/frame_view.cc
+++ b/third_party/blink/renderer/core/frame/frame_view.cc
@@ -102,7 +102,8 @@ void FrameView::UpdateViewportIntersection(unsigned flags,
 
     IntersectionGeometry geometry(nullptr, *owner_element, {} /* root_margin */,
                                   {IntersectionObserver::kMinimumThreshold},
-                                  {} /* target_margin */, geometry_flags);
+                                  {} /* target_margin */,
+                                  {} /* scroll_margin */, geometry_flags);
     PhysicalRect new_rect_in_parent = geometry.IntersectionRect();
     if (new_rect_in_parent.size != rect_in_parent_.size ||
         ((new_rect_in_parent.X() - rect_in_parent_.X()).Abs() +
