diff --git a/third_party/blink/renderer/core/html/forms/text_control_inner_elements.cc b/third_party/blink/renderer/core/html/forms/text_control_inner_elements.cc
index 839a17def3afa95baec3e8b60992f76be3cfb9e8..6ef35c8883b2af525ca64e1461b59468a5878048 100644
--- a/third_party/blink/renderer/core/html/forms/text_control_inner_elements.cc
+++ b/third_party/blink/renderer/core/html/forms/text_control_inner_elements.cc
@@ -57,6 +57,7 @@ EditingViewPortElement::CustomStyleForLayoutObject(const StyleRecalcContext&) {
 
   style_builder.SetFlexGrow(1);
   style_builder.SetMinWidth(Length::Fixed(0));
+  style_builder.SetMinHeight(Length::Fixed(0));
   style_builder.SetDisplay(EDisplay::kBlock);
   style_builder.SetDirection(TextDirection::kLtr);
 
@@ -150,6 +151,14 @@ TextControlInnerEditorElement::CustomStyleForLayoutObject(
           : EUserModify::kReadWritePlaintextOnly);
   style_builder.SetDisplay(EDisplay::kBlock);
   style_builder.SetHasLineIfEmpty(true);
+  if (!start_style.ApplyControlFixedSize(host)) {
+    Length caret_width(GetDocument().View()->CaretWidth(), Length::kFixed);
+    if (IsHorizontalWritingMode(style_builder.GetWritingMode())) {
+      style_builder.SetMinWidth(caret_width);
+    } else {
+      style_builder.SetMinHeight(caret_width);
+    }
+  }
   style_builder.SetShouldIgnoreOverflowPropertyForInlineBlockBaseline();
 
   if (!IsA<HTMLTextAreaElement>(host)) {
