diff --git a/third_party/blink/renderer/core/html/forms/base_text_input_type.h b/third_party/blink/renderer/core/html/forms/base_text_input_type.h
index cf2d32c572ae6c2c9b7056f7339ffc3836281d3b..23a4a805ec468d9c99140cbdd429bfe6aabd4533 100644
--- a/third_party/blink/renderer/core/html/forms/base_text_input_type.h
+++ b/third_party/blink/renderer/core/html/forms/base_text_input_type.h
@@ -58,6 +58,7 @@ class BaseTextInputType : public TextFieldInputType {
   bool SupportsPlaceholder() const final;
   bool SupportsSelectionAPI() const override;
   bool PatternMismatchPerValue(const String&) const;
+  bool ShouldAutoDirUseValue() const override;
 
   // regexp_ and pattern_for_regexp_ are mutable because they are kinds of
   // cache.
