diff --git a/third_party/blink/renderer/core/html/forms/html_input_element.h b/third_party/blink/renderer/core/html/forms/html_input_element.h
index cea17f8726686050f3d0dee134967a56d34af4db..7cfa1c29f50579a87d92f302c9ddcc52b2bc6b53 100644
--- a/third_party/blink/renderer/core/html/forms/html_input_element.h
+++ b/third_party/blink/renderer/core/html/forms/html_input_element.h
@@ -107,6 +107,9 @@ class CORE_EXPORT HTMLInputElement
   // Returns true if the type is email, number, password, search, tel, text,
   // or url.
   bool IsTextField() const;
+  bool IsTelephone() const;
+  // To override from TextControlElement
+  bool ShouldAutoDirUseValue() const final;
   // Do not add type check predicates for concrete input types; e.g.  isImage,
   // isRadio, isFile.  If you want to check the input type, you may use
   // |input->type() == input_type_names::kImage|, etc.
