diff --git a/third_party/blink/renderer/core/css/css_selector.h b/third_party/blink/renderer/core/css/css_selector.h
index d4dfa0e334d7c82ccf0f51f4dd26160268ae6edf..e67beabcdb911aa642789636297fc2ab6c4425e7 100644
--- a/third_party/blink/renderer/core/css/css_selector.h
+++ b/third_party/blink/renderer/core/css/css_selector.h
@@ -293,6 +293,8 @@ class CORE_EXPORT CSSSelector {
     // selector (&), or a :scope pseudo-class, and must therefore be kept
     // for serialization purposes.
     kPseudoUnparsed,
+    kPseudoUserInvalid,
+    kPseudoUserValid,
     kPseudoValid,
     kPseudoVertical,
     kPseudoVisited,
