diff --git a/third_party/blink/renderer/core/animation/interpolable_value.h b/third_party/blink/renderer/core/animation/interpolable_value.h
index 45e369e5afe4b5100f7a42202a3ecf09b5cf8c9b..c3f05449fe5f2049d6c5c342e560cd4383af02be 100644
--- a/third_party/blink/renderer/core/animation/interpolable_value.h
+++ b/third_party/blink/renderer/core/animation/interpolable_value.h
@@ -38,6 +38,7 @@ class CORE_EXPORT InterpolableValue {
   virtual bool IsNumber() const { return false; }
   virtual bool IsBool() const { return false; }
   virtual bool IsColor() const { return false; }
+  virtual bool IsScrollbarColor() const { return false; }
   virtual bool IsList() const { return false; }
   virtual bool IsLength() const { return false; }
   virtual bool IsAspectRatio() const { return false; }
