diff --git a/third_party/blink/renderer/core/css/css_value_keywords.json5 b/third_party/blink/renderer/core/css/css_value_keywords.json5
index c6d27b511248e693bb0c36eaabe0da4a92f1f4ad..4d818139a626bb50399f8bd78cca0e7b1c7db7a5 100644
--- a/third_party/blink/renderer/core/css/css_value_keywords.json5
+++ b/third_party/blink/renderer/core/css/css_value_keywords.json5
@@ -1344,6 +1344,8 @@
     "oklch",
     // "color",
     "-internal-light-dark",
+    // FlyWeb: public light-dark() for colors (Chromium 123).
+    "light-dark",
 
     // color spaces
     // "srgb", included in color-interpolation
