diff --git a/third_party/blink/renderer/core/events/pop_state_event.cc b/third_party/blink/renderer/core/events/pop_state_event.cc
index 2115046b035fbfffedcb521fcddd322e92897f44..7e1e4988b9368f87d2a9750a867f63343e4642e0 100644
--- a/third_party/blink/renderer/core/events/pop_state_event.cc
+++ b/third_party/blink/renderer/core/events/pop_state_event.cc
@@ -53,7 +53,8 @@ PopStateEvent* PopStateEvent::Create(
 PopStateEvent::PopStateEvent(ScriptState* script_state,
                              const AtomicString& type,
                              const PopStateEventInit* initializer)
-    : Event(type, initializer) {
+    : Event(type, initializer),
+      has_ua_visual_transition_(initializer->hasUAVisualTransition()) {
   v8::Isolate* isolate = script_state->GetIsolate();
   if (initializer->hasState()) {
     state_.Set(isolate, initializer->state().V8Value());
