diff --git a/third_party/blink/renderer/core/animation/css_display_interpolation_type.h b/third_party/blink/renderer/core/animation/css_display_interpolation_type.h
index 42607b6e37152b14debbb8453f744436a5d1f851..7d1f78b1e714b1a16bcda8b3822ed718b508116a 100644
--- a/third_party/blink/renderer/core/animation/css_display_interpolation_type.h
+++ b/third_party/blink/renderer/core/animation/css_display_interpolation_type.h
@@ -30,6 +30,7 @@ class CSSDisplayInterpolationType : public CSSInterpolationType {
   void ApplyStandardPropertyValue(const InterpolableValue&,
                                   const NonInterpolableValue*,
                                   StyleResolverState&) const final;
+  bool IsDiscrete() const final { return true; }
 
  private:
   InterpolationValue CreateDisplayValue(EDisplay) const;
