diff --git a/third_party/blink/renderer/core/paint/ng/ng_highlight_painter.cc b/third_party/blink/renderer/core/paint/ng/ng_highlight_painter.cc
index cc671bb5ea413ec5a1c054dff13d1cf5b6111d5e..3b7677a5c6d531b99ca78600d45e6c23b825e45d 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_highlight_painter.cc
+++ b/third_party/blink/renderer/core/paint/ng/ng_highlight_painter.cc
@@ -204,8 +204,8 @@ bool HasNonTrivialSpellingGrammarStyles(const NGFragmentItem& fragment_item,
 
     // If the ‘text-decoration-line’ is not ‘spelling-error’ or ‘grammar-error’,
     // depending on the pseudo. ‘text-decoration-color’ can vary without hurting
-    // the optimisation, and for these line types, we ignore ‘-thickness’ and
-    // ‘-style’ and ‘text-decoration-skip-ink’ anyway.
+    // the optimisation, and for these line types, we ignore all other text
+    // decoration related properties anyway.
     if (pseudo_style->TextDecorationsInEffect() !=
         (pseudo == kPseudoIdSpellingError
              ? TextDecorationLine::kSpellingError
