diff --git a/third_party/blink/renderer/core/layout/ng/svg/layout_ng_svg_foreign_object.h b/third_party/blink/renderer/core/layout/ng/svg/layout_ng_svg_foreign_object.h
index 3f611f01e3eb696cbeec4926fa319a74d5cff824..044b25e3af78850848e5a31959fa366077d859ec 100644
--- a/third_party/blink/renderer/core/layout/ng/svg/layout_ng_svg_foreign_object.h
+++ b/third_party/blink/renderer/core/layout/ng/svg/layout_ng_svg_foreign_object.h
@@ -62,6 +62,7 @@ class LayoutNGSVGForeignObject final
                       const ComputedStyle& style) const override;
   gfx::RectF ObjectBoundingBox() const override;
   gfx::RectF StrokeBoundingBox() const override;
+  gfx::RectF DecoratedBoundingBox() const override;
   gfx::RectF VisualRectInLocalSVGCoordinates() const override;
   AffineTransform LocalToSVGParentTransform() const override;
 
