diff --git a/third_party/blink/renderer/core/intersection_observer/intersection_geometry.h b/third_party/blink/renderer/core/intersection_observer/intersection_geometry.h
index a9f332e9af5479ad0669cdd94dfc624fad6130d6..3d52bcccec31e7f1f1780b70a230d8962946f563 100644
--- a/third_party/blink/renderer/core/intersection_observer/intersection_geometry.h
+++ b/third_party/blink/renderer/core/intersection_observer/intersection_geometry.h
@@ -86,6 +86,7 @@ class CORE_EXPORT IntersectionGeometry {
                        const Vector<Length>& root_margin,
                        const Vector<float>& thresholds,
                        const Vector<Length>& target_margin,
+                       const Vector<Length>& scroll_margin,
                        unsigned flags,
                        CachedRects* cached_rects = nullptr);
 
@@ -94,6 +95,7 @@ class CORE_EXPORT IntersectionGeometry {
                        const Element& target,
                        const Vector<float>& thresholds,
                        const Vector<Length>& target_margin,
+                       const Vector<Length>& scroll_margin,
                        unsigned flags,
                        CachedRects* cached_rects = nullptr);
 
@@ -167,11 +169,14 @@ class CORE_EXPORT IntersectionGeometry {
   RootAndTarget PrepareComputeGeometry(const Node* root_node,
                                        const Element& target_element,
                                        CachedRects* cached_rects);
+
   void ComputeGeometry(const RootGeometry& root_geometry,
                        const RootAndTarget& root_and_target,
                        const Vector<float>& thresholds,
                        const Vector<Length>& target_margin,
+                       const Vector<Length>& scroll_margin,
                        CachedRects* cached_rects);
+
   // Map intersection_rect from the coordinate system of the target to the
   // coordinate system of the root, applying intervening clips.
   bool ClipToRoot(const LayoutObject* root,
@@ -179,6 +184,7 @@ class CORE_EXPORT IntersectionGeometry {
                   const PhysicalRect& root_rect,
                   PhysicalRect& unclipped_intersection_rect,
                   PhysicalRect& intersection_rect,
+                  const Vector<Length>& scroll_margin,
                   CachedRects* cached_rects = nullptr);
   unsigned FirstThresholdGreaterThan(float ratio,
                                      const Vector<float>& thresholds) const;
@@ -187,6 +193,7 @@ class CORE_EXPORT IntersectionGeometry {
       const gfx::Transform& target_to_document_transform,
       const gfx::Transform& root_to_document_transform,
       const Vector<float>& thresholds,
+      const Vector<Length>& scroll_margin,
       CachedRects* cached_rects) const;
 
   PhysicalRect target_rect_;
