diff --git a/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.cc b/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.cc
index d1c02787f5569779f0e2b01c06678ad5520ac17a..0b0d573c50dbfe064a492c6f9c358a093ec4ceb3 100644
--- a/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.cc
+++ b/third_party/blink/renderer/core/paint/ng/ng_box_fragment_painter.cc
@@ -1880,6 +1880,10 @@ NGPhysicalBoxStrut NGBoxFragmentPainter::ComputePadding() const {
   return PhysicalFragment().Padding();
 }
 
+NGPhysicalBoxStrut NGBoxFragmentPainter::ComputeMargins() const {
+  return PhysicalFragment().Margins();
+}
+
 BoxPainterBase::FillLayerInfo NGBoxFragmentPainter::GetFillLayerInfo(
     const Color& color,
     const FillLayer& bg_layer,
