diff --git a/third_party/blink/renderer/core/mathml/mathml_element.cc b/third_party/blink/renderer/core/mathml/mathml_element.cc
index 34810c363f879969a7ea5cfd1519f89f1f2b9e72..342e2075a96dc5b40bee19a4535647d77874740a 100644
--- a/third_party/blink/renderer/core/mathml/mathml_element.cc
+++ b/third_party/blink/renderer/core/mathml/mathml_element.cc
@@ -10,10 +10,8 @@
 #include "third_party/blink/renderer/core/css/parser/css_parser.h"
 #include "third_party/blink/renderer/core/css_value_keywords.h"
 #include "third_party/blink/renderer/core/dom/document.h"
-#include "third_party/blink/renderer/core/dom/node_computed_style.h"
 #include "third_party/blink/renderer/core/execution_context/execution_context.h"
 #include "third_party/blink/renderer/core/html/html_element.h"
-#include "third_party/blink/renderer/core/layout/ng/mathml/layout_ng_table_cell_with_anonymous_mrow.h"
 #include "third_party/blink/renderer/platform/wtf/text/character_visitor.h"
 #include "third_party/blink/renderer/platform/wtf/text/string_to_number.h"
 
@@ -188,13 +186,4 @@ absl::optional<Length> MathMLElement::AddMathLengthToComputedStyle(
   return absl::nullopt;
 }
 
-LayoutObject* MathMLElement::CreateLayoutObject(const ComputedStyle& style) {
-  if (RuntimeEnabledFeatures::MathMLCoreEnabled() &&
-      Node::HasTagName(mathml_names::kMtdTag) &&
-      style.Display() == EDisplay::kTableCell) {
-    return MakeGarbageCollected<LayoutNGTableCellWithAnonymousMrow>(this);
-  }
-  return Element::CreateLayoutObject(style);
-}
-
 }  // namespace blink
