diff --git a/third_party/blink/renderer/core/dom/get_html_options.idl b/third_party/blink/renderer/core/dom/get_html_options.idl
new file mode 100644
index 0000000000000000000000000000000000000000..98e71a133a207e51ed2c57eb4a2764d47d6d98d8
--- /dev/null
+++ b/third_party/blink/renderer/core/dom/get_html_options.idl
@@ -0,0 +1,10 @@
+// Copyright 2024 The Chromium Authors
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// https://github.com/whatwg/html/pull/10139
+
+dictionary GetHTMLOptions {
+  boolean serializableShadowRoots = false;
+  sequence<ShadowRoot> shadowRoots = [];
+};
