diff --git a/third_party/blink/renderer/core/html/html_template_element.idl b/third_party/blink/renderer/core/html/html_template_element.idl
index 8f5ca1474c7182d1c80bca161d907aa00b26bbd5..405db3426de139aa90968181405324fdef65cf50 100644
--- a/third_party/blink/renderer/core/html/html_template_element.idl
+++ b/third_party/blink/renderer/core/html/html_template_element.idl
@@ -52,5 +52,8 @@
 
     // This is the "new" declarative Shadow DOM IDL attribute, which reflects
     // the 'shadowrootmode' content attribute.
-    [RuntimeEnabled=StreamingDeclarativeShadowDOM,Reflect,ReflectOnly=("open","closed")] attribute DOMString? shadowRootMode;
+    [RuntimeEnabled=StreamingDeclarativeShadowDOM,Reflect,ReflectOnly=("open","closed")] attribute DOMString shadowRootMode;
+    [CEReactions,Reflect,RuntimeEnabled=TemplateDelegatesFocusReflection] attribute boolean shadowRootDelegatesFocus;
+    [CEReactions,Reflect,RuntimeEnabled=ShadowRootClonable] attribute boolean shadowRootClonable;
+    [CEReactions,Reflect,RuntimeEnabled=ElementGetHTML] attribute boolean shadowRootSerializable;
 };
