diff --git a/third_party/blink/renderer/core/svg/svg_filter_element.cc b/third_party/blink/renderer/core/svg/svg_filter_element.cc
index f07249666b68fa12a6d59759996dd74caf666133..0e832c91c824e368c0501508561d177c86b874ab 100644
--- a/third_party/blink/renderer/core/svg/svg_filter_element.cc
+++ b/third_party/blink/renderer/core/svg/svg_filter_element.cc
@@ -124,10 +124,6 @@ void SVGFilterElement::ChildrenChanged(const ChildrenChange& change) {
   if (change.ByParser() && !AssociatedResource())
     return;
 
-  if (LayoutObject* object = GetLayoutObject()) {
-    object->SetNeedsLayoutAndFullPaintInvalidation(
-        layout_invalidation_reason::kChildChanged);
-  }
   InvalidateFilterChain();
 }
 
