diff --git a/third_party/blink/renderer/core/svg/svg_length_context.cc b/third_party/blink/renderer/core/svg/svg_length_context.cc
index 44c8e568bd4515141d2af1489373b0a7c6acc5ea..650b1241edeb339519ea5cc7c7bc8d78a8c412a9 100644
--- a/third_party/blink/renderer/core/svg/svg_length_context.cc
+++ b/third_party/blink/renderer/core/svg/svg_length_context.cc
@@ -122,7 +122,9 @@ gfx::RectF SVGLengthContext::ResolveRectangle(const SVGElement* context,
                                               const SVGLength& x,
                                               const SVGLength& y,
                                               const SVGLength& width,
-                                              const SVGLength& height) {
+                                              const SVGLength& height,
+                                              const absl::optional<gfx::SizeF>&
+                                                  override_viewport) {
   DCHECK_NE(SVGUnitTypes::kSvgUnitTypeUnknown, type);
   const ComputedStyle* style = ComputedStyleForLengthResolving(context);
   if (!style) {
@@ -155,7 +157,9 @@ gfx::RectF SVGLengthContext::ResolveRectangle(const SVGElement* context,
     gfx::SizeF viewport_size_for_resolve;
     if (size.Width().IsPercentOrCalc() || size.Height().IsPercentOrCalc() ||
         point.X().IsPercentOrCalc() || point.Y().IsPercentOrCalc()) {
-      viewport_size_for_resolve = SVGLengthContext(context).ResolveViewport();
+      viewport_size_for_resolve =
+          override_viewport ? *override_viewport
+                            : SVGLengthContext(context).ResolveViewport();
     }
     // Resolve the Lengths to user units.
     resolved_rect =
