diff --git a/third_party/blink/renderer/core/css/style_request.h b/third_party/blink/renderer/core/css/style_request.h
index 6a8f758b144abf36579dd18883e1d984de3cbe85..8576c9686c3ef6727bc00c43fb896dc8a1e52541 100644
--- a/third_party/blink/renderer/core/css/style_request.h
+++ b/third_party/blink/renderer/core/css/style_request.h
@@ -32,6 +32,7 @@
 namespace blink {
 
 class ComputedStyle;
+class Element;
 
 enum RuleMatchingBehavior { kMatchAllRules, kMatchAllRulesExcludingSMIL };
 
