diff --git a/third_party/blink/renderer/core/animation/css_basic_shape_interpolation_type.cc b/third_party/blink/renderer/core/animation/css_basic_shape_interpolation_type.cc
index 48ad44b75429038b557bbe308a9005379927c290..fed1b3db7609b7f59938617d6da8b68cd710ea2e 100644
--- a/third_party/blink/renderer/core/animation/css_basic_shape_interpolation_type.cc
+++ b/third_party/blink/renderer/core/animation/css_basic_shape_interpolation_type.cc
@@ -16,6 +16,7 @@
 #include "third_party/blink/renderer/core/css/resolver/style_resolver_state.h"
 #include "third_party/blink/renderer/core/style/basic_shapes.h"
 #include "third_party/blink/renderer/core/style/computed_style.h"
+#include "third_party/blink/renderer/core/style/computed_style_constants.h"
 #include "third_party/blink/renderer/core/style/shape_clip_path_operation.h"
 #include "third_party/blink/renderer/core/style/shape_offset_path_operation.h"
 #include "third_party/blink/renderer/platform/heap/garbage_collected.h"
@@ -217,8 +218,9 @@ void CSSBasicShapeInterpolationType::ApplyStandardPropertyValue(
           std::move(shape), CoordBox::kBorderBox));
       break;
     case CSSPropertyID::kClipPath:
-      state.StyleBuilder().SetClipPath(
-          ShapeClipPathOperation::Create(std::move(shape)));
+      // TODO(pdr): Handle geometry box.
+      state.StyleBuilder().SetClipPath(ShapeClipPathOperation::Create(
+          std::move(shape), GeometryBox::kBorderBox));
       break;
     case CSSPropertyID::kObjectViewBox:
       state.StyleBuilder().SetObjectViewBox(std::move(shape));
