diff --git a/third_party/blink/renderer/core/css/media_query_exp.cc b/third_party/blink/renderer/core/css/media_query_exp.cc
index f7f7083d0be44d20b4026f49cec07f010e9d22cc..8cae9ba714ac44477a313837516f7f9f50fd8366 100644
--- a/third_party/blink/renderer/core/css/media_query_exp.cc
+++ b/third_party/blink/renderer/core/css/media_query_exp.cc
@@ -649,6 +649,7 @@ unsigned MediaQueryExpValue::GetUnitFlags() const {
   if (length_type_flags.test(CSSPrimitiveValue::kUnitTypeFontSize) ||
       length_type_flags.test(CSSPrimitiveValue::kUnitTypeFontXSize) ||
       length_type_flags.test(CSSPrimitiveValue::kUnitTypeZeroCharacterWidth) ||
+      length_type_flags.test(CSSPrimitiveValue::kUnitTypeFontCapitalHeight) ||
       length_type_flags.test(
           CSSPrimitiveValue::kUnitTypeIdeographicFullWidth) ||
       length_type_flags.test(CSSPrimitiveValue::kUnitTypeLineHeight)) {
@@ -657,6 +658,8 @@ unsigned MediaQueryExpValue::GetUnitFlags() const {
 
   if (length_type_flags.test(CSSPrimitiveValue::kUnitTypeRootFontSize) ||
       length_type_flags.test(CSSPrimitiveValue::kUnitTypeRootFontXSize) ||
+      length_type_flags.test(
+          CSSPrimitiveValue::kUnitTypeRootFontCapitalHeight) ||
       length_type_flags.test(
           CSSPrimitiveValue::kUnitTypeRootFontZeroCharacterWidth) ||
       length_type_flags.test(
