diff --git a/third_party/blink/renderer/core/dom/node.cc b/third_party/blink/renderer/core/dom/node.cc
index 9788cd4895f9418df8c38a03468d3e199a3a15b2..8dc5bc1e5d9bd5746a13e525ccd7a436f01aec45 100644
--- a/third_party/blink/renderer/core/dom/node.cc
+++ b/third_party/blink/renderer/core/dom/node.cc
@@ -992,7 +992,9 @@ Node* Node::cloneNode(bool deep, ExceptionState& exception_state) const {
   // true, and the clone shadows flag set if this is a DocumentFragment whose
   // host is an HTML template element.
   auto* fragment = DynamicTo<DocumentFragment>(this);
-  bool clone_shadows_flag = fragment && fragment->IsTemplateContent();
+  bool clone_shadows_flag =
+      !RuntimeEnabledFeatures::ShadowRootClonableEnabled() && fragment &&
+      fragment->IsTemplateContent();
   return Clone(GetDocument(),
                deep ? (clone_shadows_flag ? CloneChildrenFlag::kCloneWithShadows
                                           : CloneChildrenFlag::kClone)
@@ -2896,8 +2898,9 @@ DispatchEventResult Node::DispatchEventInternal(Event& event) {
 }
 
 void Node::DispatchSubtreeModifiedEvent() {
-  if (IsInShadowTree())
+  if (IsInShadowTree() || GetDocument().ShouldSuppressMutationEvents()) {
     return;
+  }
 
 #if DCHECK_IS_ON()
   DCHECK(!EventDispatchForbiddenScope::IsEventDispatchForbidden());
@@ -3334,6 +3337,11 @@ void Node::FlatTreeParentChanged() {
 }
 
 void Node::AddCandidateDirectionalityForSlot() {
+  if (RuntimeEnabledFeatures::CSSPseudoDirEnabled()) {
+    // This code is not needed for the new dir=auto inheritance rules.
+    return;
+  }
+
   ShadowRoot* root = ShadowRootOfParent();
   if (!root || !root->HasSlotAssignment()) {
     // We should add this node as a candidate that needs to recalculate its
@@ -3393,6 +3401,50 @@ HashSet<Member<TreeScope>> Node::GetAncestorTreeScopes() const {
   return ancestor_tree_scopes;
 }
 
+void Node::SetCachedDirectionality(TextDirection direction) {
+  switch (direction) {
+    case TextDirection::kRtl:
+      SetFlag(kCachedDirectionalityIsRtl);
+      break;
+    case TextDirection::kLtr:
+      ClearFlag(kCachedDirectionalityIsRtl);
+      break;
+  }
+  if (!RuntimeEnabledFeatures::CSSPseudoDirEnabled()) {
+    ClearFlag(kNeedsInheritDirectionalityFromParent);
+  }
+}
+
+bool Node::NeedsInheritDirectionalityFromParent() const {
+  CHECK(!RuntimeEnabledFeatures::CSSPseudoDirEnabled());
+  return GetFlag(kNeedsInheritDirectionalityFromParent);
+}
+
+void Node::SetNeedsInheritDirectionalityFromParent() {
+  CHECK(!RuntimeEnabledFeatures::CSSPseudoDirEnabled());
+  SetFlag(kNeedsInheritDirectionalityFromParent);
+}
+
+void Node::ClearNeedsInheritDirectionalityFromParent() {
+  CHECK(!RuntimeEnabledFeatures::CSSPseudoDirEnabled());
+  ClearFlag(kNeedsInheritDirectionalityFromParent);
+}
+
+bool Node::DirAutoInheritsFromParent() const {
+  return RuntimeEnabledFeatures::CSSPseudoDirEnabled() &&
+         GetFlag(kDirAutoInheritsFromParent);
+}
+
+void Node::SetDirAutoInheritsFromParent() {
+  CHECK(RuntimeEnabledFeatures::CSSPseudoDirEnabled());
+  return SetFlag(kDirAutoInheritsFromParent);
+}
+
+void Node::ClearDirAutoInheritsFromParent() {
+  CHECK(RuntimeEnabledFeatures::CSSPseudoDirEnabled());
+  return ClearFlag(kDirAutoInheritsFromParent);
+}
+
 void Node::Trace(Visitor* visitor) const {
   visitor->Trace(parent_or_shadow_host_node_);
   visitor->Trace(previous_);
