diff --git a/third_party/blink/renderer/core/css/css_math_expression_node.h b/third_party/blink/renderer/core/css/css_math_expression_node.h
index 146a6d9c97bafa16ba2132294c1dde39952bf1b4..63082cc494460f66e7c38686ebf8648af59556c3 100644
--- a/third_party/blink/renderer/core/css/css_math_expression_node.h
+++ b/third_party/blink/renderer/core/css/css_math_expression_node.h
@@ -80,7 +80,10 @@ class CORE_EXPORT CSSMathExpressionNode
       CSSParserTokenRange tokens,
       const CSSParserContext&,
       const bool is_percentage_allowed,
-      CSSAnchorQueryTypes allowed_anchor_queries);
+      CSSAnchorQueryTypes allowed_anchor_queries,
+      // Variable substitutions for relative color syntax.
+      // https://www.w3.org/TR/css-color-5/#relative-colors
+      HashMap<CSSValueID, double> color_channel_keyword_values = {});
 
   virtual bool IsNumericLiteral() const { return false; }
   virtual bool IsOperation() const { return false; }
