diff --git a/third_party/blink/renderer/core/html/forms/date_time_field_element.cc b/third_party/blink/renderer/core/html/forms/date_time_field_element.cc
index d81cf844dc80e0fd12ed2859cfeb36d717f485fc..c760af3cf78cd79ddf2534c8e948129c4d4377cb 100644
--- a/third_party/blink/renderer/core/html/forms/date_time_field_element.cc
+++ b/third_party/blink/renderer/core/html/forms/date_time_field_element.cc
@@ -27,6 +27,7 @@
 
 #include "third_party/blink/renderer/core/css/style_change_reason.h"
 #include "third_party/blink/renderer/core/dom/document.h"
+#include "third_party/blink/renderer/core/dom/node_computed_style.h"
 #include "third_party/blink/renderer/core/dom/text.h"
 #include "third_party/blink/renderer/core/editing/frame_selection.h"
 #include "third_party/blink/renderer/core/editing/position.h"
@@ -81,8 +82,10 @@ void DateTimeFieldElement::DefaultKeyboardEventHandler(
     return;
 
   const String& key = keyboard_event.key();
+  bool is_horizontal = GetComputedStyle()->IsHorizontalWritingMode();
 
-  if (key == "ArrowLeft") {
+  if ((is_horizontal && key == "ArrowLeft") ||
+      (!is_horizontal && key == "ArrowUp")) {
     if (!field_owner_)
       return;
     // FIXME: We'd like to use FocusController::advanceFocus(FocusDirectionLeft,
@@ -92,7 +95,8 @@ void DateTimeFieldElement::DefaultKeyboardEventHandler(
     return;
   }
 
-  if (key == "ArrowRight") {
+  if ((is_horizontal && key == "ArrowRight") ||
+      (!is_horizontal && key == "ArrowDown")) {
     if (!field_owner_)
       return;
     // FIXME: We'd like to use
@@ -106,7 +110,8 @@ void DateTimeFieldElement::DefaultKeyboardEventHandler(
   if (IsFieldOwnerReadOnly())
     return;
 
-  if (key == "ArrowDown") {
+  if ((is_horizontal && key == "ArrowDown") ||
+      (!is_horizontal && key == "ArrowLeft")) {
     if (keyboard_event.getModifierState("Alt"))
       return;
     keyboard_event.SetDefaultHandled();
@@ -114,7 +119,8 @@ void DateTimeFieldElement::DefaultKeyboardEventHandler(
     return;
   }
 
-  if (key == "ArrowUp") {
+  if ((is_horizontal && key == "ArrowUp") ||
+      (!is_horizontal && key == "ArrowRight")) {
     keyboard_event.SetDefaultHandled();
     StepUp();
     return;
