diff --git a/third_party/blink/renderer/core/dom/shadow_root.idl b/third_party/blink/renderer/core/dom/shadow_root.idl
index c07f58479455acd7571fa8415cefcfe907e134cd..e22da3cc266065700733caf59aa9bc765162039d 100644
--- a/third_party/blink/renderer/core/dom/shadow_root.idl
+++ b/third_party/blink/renderer/core/dom/shadow_root.idl
@@ -37,11 +37,22 @@ interface ShadowRoot : DocumentFragment {
     readonly attribute boolean delegatesFocus;
     readonly attribute SlotAssignmentMode slotAssignment;
 
+    // The clonable attribute controls whether the shadow root will be
+    // cloned by DOM cloning operations.
+    [RuntimeEnabled=ShadowRootClonable] readonly attribute boolean clonable;
+
     [RuntimeEnabled=ScopedCustomElementRegistry]
     readonly attribute CustomElementRegistry? registry;
 
     // Declarative Shadow DOM getInnerHTML() function.
     [Affects=Nothing, MeasureAs=ElementGetInnerHTML] HTMLString getInnerHTML(optional GetInnerHTMLOptions options = {});
+    // Declarative Shadow DOM getHTML() function (Chromium 125).
+    [Affects=Nothing, RuntimeEnabled=ElementGetHTML] HTMLString getHTML(optional GetHTMLOptions options = {});
+    // The serializable attribute controls whether the shadow root will be
+    // serialized by getHTML({serializableShadowRoots:true}).
+    [RuntimeEnabled=ElementGetHTML] attribute boolean serializable;
+
+    [RuntimeEnabled=HTMLUnsafeMethods,RaisesException,MeasureAs=SetHTMLUnsafe,CEReactions] void setHTMLUnsafe(HTMLString string);
 };
 
 ShadowRoot includes DocumentOrShadowRoot;
