diff --git a/third_party/blink/renderer/platform/fonts/font_size_adjust.cc b/third_party/blink/renderer/platform/fonts/font_size_adjust.cc
index 976ef002d83d2a91a067263cfb5055f7e258a2b3..e0111769dd0244475703b5e3c697250b3412fbf4 100644
--- a/third_party/blink/renderer/platform/fonts/font_size_adjust.cc
+++ b/third_party/blink/renderer/platform/fonts/font_size_adjust.cc
@@ -13,6 +13,7 @@ unsigned FontSizeAdjust::GetHash() const {
   // Normalize negative zero.
   WTF::AddFloatToHash(computed_hash, value_ == 0.0 ? 0.0 : value_);
   WTF::AddIntToHash(computed_hash, static_cast<const unsigned>(metric_));
+  WTF::AddIntToHash(computed_hash, static_cast<const unsigned>(type_));
   return computed_hash;
 }
 
@@ -24,6 +25,8 @@ String FontSizeAdjust::ToString(Metric metric) const {
       return "ch-width";
     case Metric::kIcWidth:
       return "ic-width";
+    case Metric::kIcHeight:
+      return "ic-height";
     case Metric::kExHeight:
       return "ex-height";
   }
@@ -34,8 +37,15 @@ String FontSizeAdjust::ToString() const {
   if (value_ == kFontSizeAdjustNone) {
     return "none";
   }
-  return metric_ == Metric::kExHeight
-             ? String::Format("%s", String::Number(value_).Ascii().c_str())
+
+  if (metric_ == Metric::kExHeight) {
+    return IsFromFont()
+               ? "from-font"
+               : String::Format("%s", String::Number(value_).Ascii().c_str());
+  }
+
+  return IsFromFont()
+             ? String::Format("%s from-font", ToString(metric_).Ascii().c_str())
              : String::Format("%s %s", ToString(metric_).Ascii().c_str(),
                               String::Number(value_).Ascii().c_str());
 }
