diff --git a/third_party/blink/renderer/core/css/css_property_equality.cc b/third_party/blink/renderer/core/css/css_property_equality.cc
index 13ae48a03df3d22ad52dc3c103159816c93e7655..97c2b2c99e75e425ca360a612235a84948a65b95 100644
--- a/third_party/blink/renderer/core/css/css_property_equality.cc
+++ b/third_party/blink/renderer/core/css/css_property_equality.cc
@@ -80,17 +80,30 @@ bool FillLayersEqual(const FillLayer& a_layers, const FillLayer& b_layers) {
         }
         break;
       case CSSPropertyID::kBackgroundClip:
+      case CSSPropertyID::kMaskClip:
       case CSSPropertyID::kWebkitMaskClip:
         if (a_layer->Clip() != b_layer->Clip()) {
           return false;
         }
         break;
+      case CSSPropertyID::kMaskComposite:
+      case CSSPropertyID::kWebkitMaskComposite:
+        if (a_layer->CompositingOperator() != b_layer->CompositingOperator()) {
+          return false;
+        }
+        break;
       case CSSPropertyID::kBackgroundOrigin:
+      case CSSPropertyID::kMaskOrigin:
       case CSSPropertyID::kWebkitMaskOrigin:
         if (a_layer->Origin() != b_layer->Origin()) {
           return false;
         }
         break;
+      case CSSPropertyID::kMaskMode:
+        if (a_layer->MaskMode() != b_layer->MaskMode()) {
+          return false;
+        }
+        break;
       case CSSPropertyID::kBackgroundPositionX:
       case CSSPropertyID::kWebkitMaskPositionX:
         if (a_layer->PositionX() != b_layer->PositionX()) {
@@ -109,25 +122,22 @@ bool FillLayersEqual(const FillLayer& a_layers, const FillLayer& b_layers) {
           return false;
         }
         break;
-      case CSSPropertyID::kWebkitMaskRepeatX:
-      case CSSPropertyID::kBackgroundRepeatX:
-        if (a_layer->RepeatX() != b_layer->RepeatX()) {
-          return false;
-        }
-        break;
-      case CSSPropertyID::kWebkitMaskRepeatY:
-      case CSSPropertyID::kBackgroundRepeatY:
-        if (a_layer->RepeatY() != b_layer->RepeatY()) {
+      case CSSPropertyID::kMaskRepeat:
+      case CSSPropertyID::kWebkitMaskRepeat:
+      case CSSPropertyID::kBackgroundRepeat:
+        if (a_layer->Repeat() != b_layer->Repeat()) {
           return false;
         }
         break;
       case CSSPropertyID::kBackgroundSize:
       case CSSPropertyID::kWebkitMaskSize:
+      case CSSPropertyID::kMaskSize:
         if (!(a_layer->SizeLength() == b_layer->SizeLength())) {
           return false;
         }
         break;
       case CSSPropertyID::kBackgroundImage:
+      case CSSPropertyID::kMaskImage:
         if (!base::ValuesEquivalent(a_layer->GetImage(), b_layer->GetImage())) {
           return false;
         }
@@ -201,11 +211,8 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kBackgroundPositionY:
       return FillLayersEqual<CSSPropertyID::kBackgroundPositionY>(
           a.BackgroundLayers(), b.BackgroundLayers());
-    case CSSPropertyID::kBackgroundRepeatX:
-      return FillLayersEqual<CSSPropertyID::kBackgroundRepeatX>(
-          a.BackgroundLayers(), b.BackgroundLayers());
-    case CSSPropertyID::kBackgroundRepeatY:
-      return FillLayersEqual<CSSPropertyID::kBackgroundRepeatY>(
+    case CSSPropertyID::kBackgroundRepeat:
+      return FillLayersEqual<CSSPropertyID::kBackgroundRepeat>(
           a.BackgroundLayers(), b.BackgroundLayers());
     case CSSPropertyID::kBackgroundSize:
       return FillLayersEqual<CSSPropertyID::kBackgroundSize>(
@@ -417,6 +424,8 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
       return a.GetFontWeight() == b.GetFontWeight();
     case CSSPropertyID::kForcedColorAdjust:
       return a.ForcedColorAdjust() == b.ForcedColorAdjust();
+    case CSSPropertyID::kFieldSizing:
+      return a.FieldSizing() == b.FieldSizing();
     case CSSPropertyID::kGridAutoColumns:
       return a.GridAutoColumns() == b.GridAutoColumns();
     case CSSPropertyID::kGridAutoFlow:
@@ -776,11 +785,28 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
                                     b.MaskBoxImageSource());
     case CSSPropertyID::kWebkitMaskBoxImageWidth:
       return a.MaskBoxImageWidth() == b.MaskBoxImageWidth();
+    case CSSPropertyID::kMaskClip:
+      return FillLayersEqual<CSSPropertyID::kMaskClip>(a.MaskLayers(),
+                                                       b.MaskLayers());
     case CSSPropertyID::kWebkitMaskClip:
       return FillLayersEqual<CSSPropertyID::kWebkitMaskClip>(a.MaskLayers(),
                                                              b.MaskLayers());
+    case CSSPropertyID::kMaskComposite:
+      return FillLayersEqual<CSSPropertyID::kMaskComposite>(a.MaskLayers(),
+                                                            b.MaskLayers());
+    case CSSPropertyID::kWebkitMaskComposite:
+      return FillLayersEqual<CSSPropertyID::kWebkitMaskComposite>(
+          a.MaskLayers(), b.MaskLayers());
+    case CSSPropertyID::kMaskImage:
     case CSSPropertyID::kWebkitMaskImage:
-      return base::ValuesEquivalent(a.MaskImage(), b.MaskImage());
+      return FillLayersEqual<CSSPropertyID::kMaskImage>(a.MaskLayers(),
+                                                        b.MaskLayers());
+    case CSSPropertyID::kMaskOrigin:
+      return FillLayersEqual<CSSPropertyID::kMaskOrigin>(a.MaskLayers(),
+                                                         b.MaskLayers());
+    case CSSPropertyID::kMaskMode:
+      return FillLayersEqual<CSSPropertyID::kMaskMode>(a.MaskLayers(),
+                                                       b.MaskLayers());
     case CSSPropertyID::kWebkitMaskOrigin:
       return FillLayersEqual<CSSPropertyID::kWebkitMaskOrigin>(a.MaskLayers(),
                                                                b.MaskLayers());
@@ -790,12 +816,15 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kWebkitMaskPositionY:
       return FillLayersEqual<CSSPropertyID::kWebkitMaskPositionY>(
           a.MaskLayers(), b.MaskLayers());
-    case CSSPropertyID::kWebkitMaskRepeatX:
-      return FillLayersEqual<CSSPropertyID::kWebkitMaskRepeatX>(a.MaskLayers(),
-                                                                b.MaskLayers());
-    case CSSPropertyID::kWebkitMaskRepeatY:
-      return FillLayersEqual<CSSPropertyID::kWebkitMaskRepeatY>(a.MaskLayers(),
-                                                                b.MaskLayers());
+    case CSSPropertyID::kMaskRepeat:
+      return FillLayersEqual<CSSPropertyID::kMaskRepeat>(a.MaskLayers(),
+                                                         b.MaskLayers());
+    case CSSPropertyID::kWebkitMaskRepeat:
+      return FillLayersEqual<CSSPropertyID::kWebkitMaskRepeat>(a.MaskLayers(),
+                                                               b.MaskLayers());
+    case CSSPropertyID::kMaskSize:
+      return FillLayersEqual<CSSPropertyID::kMaskSize>(a.MaskLayers(),
+                                                       b.MaskLayers());
     case CSSPropertyID::kWebkitMaskSize:
       return FillLayersEqual<CSSPropertyID::kWebkitMaskSize>(a.MaskLayers(),
                                                              b.MaskLayers());
@@ -822,6 +851,9 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
       return base::ValuesEquivalent<TransformOperation>(a.Rotate(), b.Rotate());
     case CSSPropertyID::kScale:
       return base::ValuesEquivalent<TransformOperation>(a.Scale(), b.Scale());
+    case CSSPropertyID::kSize:
+      return a.GetPageSizeType() == b.GetPageSizeType() &&
+             a.PageSize() == b.PageSize();
     case CSSPropertyID::kTransformOrigin:
       return a.GetTransformOrigin() == b.GetTransformOrigin();
     case CSSPropertyID::kWebkitPerspectiveOriginX:
@@ -909,6 +941,14 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kAliasWebkitAlternativeAnimationDelay:
     case CSSPropertyID::kAliasWebkitAlternativeAnimationWithDelayStartEnd:
     case CSSPropertyID::kAliasWebkitAlternativeAnimationWithTimeline:
+    case CSSPropertyID::kAliasWebkitAlternativeMask:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskClip:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskComposite:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskImage:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskOrigin:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskPosition:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskRepeat:
+    case CSSPropertyID::kAliasWebkitAlternativeMaskSize:
     case CSSPropertyID::kAliasWebkitAnimation:
     case CSSPropertyID::kAliasWebkitAnimationDelay:
     case CSSPropertyID::kAliasWebkitAnimationDirection:
@@ -1025,7 +1065,6 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kWebkitLineBreak:
     case CSSPropertyID::kWebkitLineClamp:
     case CSSPropertyID::kWebkitMaskBoxImageRepeat:
-    case CSSPropertyID::kWebkitMaskComposite:
     case CSSPropertyID::kWebkitPrintColorAdjust:
     case CSSPropertyID::kWebkitRtlOrdering:
     case CSSPropertyID::kWebkitRubyPosition:
@@ -1150,7 +1189,6 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     // Shorthand properties shouldn't be compared, use their longhands.
     case CSSPropertyID::kBackground:
     case CSSPropertyID::kBackgroundPosition:
-    case CSSPropertyID::kBackgroundRepeat:
     case CSSPropertyID::kBorder:
     case CSSPropertyID::kBorderBottom:
     case CSSPropertyID::kBorderColor:
@@ -1205,13 +1243,14 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kToggle:
     case CSSPropertyID::kTransition:
     case CSSPropertyID::kViewTimeline:
+    case CSSPropertyID::kAlternativeMask:
     case CSSPropertyID::kWebkitColumnBreakAfter:
     case CSSPropertyID::kWebkitColumnBreakBefore:
     case CSSPropertyID::kWebkitColumnBreakInside:
     case CSSPropertyID::kWebkitMask:
     case CSSPropertyID::kWebkitMaskBoxImage:
+    case CSSPropertyID::kMaskPosition:
     case CSSPropertyID::kWebkitMaskPosition:
-    case CSSPropertyID::kWebkitMaskRepeat:
     case CSSPropertyID::kWebkitTextStroke:
     case CSSPropertyID::kAlternativeWhiteSpace:
       NOTREACHED() << property.GetCSSPropertyName().ToAtomicString().Ascii();
@@ -1247,6 +1286,7 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kToggleGroup:
     case CSSPropertyID::kToggleRoot:
     case CSSPropertyID::kToggleTrigger:
+    case CSSPropertyID::kTransitionBehavior:
     case CSSPropertyID::kTransitionDelay:
     case CSSPropertyID::kTransitionDuration:
     case CSSPropertyID::kTransitionProperty:
@@ -1258,13 +1298,12 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
       NOTREACHED() << property.GetCSSPropertyName().ToAtomicString().Ascii();
       return true;
 
-    // Invalid properties.
+    // TODO(crbug.com/1459374): Implement comparison for these properties. They
+    // are reachable via transitions now.
     case CSSPropertyID::kAdditiveSymbols:
-    case CSSPropertyID::kAll:
     case CSSPropertyID::kAscentOverride:
     case CSSPropertyID::kBasePalette:
     case CSSPropertyID::kDescentOverride:
-    case CSSPropertyID::kInvalid:
     case CSSPropertyID::kFallback:
     case CSSPropertyID::kFontDisplay:
     case CSSPropertyID::kInherits:
@@ -1275,7 +1314,6 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kPad:
     case CSSPropertyID::kPrefix:
     case CSSPropertyID::kRange:
-    case CSSPropertyID::kSize:
     case CSSPropertyID::kSpeakAs:
     case CSSPropertyID::kSrc:
     case CSSPropertyID::kSuffix:
@@ -1283,6 +1321,11 @@ bool CSSPropertyEquality::PropertiesEqual(const PropertyHandle& property,
     case CSSPropertyID::kSyntax:
     case CSSPropertyID::kSystem:
     case CSSPropertyID::kUnicodeRange:
+      return true;
+
+    // Invalid properties.
+    case CSSPropertyID::kAll:
+    case CSSPropertyID::kInvalid:
     case CSSPropertyID::kVariable:
       NOTREACHED() << property.GetCSSPropertyName().ToAtomicString().Ascii();
       return true;
