diff --git a/third_party/blink/renderer/core/css/resolver/style_builder_converter.h b/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
index 07090144d0700e9f63045939c9fa5f3b2ad090b7..6741b482b9cbe6e26ae1ee733ca6cad153e7bf3f 100644
--- a/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
+++ b/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
@@ -350,6 +350,8 @@ class StyleBuilderConverter {
 
   static AtomicString ConvertViewTransitionName(StyleResolverState&,
                                                 const CSSValue&);
+  static Vector<AtomicString> ConvertViewTransitionClass(StyleResolverState&,
+                                                         const CSSValue&);
 
   // Take a list value for a specified color-scheme, extract flags for known
   // color-schemes and the 'only' modifier, and push the list items into a
