diff --git a/third_party/blink/renderer/core/html/forms/menu_list_inner_element.cc b/third_party/blink/renderer/core/html/forms/menu_list_inner_element.cc
index 6aab4c7abef37f755354f8810c3bb5774da9c264..b43377164702509532868ae2b45524f06c28a837 100644
--- a/third_party/blink/renderer/core/html/forms/menu_list_inner_element.cc
+++ b/third_party/blink/renderer/core/html/forms/menu_list_inner_element.cc
@@ -29,7 +29,9 @@ MenuListInnerElement::CustomStyleForLayoutObject(
   style_builder.SetFlexShrink(1);
   // min-width: 0; is needed for correct shrinking.
   style_builder.SetMinWidth(Length::Fixed(0));
-  style_builder.SetHasLineIfEmpty(true);
+  if (parent_style.ApplyControlFixedSize(OwnerShadowHost())) {
+    style_builder.SetHasLineIfEmpty(true);
+  }
   style_builder.SetOverflowX(EOverflow::kHidden);
   style_builder.SetOverflowY(EOverflow::kHidden);
   style_builder.SetShouldIgnoreOverflowPropertyForInlineBlockBaseline();
