diff --git a/third_party/blink/renderer/modules/geolocation/geolocation_coordinates.idl b/third_party/blink/renderer/modules/geolocation/geolocation_coordinates.idl
index f2916d2b3a27000fc8673304c3803468bcdc4be0..3d48bf736c99b50ed644c3f193946d91b1aee5f6 100644
--- a/third_party/blink/renderer/modules/geolocation/geolocation_coordinates.idl
+++ b/third_party/blink/renderer/modules/geolocation/geolocation_coordinates.idl
@@ -33,4 +33,5 @@
     readonly attribute double? altitudeAccuracy;
     readonly attribute double? heading;
     readonly attribute double? speed;
+    [CallWith=ScriptState] object toJSON();
 };
