diff --git a/third_party/blink/renderer/core/dom/build.gni b/third_party/blink/renderer/core/dom/build.gni
index 158f5d301064252fd1df81284fff3b7a61a5b603..265e6dd91ab8fc2ffc366665c8852178882da99b 100644
--- a/third_party/blink/renderer/core/dom/build.gni
+++ b/third_party/blink/renderer/core/dom/build.gni
@@ -142,6 +142,7 @@ blink_core_sources_dom = [
   "events/event_target.h",
   "events/event_target_impl.cc",
   "events/event_target_impl.h",
+  "events/mutation_event_suppression_scope.h",
   "events/native_event_listener.h",
   "events/node_event_context.cc",
   "events/node_event_context.h",
