diff --git a/third_party/blink/renderer/platform/geometry/length.h b/third_party/blink/renderer/platform/geometry/length.h
index 632ed79e31f2513328f8bcd0fa5dd2de9c6cc4e6..55c85be3d98c4d745fa29c8c8c5383677229ae07 100644
--- a/third_party/blink/renderer/platform/geometry/length.h
+++ b/third_party/blink/renderer/platform/geometry/length.h
@@ -318,6 +318,8 @@ class PLATFORM_EXPORT Length {
 
   Length SubtractFromOneHundredPercent() const;
 
+  Length Add(const Length& other) const;
+
   Length Zoom(double factor) const;
 
   String ToString() const;
