diff --git a/third_party/blink/renderer/build/scripts/core/css/css_properties.py b/third_party/blink/renderer/build/scripts/core/css/css_properties.py
index e6d9b7e13aae79bc2ef280bcebe5e118f4b559c3..b05ec9610325c97ca469315811686addb3fa3c23 100644
--- a/third_party/blink/renderer/build/scripts/core/css/css_properties.py
+++ b/third_party/blink/renderer/build/scripts/core/css/css_properties.py
@@ -400,6 +400,7 @@ class CSSProperties(object):
             updated_alias.alternative = alias.alternative
             updated_alias.aliased_property = aliased_property.name.to_upper_camel_case(
             )
+            updated_alias.computable = alias.computable
             updated_alias.property_id = id_for_css_property_alias(alias.name)
             updated_alias.enum_key = enum_key_for_css_property_alias(
                 alias.name)
@@ -410,17 +411,23 @@ class CSSProperties(object):
                 'Shorthand' if aliased_property.longhands else 'Longhand'
             self._aliases[i] = updated_alias
 
-        # The above loop produces an "updated" alias for each (incoming) alias.
-        # The alternative_of/alternative references must be updated to point to
-        # the respective "updated" aliases.
         updated_aliases_by_name = {a.name: a for a in self._aliases}
-        for alias in self._aliases:
-            if alias.alternative_of:
-                alias.alternative_of = updated_aliases_by_name[
-                    alias.alternative_of.name]
-            if alias.alternative:
-                alias.alternative = updated_aliases_by_name[
-                    alias.alternative.name]
+
+        # The above loop produces an "updated" alias for each (incoming) alias.
+        # Any alternative_of/alternative references that point to aliases
+        # must be updated to point to the respective "updated" aliases.
+        def update_alternatives(properties):
+            for _property in properties:
+                if _property.alternative_of and _property.alternative_of.alias_for:
+                    _property.alternative_of = updated_aliases_by_name[
+                        _property.alternative_of.name]
+                if _property.alternative and _property.alternative.alias_for:
+                    _property.alternative = updated_aliases_by_name[
+                        _property.alternative.name]
+
+        update_alternatives(self.longhands)
+        update_alternatives(self.shorthands)
+        update_alternatives(self.aliases)
 
     def set_derived_attributes(self, property_):
         """Set new attributes on 'property_', based on existing attribute values
@@ -567,7 +574,12 @@ class CSSProperties(object):
 
     @property
     def computable(self):
-        is_prefixed = lambda p: p.name.original.startswith('-')
+        # Use the name of the ultimate property as the sorting key,
+        # otherwise '-alternative-foo' will sort according to
+        # '-alternative-...', when it will really be exposed to
+        # parsing/serialization as just 'foo'.
+        sorting_name = lambda p: p.ultimate_property.name.original
+        is_prefixed = lambda p: sorting_name(p).startswith('-')
         is_not_prefixed = lambda p: not is_prefixed(p)
 
         prefixed = filter(is_prefixed, self._properties_including_aliases)
@@ -590,10 +602,8 @@ class CSSProperties(object):
         prefixed = filter(is_computable, prefixed)
         unprefixed = filter(is_computable, unprefixed)
 
-        original_name = lambda x: x.name.original
-
-        return sorted(unprefixed, key=original_name) + \
-            sorted(prefixed, key=original_name)
+        return sorted(unprefixed, key=sorting_name) + \
+            sorted(prefixed, key=sorting_name)
 
     @property
     def shorthands(self):
