diff --git a/third_party/blink/renderer/core/script/value_wrapper_synthetic_module_script.cc b/third_party/blink/renderer/core/script/value_wrapper_synthetic_module_script.cc
index 5f13587384c31656abb8ef60f9644b9d0b69b9d5..3e8530966d5cbd7a04d1ac97fad75a4be6b94e44 100644
--- a/third_party/blink/renderer/core/script/value_wrapper_synthetic_module_script.cc
+++ b/third_party/blink/renderer/core/script/value_wrapper_synthetic_module_script.cc
@@ -140,7 +140,8 @@ ValueWrapperSyntheticModuleScript::CreateWithDefaultExport(
     const ScriptFetchOptions& fetch_options,
     const TextPosition& start_position) {
   v8::Isolate* isolate = settings_object->GetScriptState()->GetIsolate();
-  std::vector<v8::Local<v8::String>> export_names{V8String(isolate, "default")};
+  auto export_names =
+      v8::to_array<v8::Local<v8::String>>({V8String(isolate, "default")});
   v8::Local<v8::Module> v8_synthetic_module = v8::Module::CreateSyntheticModule(
       isolate, V8String(isolate, source_url.GetString()), export_names,
       ValueWrapperSyntheticModuleScript::EvaluationSteps);
