diff --git a/third_party/blink/renderer/core/css/resolver/style_builder_converter.h b/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
index 07090144d0700e9f63045939c9fa5f3b2ad090b7..a4f01d78ad0753861d4ff28dcf10bfb2122e1eff 100644
--- a/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
+++ b/third_party/blink/renderer/core/css/resolver/style_builder_converter.h
@@ -100,7 +100,8 @@ class StyleBuilderConverterBase {
       const CSSToLengthConversionData&,
       FontDescription::Size parent_size,
       const Document*);
-  static FontSizeAdjust ConvertFontSizeAdjust(const CSSValue&);
+  static FontSizeAdjust ConvertFontSizeAdjust(const StyleResolverState&,
+                                              const CSSValue&);
   static scoped_refptr<FontPalette> ConvertFontPalette(const CSSValue&);
   static scoped_refptr<FontPalette> ConvertPaletteMix(const CSSValue&);
 };
@@ -350,6 +351,8 @@ class StyleBuilderConverter {
 
   static AtomicString ConvertViewTransitionName(StyleResolverState&,
                                                 const CSSValue&);
+  static Vector<AtomicString> ConvertViewTransitionClass(StyleResolverState&,
+                                                         const CSSValue&);
 
   // Take a list value for a specified color-scheme, extract flags for known
   // color-schemes and the 'only' modifier, and push the list items into a
