diff --git a/third_party/blink/renderer/core/html/html_attribute_names.json5 b/third_party/blink/renderer/core/html/html_attribute_names.json5
index ca28e7b550cb0f3fc2b7011cd34e8372e10138a9..5d88c755b7ec2023df59fcdd2a1197f419d392ee 100644
--- a/third_party/blink/renderer/core/html/html_attribute_names.json5
+++ b/third_party/blink/renderer/core/html/html_attribute_names.json5
@@ -310,6 +310,7 @@
     "select",
     "selected",
     "shadowroot",
+    "shadowrootclonable",
     "shadowrootmode",
     "shadowrootdelegatesfocus",
     "shape",
