diff --git a/third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc b/third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc
index 6e8e5b7cf70d3df99cfd814e475ad97e97f74682..51f3c51d04ed4fdbc0db9b159edcb4b9332f9970 100644
--- a/third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc
+++ b/third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc
@@ -203,7 +203,7 @@ v8::Local<v8::Value> JSEventHandlerForContentAttribute::GetCompiledHandler(
   DCHECK_LE(scopes_size, std::size(scopes));
 
   v8::ScriptOrigin origin(
-      isolate, V8String(isolate, source_url_), position_.line_.ZeroBasedInt(),
+      V8String(isolate, source_url_), position_.line_.ZeroBasedInt(),
       position_.column_.ZeroBasedInt(),
       true);  // true as |SanitizeScriptErrors::kDoNotSanitize|
   v8::ScriptCompiler::Source source(V8String(isolate, script_body_), origin);
