diff --git a/third_party/blink/renderer/core/css/css_primitive_value.h b/third_party/blink/renderer/core/css/css_primitive_value.h
index 1266c686e545c2b3bdcddb465773d103429dd9d4..4e0d80a289c56a1be5bcd7430c6dbd952cb4e94e 100644
--- a/third_party/blink/renderer/core/css/css_primitive_value.h
+++ b/third_party/blink/renderer/core/css/css_primitive_value.h
@@ -129,6 +129,8 @@ class CORE_EXPORT CSSPrimitiveValue : public CSSValue {
     kIcs,
     kLhs,
     kRlhs,
+    kCaps,
+    kRcaps,
     kUserUnits,  // The SVG term for unitless lengths
     // Angle units
     kDegrees,
@@ -163,6 +165,8 @@ class CORE_EXPORT CSSPrimitiveValue : public CSSValue {
     kUnitTypePercentage,
     kUnitTypeFontSize,
     kUnitTypeFontXSize,
+    kUnitTypeFontCapitalHeight,
+    kUnitTypeRootFontCapitalHeight,
     kUnitTypeRootFontSize,
     kUnitTypeRootFontXSize,
     kUnitTypeRootFontZeroCharacterWidth,
@@ -231,6 +235,9 @@ class CORE_EXPORT CSSPrimitiveValue : public CSSValue {
     static_assert(kUnitTypeZeroCharacterWidth < kSize, "ch unit supported");
     static_assert(kUnitTypeRootFontZeroCharacterWidth < kSize,
                   "rch unit supported");
+    static_assert(kUnitTypeFontCapitalHeight < kSize, "cap unit supported");
+    static_assert(kUnitTypeRootFontCapitalHeight < kSize,
+                  "rcap unit supported");
     static_assert(kUnitTypeViewportWidth < kSize, "vw unit supported");
     static_assert(kUnitTypeViewportHeight < kSize, "vh unit supported");
     static_assert(kUnitTypeViewportInlineSize < kSize, "vi unit supported");
@@ -303,7 +310,8 @@ class CORE_EXPORT CSSPrimitiveValue : public CSSValue {
            type == UnitType::kChs || type == UnitType::kIcs ||
            type == UnitType::kLhs || type == UnitType::kRexs ||
            type == UnitType::kRchs || type == UnitType::kRics ||
-           type == UnitType::kRlhs || IsViewportPercentageLength(type) ||
+           type == UnitType::kRlhs || type == UnitType::kCaps ||
+           type == UnitType::kRcaps || IsViewportPercentageLength(type) ||
            IsContainerPercentageLength(type);
   }
   bool IsLength() const;
