diff --git a/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom b/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
index a9d8b17d029ff11fcd5f0bd1f3c25bdaafafe63b..b2b0e3966bc85ddbeea84b7f6f7480777adf9f5a 100644
--- a/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
+++ b/third_party/blink/public/mojom/use_counter/metrics/css_property_id.mojom
@@ -816,6 +816,8 @@ enum CSSSampleId {
     kTimelineScope = 762,
     kScrollbarColor = 763,
     kWordBoundaryDetection = 764,
+    // FlyWeb: same number as Chrome 117 (765 is not in the 116).
+    kTransitionBehavior = 766,
     // 1. Add new features above this line (don't change the assigned numbers of
     //    the existing items).
     // 2. Run the src/tools/metrics/histograms/update_use_counter_css.py script
