diff --git a/third_party/blink/renderer/core/css/css_syntax_definition.cc b/third_party/blink/renderer/core/css/css_syntax_definition.cc
index 2a2d5945f39947d2013047381b3b560a2076beb0..33fe320c10ba503f35d9b657a3e150c0d6960730 100644
--- a/third_party/blink/renderer/core/css/css_syntax_definition.cc
+++ b/third_party/blink/renderer/core/css/css_syntax_definition.cc
@@ -119,8 +119,8 @@ const CSSValue* CSSSyntaxDefinition::Parse(CSSTokenizedValue value,
                                            const CSSParserContext& context,
                                            bool is_animation_tainted) const {
   if (IsUniversal()) {
-    return CSSVariableParser::ParseVariableReferenceValue(value, context,
-                                                          is_animation_tainted);
+    return CSSVariableParser::ParseUniversalSyntaxValue(value, context,
+                                                        is_animation_tainted);
   }
   value.range.ConsumeWhitespace();
   for (const CSSSyntaxComponent& component : syntax_components_) {
