diff --git a/third_party/blink/renderer/core/animation/css_path_interpolation_type.cc b/third_party/blink/renderer/core/animation/css_path_interpolation_type.cc
index c99d16a14208e12b9acb22cb6296d3e3d1130cbc..9f23db1dbca40d3f0499dbe337abc6fe9158be24 100644
--- a/third_party/blink/renderer/core/animation/css_path_interpolation_type.cc
+++ b/third_party/blink/renderer/core/animation/css_path_interpolation_type.cc
@@ -13,6 +13,7 @@
 #include "third_party/blink/renderer/core/css/css_value_list.h"
 #include "third_party/blink/renderer/core/css/resolver/style_resolver_state.h"
 #include "third_party/blink/renderer/core/style/computed_style.h"
+#include "third_party/blink/renderer/core/style/computed_style_constants.h"
 #include "third_party/blink/renderer/core/style/shape_clip_path_operation.h"
 #include "third_party/blink/renderer/core/style/shape_offset_path_operation.h"
 
@@ -60,7 +61,9 @@ void SetPath(const CSSProperty& property,
           std::move(path), CoordBox::kBorderBox));
       return;
     case CSSPropertyID::kClipPath:
-      builder.SetClipPath(ShapeClipPathOperation::Create(std::move(path)));
+      // TODO(pdr): Handle geometry box.
+      builder.SetClipPath(ShapeClipPathOperation::Create(
+          std::move(path), GeometryBox::kBorderBox));
       return;
     default:
       NOTREACHED();
