diff --git a/third_party/blink/renderer/core/style/style_image.h b/third_party/blink/renderer/core/style/style_image.h
index 91713f58bea5070ad50726ed58dadb5190ebcae7..673fefec36278bfe18c30055498759bc951d0bea 100644
--- a/third_party/blink/renderer/core/style/style_image.h
+++ b/third_party/blink/renderer/core/style/style_image.h
@@ -31,7 +31,7 @@
 
 namespace gfx {
 class SizeF;
-}
+}  // namespace gfx
 
 namespace blink {
 
@@ -121,9 +121,9 @@ class CORE_EXPORT StyleImage : public GarbageCollected<StyleImage> {
   // and can have a variety of relationships to the container's size. Hence
   // it requires float resolution.
   //
-  // Note that the |target_size| is in the effective zoom level of the
+  // Note that the `target_size` is in the effective zoom level of the
   // computed style, i.e if the style has an effective zoom level of 1.0 the
-  // |target_size| is not zoomed.
+  // `target_size` is not zoomed.
   virtual scoped_refptr<Image> GetImage(
       const ImageResourceObserver&,
       const Document&,
@@ -159,6 +159,9 @@ class CORE_EXPORT StyleImage : public GarbageCollected<StyleImage> {
   ALWAYS_INLINE bool IsImageResourceSet() const {
     return is_image_resource_set_;
   }
+  ALWAYS_INLINE bool IsSVGMaskReference() const {
+    return is_svg_mask_reference_;
+  }
   ALWAYS_INLINE bool IsPaintImage() const { return is_paint_image_; }
   ALWAYS_INLINE bool IsCrossfadeImage() const { return is_crossfade_; }
 
@@ -175,6 +178,7 @@ class CORE_EXPORT StyleImage : public GarbageCollected<StyleImage> {
         is_generated_image_(false),
         is_image_resource_set_(false),
         is_crossfade_(false),
+        is_svg_mask_reference_(false),
         is_paint_image_(false),
         is_lazyload_possibly_deferred_(false) {}
   bool is_image_resource_ : 1;
@@ -182,6 +186,7 @@ class CORE_EXPORT StyleImage : public GarbageCollected<StyleImage> {
   bool is_generated_image_ : 1;
   bool is_image_resource_set_ : 1;
   bool is_crossfade_ : 1;
+  bool is_svg_mask_reference_ : 1;
   bool is_paint_image_ : 1;
   bool is_lazyload_possibly_deferred_ : 1;
 
