diff --git a/third_party/blink/renderer/core/html/forms/html_text_area_element.h b/third_party/blink/renderer/core/html/forms/html_text_area_element.h
index cc4528465d7a827d31bf5e3f3e7e258f874b2e5a..721682233e690d3385e9c42709003708ea009037 100644
--- a/third_party/blink/renderer/core/html/forms/html_text_area_element.h
+++ b/third_party/blink/renderer/core/html/forms/html_text_area_element.h
@@ -118,6 +118,7 @@ class CORE_EXPORT HTMLTextAreaElement final : public TextControlElement {
   void RestoreFormControlState(const FormControlState&) override;
 
   bool IsTextControl() const override { return true; }
+  bool ShouldAutoDirUseValue() const final { return true; }
   int scrollWidth() override;
   int scrollHeight() override;
   void ChildrenChanged(const ChildrenChange&) override;
