diff --git a/third_party/blink/renderer/build/scripts/core/css/parser/templates/css.proto.tmpl b/third_party/blink/renderer/build/scripts/core/css/parser/templates/css.proto.tmpl
index 435105675ef831f7c6d456fc48d9fef698658d7e..c00a7ae14c6e9f0aeff2d1103eb9300befbac427 100644
--- a/third_party/blink/renderer/build/scripts/core/css/parser/templates/css.proto.tmpl
+++ b/third_party/blink/renderer/build/scripts/core/css/parser/templates/css.proto.tmpl
@@ -1026,21 +1026,23 @@ message Selector {
     START = 79;
     TARGET = 80;
     TOP_LAYER = 81;
-    VALID = 82;
-    VERTICAL = 83;
-    VISITED = 84;
-    WINDOW_INACTIVE = 85;
-    _WEBKIT_ANY = 86;
-    HOST_CONTEXT = 87;
-    LANG = 88;
-    NOT = 89;
-    NTH_CHILD = 90;
-    NTH_LAST_CHILD = 91;
-    NTH_LAST_OF_TYPE = 92;
-    NTH_OF_TYPE = 93;
-    SLOTTED = 94;
-    XR_OVERLAY = 95;
-    INVALID_PSEUDO_VALUE = 96;
+    USER_INVALID = 82;
+    USER_VALID = 83;
+    VALID = 84;
+    VERTICAL = 85;
+    VISITED = 86;
+    WINDOW_INACTIVE = 87;
+    _WEBKIT_ANY = 88;
+    HOST_CONTEXT = 89;
+    LANG = 90;
+    NOT = 91;
+    NTH_CHILD = 92;
+    NTH_LAST_CHILD = 93;
+    NTH_LAST_OF_TYPE = 94;
+    NTH_OF_TYPE = 95;
+    SLOTTED = 96;
+    XR_OVERLAY = 97;
+    INVALID_PSEUDO_VALUE = 98;
   }
   optional PseudoValueId pseudo_value_id = 4;
   required Combinator combinator = 5;
