diff --git a/third_party/blink/renderer/core/url_pattern/url_pattern.h b/third_party/blink/renderer/core/url_pattern/url_pattern.h
index 98bb97334747eb5b861cebff19edfa2cef0708f0..546628b954c47a64b2cc057c6f6e5e1dd5d9a75d 100644
--- a/third_party/blink/renderer/core/url_pattern/url_pattern.h
+++ b/third_party/blink/renderer/core/url_pattern/url_pattern.h
@@ -84,6 +84,8 @@ class CORE_EXPORT URLPattern : public ScriptWrappable {
   String search() const;
   String hash() const;
 
+  bool hasRegExpGroups() const;
+
   static int compareComponent(const V8URLPatternComponent& component,
                               const URLPattern* left,
                               const URLPattern* right);
