diff --git a/third_party/blink/renderer/platform/bindings/wrapper_type_info.cc b/third_party/blink/renderer/platform/bindings/wrapper_type_info.cc
index ce61e53d8b35cd40a51be5e59d6710deaf92c03c..26f85768e33a5ac91c158a4e77101a051596c403 100644
--- a/third_party/blink/renderer/platform/bindings/wrapper_type_info.cc
+++ b/third_party/blink/renderer/platform/bindings/wrapper_type_info.cc
@@ -46,6 +46,9 @@ v8::Local<v8::Template> WrapperTypeInfo::GetV8ClassTemplate(
     case kIdlSyncIterator:
       v8_template = v8::FunctionTemplate::New(isolate);
       break;
+    case kIdlAsyncIterator:
+      v8_template = v8::FunctionTemplate::New(isolate);
+      break;
     case kCustomWrappableKind:
       v8_template = v8::FunctionTemplate::New(isolate);
       break;
