diff --git a/third_party/blink/renderer/bindings/bindings.gni b/third_party/blink/renderer/bindings/bindings.gni
index 4e49064e5b44777fd85bb01378f84be1917269ce..a7187c4a48e99eafc5effd0918007f14854e75ab 100644
--- a/third_party/blink/renderer/bindings/bindings.gni
+++ b/third_party/blink/renderer/bindings/bindings.gni
@@ -16,6 +16,8 @@ blink_core_sources_bindings =
                     "core/v8/active_script_wrappable_creation_key.h",
                     "core/v8/array_value.cc",
                     "core/v8/array_value.h",
+                    "core/v8/async_iterable.cc",
+                    "core/v8/async_iterable.h",
                     "core/v8/binding_security.cc",
                     "core/v8/binding_security.h",
                     "core/v8/boxed_v8_module.h",
