diff --git a/third_party/blink/renderer/platform/runtime_enabled_features.json5 b/third_party/blink/renderer/platform/runtime_enabled_features.json5
index e77f3afb2cc9e94b35d7456daa7f7e7554d56bd8..091f289612c5ccc7ab946f1c39ec36b98f59fa92 100644
--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -276,8 +276,10 @@
       public: true,
     },
     {
+      // Shipping in M120.  Flag should be removed after M120 has been
+      // shipping to stable for a few weeks.
       name: "AccordionPattern",
-      status: "experimental",
+      status: "stable",
     },
     {
       name: "AddIdentityInCanMakePaymentEvent",
@@ -621,6 +623,16 @@
       status: "experimental",
       base_feature: "none",
     },
+    {
+      // https://drafts.fxtf.org/css-masking/#typedef-geometry-box
+      name: "ClipPathGeometryBox",
+      status: "stable",
+    },
+    {
+      // https://www.w3.org/TR/css-shapes/#supported-basic-shapes
+      name: "ClipPathXYWHAndRect",
+      status: "stable",
+    },
     {
       name: "CloseWatcher",
     },
@@ -796,6 +808,10 @@
       name: "CSSCalcSimplificationAndSerialization",
       base_feature: "none",
     },
+    {
+      name: "CSSCapFontUnits",
+      status: "stable",
+    },
     {
       // Support case-sensitive attribute selector modifier
       // https://drafts.csswg.org/selectors-4/#attribute-case
@@ -815,7 +831,8 @@
     },
     {
       name: "CSSContainIntrinsicSizeAutoNone",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     // Ablation study to try to measure very roughly memory savings from not saving
     // pretokenized strings in CSS custom properties.
@@ -846,7 +863,7 @@
     },
     {
       name: "CSSExponentialFunctions",
-      status: "experimental",
+      status: "stable",
     },
     {
       name: "CSSFocusVisible",
@@ -909,7 +926,8 @@
     },
     {
       name: "CSSInitialPseudo",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -960,6 +978,14 @@
       implied_by: ["MathMLCore"],
       base_feature: "none",
     },
+    {
+      // Un-prefixed and interoperable mask implementation, including
+      // sub-features such as mask-image, mask-mode, etc.
+      // Tracking bug: https://crbug.com/1418401
+      // Spec: https://drafts.fxtf.org/css-masking/#the-mask
+      name: "CSSMaskingInterop",
+      status: "stable",
+    },
     {
       name: "CSSMixBlendModePlusLighter",
       status: "stable",
@@ -975,6 +1001,14 @@
     {
       // Support nested selectors that start with idents.
       name: "CSSNestingIdent",
+      status: "stable",
+    },
+    {
+      // Support for typed om numeric factory methods for various units that
+      // were otherwise shipped: rex, rch, ic, ric, lh, rlh, x (cap and rcap
+      // covered by separate runtime flag).
+      name: "CSSNumericFactoryCompleteness",
+      status: "stable",
     },
     {
       name: "CSSObjectViewBox",
@@ -1042,8 +1076,9 @@
       base_feature: "none",
     },
     {
+      // FlyWeb: shipped in M120 (level 120).
       name: "CSSPseudoDir",
-      status: "experimental",
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -1099,8 +1134,9 @@
       // https://drafts.csswg.org/css-pseudo-4/#selectordef-grammar-error
       // https://drafts.csswg.org/css-text-decor-4/#valdef-text-decoration-line-spelling-error
       // https://drafts.csswg.org/css-text-decor-4/#valdef-text-decoration-line-grammar-error
+      // FlyWeb: shipped in M121 (level 121).
       name: "CSSSpellingGrammarErrors",
-      status: "experimental",
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -1152,7 +1188,8 @@
     // `text-wrap: pretty`. crbug.com/1432798
     {
       name: "CSSTextWrapPretty",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       depends_on: ["CSSTextWrap", "CSSWhiteSpaceShorthand"],
     },
     // Support for CSS Toggles, https://tabatkins.github.io/css-toggle/
@@ -1170,12 +1207,21 @@
       //
       // See: https://github.com/w3c/csswg-drafts/issues/8189
       name: "CSSTopLayerForTransitions",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
+    {
+      // Support for the 'border-box', 'content-box' and 'stroke-box' keywords
+      // for the 'transform-box' property.
+      // FlyWeb: as shipped in Chromium 118 (engine level 118).
+      name: "CSSTransformBoxAdditionalKeywords",
+      status: "stable",
+    },
     {
       name: "CSSTransitionDiscrete",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
     },
     {
       // A kill switch for the fix for part of https://crbug.com/1180387 .
@@ -1303,6 +1349,14 @@
       name: "DesktopPWAsSubApps",
       status: "test",
     },
+    {
+      // This flag makes the details element's toggle events use the
+      // ToggleEvent class instead of being a regular Event.
+      // http://crbug.com/1416738
+      // https://github.com/whatwg/html/pull/8893
+      name: "DetailsElementToggleEvent",
+      status: "stable",
+    },
     {
       name: "DeviceAttributes",
       origin_trial_feature_name: "DeviceAttributes",
@@ -1996,6 +2050,11 @@
       name: "HTMLPopoverHint",
       status: "experimental",
     },
+    {
+      // FlyWeb: <search>, Chrome 118 (engine level 118).
+      name: "HTMLSearchElement",
+      status: "stable",
+    },
     {
       name: "HTMLSelectMenuElement",
       status: "experimental",
@@ -2137,7 +2196,8 @@
     },
     {
       name: "LayoutNGSubgrid",
-      status: "experimental",
+      // FlyWeb: on, as shipped in Chromium 117 (engine level 117).
+      status: "stable",
       base_feature: "none",
     },
     {
@@ -3109,18 +3169,23 @@
       status: "stable",
       base_feature: "none",
     },
+    {
+      name: "ScriptingMediaFeature",
+      status: "stable",
+    },
     {
       name: "ScrollableAreaNoSnapping",
       status: "stable",
     },
     {
+      // FlyWeb: shipped in M121 (level 121).
       name: "ScrollbarColor",
-      status: "test",
+      status: "stable",
     },
     {
+      // FlyWeb: shipped in M121 (level 121).
       name: "ScrollbarWidth",
-      status: "experimental",
-      base_feature: "none",
+      status: "stable",
     },
     {
       name: "ScrollCustomization",
@@ -3453,6 +3518,10 @@
       name: "SvgMapLocalToAncestorFix",
       status: "stable",
     },
+    {
+      name: "SvgNoPixelSnappingScaleAdjustment",
+      status: "stable",
+    },
     {
       name: "SvgRasterOptimizations",
       status: "experimental",
@@ -3638,6 +3707,10 @@
       origin_trial_feature_name: "UnrestrictedSharedArrayBuffer",
       origin_trial_os: ["win", "mac", "linux", "fuchsia", "chromeos"],
     },
+    {
+      name: "URLCanParse",
+      status: "stable",
+    },
     {
       name: "URLPatternCompareComponent",
       status: "experimental",
@@ -3657,6 +3730,11 @@
       name: "UserAgentClientHint",
       status: "stable",
     },
+    {
+      // https://chromestatus.com/feature/5132477781245952
+      name: "UserValidUserInvalid",
+      status: "stable",
+    },
     {
       name: "V8IdleTasks",
       base_feature: "none",
@@ -4167,6 +4245,12 @@
       name: "WindowPlacementFullscreenOnScreensChange",
       status: "experimental",
     },
+    {
+      // Update xywh()/rect() computed value to use inset() per the spec:
+      // https://drafts.csswg.org/css-shapes-1/#basic-shape-computed-values
+      name: "XYWHAndRectComputedValue",
+      status: "stable",
+    },
     {
       // If enabled, the `getDisplayMedia()` family of APIs will ask for NV12
       // frames, which should trigger a zero-copy path in the tab capture code.
