diff --git a/third_party/liburlpattern/pattern.h b/third_party/liburlpattern/pattern.h
index 803264ce2536eec3d1b8125c7bac34b1b09f72de..2ad43e5399e7039976809e642c55fe82aa698e94 100644
--- a/third_party/liburlpattern/pattern.h
+++ b/third_party/liburlpattern/pattern.h
@@ -135,6 +135,9 @@ class COMPONENT_EXPORT(LIBURLPATTERN) Pattern {
 
   const std::vector<Part>& PartList() const { return part_list_; }
 
+  // Returns true if the pattern has at least one kRegex part.
+  bool HasRegexGroups() const;
+
   // Returns true if the pattern can match input strings using `DirectMatch()`.
   bool CanDirectMatch() const;
 
