diff --git a/third_party/blink/renderer/core/css/css_value.h b/third_party/blink/renderer/core/css/css_value.h
index ae4527ce78bd2238489b9e7af95d1cac489528e9..fc38be2fcaf6f2eee9d44ba73b753cfe51b5e365 100644
--- a/third_party/blink/renderer/core/css/css_value.h
+++ b/third_party/blink/renderer/core/css/css_value.h
@@ -197,6 +197,8 @@ class CORE_EXPORT CSSValue : public GarbageCollected<CSSValue> {
   bool IsViewValue() const { return class_type_ == kViewClass; }
   bool IsRatioValue() const { return class_type_ == kRatioClass; }
 
+  bool IsRepeatStyleValue() const { return class_type_ == kRepeatStyleClass; }
+
   bool HasFailedOrCanceledSubresources() const;
   bool MayContainUrl() const;
   void ReResolveUrl(const Document&) const;
@@ -302,6 +304,8 @@ class CORE_EXPORT CSSValue : public GarbageCollected<CSSValue> {
     kImageSetOptionClass,
     kImageSetTypeClass,
 
+    kRepeatStyleClass,
+
     // List class types must appear after ValueListClass.
     kValueListClass,
     kFunctionClass,
