diff --git a/third_party/blink/renderer/core/scroll/scrollbar_theme_mac.cc b/third_party/blink/renderer/core/scroll/scrollbar_theme_mac.cc
index 19cd73f608b452fa51197ada958e9809bc44d059..b951efc3c80382647c215c15958b850f80319ad0 100644
--- a/third_party/blink/renderer/core/scroll/scrollbar_theme_mac.cc
+++ b/third_party/blink/renderer/core/scroll/scrollbar_theme_mac.cc
@@ -186,6 +186,17 @@ WebThemeEngine::ExtraParams GetPaintParams(const Scrollbar& scrollbar,
   params.scrollbar_extra.is_hovering =
       scrollbar.HoveredPart() != ScrollbarPart::kNoPart;
   params.scrollbar_extra.scale_from_dip = scrollbar.ScaleFromDIP();
+
+  if (scrollbar.ScrollbarThumbColor().has_value()) {
+    params.scrollbar_extra.thumb_color =
+        scrollbar.ScrollbarThumbColor().value().toSkColor4f().toSkColor();
+  }
+
+  if (scrollbar.ScrollbarTrackColor().has_value()) {
+    params.scrollbar_extra.track_color =
+        scrollbar.ScrollbarTrackColor().value().toSkColor4f().toSkColor();
+  }
+
   return params;
 }
 
