diff --git a/third_party/blink/renderer/core/css/cssom/style_value_factory.cc b/third_party/blink/renderer/core/css/cssom/style_value_factory.cc
index ac1c9fb574d4a7ec1ed35bfb567fc730bd117914..4974db45a1b7cf059157cf2672f6a8c825aef43f 100644
--- a/third_party/blink/renderer/core/css/cssom/style_value_factory.cc
+++ b/third_party/blink/renderer/core/css/cssom/style_value_factory.cc
@@ -123,6 +123,19 @@ CSSStyleValue* CreateStyleValueWithPropertyInternal(CSSPropertyID property_id,
       return MakeGarbageCollected<CSSUnsupportedStyleValue>(
           CSSPropertyName(property_id), value);
     }
+    case CSSPropertyID::kClipPath: {
+      if (value.IsIdentifierValue()) {
+        return CreateStyleValue(value);
+      }
+
+      if (const auto* value_list = DynamicTo<CSSValueList>(&value)) {
+        // Only single keywords are supported in level 1.
+        if (value_list->length() == 1U) {
+          return CreateStyleValue(value_list->Item(0));
+        }
+      }
+      return nullptr;
+    }
     case CSSPropertyID::kContain:
     case CSSPropertyID::kContainerType: {
       if (value.IsIdentifierValue()) {
