diff --git a/third_party/blink/renderer/core/events/pop_state_event.idl b/third_party/blink/renderer/core/events/pop_state_event.idl
index 16624a90559f046509fd9bc84a8003b77d930b66..c424187ee36d5e7d1a222316207507aeb0080488 100644
--- a/third_party/blink/renderer/core/events/pop_state_event.idl
+++ b/third_party/blink/renderer/core/events/pop_state_event.idl
@@ -32,4 +32,5 @@
 ] interface PopStateEvent : Event {
     [CallWith=ScriptState] constructor(DOMString type, optional PopStateEventInit eventInitDict = {});
     [CallWith=ScriptState, RaisesException, CachedAttribute=IsStateDirty] readonly attribute any state;
+    [RuntimeEnabled=HasUAVisualTransition] readonly attribute boolean hasUAVisualTransition;
 };
