diff --git a/third_party/blink/public/common/privacy_budget/identifiable_surface.h b/third_party/blink/public/common/privacy_budget/identifiable_surface.h
index 8ad570b10e702988db11273d2e8d33260de9afec..bd087758b363b7e2fad868887b54f48600e9fb01 100644
--- a/third_party/blink/public/common/privacy_budget/identifiable_surface.h
+++ b/third_party/blink/public/common/privacy_budget/identifiable_surface.h
@@ -352,8 +352,9 @@ class IdentifiableSurface {
     kHorizontalViewportSegments = 23,
     kVerticalViewportSegments = 24,
     kAspectRatioNormalized = 25,
+    kScripting = 28,
     // We can use enum values up to and including 63, see static_assert below.
-    kMax = kAspectRatioNormalized
+    kMax = kScripting
   };
   static_assert(static_cast<int>(MediaFeatureName::kMax) < 64,
                 "MediaFeatureName only allows values < 64 since we use it in "
