diff --git a/third_party/blink/renderer/core/html/forms/password_input_type.h b/third_party/blink/renderer/core/html/forms/password_input_type.h
index d8ea947fce17974a66c916f38d6cda969e1827dd..63610dc8c715906621de5a897b84e16b1dbf7520 100644
--- a/third_party/blink/renderer/core/html/forms/password_input_type.h
+++ b/third_party/blink/renderer/core/html/forms/password_input_type.h
@@ -66,6 +66,7 @@ class PasswordInputType final : public BaseTextInputType {
 
   void HandleBlurEvent() override;
   bool SupportsInputModeAttribute() const override;
+  bool ShouldAutoDirUseValue() const override;
 
   bool should_draw_caps_lock_indicator_ = false;
   bool should_show_reveal_button_ = false;
