diff --git a/third_party/blink/renderer/core/paint/box_painter_base.h b/third_party/blink/renderer/core/paint/box_painter_base.h
index 286f4abf9ef6d84dfc7e0ba56c77d1e66a2485c3..5e0afefa8dbf42e4c332bcd50be780804c589faa 100644
--- a/third_party/blink/renderer/core/paint/box_painter_base.h
+++ b/third_party/blink/renderer/core/paint/box_painter_base.h
@@ -146,6 +146,7 @@ class BoxPainterBase {
  protected:
   virtual NGPhysicalBoxStrut ComputeBorders() const = 0;
   virtual NGPhysicalBoxStrut ComputePadding() const = 0;
+  virtual NGPhysicalBoxStrut ComputeMargins() const = 0;
   NGPhysicalBoxStrut AdjustedBorderOutsets(const FillLayerInfo&) const;
   void PaintFillLayerTextFillBox(const PaintInfo&,
                                  const FillLayerInfo&,
