diff --git a/third_party/blink/renderer/core/paint/svg_root_painter.cc b/third_party/blink/renderer/core/paint/svg_root_painter.cc
index 23a2c881ac43ae1f47cd1a5766eed5ae058496f1..399be186ebf3887bc661cd6e8bb1a7a7f976b74b 100644
--- a/third_party/blink/renderer/core/paint/svg_root_painter.cc
+++ b/third_party/blink/renderer/core/paint/svg_root_painter.cc
@@ -14,9 +14,24 @@
 #include "third_party/blink/renderer/core/paint/scoped_svg_paint_state.h"
 #include "third_party/blink/renderer/core/paint/svg_foreign_object_painter.h"
 #include "third_party/blink/renderer/core/svg/svg_svg_element.h"
+#include "third_party/blink/renderer/platform/runtime_enabled_features.h"
 
 namespace blink {
 
+namespace {
+
+bool ShouldApplySnappingScaleAdjustment(const LayoutSVGRoot& layout_svg_root) {
+  // If the RuntimeEnabledFeatures flag isn't set then apply scale adjustment.
+  if (!RuntimeEnabledFeatures::SvgNoPixelSnappingScaleAdjustmentEnabled()) {
+    return true;
+  }
+  // Apply scale adjustment if the SVG root is the document root - i.e it is
+  // not an inline SVG.
+  return layout_svg_root.IsDocumentElement();
+}
+
+}  // namespace
+
 gfx::Rect SVGRootPainter::PixelSnappedSize(
     const PhysicalOffset& paint_offset) const {
   return ToPixelSnappedRect(
@@ -28,11 +43,13 @@ AffineTransform SVGRootPainter::TransformToPixelSnappedBorderBox(
   const gfx::Rect snapped_size = PixelSnappedSize(paint_offset);
   AffineTransform paint_offset_to_border_box =
       AffineTransform::Translation(snapped_size.x(), snapped_size.y());
-  LayoutSize size = layout_svg_root_.Size();
-  if (!size.IsEmpty()) {
-    paint_offset_to_border_box.Scale(
-        snapped_size.width() / size.Width().ToFloat(),
-        snapped_size.height() / size.Height().ToFloat());
+  if (ShouldApplySnappingScaleAdjustment(layout_svg_root_)) {
+    LayoutSize size = layout_svg_root_.Size();
+    if (!size.IsEmpty()) {
+      paint_offset_to_border_box.Scale(
+          snapped_size.width() / size.Width().ToFloat(),
+          snapped_size.height() / size.Height().ToFloat());
+    }
   }
   paint_offset_to_border_box.PreConcat(
       layout_svg_root_.LocalToBorderBoxTransform());
