diff --git a/third_party/blink/renderer/core/events/pop_state_event.h b/third_party/blink/renderer/core/events/pop_state_event.h
index 0cd8f979d45284429fb00d1b7aa711332c88c054..8cb6e17463c3d96f97e7dab9f8d688a6d4b1f664 100644
--- a/third_party/blink/renderer/core/events/pop_state_event.h
+++ b/third_party/blink/renderer/core/events/pop_state_event.h
@@ -59,6 +59,7 @@ class CORE_EXPORT PopStateEvent final : public Event {
   ~PopStateEvent() override = default;
 
   ScriptValue state(ScriptState* script_state, ExceptionState& exception_state);
+  bool hasUAVisualTransition() const { return has_ua_visual_transition_; }
   bool IsStateDirty() const { return false; }
 
   const AtomicString& InterfaceName() const override;
@@ -69,6 +70,7 @@ class CORE_EXPORT PopStateEvent final : public Event {
   WorldSafeV8Reference<v8::Value> state_;
   scoped_refptr<SerializedScriptValue> serialized_state_;
   Member<History> history_;
+  const bool has_ua_visual_transition_ = false;
 };
 
 }  // namespace blink
