diff --git a/third_party/blink/renderer/core/layout/svg/layout_svg_inline.h b/third_party/blink/renderer/core/layout/svg/layout_svg_inline.h
index 24ac2561844bd16865f98faafbd48031d00478f4..b792177cbe195dc6929aed4b42be821cb6ca1302 100644
--- a/third_party/blink/renderer/core/layout/svg/layout_svg_inline.h
+++ b/third_party/blink/renderer/core/layout/svg/layout_svg_inline.h
@@ -48,7 +48,7 @@ class LayoutSVGInline : public LayoutInline {
   bool IsChildAllowed(LayoutObject*, const ComputedStyle&) const override;
 
   gfx::RectF ObjectBoundingBox() const final;
-  gfx::RectF StrokeBoundingBox() const final;
+  gfx::RectF DecoratedBoundingBox() const final;
   gfx::RectF VisualRectInLocalSVGCoordinates() const final;
 
   PhysicalRect VisualRectInDocument(
