diff --git a/third_party/blink/renderer/core/navigation_api/build.gni b/third_party/blink/renderer/core/navigation_api/build.gni
index 8e9fa7f4c49d77a73cbb76dccfd5896c9be8aad2..5c8e059694978916ef603b7df934fe77ee97cda2 100644
--- a/third_party/blink/renderer/core/navigation_api/build.gni
+++ b/third_party/blink/renderer/core/navigation_api/build.gni
@@ -7,6 +7,7 @@ blink_core_sources_navigation_api = [
   "navigate_event.h",
   "navigate_event_dispatch_params.cc",
   "navigate_event_dispatch_params.h",
+  "navigation_activation.h",
   "navigation_api_navigation.cc",
   "navigation_api_navigation.h",
   "navigation_api.cc",
