diff --git a/ui/native_theme/native_theme.h b/ui/native_theme/native_theme.h
index 5ee7cd7952f7b956c32e793e251d5b0cb457026a..31d9b0d5eb601cfe8c06558412ca217ea0efca2d 100644
--- a/ui/native_theme/native_theme.h
+++ b/ui/native_theme/native_theme.h
@@ -294,6 +294,11 @@ class NATIVE_THEME_EXPORT NativeTheme {
     ScrollbarOverlayColorTheme scrollbar_theme;
     ScrollbarOrientation orientation;  // Used on Mac for drawing gradients.
     float scale_from_dip;
+    absl::optional<SkColor> thumb_color;
+    absl::optional<SkColor> track_color;
+
+    ScrollbarExtraParams()
+        : thumb_color(absl::nullopt), track_color(absl::nullopt) {}
   };
 #endif
 
