diff --git a/ui/native_theme/native_theme_mac.h b/ui/native_theme/native_theme_mac.h
index 80914c653dda70600c4a5dc99b7277f0b152383f..7d08f4cd77d0c6c2ea9da54ea6e699d8049e799d 100644
--- a/ui/native_theme/native_theme_mac.h
+++ b/ui/native_theme/native_theme_mac.h
@@ -129,6 +129,7 @@ class NATIVE_THEME_EXPORT NativeThemeMac : public NativeThemeBase {
 
   enum ScrollbarPart {
     kThumb,
+    kTrack,
     kTrackInnerBorder,
     kTrackOuterBorder,
   };
