diff --git a/ui/native_theme/native_theme_mac.mm b/ui/native_theme/native_theme_mac.mm
index 37a21af908449a7256baf1db213d70fb693f68e5..d0229c5f30d33ef9695a3dcfb0fae2cb048ff54c 100644
--- a/ui/native_theme/native_theme_mac.mm
+++ b/ui/native_theme/native_theme_mac.mm
@@ -280,11 +280,19 @@ void NativeThemeMac::PaintScrollBarTrackGradient(
   }
 
   // And draw.
-  cc::PaintFlags gradient;
-  gradient.setShader(cc::PaintShader::MakeLinearGradient(
-      gradient_bounds.data(), gradient_colors.data(), nullptr,
-      gradient_colors.size(), SkTileMode::kClamp));
-  paint_canvas.DrawRect(rect, gradient);
+  cc::PaintFlags flags;
+  absl::optional<SkColor> track_color =
+      GetScrollbarColor(ScrollbarPart::kTrack, color_scheme, extra_params);
+  if (track_color.has_value()) {
+    flags.setAntiAlias(true);
+    flags.setStyle(cc::PaintFlags::kFill_Style);
+    flags.setColor(track_color.value());
+  } else {
+    flags.setShader(cc::PaintShader::MakeLinearGradient(
+        gradient_bounds.data(), gradient_colors.data(), nullptr,
+        gradient_colors.size(), SkTileMode::kClamp));
+  }
+  paint_canvas.DrawRect(rect, flags);
 }
 
 void NativeThemeMac::PaintScrollbarTrackInnerBorder(
@@ -420,6 +428,9 @@ absl::optional<SkColor> NativeThemeMac::GetScrollbarColor(
   // colors.
   bool dark_mode = color_scheme == ColorScheme::kDark;
   if (part == ScrollbarPart::kThumb) {
+    if (extra_params.thumb_color.has_value()) {
+      return extra_params.thumb_color.value();
+    }
     if (extra_params.is_overlay)
       return dark_mode ? SkColorSetARGB(0x80, 0xFF, 0xFF, 0xFF)
                        : SkColorSetARGB(0x80, 0, 0, 0);
@@ -431,6 +442,10 @@ absl::optional<SkColor> NativeThemeMac::GetScrollbarColor(
     return extra_params.is_hovering ? SkColorSetARGB(0x80, 0, 0, 0)
                                     : SkColorSetARGB(0x3A, 0, 0, 0);
   } else if (part == ScrollbarPart::kTrackInnerBorder) {
+    if (extra_params.track_color.has_value()) {
+      return extra_params.track_color.value();
+    }
+
     if (extra_params.is_overlay)
       return dark_mode ? SkColorSetARGB(0x33, 0xE5, 0xE5, 0xE5)
                        : SkColorSetARGB(0xF9, 0xDF, 0xDF, 0xDF);
@@ -438,12 +453,19 @@ absl::optional<SkColor> NativeThemeMac::GetScrollbarColor(
     return dark_mode ? SkColorSetRGB(0x3D, 0x3D, 0x3D)
                      : SkColorSetRGB(0xE8, 0xE8, 0xE8);
   } else if (part == ScrollbarPart::kTrackOuterBorder) {
+    if (extra_params.track_color.has_value()) {
+      return extra_params.track_color.value();
+    }
     if (extra_params.is_overlay)
       return dark_mode ? SkColorSetARGB(0x28, 0xD8, 0xD8, 0xD8)
                        : SkColorSetARGB(0xC6, 0xE8, 0xE8, 0xE8);
 
     return dark_mode ? SkColorSetRGB(0x51, 0x51, 0x51)
                      : SkColorSetRGB(0xED, 0xED, 0xED);
+  } else if (part == ScrollbarPart::kTrack) {
+    if (extra_params.track_color.has_value()) {
+      return extra_params.track_color.value();
+    }
   }
 
   return absl::nullopt;
