diff --git a/fxjs/cfxjs_engine.h b/fxjs/cfxjs_engine.h
index b0011fc025b21ee4431e9a3f222c12894fdcf6cf..724391c19ffd5c1da88468d7106fc8eb102e0fe3 100644
--- a/fxjs/cfxjs_engine.h
+++ b/fxjs/cfxjs_engine.h
@@ -108,8 +108,8 @@ class CFXJS_Engine : public CFX_V8 {
                        v8::FunctionCallback pMethodCall);
   void DefineObjProperty(uint32_t nObjDefnID,
                          const char* sPropName,
-                         v8::AccessorGetterCallback pPropGet,
-                         v8::AccessorSetterCallback pPropPut);
+                         v8::AccessorNameGetterCallback pPropGet,
+                         v8::AccessorNameSetterCallback pPropPut);
   void DefineObjAllProperties(
       uint32_t nObjDefnID,
       v8::GenericNamedPropertyQueryCallback pPropQurey,
