diff --git a/fxjs/cjs_object.h b/fxjs/cjs_object.h
index c53568eee60e7a943c352a1e26e8532c9d33ccc8..def70ed4ab3a24e5f366d285ab0b7ddecd4975d1 100644
--- a/fxjs/cjs_object.h
+++ b/fxjs/cjs_object.h
@@ -24,8 +24,8 @@ struct JSConstSpec {
 
 struct JSPropertySpec {
   const char* pName;
-  v8::AccessorGetterCallback pPropGet;
-  v8::AccessorSetterCallback pPropPut;
+  v8::AccessorNameGetterCallback pPropGet;
+  v8::AccessorNameSetterCallback pPropPut;
 };
 
 struct JSMethodSpec {
