diff --git a/src/text/gpu/GlyphVector.h b/src/text/gpu/GlyphVector.h
index 6bb2964f681263357cc1a8e9115feb4d7d294c05..67aa521551e0accb746c476200f84055ec9b0637 100644
--- a/src/text/gpu/GlyphVector.h
+++ b/src/text/gpu/GlyphVector.h
@@ -67,7 +67,7 @@ public:
     // the sub runs.
     int unflattenSize() const { return GlyphVectorSize(fGlyphs.size()); }
 
-    void packedGlyphIDToGlyph(StrikeCache* cache);
+    void packedGlyphIDToGlyph(StrikeCache* cache, skgpu::MaskFormat maskFormat);
 
     static size_t GlyphVectorSize(size_t count) {
         return sizeof(Variant) * count;
