diff --git a/src/libANGLE/TransformFeedback.h b/src/libANGLE/TransformFeedback.h
index e55d0054c149cefbc3f560384b81bf06418924ed..5bca135486771620559284dfac4fda445f93d8b2 100644
--- a/src/libANGLE/TransformFeedback.h
+++ b/src/libANGLE/TransformFeedback.h
@@ -102,6 +102,10 @@ class TransformFeedback final : public RefCountObject<TransformFeedbackID>, publ
     // Returns true if any buffer bound to this object is also bound to another target.
     bool buffersBoundForOtherUseInWebGL() const;
 
+    // Returns true if the buffer is bound to any of the indexed binding points in this transform
+    // feedback.
+    bool isBufferBound(BufferID bufferID) const;
+
     angle::Result detachBuffer(const Context *context, BufferID bufferID);
 
     rx::TransformFeedbackImpl *getImplementation() const;
