diff --git a/BUILD.gn b/BUILD.gn
index c13cf053a80d212b836a9be3ec6e76b11627a19d..226191895bf04099721092350c4826bab7bacd86 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -700,6 +700,7 @@ config("internal_config_base") {
     "$target_gen_dir",
     "$target_gen_dir/include",
   ]
+  _include_dirs = include_dirs include_dirs = [] include_dirs = [ "//brave/chromium_src/v8" ] + _include_dirs
 }
 
 config("internal_config") {
@@ -1844,12 +1845,13 @@ torque_files = [
   "src/builtins/promise-any.tq",
   "src/builtins/promise-constructor.tq",
   "src/builtins/promise-finally.tq",
+  "src/builtins/promise-jobs.tq",
   "src/builtins/promise-misc.tq",
   "src/builtins/promise-race.tq",
   "src/builtins/promise-reaction-job.tq",
   "src/builtins/promise-resolve.tq",
   "src/builtins/promise-then.tq",
-  "src/builtins/promise-jobs.tq",
+  "src/builtins/promise-withresolvers.tq",
   "src/builtins/proxy-constructor.tq",
   "src/builtins/proxy-delete-property.tq",
   "src/builtins/proxy-get-property.tq",
@@ -2851,6 +2853,7 @@ v8_header_set("v8_headers") {
     ":cppgc_headers",
     ":v8_version",
   ]
+  import("//brave/v8/sources.gni") sources += brave_v8_headers_sources public_deps += brave_v8_headers_public_deps
 }
 
 if (v8_generate_external_defines_header) {
