diff --git a/include/v8-primitive.h b/include/v8-primitive.h
index eb0a791cf73946c0955f6ccadbb36f0c4999cbab..d025467aab8d6bb0a6c97bb57d133d23093657eb 100644
--- a/include/v8-primitive.h
+++ b/include/v8-primitive.h
@@ -510,7 +510,8 @@ class V8_EXPORT String : public Name {
    */
   class V8_EXPORT Utf8Value {
    public:
-    Utf8Value(Isolate* isolate, Local<v8::Value> obj);
+    Utf8Value(Isolate* isolate, Local<v8::Value> obj,
+              WriteOptions options = REPLACE_INVALID_UTF8);
     ~Utf8Value();
     char* operator*() { return str_; }
     const char* operator*() const { return str_; }
