diff --git a/src/builtins/builtins-collections-gen.h b/src/builtins/builtins-collections-gen.h
index d69534dfa6708db642518bf3d6e7cf223992b5f9..867d558144a3c2dd9772f3d8f285f0dc5dc1e3cb 100644
--- a/src/builtins/builtins-collections-gen.h
+++ b/src/builtins/builtins-collections-gen.h
@@ -198,8 +198,9 @@ class CollectionsBuiltinsAssembler : public BaseCollectionsAssembler {
   //
   // Utility used by Object.groupBy and Map.groupBy.
   const TNode<OrderedHashMap> AddValueToKeyedGroup(
-      const TNode<OrderedHashMap> groups, const TNode<Object> key,
-      const TNode<Object> value, const TNode<String> methodName);
+      const TNode<Context> context, const TNode<OrderedHashMap> groups,
+      const TNode<Object> key, const TNode<Object> value,
+      const TNode<String> methodName);
 
   // Normalizes -0 to +0.
   const TNode<Object> NormalizeNumberKey(const TNode<Object> key);
