diff --git a/src/ast/scopes.cc b/src/ast/scopes.cc
index 39e3a8d5d59acbe9405f351fa6c57121c1900657..2aa1bb6c0f8f6e098a3ddf3f54ec2d0d5af9dc24 100644
--- a/src/ast/scopes.cc
+++ b/src/ast/scopes.cc
@@ -2445,7 +2445,7 @@ bool Scope::MustAllocate(Variable* var) {
     var->set_is_used();
     if (inner_scope_calls_eval_ && !var->is_this()) var->SetMaybeAssigned();
   }
-  DCHECK(!var->has_forced_context_allocation() || var->is_used());
+  CHECK(!var->has_forced_context_allocation() || var->is_used());
   // Global variables do not need to be allocated.
   return !var->IsGlobalObjectProperty() && var->is_used();
 }
