diff --git a/src/codegen/handler-table.cc b/src/codegen/handler-table.cc
index e1c1f6aab01da10b90f0176c3f8c0afc3c854990..b815ff27f180a0152dcc8bd123be013a45d1f32f 100644
--- a/src/codegen/handler-table.cc
+++ b/src/codegen/handler-table.cc
@@ -141,6 +141,7 @@ void HandlerTable::SetRangeEnd(int index, int value) {
 
 void HandlerTable::SetRangeHandler(int index, int handler_offset,
                                    CatchPrediction prediction) {
+  CHECK(HandlerOffsetField::is_valid(handler_offset));
   int value = HandlerOffsetField::encode(handler_offset) |
               HandlerWasUsedField::encode(false) |
               HandlerPredictionField::encode(prediction);
