diff --git a/src/builtins/object-groupby.tq b/src/builtins/object-groupby.tq
index 9c0b53d345d2fe53196a53d92e86f73d5a38598f..f3b23d80d1361c7dade0432c7fd3cbaa5d5dd46f 100644
--- a/src/builtins/object-groupby.tq
+++ b/src/builtins/object-groupby.tq
@@ -11,12 +11,25 @@ extern macro CollectionsBuiltinsAssembler::NormalizeNumberKey(JSAny): JSAny;
 
 }  // namespace collections
 
+// https://tc39.es/proposal-array-grouping/#sec-group-by
+transitioning macro CoerceGroupKey(implicit context: Context)(
+    key: JSAny, coerceToProperty: Boolean): JSAny {
+  // 6.g. If coercion is property, then
+  if (coerceToProperty == True) {
+    // i. Set key to Completion(ToPropertyKey(key)).
+    return ToName(key);
+  }
+  // 6.h. Else,
+  //    i. Assert: coercion is zero.
+  //   ii. If key is -0𝔽, set key to +0𝔽.
+  return collections::NormalizeNumberKey(key);
+}
+
 // https://tc39.es/proposal-array-grouping/#sec-group-by
 transitioning builtin GroupByGeneric(implicit context: Context)(
-    items: JSAny, callbackfn: Callable, coerceToProperty: Boolean,
-    methodName: String): OrderedHashMap {
-  // 3. Let groups be a new empty List.
-  let groups = AllocateOrderedHashMap();
+    items: JSAny, initialGroups: OrderedHashMap, callbackfn: Callable,
+    coerceToProperty: Boolean, methodName: String): OrderedHashMap {
+  let groups = initialGroups;
 
   // 4. Let iteratorRecord be ? GetIterator(items, sync).
   const fastIteratorResultMap = GetIteratorResultMap();
@@ -58,18 +71,7 @@ transitioning builtin GroupByGeneric(implicit context: Context)(
     let key: JSAny;
     try {
       key = Call(context, callbackfn, Undefined, value, k);
-
-      // g. If coercion is property, then
-      if (coerceToProperty == True) {
-        // i. Set key to Completion(ToPropertyKey(key)).
-        key = ToName(key);
-      }
-      // h. Else,
-      else {
-        // i. Assert: coercion is zero.
-        // ii. If key is -0𝔽, set key to +0𝔽.
-        key = collections::NormalizeNumberKey(key);
-      }
+      key = CoerceGroupKey(key, coerceToProperty);
     } catch (e, message) {
       // f. and g.ii.
       // IfAbruptCloseIterator(key, iteratorRecord).
@@ -89,7 +91,7 @@ transitioning builtin GroupByGeneric(implicit context: Context)(
 
 // https://tc39.es/proposal-array-grouping/#sec-group-by
 transitioning macro GroupByImpl(implicit context: Context)(
-    items: JSAny, callback: JSAny,
+    items: JSAny, callback: JSAny, coerceToProperty: Boolean,
     methodName: constexpr string): OrderedHashMap {
   // 1. Perform ? RequireObjectCoercible(items).
   RequireObjectCoercible(items, methodName);
@@ -98,16 +100,74 @@ transitioning macro GroupByImpl(implicit context: Context)(
   const callbackfn = Cast<Callable>(callback)
       otherwise ThrowTypeError(MessageTemplate::kCalledNonCallable, callback);
 
-  // TODO(v8:12499): Array fast path.
-  return GroupByGeneric(
-      items, callbackfn, /* coerceToProperty */ True, methodName);
+  // 3. Let groups be a new empty List.
+  let groups = AllocateOrderedHashMap();
+
+  try {
+    typeswitch (items) {
+      case (array: FastJSArrayForReadWithNoCustomIteration): {
+        // Per spec, the iterator and its next method are cached up front. This
+        // means that we only need to check for no custom iteration once up
+        // front. Even though the grouping callback has arbitrary side effects,
+        // mutations to %ArrayIteratorPrototype% will not be reflected during
+        // the iteration itself. Therefore we don't need a "no custom iteration"
+        // witness.
+        let fastArrayWitness = NewFastJSArrayForReadWitness(array);
+        const stableArray = fastArrayWitness.stable;
+        let k: Smi = 0;
+
+        try {
+          while (k < stableArray.length) {
+            fastArrayWitness.Recheck() otherwise goto SlowArrayContinuation;
+            let value: JSAny;
+            try {
+              value =
+                  fastArrayWitness.LoadElementNoHole(k) otherwise IsUndefined;
+            } label IsUndefined {
+              value = Undefined;
+            }
+            const key = CoerceGroupKey(
+                Call(context, callbackfn, Undefined, value, k),
+                coerceToProperty);
+            groups = collections::AddValueToKeyedGroup(
+                groups, key, value, methodName);
+            ++k;
+          }
+        } label SlowArrayContinuation deferred {
+          // The grouping callback can mutate the array such that it is no
+          // longer fast, but it is still a JSArray. Since the spec caches the
+          // iterator up front, a fully generic fallback is not needed. Instead
+          // we encode the array iterator logic here directly for the rest of
+          // the loop.
+          while (k < stableArray.length) {
+            const value = GetProperty(stableArray, k);
+            const key = CoerceGroupKey(
+                Call(context, callbackfn, Undefined, value, k),
+                coerceToProperty);
+            groups = collections::AddValueToKeyedGroup(
+                groups, key, value, methodName);
+            ++k;
+          }
+        }
+
+        return groups;
+      }
+      case (JSAny): {
+        goto SlowGeneric;
+      }
+    }
+  } label SlowGeneric {
+    return GroupByGeneric(
+        items, groups, callbackfn, coerceToProperty, methodName);
+  }
 }
 
 transitioning javascript builtin
 ObjectGroupBy(js-implicit context: NativeContext, receiver: JSAny)(
     items: JSAny, callback: JSAny): JSAny {
   // 1. Let groups be ? GroupBy(items, callbackfn, property).
-  const groups: OrderedHashMap = GroupByImpl(items, callback, 'Object.groupBy');
+  const groups: OrderedHashMap = GroupByImpl(
+      items, callback, /* coerceToProperty */ True, 'Object.groupBy');
 
   let iter = collections::NewUnmodifiedOrderedHashMapIterator(groups);
 
@@ -122,7 +182,9 @@ ObjectGroupBy(js-implicit context: NativeContext, receiver: JSAny)(
         AllocateSwissNameDictionary(Convert<intptr>(iter.usedCapacity));
   }
   @ifnot(V8_ENABLE_SWISS_NAME_DICTIONARY) {
-    properties = AllocateNameDictionary(Convert<intptr>(iter.usedCapacity));
+    properties = AllocateNameDictionary(
+        Convert<intptr>(iter.usedCapacity),
+        AllocationFlag::kAllowLargeObjectAllocation);
   }
   const nullProtoMap = LoadSlowObjectWithNullPrototypeMap(context);
   const obj = AllocateJSObjectFromMap(nullProtoMap, properties);
