diff --git a/src/maglev/maglev-interpreter-frame-state.h b/src/maglev/maglev-interpreter-frame-state.h
index 2406d1d8fbdca10221ed22e7636fe4b7c660bece..1120fb94008e01771f02a87d90893a8d9ebc8937 100644
--- a/src/maglev/maglev-interpreter-frame-state.h
+++ b/src/maglev/maglev-interpreter-frame-state.h
@@ -877,6 +877,8 @@ class MergePointInterpreterFrameState {
            predecessors_so_far_ == 0;
   }
 
+  bool IsUnreachable() const;
+
   BasicBlockType basic_block_type() const {
     return kBasicBlockTypeBits::decode(bitfield_);
   }
