diff --git a/src/execution/frames.h b/src/execution/frames.h
index 400d8a4df85d6ae1d0fed7faeadeca966cd00030..56f389aeb2347cb3d3b6d7bcead9fe3bc29007f0 100644
--- a/src/execution/frames.h
+++ b/src/execution/frames.h
@@ -626,7 +626,8 @@ class TypedFrame : public CommonFrame {
   Tagged<HeapObject> unchecked_code() const override { return {}; }
   void Iterate(RootVisitor* v) const override;
 
-  void IterateParamsOfWasmToJSWrapper(RootVisitor* v) const;
+  void IterateParamsOfGenericWasmToJSWrapper(RootVisitor* v) const;
+  void IterateParamsOfOptimizedWasmToJSWrapper(RootVisitor* v) const;
 
  protected:
   inline explicit TypedFrame(StackFrameIteratorBase* iterator);
