diff --git a/src/compiler/access-info.cc b/src/compiler/access-info.cc
index 9d022ba402d7bfdd5ca745a4193ee0de0e3d755e..54729abd77742fdd244e691606c86b308fdfa29c 100644
--- a/src/compiler/access-info.cc
+++ b/src/compiler/access-info.cc
@@ -915,6 +915,7 @@ PropertyAccessInfo AccessInfoFactory::ComputePropertyAccessInfo(
       return PropertyAccessInfo::NotFound(zone(), receiver_map, holder);
     }
 
+    CHECK(prototype.IsJSObject());
     holder = prototype.AsJSObject();
     map = map_prototype_map;
 
