diff --git a/src/wasm/wasm-objects.tq b/src/wasm/wasm-objects.tq
index 9f204366041e49837252736ad1b83d9f5a9f815d..9287a5066846db7a775c13ae7889dc8c38dd4129 100644
--- a/src/wasm/wasm-objects.tq
+++ b/src/wasm/wasm-objects.tq
@@ -185,6 +185,7 @@ extern class WasmGlobalObject extends JSObject {
 extern class WasmTagObject extends JSObject {
   serialized_signature: PodArrayOfWasmValueType;
   tag: HeapObject;
+  instance: WasmInstanceObject|Undefined;
   canonical_type_index: Smi;
 }
 
