diff --git a/src/compiler/access-builder.cc b/src/compiler/access-builder.cc
index b35afc98bd540c3771b70fb7d529ce5a6bd9a40f..44add0632a287bed22195e90a46ee9dd603f4b20 100644
--- a/src/compiler/access-builder.cc
+++ b/src/compiler/access-builder.cc
@@ -4,6 +4,7 @@
 
 #include "src/compiler/access-builder.h"
 
+#include "src/codegen/machine-type.h"
 #include "src/compiler/type-cache.h"
 #include "src/handles/handles-inl.h"
 #include "src/objects/arguments.h"
@@ -1019,6 +1020,20 @@ FieldAccess AccessBuilder::ForFeedbackVectorSlot(int index) {
   return access;
 }
 
+// static
+FieldAccess AccessBuilder::ForPropertyArraySlot(int index,
+                                                Representation representation) {
+  int offset = PropertyArray::OffsetOfElementAt(index);
+  MachineType machine_type =
+      representation.IsHeapObject() || representation.IsDouble()
+          ? MachineType::TaggedPointer()
+          : MachineType::AnyTagged();
+  FieldAccess access = {
+      kTaggedBase, offset,       Handle<Name>(),    OptionalMapRef(),
+      Type::Any(), machine_type, kFullWriteBarrier, "PropertyArraySlot"};
+  return access;
+}
+
 // static
 FieldAccess AccessBuilder::ForWeakFixedArraySlot(int index) {
   int offset = WeakFixedArray::OffsetOfElementAt(index);
