diff --git a/src/compiler/heap-refs.cc b/src/compiler/heap-refs.cc
index 9f3f3fd80631183e9178c4a0676c55bfaeb0bcba..bb3b57b88a136e3b734f5070d091149e1f28b271 100644
--- a/src/compiler/heap-refs.cc
+++ b/src/compiler/heap-refs.cc
@@ -1662,6 +1662,7 @@ HolderLookupResult FunctionTemplateInfoRef::LookupHolderOfExpectedType(
   if (!expected_receiver_type->IsTemplateFor(prototype.object()->map())) {
     return not_found;
   }
+  CHECK(prototype.IsJSObject());
   return HolderLookupResult(CallOptimization::kHolderFound,
                             prototype.AsJSObject());
 }
