diff --git a/src/builtins/builtins-collections-gen.cc b/src/builtins/builtins-collections-gen.cc
index d22faf86aab33344c1ba14eb06e6ec6c34754c95..6f1fbc40d797b86315f7572b98961df18bed4991 100644
--- a/src/builtins/builtins-collections-gen.cc
+++ b/src/builtins/builtins-collections-gen.cc
@@ -1758,8 +1758,8 @@ TNode<OrderedHashSet> CollectionsBuiltinsAssembler::AddToSetTable(
   key = NormalizeNumberKey(key);
 
   GrowCollection<OrderedHashSet> grow = [this, context, table, method_name]() {
-    TNode<OrderedHashSet> new_table = Cast(CallRuntime(
-        Runtime::kOrderedHashSetEnsureGrowable, context, table, method_name));
+    TNode<OrderedHashSet> new_table = Cast(
+        CallRuntime(Runtime::kOrderedHashSetGrow, context, table, method_name));
     // TODO(v8:13556): check if the table is updated and remove pointer to the
     // new table.
     return new_table;
@@ -2295,12 +2295,12 @@ TF_BUILTIN(FindOrderedHashSetEntry, CollectionsBuiltinsAssembler) {
 }
 
 const TNode<OrderedHashMap> CollectionsBuiltinsAssembler::AddValueToKeyedGroup(
-    const TNode<OrderedHashMap> groups, const TNode<Object> key,
-    const TNode<Object> value, const TNode<String> methodName) {
-  GrowCollection<OrderedHashMap> grow = [this, groups, methodName]() {
-    TNode<OrderedHashMap> new_groups =
-        CAST(CallRuntime(Runtime::kOrderedHashMapEnsureGrowable,
-                         NoContextConstant(), groups, methodName));
+    const TNode<Context> context, const TNode<OrderedHashMap> groups,
+    const TNode<Object> key, const TNode<Object> value,
+    const TNode<String> methodName) {
+  GrowCollection<OrderedHashMap> grow = [&]() {
+    TNode<OrderedHashMap> new_groups = CAST(
+        CallRuntime(Runtime::kOrderedHashMapGrow, context, groups, methodName));
     // The groups OrderedHashMap is not escaped to user script while grouping
     // items, so there can't be live iterators. So we don't need to keep the
     // pointer from the old table to the new one.
@@ -2308,7 +2308,7 @@ const TNode<OrderedHashMap> CollectionsBuiltinsAssembler::AddValueToKeyedGroup(
     Branch(TaggedEqual(groups, new_groups), &done, &did_grow);
     BIND(&did_grow);
     {
-      StoreObjectFieldNoWriteBarrier(groups, OrderedHashMap::NextTableIndex(),
+      StoreObjectFieldNoWriteBarrier(groups, OrderedHashMap::NextTableOffset(),
                                      SmiConstant(0));
       Goto(&done);
     }
