diff --git a/src/builtins/js-to-wasm.tq b/src/builtins/js-to-wasm.tq
index 2a5a4421283454fae63bfcf62194339b362e844c..7961f7176148f178f6ed38ebd70f7d5036da03df 100644
--- a/src/builtins/js-to-wasm.tq
+++ b/src/builtins/js-to-wasm.tq
@@ -559,6 +559,15 @@ macro JSToWasmWrapperHelper(
 
   let locationAllocator = LocationAllocatorForParams(paramBuffer);
   let hasRefParam: bool = false;
+  // A storage for converted reference parameters, so that they don't get
+  // garbage collected if the conversion of primitive parameters causes a GC.
+  // The storage gets allocated lazily when the first reference parameter gets
+  // converted to avoid performance regressions for signatures without tagged
+  // parameters. An old implementation used the `arguments` array as the storage
+  // for converted reference parameters, but this does not work because then
+  // converted reference parameters can be accessed from JavaScript using
+  // `Function.prototype.arguments`.
+  let convertedTagged: FixedArray|Smi = Convert<Smi>(0);
 
   let paramTypeIndex: int32 = 0;
   // For stack switching paramTypeIndex and paramIndex diverges,
@@ -612,7 +621,12 @@ macro JSToWasmWrapperHelper(
       // primitive types are finished. For now we write the converted parameter
       // back to the stack.
       hasRefParam = true;
-      arguments[Convert<intptr>(paramIndex)] =
+      if (TaggedIsSmi(convertedTagged)) {
+        convertedTagged =
+            WasmAllocateZeroedFixedArray(Convert<intptr>(paramCount));
+      }
+      UnsafeCast<FixedArray>(convertedTagged)
+          .objects[Convert<intptr>(paramIndex)] =
           JSToWasmObject(context, trustedInstanceData, paramType, param);
     }
   }
@@ -642,7 +656,8 @@ macro JSToWasmWrapperHelper(
         paramIndex++;
         continue;
       }
-      const param = arguments[Convert<intptr>(paramIndex++)];
+      const param = UnsafeCast<FixedArray>(convertedTagged)
+                        .objects[Convert<intptr>(paramIndex++)];
       let toRef = locationAllocator.GetGPSlot();
       *toRef = BitcastTaggedToWord(param);
       k++;
