diff --git a/src/api/api.h b/src/api/api.h
index 40b419db777f96a9328617b3fa854cf8fd27476e..fcc588e50158fe52b2876018d0e74aa7a99ce410 100644
--- a/src/api/api.h
+++ b/src/api/api.h
@@ -127,7 +127,8 @@ class RegisteredExtension {
   V(ToLocalPrimitive, Object, Primitive)                 \
   V(FixedArrayToLocal, FixedArray, FixedArray)           \
   V(PrimitiveArrayToLocal, FixedArray, PrimitiveArray)   \
-  V(ToLocal, ScriptOrModule, ScriptOrModule)
+  V(ToLocal, ScriptOrModule, ScriptOrModule)             \
+  IF_WASM(V, ToLocal, WasmModuleObject, WasmModuleObject)
 
 #define OPEN_HANDLE_LIST(V)                     \
   V(Template, TemplateInfo)                     \
