diff --git a/src/wasm/wasm-objects.tq b/src/wasm/wasm-objects.tq
index 386114af70e17ad02d24fac63dc5f088edfab0f1..5ae9c5db18888f749318eab0d4dd2a8cb00e3138 100644
--- a/src/wasm/wasm-objects.tq
+++ b/src/wasm/wasm-objects.tq
@@ -205,6 +205,7 @@ extern class WasmGlobalObject extends JSObject {
 extern class WasmTagObject extends JSObject {
   serialized_signature: PodArrayOfWasmValueType;
   tag: HeapObject;
+  instance: WasmInstanceObject|Undefined;
   canonical_type_index: Smi;
 }
 
