diff --git a/src/objects/objects.cc b/src/objects/objects.cc
index 99446acd540ac1fb3f4d2f33f3a81ac3ab8e251a..7de54535cb77d0f4ba089c017abf6442bfd98d7d 100644
--- a/src/objects/objects.cc
+++ b/src/objects/objects.cc
@@ -1781,7 +1781,7 @@ bool Object::IterationHasObservableEffects(Tagged<Object> obj) {
   if (!IsJSObject(array_proto)) return true;
   Tagged<NativeContext> native_context = array->GetCreationContext().value();
   auto initial_array_prototype = native_context->initial_array_prototype();
-  if (initial_array_prototype != JSObject::cast(array_proto)) return true;
+  if (initial_array_prototype != array_proto) return true;
 
   Isolate* isolate = array->GetIsolate();
   // Check that the ArrayPrototype hasn't been modified in a way that would
