diff --git a/src/wasm/wasm-objects.h b/src/wasm/wasm-objects.h
index 502324d54d02ce26f179e0070b30986b482c5e83..5eee7553f7fab63125c601d6c37c994f9cd397b9 100644
--- a/src/wasm/wasm-objects.h
+++ b/src/wasm/wasm-objects.h
@@ -591,7 +591,8 @@ class WasmTagObject
   static Handle<WasmTagObject> New(Isolate* isolate,
                                    const wasm::FunctionSig* sig,
                                    uint32_t canonical_type_index,
-                                   Handle<HeapObject> tag);
+                                   Handle<HeapObject> tag,
+                                   Handle<HeapObject> instance);
 
   TQ_OBJECT_CONSTRUCTORS(WasmTagObject)
 };
