diff --git a/src/heap/incremental-marking.h b/src/heap/incremental-marking.h
index 3466e41f6acd7905cec5ed350dbcc80ea14d55b0..a9b3a1ce17e10baa3d173f93823b4fc8ba5dac3b 100644
--- a/src/heap/incremental-marking.h
+++ b/src/heap/incremental-marking.h
@@ -101,6 +101,7 @@ class V8_EXPORT_PRIVATE IncrementalMarking final {
   bool Stop();
 
   void UpdateMarkingWorklistAfterScavenge();
+  void UpdateExternalPointerTableAfterScavenge();
   void UpdateMarkedBytesAfterScavenge(size_t dead_bytes_in_new_space);
 
   // Performs incremental marking step and finalizes marking if complete.
