diff --git a/src/wasm/module-instantiate.cc b/src/wasm/module-instantiate.cc
index 436aa63f78eb9816409239110bfe307e56f57c00..0dbe14596b9a7b4e0b3a67a47f36668219c83573 100644
--- a/src/wasm/module-instantiate.cc
+++ b/src/wasm/module-instantiate.cc
@@ -2589,7 +2589,7 @@ void InstanceBuilder::ProcessExports(Handle<WasmInstanceObject> instance) {
           uint32_t canonical_sig_index =
               module_->isorecursive_canonical_type_ids[tag.sig_index];
           wrapper = WasmTagObject::New(isolate_, tag.sig, canonical_sig_index,
-                                       tag_object);
+                                       tag_object, instance);
           tags_wrappers_[exp.index] = wrapper;
         }
         desc.set_value(wrapper);
