diff --git a/src/objects/elements.cc b/src/objects/elements.cc
index 3f770292fbe13f84b1b678c49b7107aa6e717b22..1ffabbcc2394f40ab7d2f9c7d89742151014e875 100644
--- a/src/objects/elements.cc
+++ b/src/objects/elements.cc
@@ -3941,7 +3941,8 @@ class TypedElementsAccessor
     // them.
     if (IsNull(source_proto, isolate)) return false;
     if (IsJSProxy(source_proto)) return true;
-    if (!context->native_context()->is_initial_array_prototype(
+    if (IsJSObject(source_proto) &&
+        !context->native_context()->is_initial_array_prototype(
             JSObject::cast(source_proto))) {
       return true;
     }
