diff --git a/src/compiler/turboshaft/operations.cc b/src/compiler/turboshaft/operations.cc
index a025e3ea8c4d8cab3020e56f73251ff230277188..64d303bf315f6220b22ac3d3cb2aeb6f12c71593 100644
--- a/src/compiler/turboshaft/operations.cc
+++ b/src/compiler/turboshaft/operations.cc
@@ -1868,6 +1868,7 @@ bool Operation::IsOnlyUserOf(const Operation& value, const Graph& graph) const {
   DCHECK_GE(std::count(inputs().begin(), inputs().end(), graph.Index(value)),
             1);
   if (value.saturated_use_count.IsOne()) return true;
+  if (value.saturated_use_count.IsSaturated()) return false;
   return std::count(inputs().begin(), inputs().end(), graph.Index(value)) ==
          value.saturated_use_count.Get();
 }
