diff --git a/src/wasm/module-instantiate.cc b/src/wasm/module-instantiate.cc
index 4521d49fc73a3382f4a1c89a78a206b94202b229..93fef1b1c382281599986d1b7e350d1da8cf03ca 100644
--- a/src/wasm/module-instantiate.cc
+++ b/src/wasm/module-instantiate.cc
@@ -2629,7 +2629,7 @@ void InstanceBuilder::ProcessExports(Handle<WasmInstanceObject> instance) {
           uint32_t canonical_sig_index =
               module_->isorecursive_canonical_type_ids[tag.sig_index];
           wrapper = WasmTagObject::New(isolate_, tag.sig, canonical_sig_index,
-                                       tag_object);
+                                       tag_object, instance);
           tags_wrappers_[exp.index] = wrapper;
         }
         value = wrapper;
