diff --git a/src/compiler/turboshaft/loop-unrolling-reducer.cc b/src/compiler/turboshaft/loop-unrolling-reducer.cc
index cf199b251e0efe102960993f78032597243b81ff..0c912154bb7e69f22e4c9f3e0d6ca4783f25ece0 100644
--- a/src/compiler/turboshaft/loop-unrolling-reducer.cc
+++ b/src/compiler/turboshaft/loop-unrolling-reducer.cc
@@ -362,10 +362,20 @@ bool SubWillOverflow(Int lhs, Int rhs) {
   }
 }
 
+template <class Int>
+bool DivWillOverflow(Int dividend, Int divisor) {
+  if constexpr (std::is_unsigned_v<Int>) {
+    return false;
+  } else {
+    return dividend == std::numeric_limits<Int>::min() && divisor == -1;
+  }
+}
+
 }  // namespace
 
-// Returns true if the loop `for (i = init, i cmp_op max; i = i binop_cst
-// binop_op)` has fewer than `max_iter_` iterations.
+// Returns true if the loop
+// `for (i = init, i cmp_op max; i = i binop_op binop_cst)` has fewer than
+// `max_iter_` iterations.
 template <class Int>
 IterationCount StaticCanonicalForLoopMatcher::CountIterationsImpl(
     Int init, Int max, CmpOp cmp_op, Int binop_cst, BinOp binop_op,
@@ -419,6 +429,7 @@ IterationCount StaticCanonicalForLoopMatcher::CountIterationsImpl(
         // eventually stop.
         return {};
       }
+      DCHECK(!DivWillOverflow(max - init, binop_cst));
       Int quotient = (max - init) / binop_cst;
       DCHECK_GE(quotient, 0);
       return IterationCount::Approx(quotient);
@@ -434,6 +445,7 @@ IterationCount StaticCanonicalForLoopMatcher::CountIterationsImpl(
         // eventually stop.
         return {};
       }
+      if (DivWillOverflow(max - init, binop_cst)) return {};
       Int quotient = (max - init) / binop_cst;
       DCHECK_GE(quotient, 0);
       return IterationCount::Approx(quotient);
@@ -469,8 +481,9 @@ IterationCount StaticCanonicalForLoopMatcher::CountIterationsImpl(
   return {};
 }
 
-// Returns true if the loop `for (i = init, i cmp_op max; i = i binop_cst
-// binop_op)` has fewer than `max_iter_` iterations.
+// Returns true if the loop
+// `for (i = initial_input, i cmp_op cmp_cst; i = i binop_op binop_cst)` has
+// fewer than `max_iter_` iterations.
 IterationCount StaticCanonicalForLoopMatcher::CountIterations(
     uint64_t cmp_cst, CmpOp cmp_op, uint64_t initial_input, uint64_t binop_cst,
     BinOp binop_op, WordRepresentation binop_rep, bool loop_if_cond_is) const {
