diff --git a/src/compiler/property-access-builder.cc b/src/compiler/property-access-builder.cc
index 15fbff9449dfabc325869f943957f8608d523feb..9dca5b9184938e461d65745d29eb5effbb2f55a2 100644
--- a/src/compiler/property-access-builder.cc
+++ b/src/compiler/property-access-builder.cc
@@ -317,6 +317,7 @@ Node* PropertyAccessBuilder::BuildLoadDataField(
       if (field_map->is_stable()) {
         dependencies()->DependOnStableMap(field_map.value());
         field_access.map = field_map;
+        field_access.type = Type::For(*field_map, broker());
       }
     }
   }
