diff --git a/src/codegen/code-stub-assembler.h b/src/codegen/code-stub-assembler.h
index 4e6d3c0dfaece96e6a63b163bd3868fdfd621ed1..b8e18f02dcffb9a7d9b987bc4361d28c43c09685 100644
--- a/src/codegen/code-stub-assembler.h
+++ b/src/codegen/code-stub-assembler.h
@@ -2115,6 +2115,11 @@ class V8_EXPORT_PRIVATE CodeStubAssembler
                                                    TNode<Object> key,
                                                    TNode<Object> value);
 
+  TNode<JSObject> AllocatePromiseWithResolversResult(TNode<Context> context,
+                                                     TNode<Object> promise,
+                                                     TNode<Object> resolve,
+                                                     TNode<Object> reject);
+
   TNode<JSReceiver> ArraySpeciesCreate(TNode<Context> context,
                                        TNode<Object> originalArray,
                                        TNode<Number> len);
