diff --git a/src/json/json-stringifier.cc b/src/json/json-stringifier.cc
index 30dfb256231a9a094b224548598ee62787f5ce16..f65645b36a291aece369883c2cf66323e3383bad 100644
--- a/src/json/json-stringifier.cc
+++ b/src/json/json-stringifier.cc
@@ -932,6 +932,8 @@ JsonStringifier::Result JsonStringifier::Serialize_(Handle<Object> object,
         if (InstanceTypeChecker::IsJSProxy(instance_type)) {
           return SerializeJSProxy(Handle<JSProxy>::cast(object), key);
         }
+        // WASM_{STRUCT,ARRAY}_TYPE are handled in `case:` blocks above.
+        DCHECK(IsJSObject(*object));
         return SerializeJSObject(Handle<JSObject>::cast(object), key);
       }
   }
