diff --git a/src/ast/scopes.cc b/src/ast/scopes.cc
index 003bd0f27368a26893c1e2a16a7a914036ec6114..458ca764cc3297831622e8ce885ec933875a98e7 100644
--- a/src/ast/scopes.cc
+++ b/src/ast/scopes.cc
@@ -2411,7 +2411,7 @@ bool Scope::MustAllocate(Variable* var) {
     var->set_is_used();
     if (inner_scope_calls_eval_ && !var->is_this()) var->SetMaybeAssigned();
   }
-  DCHECK(!var->has_forced_context_allocation() || var->is_used());
+  CHECK(!var->has_forced_context_allocation() || var->is_used());
   // Global variables do not need to be allocated.
   return !var->IsGlobalObjectProperty() && var->is_used();
 }
