diff --git a/src/compiler/heap-refs.cc b/src/compiler/heap-refs.cc
index 3187640efcc1e44f2a304e614566ea754adc8fbd..f3713e28f0475ff04109719f72e57284a2668b23 100644
--- a/src/compiler/heap-refs.cc
+++ b/src/compiler/heap-refs.cc
@@ -1636,6 +1636,7 @@ HolderLookupResult FunctionTemplateInfoRef::LookupHolderOfExpectedType(
   if (!expected_receiver_type->IsTemplateFor(prototype.object()->map())) {
     return not_found;
   }
+  CHECK(prototype.IsJSObject());
   return HolderLookupResult(CallOptimization::kHolderFound,
                             prototype.AsJSObject());
 }
