diff --git a/src/compiler/access-builder.cc b/src/compiler/access-builder.cc
index 36db2befb31aebed9d392bfa3a7f9ee47da99545..249e0168885c1caffacd71a98a0f45307c8ba0db 100644
--- a/src/compiler/access-builder.cc
+++ b/src/compiler/access-builder.cc
@@ -4,6 +4,7 @@
 
 #include "src/compiler/access-builder.h"
 
+#include "src/codegen/machine-type.h"
 #include "src/compiler/type-cache.h"
 #include "src/handles/handles-inl.h"
 #include "src/objects/arguments.h"
@@ -1012,6 +1013,20 @@ FieldAccess AccessBuilder::ForFeedbackVectorSlot(int index) {
   return access;
 }
 
+// static
+FieldAccess AccessBuilder::ForPropertyArraySlot(int index,
+                                                Representation representation) {
+  int offset = PropertyArray::OffsetOfElementAt(index);
+  MachineType machine_type =
+      representation.IsHeapObject() || representation.IsDouble()
+          ? MachineType::TaggedPointer()
+          : MachineType::AnyTagged();
+  FieldAccess access = {
+      kTaggedBase, offset,       Handle<Name>(),    OptionalMapRef(),
+      Type::Any(), machine_type, kFullWriteBarrier, "PropertyArraySlot"};
+  return access;
+}
+
 // static
 FieldAccess AccessBuilder::ForWeakFixedArraySlot(int index) {
   int offset = WeakFixedArray::OffsetOfElementAt(index);
