diff --git a/src/wasm/wasm-objects.h b/src/wasm/wasm-objects.h
index dc43d5e686ae63000f02eaa8df8c4fb3caf7b7eb..794b56c1f9a8e39b1b71d8d39f2fe417e0bf54e1 100644
--- a/src/wasm/wasm-objects.h
+++ b/src/wasm/wasm-objects.h
@@ -607,7 +607,8 @@ class WasmTagObject
   static Handle<WasmTagObject> New(Isolate* isolate,
                                    const wasm::FunctionSig* sig,
                                    uint32_t canonical_type_index,
-                                   Handle<HeapObject> tag);
+                                   Handle<HeapObject> tag,
+                                   Handle<HeapObject> instance);
 
   TQ_OBJECT_CONSTRUCTORS(WasmTagObject)
 };
