diff --git a/src/compiler/access-builder.h b/src/compiler/access-builder.h
index 62a71e7a913b2f29db829cbfa7c805302d78ba2b..21da4a53ad71952f39dcf53633520c8f0d2fcf32 100644
--- a/src/compiler/access-builder.h
+++ b/src/compiler/access-builder.h
@@ -11,6 +11,7 @@
 #include "src/compiler/write-barrier-kind.h"
 #include "src/objects/elements-kind.h"
 #include "src/objects/js-objects.h"
+#include "src/objects/property-details.h"
 
 namespace v8 {
 namespace internal {
@@ -308,6 +309,10 @@ class V8_EXPORT_PRIVATE AccessBuilder final
 
   static FieldAccess ForFeedbackVectorSlot(int index);
 
+  // Provides access to PropertyArray slots.
+  static FieldAccess ForPropertyArraySlot(int index,
+                                          Representation representation);
+
   // Provides access to ScopeInfo flags.
   static FieldAccess ForScopeInfoFlags();
 
