diff --git a/src/objects/contexts.tq b/src/objects/contexts.tq
index 062b516608f0a58c661aaa7d162019677934f096..f438fda47e98145b0bde0e3527fab3f09384bba7 100644
--- a/src/objects/contexts.tq
+++ b/src/objects/contexts.tq
@@ -110,6 +110,7 @@ extern enum ContextSlot extends intptr constexpr 'Context::Field' {
   VALID_ITERATOR_WRAPPER_MAP_INDEX: Slot<NativeContext, Map>,
   JS_ARRAY_PACKED_ELEMENTS_MAP_INDEX: Slot<NativeContext, Map>,
   JS_ARRAY_PACKED_SMI_ELEMENTS_MAP_INDEX: Slot<NativeContext, Map>,
+  JS_MAP_MAP_INDEX: Slot<NativeContext, Map>,
   JS_SET_MAP_INDEX: Slot<NativeContext, Map>,
   MATH_RANDOM_CACHE_INDEX: Slot<NativeContext, FixedDoubleArray>,
   MATH_RANDOM_INDEX_INDEX: Slot<NativeContext, Smi>,
