diff --git a/src/wasm/module-instantiate.cc b/src/wasm/module-instantiate.cc
index c74dc117ee21b999125a59549fa1f56a69313560..066ae4a442394167f2e8cdd3c64ebb057866e57c 100644
--- a/src/wasm/module-instantiate.cc
+++ b/src/wasm/module-instantiate.cc
@@ -2656,7 +2656,9 @@ void InstanceBuilder::ProcessExports(
           uint32_t canonical_sig_index =
               module_->isorecursive_canonical_type_ids[tag.sig_index];
           wrapper = WasmTagObject::New(isolate_, tag.sig, canonical_sig_index,
-                                       tag_object);
+                                       tag_object,
+                                       handle(trusted_instance_data->instance_object(),
+                                              isolate_));
           tags_wrappers_[exp.index] = wrapper;
         }
         value = wrapper;
