diff --git a/src/objects/contexts.h b/src/objects/contexts.h
index 79b916829e08fa8ada69e6c697334328a29ae69b..3af3f0c110a9201f526161f1977cc850618de2bc 100644
--- a/src/objects/contexts.h
+++ b/src/objects/contexts.h
@@ -219,6 +219,8 @@ enum ContextLookupFlags {
   V(JS_TEMPORAL_ZONED_DATE_TIME_FUNCTION_INDEX, JSFunction,                    \
     temporal_zoned_date_time_function)                                         \
   V(JSON_OBJECT, JSObject, json_object)                                        \
+  V(PROMISE_WITHRESOLVERS_RESULT_MAP_INDEX, Map,                               \
+    promise_withresolvers_result_map)                                          \
   V(TEMPORAL_INSTANT_FIXED_ARRAY_FROM_ITERABLE_FUNCTION_INDEX, JSFunction,     \
     temporal_instant_fixed_array_from_iterable)                                \
   V(STRING_FIXED_ARRAY_FROM_ITERABLE_FUNCTION_INDEX, JSFunction,               \
