diff --git a/src/api/api.h b/src/api/api.h
index bbc42ef09680bd41d1cfcc20a494838a33ac2f98..bfcaedac83ef89d5cea67e14cb66b67cc496b688 100644
--- a/src/api/api.h
+++ b/src/api/api.h
@@ -136,7 +136,8 @@ class RegisteredExtension {
   V(ToLocalPrimitive, Object, Primitive)                 \
   V(FixedArrayToLocal, FixedArray, FixedArray)           \
   V(PrimitiveArrayToLocal, FixedArray, PrimitiveArray)   \
-  V(ToLocal, ScriptOrModule, ScriptOrModule)
+  V(ToLocal, ScriptOrModule, ScriptOrModule)             \
+  IF_WASM(V, ToLocal, WasmModuleObject, WasmModuleObject)
 
 #define OPEN_HANDLE_LIST(V)                     \
   V(Template, TemplateInfo)                     \
