diff --git a/src/parsing/parser-base.h b/src/parsing/parser-base.h
index 8d77440e96afb5877df800226749600e478aee28..a6fcba4e62093ac583bc7abd57eacec49121fc4a 100644
--- a/src/parsing/parser-base.h
+++ b/src/parsing/parser-base.h
@@ -2606,6 +2606,7 @@ typename ParserBase<Impl>::BlockT ParserBase<Impl>::ParseClassStaticBlock(
   }
 
   FunctionState initializer_state(&function_state_, &scope_, initializer_scope);
+  FunctionParsingScope body_parsing_scope(impl());
   AcceptINScope accept_in(this, true);
 
   // Each static block has its own var and lexical scope, so make a new var
