diff --git a/src/wasm/graph-builder-interface.cc b/src/wasm/graph-builder-interface.cc
index c5a420e812951769dd2b4342d84563ba29914e98..05a8d12cf3c7e7d12b6951da03531baf03aca5d6 100644
--- a/src/wasm/graph-builder-interface.cc
+++ b/src/wasm/graph-builder-interface.cc
@@ -1206,8 +1206,6 @@ class WasmGraphBuildingInterface {
 
   void BrOnNonNull(FullDecoder* decoder, const Value& ref_object, Value* result,
                    uint32_t depth, bool /* drop_null_on_fallthrough */) {
-    result->node =
-        builder_->TypeGuard(ref_object.node, ref_object.type.AsNonNull());
     SsaEnv* false_env = ssa_env_;
     SsaEnv* true_env = Split(decoder->zone(), false_env);
     false_env->SetNotMerged();
@@ -1215,6 +1213,8 @@ class WasmGraphBuildingInterface {
         builder_->BrOnNull(ref_object.node, ref_object.type);
     builder_->SetControl(false_env->control);
     ScopedSsaEnv scoped_env(this, true_env);
+    // Make sure the TypeGuard has the right Control dependency.
+    SetAndTypeNode(result, builder_->TypeGuard(ref_object.node, result->type));
     BrOrRet(decoder, depth);
   }
 
@@ -1858,8 +1858,13 @@ class WasmGraphBuildingInterface {
         // and BrOnNull interacts with the values on the stack.
         // TODO(14034): The compiler shouldn't have to access the stack used by
         // the decoder ideally.
+        // Note: This TypeGuard doesn't add any new type information but we need
+        // a non-const Value that we can return to the decoder for
+        // value_on_branch. The clean solution would be to emit the TypeGuard
+        // with the branch-type in the effect edge of the branch and use
+        // value_on_branch->type as type there.
         SetAndTypeNode(value_on_branch,
-                       builder_->TypeGuard(object.node, value_on_branch->type));
+                       builder_->TypeGuard(object.node, object.type));
         return BrOnNull(decoder, object, br_depth, true, value_on_branch);
       case HeapType::kAny:
         // Any may never need a cast as it is either implicitly convertible or
