diff --git a/src/objects/module.cc b/src/objects/module.cc
index 1166a4af3603daf4ff8c4ccd85b714ca91cf11c9..0353d04132e6b668d3f34ca916cd1ce42263be56 100644
--- a/src/objects/module.cc
+++ b/src/objects/module.cc
@@ -192,7 +192,7 @@ bool Module::Instantiate(Isolate* isolate, Handle<Module> module,
     return false;
   }
   DCHECK(module->status() == kLinked || module->status() == kEvaluated ||
-         module->status() == kErrored);
+         module->status() == kEvaluatingAsync || module->status() == kErrored);
   DCHECK(stack.empty());
   return true;
 }
