diff --git a/src/compiler/turboshaft/snapshot-table-opindex.h b/src/compiler/turboshaft/snapshot-table-opindex.h
index e8db69f188a1924e76e6cd3e6168db2339db77ee..c455aec9a6d473712248d778f804d3b8b64306eb 100644
--- a/src/compiler/turboshaft/snapshot-table-opindex.h
+++ b/src/compiler/turboshaft/snapshot-table-opindex.h
@@ -58,6 +58,9 @@ class SparseOpIndexSnapshotTable : public SnapshotTable<Value, KeyData> {
     return base::nullopt;
   }
 
+  auto begin() { return indices_to_keys_.begin(); }
+  auto end() { return indices_to_keys_.end(); }
+
  private:
   Key GetOrCreateKey(OpIndex idx) {
     auto it = indices_to_keys_.find(idx);
