diff --git a/src/objects/module.cc b/src/objects/module.cc
index 22b0e592240f8daf30fe4193709d0976d0f7a999..54f2c553a832702f283870924c87ecf42857dfce 100644
--- a/src/objects/module.cc
+++ b/src/objects/module.cc
@@ -355,8 +355,7 @@ Handle<JSModuleNamespace> Module::GetModuleNamespace(Isolate* isolate,
   //   Turbofan can use this for inlining the access.
   JSObject::OptimizeAsPrototype(ns);
 
-  Handle<PrototypeInfo> proto_info =
-      Map::GetOrCreatePrototypeInfo(Handle<JSObject>::cast(ns), isolate);
+  Handle<PrototypeInfo> proto_info = Map::GetOrCreatePrototypeInfo(ns, isolate);
   proto_info->set_module_namespace(*ns);
   return ns;
 }
