diff --git a/src/builtins/base.tq b/src/builtins/base.tq
index 97e9f0457228ab1622b1bfce4b39692e68bebc49..c3e8898b71846a6b99d4f62944440f06c9d24fe4 100644
--- a/src/builtins/base.tq
+++ b/src/builtins/base.tq
@@ -782,6 +782,9 @@ extern macro ConstructorBuiltinsAssembler::IsDictionaryMap(Map): bool;
 extern macro CodeStubAssembler::AllocateNameDictionary(constexpr int32):
     NameDictionary;
 extern macro CodeStubAssembler::AllocateNameDictionary(intptr): NameDictionary;
+// FlyWeb: for Object.groupBy with many groups (from V8 11.7).
+extern macro CodeStubAssembler::AllocateNameDictionary(
+    intptr, constexpr AllocationFlag): NameDictionary;
 extern macro CodeStubAssembler::AllocateOrderedNameDictionary(constexpr int32):
     OrderedNameDictionary;
 extern macro CodeStubAssembler::AllocateSwissNameDictionary(constexpr int32):
