diff --git a/src/heap/scavenger.cc b/src/heap/scavenger.cc
index 9b8ecc342d022749e70d7d7cd4a4d336566a2555..b78ad63c48803dcd36f520b453d04aa1c7518613 100644
--- a/src/heap/scavenger.cc
+++ b/src/heap/scavenger.cc
@@ -505,6 +505,7 @@ void ScavengerCollector::CollectGarbage() {
         &Heap::UpdateYoungReferenceInExternalStringTableEntry);
 
     heap_->incremental_marking()->UpdateMarkingWorklistAfterScavenge();
+    heap_->incremental_marking()->UpdateExternalPointerTableAfterScavenge();
 
     if (V8_UNLIKELY(v8_flags.always_use_string_forwarding_table)) {
       isolate_->string_forwarding_table()->UpdateAfterYoungEvacuation();
