diff --git a/src/parsing/parser-base.h b/src/parsing/parser-base.h
index ab01fa8f1ae36c3ec2376315e5bc3396169ed784..0f8e958143cdf152ebc5a7f8699b926a3e93bc58 100644
--- a/src/parsing/parser-base.h
+++ b/src/parsing/parser-base.h
@@ -2661,6 +2661,7 @@ typename ParserBase<Impl>::BlockT ParserBase<Impl>::ParseClassStaticBlock(
   }
 
   FunctionState initializer_state(&function_state_, &scope_, initializer_scope);
+  FunctionParsingScope body_parsing_scope(impl());
   AcceptINScope accept_in(this, true);
 
   // Each static block has its own var and lexical scope, so make a new var
