diff --git a/src/wasm/canonical-types.h b/src/wasm/canonical-types.h
index dd0150efd41efc0c852e730b1654079707755eef..df98237a241c88dc69f88d6a891f4caa2cd60f63 100644
--- a/src/wasm/canonical-types.h
+++ b/src/wasm/canonical-types.h
@@ -169,6 +169,8 @@ class TypeCanonicalizer {
   ValueType CanonicalizeValueType(const WasmModule* module, ValueType type,
                                   uint32_t recursive_group_start) const;
 
+  void CheckMaxCanonicalIndex() const;
+
   std::vector<uint32_t> canonical_supertypes_;
   // Maps groups of size >=2 to the canonical id of the first type.
   std::unordered_map<CanonicalGroup, uint32_t, CanonicalGroup::hash>
