diff --git a/src/builtins/builtins-collections-gen.h b/src/builtins/builtins-collections-gen.h
index 90eb21df3af0682e009fbdd7aeabdef6f5a3cb5e..537d11d91878bc664c072eb6e28c93ee507b6d14 100644
--- a/src/builtins/builtins-collections-gen.h
+++ b/src/builtins/builtins-collections-gen.h
@@ -198,12 +198,21 @@ class CollectionsBuiltinsAssembler : public BaseCollectionsAssembler {
   //
   // Utility used by Object.groupBy and Map.groupBy.
   const TNode<OrderedHashMap> AddValueToKeyedGroup(
-      const TNode<OrderedHashMap> groups, const TNode<Object> key,
-      const TNode<Object> value, const TNode<String> methodName);
+      const TNode<Context> context, const TNode<OrderedHashMap> groups,
+      const TNode<Object> key, const TNode<Object> value,
+      const TNode<String> methodName);
 
   // Normalizes -0 to +0.
   const TNode<Object> NormalizeNumberKey(const TNode<Object> key);
 
+  // Exposed for MapGroupBy in Torque.
+  void UnsafeStoreValueInOrderedHashMapEntry(const TNode<OrderedHashMap> table,
+                                             const TNode<Object> value,
+                                             const TNode<IntPtrT> entry) {
+    return StoreValueInOrderedHashMapEntry(table, value, entry,
+                                           CheckBounds::kDebugOnly);
+  }
+
  protected:
   template <typename IteratorType>
   TNode<HeapObject> AllocateJSCollectionIterator(
@@ -352,12 +361,6 @@ class CollectionsBuiltinsAssembler : public BaseCollectionsAssembler {
       const TNode<IntPtrT> entry,
       CheckBounds check_bounds = CheckBounds::kAlways);
 
-  void UnsafeStoreValueInOrderedHashMapEntry(const TNode<OrderedHashMap> table,
-                                             const TNode<Object> value,
-                                             const TNode<IntPtrT> entry) {
-    return StoreValueInOrderedHashMapEntry(table, value, entry,
-                                           CheckBounds::kDebugOnly);
-  }
   void UnsafeStoreKeyValueInOrderedHashMapEntry(
       const TNode<OrderedHashMap> table, const TNode<Object> key,
       const TNode<Object> value, const TNode<IntPtrT> entry) {
