diff --git a/src/maglev/maglev-ir.cc b/src/maglev/maglev-ir.cc
index 3f96dfd1927f9dd835c06b5541079e8d0adb20b4..90a6968cad5171a50f14a67f9b619376f610bb62 100644
--- a/src/maglev/maglev-ir.cc
+++ b/src/maglev/maglev-ir.cc
@@ -3311,6 +3311,7 @@ void CreateRegExpLiteral::GenerateCode(MaglevAssembler* masm,
       pattern().object(),                           // pattern
       Smi::FromInt(flags())                         // flags
   );
+  masm->DefineLazyDeoptPoint(lazy_deopt_info());
 }
 
 int GetTemplateObject::MaxCallStackArgs() const {
