diff --git a/src/wasm/wasm-objects.tq b/src/wasm/wasm-objects.tq
index 806cbb11bb4fe33e643681023ffdc42b8c464b9f..3e9f0dd6075dd8da1f7d0cc541aeb658b7bd1869 100644
--- a/src/wasm/wasm-objects.tq
+++ b/src/wasm/wasm-objects.tq
@@ -178,6 +178,7 @@ extern class WasmGlobalObject extends JSObject {
 extern class WasmTagObject extends JSObject {
   serialized_signature: PodArrayOfWasmValueType;
   tag: HeapObject;
+  instance: WasmInstanceObject|Undefined;
   canonical_type_index: Smi;
 }
 
