diff --git a/src/ast/scopes.cc b/src/ast/scopes.cc
index 581156baf34e6d470a205a3d0347ee4a2760cd78..91130ac73eab0e6d052bb8877dffbb394c03aa13 100644
--- a/src/ast/scopes.cc
+++ b/src/ast/scopes.cc
@@ -2442,7 +2442,7 @@ bool Scope::MustAllocate(Variable* var) {
     var->set_is_used();
     if (inner_scope_calls_eval_ && !var->is_this()) var->SetMaybeAssigned();
   }
-  DCHECK(!var->has_forced_context_allocation() || var->is_used());
+  CHECK(!var->has_forced_context_allocation() || var->is_used());
   // Global variables do not need to be allocated.
   return !var->IsGlobalObjectProperty() && var->is_used();
 }
