diff --git a/src/wasm/wasm-objects.tq b/src/wasm/wasm-objects.tq
index 8d931d7d402e396a1d2e61228a905efcc3b43cbd..73a0cf50d3e3f8dc2588a80cee01b454103247a5 100644
--- a/src/wasm/wasm-objects.tq
+++ b/src/wasm/wasm-objects.tq
@@ -186,6 +186,7 @@ extern class WasmGlobalObject extends JSObject {
 extern class WasmTagObject extends JSObject {
   serialized_signature: PodArrayOfWasmValueType;
   tag: HeapObject;
+  instance: WasmInstanceObject|Undefined;
   canonical_type_index: Smi;
 }
 
