diff --git a/src/maglev/maglev-ir.h b/src/maglev/maglev-ir.h
index 68a751b775190e444d4d0522c4e3cb06ee267606..6fabb8829b6a0c2ba6f4ea7097cea6b7a752241a 100644
--- a/src/maglev/maglev-ir.h
+++ b/src/maglev/maglev-ir.h
@@ -4704,7 +4704,8 @@ class CreateRegExpLiteral
   int flags() const { return flags_; }
 
   // The implementation currently calls runtime.
-  static constexpr OpProperties kProperties = OpProperties::Call();
+  static constexpr OpProperties kProperties =
+      OpProperties::Call() | OpProperties::CanThrow();
 
   int MaxCallStackArgs() const;
   void SetValueLocationConstraints();
