diff --git a/src/wasm/graph-builder-interface.cc b/src/wasm/graph-builder-interface.cc
index f3848d5db81c7b00cecf06337813bc860b568e16..977921c0d623060a8c2285b3cb56bfcec3726768 100644
--- a/src/wasm/graph-builder-interface.cc
+++ b/src/wasm/graph-builder-interface.cc
@@ -265,17 +265,18 @@ class WasmGraphBuildingInterface {
     }
   }
 
-  // Load the instance cache entries into the Ssa Environment.
+  // Load the instance cache entries into the SSA Environment.
   void LoadInstanceCacheIntoSsa(SsaEnv* ssa_env) {
     builder_->InitInstanceCache(&ssa_env->instance_cache);
   }
 
-  // Reload the instance cache entries into the Ssa Environment, if memory can
+  // Reload the instance cache entries into the SSA Environment, if memory can
   // actually grow.
   void ReloadInstanceCacheIntoSsa(SsaEnv* ssa_env, const WasmModule* module) {
-    if (module->memories.empty()) return;
-    const WasmMemory* memory0 = &module->memories[0];
-    if (memory0->initial_pages == memory0->maximum_pages) return;
+    if (!builder_->has_cached_memory()) return;
+    const WasmMemory* cached_memory =
+        &module->memories[builder_->cached_memory_index()];
+    if (cached_memory->initial_pages == cached_memory->maximum_pages) return;
     LoadInstanceCacheIntoSsa(ssa_env);
   }
 
