diff --git a/src/objects/ordered-hash-table.h b/src/objects/ordered-hash-table.h
index 3eaf2db38754bc241881c7095aad5deca5a6b531..c15e1b9a5fd4250b74590ea6bcd7369a2aab7ce1 100644
--- a/src/objects/ordered-hash-table.h
+++ b/src/objects/ordered-hash-table.h
@@ -68,8 +68,8 @@ class OrderedHashTable : public FixedArray {
  public:
   // Returns an OrderedHashTable (possibly |table|) with enough space
   // to add at least one new element.
-  static MaybeHandle<Derived> EnsureGrowable(Isolate* isolate,
-                                             Handle<Derived> table);
+  static MaybeHandle<Derived> EnsureCapacityForAdding(Isolate* isolate,
+                                                      Handle<Derived> table);
 
   // Returns an OrderedHashTable (possibly |table|) that's shrunken
   // if possible.
