diff --git a/src/builtins/object-groupby.tq b/src/builtins/object-groupby.tq
index 48774973a1c1764bc448aba34a50c73eb741ba08..a3877d7799bad7b43ea0ef97108375e8632eecd0 100644
--- a/src/builtins/object-groupby.tq
+++ b/src/builtins/object-groupby.tq
@@ -5,7 +5,8 @@
 namespace collections {
 
 extern macro CollectionsBuiltinsAssembler::AddValueToKeyedGroup(
-    OrderedHashMap, Object, Object, String): OrderedHashMap;
+    implicit context: Context)(OrderedHashMap, Object, Object,
+    String): OrderedHashMap;
 
 extern macro CollectionsBuiltinsAssembler::NormalizeNumberKey(JSAny): JSAny;
 
