diff --git a/src/parsing/parser-base.h b/src/parsing/parser-base.h
index 66e531dfe786dd7e3ea8c52cdf58214405f4f91e..1cd50138b6a8b37151f14e41548c8236c532ebf8 100644
--- a/src/parsing/parser-base.h
+++ b/src/parsing/parser-base.h
@@ -2596,6 +2596,7 @@ typename ParserBase<Impl>::BlockT ParserBase<Impl>::ParseClassStaticBlock(
   }
 
   FunctionState initializer_state(&function_state_, &scope_, initializer_scope);
+  FunctionParsingScope body_parsing_scope(impl());
   AcceptINScope accept_in(this, true);
 
   // Each static block has its own var and lexical scope, so make a new var
