diff --git a/src/runtime/runtime.h b/src/runtime/runtime.h
index f3922edff4e1a13799585116af07741922cef7c3..ac1a0eb65e852f9be249706a9edfa5722107e710 100644
--- a/src/runtime/runtime.h
+++ b/src/runtime/runtime.h
@@ -99,13 +99,13 @@ namespace internal {
 #define FOR_EACH_INTRINSIC_COLLECTIONS(F, I) \
   F(MapGrow, 1, 1)                           \
   F(MapShrink, 1, 1)                         \
-  F(OrderedHashSetEnsureGrowable, 2, 1)      \
+  F(OrderedHashSetGrow, 2, 1)                \
   F(SetGrow, 1, 1)                           \
   F(SetShrink, 1, 1)                         \
   F(TheHole, 0, 1)                           \
   F(WeakCollectionDelete, 3, 1)              \
   F(WeakCollectionSet, 4, 1)                 \
-  F(OrderedHashMapEnsureGrowable, 2, 1)
+  F(OrderedHashMapGrow, 2, 1)
 
 #define FOR_EACH_INTRINSIC_COMPILER(F, I)       \
   F(CompileOptimizedOSR, 0, 1)                  \
