diff --git a/src/maglev/maglev-ir.cc b/src/maglev/maglev-ir.cc
index 73ba7e37c9855b4ccf4399fd9796f9a0c89fd010..b099e59b6166b1f6dbd57d5e2f16ddd69b51c443 100644
--- a/src/maglev/maglev-ir.cc
+++ b/src/maglev/maglev-ir.cc
@@ -3375,6 +3375,7 @@ void CreateRegExpLiteral::GenerateCode(MaglevAssembler* masm,
       pattern().object(),                           // pattern
       Smi::FromInt(flags())                         // flags
   );
+  masm->DefineLazyDeoptPoint(lazy_deopt_info());
 }
 
 int GetTemplateObject::MaxCallStackArgs() const {
