diff --git a/src/compiler/turboshaft/machine-optimization-reducer.h b/src/compiler/turboshaft/machine-optimization-reducer.h
index a43a31909035d7e5b2fa79d96397d88ad82ef7f4..2b4bdb266558ed9850710781294f86053c657586 100644
--- a/src/compiler/turboshaft/machine-optimization-reducer.h
+++ b/src/compiler/turboshaft/machine-optimization-reducer.h
@@ -1677,7 +1677,8 @@ class MachineOptimizationReducer : public Next {
       value = TryRemoveWord32ToWord64Conversion(value);
     }
     index =
-        ReduceMemoryIndex(index.value_or_invalid(), &offset, &element_scale);
+        ReduceMemoryIndex(index.value_or_invalid(), &offset, &element_scale,
+                          kind.tagged_base);
     switch (stored_rep) {
       case MemoryRepresentation::Uint8():
       case MemoryRepresentation::Int8():
@@ -1727,12 +1728,14 @@ class MachineOptimizationReducer : public Next {
 
     while (true) {
       index =
-          ReduceMemoryIndex(index.value_or_invalid(), &offset, &element_scale);
+          ReduceMemoryIndex(index.value_or_invalid(), &offset, &element_scale,
+                          kind.tagged_base);
       if (!kind.tagged_base && !index.valid()) {
         if (OpIndex left, right;
             matcher.MatchWordAdd(base_idx, &left, &right,
                                  WordRepresentation::PointerSized()) &&
-            TryAdjustOffset(&offset, matcher.Get(right), element_scale)) {
+            TryAdjustOffset(&offset, matcher.Get(right), element_scale,
+                            kind.tagged_base)) {
           base_idx = left;
           continue;
         }
@@ -1901,7 +1904,7 @@ class MachineOptimizationReducer : public Next {
   // Try to match a constant and add it to `offset`. Return `true` if
   // successful.
   bool TryAdjustOffset(int32_t* offset, const Operation& maybe_constant,
-                       uint8_t element_scale) {
+                       uint8_t element_scale, bool tagged_base) {
     if (!maybe_constant.Is<ConstantOp>()) return false;
     const ConstantOp& constant = maybe_constant.Cast<ConstantOp>();
     if (constant.rep != WordRepresentation::PointerSized() ||
@@ -1918,7 +1921,8 @@ class MachineOptimizationReducer : public Next {
         !base::bits::SignedAddOverflow32(
             *offset,
             static_cast<int32_t>(base::bits::Unsigned(diff) << element_scale),
-            &new_offset)) {
+            &new_offset) &&
+        LoadOp::OffsetIsValid(new_offset, tagged_base)) {
       *offset = new_offset;
       return true;
     }
@@ -1945,10 +1949,10 @@ class MachineOptimizationReducer : public Next {
   // `element_scale` and returning the updated `index`.
   // Return `OpIndex::Invalid()` if the resulting index is zero.
   OpIndex ReduceMemoryIndex(OpIndex index, int32_t* offset,
-                            uint8_t* element_scale) {
+                            uint8_t* element_scale, bool tagged_base) {
     while (index.valid()) {
       const Operation& index_op = matcher.Get(index);
-      if (TryAdjustOffset(offset, index_op, *element_scale)) {
+      if (TryAdjustOffset(offset, index_op, *element_scale, tagged_base)) {
         index = OpIndex::Invalid();
         *element_scale = 0;
       } else if (const ShiftOp* shift_op = index_op.TryCast<ShiftOp>()) {
@@ -1961,7 +1965,7 @@ class MachineOptimizationReducer : public Next {
                      index_op.TryCast<WordBinopOp>()) {
         if (binary_op->kind == WordBinopOp::Kind::kAdd &&
             TryAdjustOffset(offset, matcher.Get(binary_op->right()),
-                            *element_scale)) {
+                            *element_scale, tagged_base)) {
           index = binary_op->left();
           continue;
         }
