diff --git a/src/parsing/parser-base.h b/src/parsing/parser-base.h
index 47b31cc204476b63db520ff90289c6936104bf7b..bf947e57e414497c98fd6eac977a94ee0450edd1 100644
--- a/src/parsing/parser-base.h
+++ b/src/parsing/parser-base.h
@@ -2596,6 +2596,7 @@ typename ParserBase<Impl>::BlockT ParserBase<Impl>::ParseClassStaticBlock(
   }
 
   FunctionState initializer_state(&function_state_, &scope_, initializer_scope);
+  FunctionParsingScope body_parsing_scope(impl());
   AcceptINScope accept_in(this, true);
 
   // Each static block has its own var and lexical scope, so make a new var
